{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# GO Function Hierarchy/Ontology\n",
    "Includes molecular function, biological process, and cellular component ontologies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--2023-05-24 21:42:29--  http://purl.obolibrary.org/obo/go/go-basic.obo\n",
      "Resolving purl.obolibrary.org (purl.obolibrary.org)... 3.223.180.112\n",
      "Connecting to purl.obolibrary.org (purl.obolibrary.org)|3.223.180.112|:80... connected.\n",
      "HTTP request sent, awaiting response... 302 Found\n",
      "Location: http://current.geneontology.org/ontology/go-basic.obo [following]\n",
      "--2023-05-24 21:42:29--  http://current.geneontology.org/ontology/go-basic.obo\n",
      "Resolving current.geneontology.org (current.geneontology.org)... 204.246.191.49, 204.246.191.18, 204.246.191.83, ...\n",
      "Connecting to current.geneontology.org (current.geneontology.org)|204.246.191.49|:80... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 31087589 (30M) [text/obo]\n",
      "Saving to: ‘input/go-basic.obo’\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0% 8.09M 4s\n",
      "    50K .......... .......... .......... .......... ..........  0% 7.27M 4s\n",
      "   100K .......... .......... .......... .......... ..........  0% 10.1M 4s\n",
      "   150K .......... .......... .......... .......... ..........  0% 7.22M 4s\n",
      "   200K .......... .......... .......... .......... ..........  0% 6.02M 4s\n",
      "   250K .......... .......... .......... .......... ..........  0% 7.99M 4s\n",
      "   300K .......... .......... .......... .......... ..........  1% 8.88M 4s\n",
      "   350K .......... .......... .......... .......... ..........  1% 8.50M 4s\n",
      "   400K .......... .......... .......... .......... ..........  1% 9.48M 4s\n",
      "   450K .......... .......... .......... .......... ..........  1% 9.84M 4s\n",
      "   500K .......... .......... .......... .......... ..........  1% 12.6M 3s\n",
      "   550K .......... .......... .......... .......... ..........  1% 14.0M 3s\n",
      "   600K .......... .......... .......... .......... ..........  2% 11.2M 3s\n",
      "   650K .......... .......... .......... .......... ..........  2% 15.8M 3s\n",
      "   700K .......... .......... .......... .......... ..........  2% 16.2M 3s\n",
      "   750K .......... .......... .......... .......... ..........  2% 17.7M 3s\n",
      "   800K .......... .......... .......... .......... ..........  2% 20.1M 3s\n",
      "   850K .......... .......... .......... .......... ..........  2% 15.4M 3s\n",
      "   900K .......... .......... .......... .......... ..........  3% 19.9M 3s\n",
      "   950K .......... .......... .......... .......... ..........  3% 23.8M 3s\n",
      "  1000K .......... .......... .......... .......... ..........  3% 17.6M 3s\n",
      "  1050K .......... .......... .......... .......... ..........  3% 34.7M 3s\n",
      "  1100K .......... .......... .......... .......... ..........  3% 18.8M 2s\n",
      "  1150K .......... .......... .......... .......... ..........  3% 22.9M 2s\n",
      "  1200K .......... .......... .......... .......... ..........  4% 32.3M 2s\n",
      "  1250K .......... .......... .......... .......... ..........  4% 22.5M 2s\n",
      "  1300K .......... .......... .......... .......... ..........  4% 25.1M 2s\n",
      "  1350K .......... .......... .......... .......... ..........  4% 40.4M 2s\n",
      "  1400K .......... .......... .......... .......... ..........  4% 24.7M 2s\n",
      "  1450K .......... .......... .......... .......... ..........  4% 42.9M 2s\n",
      "  1500K .......... .......... .......... .......... ..........  5% 26.7M 2s\n",
      "  1550K .......... .......... .......... .......... ..........  5% 46.8M 2s\n",
      "  1600K .......... .......... .......... .......... ..........  5% 26.3M 2s\n",
      "  1650K .......... .......... .......... .......... ..........  5% 52.3M 2s\n",
      "  1700K .......... .......... .......... .......... ..........  5% 28.2M 2s\n",
      "  1750K .......... .......... .......... .......... ..........  5% 31.1M 2s\n",
      "  1800K .......... .......... .......... .......... ..........  6% 50.0M 2s\n",
      "  1850K .......... .......... .......... .......... ..........  6% 32.9M 2s\n",
      "  1900K .......... .......... .......... .......... ..........  6% 73.4M 2s\n",
      "  1950K .......... .......... .......... .......... ..........  6% 32.7M 2s\n",
      "  2000K .......... .......... .......... .......... ..........  6% 33.8M 2s\n",
      "  2050K .......... .......... .......... .......... ..........  6% 49.0M 2s\n",
      "  2100K .......... .......... .......... .......... ..........  7% 80.4M 2s\n",
      "  2150K .......... .......... .......... .......... ..........  7% 36.5M 2s\n",
      "  2200K .......... .......... .......... .......... ..........  7% 38.2M 2s\n",
      "  2250K .......... .......... .......... .......... ..........  7% 39.0M 2s\n",
      "  2300K .......... .......... .......... .......... ..........  7% 50.7M 2s\n",
      "  2350K .......... .......... .......... .......... ..........  7%  112M 2s\n",
      "  2400K .......... .......... .......... .......... ..........  8% 40.0M 2s\n",
      "  2450K .......... .......... .......... .......... ..........  8% 42.2M 1s\n",
      "  2500K .......... .......... .......... .......... ..........  8% 42.3M 1s\n",
      "  2550K .......... .......... .......... .......... ..........  8%  265M 1s\n",
      "  2600K .......... .......... .......... .......... ..........  8% 43.9M 1s\n",
      "  2650K .......... .......... .......... .......... ..........  8% 43.9M 1s\n",
      "  2700K .......... .......... .......... .......... ..........  9% 47.3M 1s\n",
      "  2750K .......... .......... .......... .......... ..........  9% 53.2M 1s\n",
      "  2800K .......... .......... .......... .......... ..........  9%  242M 1s\n",
      "  2850K .......... .......... .......... .......... ..........  9% 48.8M 1s\n",
      "  2900K .......... .......... .......... .......... ..........  9% 53.6M 1s\n",
      "  2950K .......... .......... .......... .......... ..........  9% 54.1M 1s\n",
      "  3000K .......... .......... .......... .......... .......... 10% 52.4M 1s\n",
      "  3050K .......... .......... .......... .......... .......... 10%  271M 1s\n",
      "  3100K .......... .......... .......... .......... .......... 10% 51.6M 1s\n",
      "  3150K .......... .......... .......... .......... .......... 10% 52.9M 1s\n",
      "  3200K .......... .......... .......... .......... .......... 10% 53.7M 1s\n",
      "  3250K .......... .......... .......... .......... .......... 10%  290M 1s\n",
      "  3300K .......... .......... .......... .......... .......... 11% 56.1M 1s\n",
      "  3350K .......... .......... .......... .......... .......... 11% 57.7M 1s\n",
      "  3400K .......... .......... .......... .......... .......... 11% 54.3M 1s\n",
      "  3450K .......... .......... .......... .......... .......... 11% 76.0M 1s\n",
      "  3500K .......... .......... .......... .......... .......... 11%  285M 1s\n",
      "  3550K .......... .......... .......... .......... .......... 11% 68.9M 1s\n",
      "  3600K .......... .......... .......... .......... .......... 12% 64.1M 1s\n",
      "  3650K .......... .......... .......... .......... .......... 12% 70.6M 1s\n",
      "  3700K .......... .......... .......... .......... .......... 12% 64.7M 1s\n",
      "  3750K .......... .......... .......... .......... .......... 12%  283M 1s\n",
      "  3800K .......... .......... .......... .......... .......... 12% 64.1M 1s\n",
      "  3850K .......... .......... .......... .......... .......... 12% 72.0M 1s\n",
      "  3900K .......... .......... .......... .......... .......... 13% 69.0M 1s\n",
      "  3950K .......... .......... .......... .......... .......... 13% 70.6M 1s\n",
      "  4000K .......... .......... .......... .......... .......... 13%  186M 1s\n",
      "  4050K .......... .......... .......... .......... .......... 13% 72.5M 1s\n",
      "  4100K .......... .......... .......... .......... .......... 13% 67.3M 1s\n",
      "  4150K .......... .......... .......... .......... .......... 13% 64.8M 1s\n",
      "  4200K .......... .......... .......... .......... .......... 13% 83.6M 1s\n",
      "  4250K .......... .......... .......... .......... .......... 14%  241M 1s\n",
      "  4300K .......... .......... .......... .......... .......... 14% 74.3M 1s\n",
      "  4350K .......... .......... .......... .......... .......... 14% 69.6M 1s\n",
      "  4400K .......... .......... .......... .......... .......... 14% 86.1M 1s\n",
      "  4450K .......... .......... .......... .......... .......... 14% 80.9M 1s\n",
      "  4500K .......... .......... .......... .......... .......... 14%  245M 1s\n",
      "  4550K .......... .......... .......... .......... .......... 15% 78.5M 1s\n",
      "  4600K .......... .......... .......... .......... .......... 15% 88.3M 1s\n",
      "  4650K .......... .......... .......... .......... .......... 15% 88.8M 1s\n",
      "  4700K .......... .......... .......... .......... .......... 15% 97.1M 1s\n",
      "  4750K .......... .......... .......... .......... .......... 15%  204M 1s\n",
      "  4800K .......... .......... .......... .......... .......... 15% 92.3M 1s\n",
      "  4850K .......... .......... .......... .......... .......... 16% 96.5M 1s\n",
      "  4900K .......... .......... .......... .......... .......... 16% 91.8M 1s\n",
      "  4950K .......... .......... .......... .......... .......... 16% 96.4M 1s\n",
      "  5000K .......... .......... .......... .......... .......... 16%  217M 1s\n",
      "  5050K .......... .......... .......... .......... .......... 16% 93.8M 1s\n",
      "  5100K .......... .......... .......... .......... .......... 16% 92.4M 1s\n",
      "  5150K .......... .......... .......... .......... .......... 17% 93.6M 1s\n",
      "  5200K .......... .......... .......... .......... .......... 17% 96.2M 1s\n",
      "  5250K .......... .......... .......... .......... .......... 17%  216M 1s\n",
      "  5300K .......... .......... .......... .......... .......... 17% 98.6M 1s\n",
      "  5350K .......... .......... .......... .......... .......... 17% 90.3M 1s\n",
      "  5400K .......... .......... .......... .......... .......... 17% 95.0M 1s\n",
      "  5450K .......... .......... .......... .......... .......... 18% 99.8M 1s\n",
      "  5500K .......... .......... .......... .......... .......... 18%  229M 1s\n",
      "  5550K .......... .......... .......... .......... .......... 18% 98.7M 1s\n",
      "  5600K .......... .......... .......... .......... .......... 18%  102M 1s\n",
      "  5650K .......... .......... .......... .......... .......... 18%  104M 1s\n",
      "  5700K .......... .......... .......... .......... .......... 18%  107M 1s\n",
      "  5750K .......... .......... .......... .......... .......... 19%  212M 1s\n",
      "  5800K .......... .......... .......... .......... .......... 19%  117M 1s\n",
      "  5850K .......... .......... .......... .......... .......... 19%  119M 1s\n",
      "  5900K .......... .......... .......... .......... .......... 19%  122M 1s\n",
      "  5950K .......... .......... .......... .......... .......... 19%  225M 1s\n",
      "  6000K .......... .......... .......... .......... .......... 19%  117M 1s\n",
      "  6050K .......... .......... .......... .......... .......... 20%  113M 1s\n",
      "  6100K .......... .......... .......... .......... .......... 20%  140M 1s\n",
      "  6150K .......... .......... .......... .......... .......... 20%  104M 1s\n",
      "  6200K .......... .......... .......... .......... .......... 20%  227M 1s\n",
      "  6250K .......... .......... .......... .......... .......... 20%  136M 1s\n",
      "  6300K .......... .......... .......... .......... .......... 20%  120M 1s\n",
      "  6350K .......... .......... .......... .......... .......... 21%  122M 1s\n",
      "  6400K .......... .......... .......... .......... .......... 21%  115M 1s\n",
      "  6450K .......... .......... .......... .......... .......... 21%  232M 1s\n",
      "  6500K .......... .......... .......... .......... .......... 21%  132M 1s\n",
      "  6550K .......... .......... .......... .......... .......... 21%  116M 1s\n",
      "  6600K .......... .......... .......... .......... .......... 21%  112M 1s\n",
      "  6650K .......... .......... .......... .......... .......... 22%  128M 1s\n",
      "  6700K .......... .......... .......... .......... .......... 22%  235M 1s\n",
      "  6750K .......... .......... .......... .......... .......... 22%  130M 1s\n",
      "  6800K .......... .......... .......... .......... .......... 22%  110M 1s\n",
      "  6850K .......... .......... .......... .......... .......... 22%  142M 1s\n",
      "  6900K .......... .......... .......... .......... .......... 22%  121M 1s\n",
      "  6950K .......... .......... .......... .......... .......... 23%  228M 1s\n",
      "  7000K .......... .......... .......... .......... .......... 23%  144M 1s\n",
      "  7050K .......... .......... .......... .......... .......... 23%  123M 1s\n",
      "  7100K .......... .......... .......... .......... .......... 23%  136M 1s\n",
      "  7150K .......... .......... .......... .......... .......... 23%  147M 1s\n",
      "  7200K .......... .......... .......... .......... .......... 23%  178M 1s\n",
      "  7250K .......... .......... .......... .......... .......... 24%  154M 1s\n",
      "  7300K .......... .......... .......... .......... .......... 24%  146M 1s\n",
      "  7350K .......... .......... .......... .......... .......... 24%  143M 1s\n",
      "  7400K .......... .......... .......... .......... .......... 24%  138M 1s\n",
      "  7450K .......... .......... .......... .......... .......... 24%  208M 1s\n",
      "  7500K .......... .......... .......... .......... .......... 24%  160M 1s\n",
      "  7550K .......... .......... .......... .......... .......... 25%  157M 1s\n",
      "  7600K .......... .......... .......... .......... .......... 25%  134M 1s\n",
      "  7650K .......... .......... .......... .......... .......... 25%  172M 1s\n",
      "  7700K .......... .......... .......... .......... .......... 25%  238M 1s\n",
      "  7750K .......... .......... .......... .......... .......... 25%  136M 1s\n",
      "  7800K .......... .......... .......... .......... .......... 25%  168M 1s\n",
      "  7850K .......... .......... .......... .......... .......... 26%  151M 1s\n",
      "  7900K .......... .......... .......... .......... .......... 26%  159M 1s\n",
      "  7950K .......... .......... .......... .......... .......... 26%  203M 1s\n",
      "  8000K .......... .......... .......... .......... .......... 26%  182M 1s\n",
      "  8050K .......... .......... .......... .......... .......... 26%  170M 1s\n",
      "  8100K .......... .......... .......... .......... .......... 26%  167M 1s\n",
      "  8150K .......... .......... .......... .......... .......... 27%  155M 1s\n",
      "  8200K .......... .......... .......... .......... .......... 27%  213M 1s\n",
      "  8250K .......... .......... .......... .......... .......... 27%  180M 1s\n",
      "  8300K .......... .......... .......... .......... .......... 27%  170M 1s\n",
      "  8350K .......... .......... .......... .......... .......... 27%  162M 0s\n",
      "  8400K .......... .......... .......... .......... .......... 27%  150M 0s\n",
      "  8450K .......... .......... .......... .......... .......... 27%  214M 0s\n",
      "  8500K .......... .......... .......... .......... .......... 28%  239M 0s\n",
      "  8550K .......... .......... .......... .......... .......... 28%  140M 0s\n",
      "  8600K .......... .......... .......... .......... .......... 28%  196M 0s\n",
      "  8650K .......... .......... .......... .......... .......... 28%  158M 0s\n",
      "  8700K .......... .......... .......... .......... .......... 28%  212M 0s\n",
      "  8750K .......... .......... .......... .......... .......... 28%  166M 0s\n",
      "  8800K .......... .......... .......... .......... .......... 29%  183M 0s\n",
      "  8850K .......... .......... .......... .......... .......... 29%  194M 0s\n",
      "  8900K .......... .......... .......... .......... .......... 29%  233M 0s\n",
      "  8950K .......... .......... .......... .......... .......... 29%  149M 0s\n",
      "  9000K .......... .......... .......... .......... .......... 29%  195M 0s\n",
      "  9050K .......... .......... .......... .......... .......... 29%  179M 0s\n",
      "  9100K .......... .......... .......... .......... .......... 30%  211M 0s\n",
      "  9150K .......... .......... .......... .......... .......... 30%  212M 0s\n",
      "  9200K .......... .......... .......... .......... .......... 30%  207M 0s\n",
      "  9250K .......... .......... .......... .......... .......... 30%  177M 0s\n",
      "  9300K .......... .......... .......... .......... .......... 30%  172M 0s\n",
      "  9350K .......... .......... .......... .......... .......... 30%  188M 0s\n",
      "  9400K .......... .......... .......... .......... .......... 31%  223M 0s\n",
      "  9450K .......... .......... .......... .......... .......... 31%  239M 0s\n",
      "  9500K .......... .......... .......... .......... .......... 31%  192M 0s\n",
      "  9550K .......... .......... .......... .......... .......... 31%  182M 0s\n",
      "  9600K .......... .......... .......... .......... .......... 31%  162M 0s\n",
      "  9650K .......... .......... .......... .......... .......... 31%  184M 0s\n",
      "  9700K .......... .......... .......... .......... .......... 32%  232M 0s\n",
      "  9750K .......... .......... .......... .......... .......... 32%  209M 0s\n",
      "  9800K .......... .......... .......... .......... .......... 32%  194M 0s\n",
      "  9850K .......... .......... .......... .......... .......... 32%  214M 0s\n",
      "  9900K .......... .......... .......... .......... .......... 32%  238M 0s\n",
      "  9950K .......... .......... .......... .......... .......... 32%  212M 0s\n",
      " 10000K .......... .......... .......... .......... .......... 33%  198M 0s\n",
      " 10050K .......... .......... .......... .......... .......... 33%  230M 0s\n",
      " 10100K .......... .......... .......... .......... .......... 33%  211M 0s\n",
      " 10150K .......... .......... .......... .......... .......... 33%  227M 0s\n",
      " 10200K .......... .......... .......... .......... .......... 33%  191M 0s\n",
      " 10250K .......... .......... .......... .......... .......... 33%  206M 0s\n",
      " 10300K .......... .......... .......... .......... .......... 34%  230M 0s\n",
      " 10350K .......... .......... .......... .......... .......... 34%  206M 0s\n",
      " 10400K .......... .......... .......... .......... .......... 34%  187M 0s\n",
      " 10450K .......... .......... .......... .......... .......... 34%  241M 0s\n",
      " 10500K .......... .......... .......... .......... .......... 34%  194M 0s\n",
      " 10550K .......... .......... .......... .......... .......... 34%  229M 0s\n",
      " 10600K .......... .......... .......... .......... .......... 35%  191M 0s\n",
      " 10650K .......... .......... .......... .......... .......... 35%  205M 0s\n",
      " 10700K .......... .......... .......... .......... .......... 35%  200M 0s\n",
      " 10750K .......... .......... .......... .......... .......... 35%  220M 0s\n",
      " 10800K .......... .......... .......... .......... .......... 35%  194M 0s\n",
      " 10850K .......... .......... .......... .......... .......... 35%  216M 0s\n",
      " 10900K .......... .......... .......... .......... .......... 36%  206M 0s\n",
      " 10950K .......... .......... .......... .......... .......... 36%  226M 0s\n",
      " 11000K .......... .......... .......... .......... .......... 36%  186M 0s\n",
      " 11050K .......... .......... .......... .......... .......... 36%  202M 0s\n",
      " 11100K .......... .......... .......... .......... .......... 36%  216M 0s\n",
      " 11150K .......... .......... .......... .......... .......... 36%  220M 0s\n",
      " 11200K .......... .......... .......... .......... .......... 37%  181M 0s\n",
      " 11250K .......... .......... .......... .......... .......... 37%  220M 0s\n",
      " 11300K .......... .......... .......... .......... .......... 37%  204M 0s\n",
      " 11350K .......... .......... .......... .......... .......... 37%  197M 0s\n",
      " 11400K .......... .......... .......... .......... .......... 37%  195M 0s\n",
      " 11450K .......... .......... .......... .......... .......... 37%  215M 0s\n",
      " 11500K .......... .......... .......... .......... .......... 38%  211M 0s\n",
      " 11550K .......... .......... .......... .......... .......... 38%  237M 0s\n",
      " 11600K .......... .......... .......... .......... .......... 38% 43.5M 0s\n",
      " 11650K .......... .......... .......... .......... .......... 38%  185M 0s\n",
      " 11700K .......... .......... .......... .......... .......... 38%  213M 0s\n",
      " 11750K .......... .......... .......... .......... .......... 38%  229M 0s\n",
      " 11800K .......... .......... .......... .......... .......... 39%  214M 0s\n",
      " 11850K .......... .......... .......... .......... .......... 39%  185M 0s\n",
      " 11900K .......... .......... .......... .......... .......... 39%  231M 0s\n",
      " 11950K .......... .......... .......... .......... .......... 39%  222M 0s\n",
      " 12000K .......... .......... .......... .......... .......... 39%  196M 0s\n",
      " 12050K .......... .......... .......... .......... .......... 39%  195M 0s\n",
      " 12100K .......... .......... .......... .......... .......... 40%  240M 0s\n",
      " 12150K .......... .......... .......... .......... .......... 40% 41.4M 0s\n",
      " 12200K .......... .......... .......... .......... .......... 40%  204M 0s\n",
      " 12250K .......... .......... .......... .......... .......... 40% 39.7M 0s\n",
      " 12300K .......... .......... .......... .......... .......... 40%  203M 0s\n",
      " 12350K .......... .......... .......... .......... .......... 40%  195M 0s\n",
      " 12400K .......... .......... .......... .......... .......... 41%  228M 0s\n",
      " 12450K .......... .......... .......... .......... .......... 41%  190M 0s\n",
      " 12500K .......... .......... .......... .......... .......... 41%  225M 0s\n",
      " 12550K .......... .......... .......... .......... .......... 41%  196M 0s\n",
      " 12600K .......... .......... .......... .......... .......... 41%  212M 0s\n",
      " 12650K .......... .......... .......... .......... .......... 41%  229M 0s\n",
      " 12700K .......... .......... .......... .......... .......... 41%  223M 0s\n",
      " 12750K .......... .......... .......... .......... .......... 42%  180M 0s\n",
      " 12800K .......... .......... .......... .......... .......... 42%  218M 0s\n",
      " 12850K .......... .......... .......... .......... .......... 42% 31.3M 0s\n",
      " 12900K .......... .......... .......... .......... .......... 42%  110M 0s\n",
      " 12950K .......... .......... .......... .......... .......... 42%  185M 0s\n",
      " 13000K .......... .......... .......... .......... .......... 42% 71.7M 0s\n",
      " 13050K .......... .......... .......... .......... .......... 43%  181M 0s\n",
      " 13100K .......... .......... .......... .......... .......... 43%  240M 0s\n",
      " 13150K .......... .......... .......... .......... .......... 43%  202M 0s\n",
      " 13200K .......... .......... .......... .......... .......... 43%  210M 0s\n",
      " 13250K .......... .......... .......... .......... .......... 43%  169M 0s\n",
      " 13300K .......... .......... .......... .......... .......... 43%  214M 0s\n",
      " 13350K .......... .......... .......... .......... .......... 44%  175M 0s\n",
      " 13400K .......... .......... .......... .......... .......... 44%  220M 0s\n",
      " 13450K .......... .......... .......... .......... .......... 44% 32.5M 0s\n",
      " 13500K .......... .......... .......... .......... .......... 44%  233M 0s\n",
      " 13550K .......... .......... .......... .......... .......... 44%  215M 0s\n",
      " 13600K .......... .......... .......... .......... .......... 44%  139M 0s\n",
      " 13650K .......... .......... .......... .......... .......... 45%  145M 0s\n",
      " 13700K .......... .......... .......... .......... .......... 45% 88.4M 0s\n",
      " 13750K .......... .......... .......... .......... .......... 45% 64.3M 0s\n",
      " 13800K .......... .......... .......... .......... .......... 45%  217M 0s\n",
      " 13850K .......... .......... .......... .......... .......... 45%  199M 0s\n",
      " 13900K .......... .......... .......... .......... .......... 45%  202M 0s\n",
      " 13950K .......... .......... .......... .......... .......... 46%  227M 0s\n",
      " 14000K .......... .......... .......... .......... .......... 46%  210M 0s\n",
      " 14050K .......... .......... .......... .......... .......... 46%  180M 0s\n",
      " 14100K .......... .......... .......... .......... .......... 46%  143M 0s\n",
      " 14150K .......... .......... .......... .......... .......... 46%  206M 0s\n",
      " 14200K .......... .......... .......... .......... .......... 46% 38.0M 0s\n",
      " 14250K .......... .......... .......... .......... .......... 47%  202M 0s\n",
      " 14300K .......... .......... .......... .......... .......... 47%  211M 0s\n",
      " 14350K .......... .......... .......... .......... .......... 47%  223M 0s\n",
      " 14400K .......... .......... .......... .......... .......... 47%  226M 0s\n",
      " 14450K .......... .......... .......... .......... .......... 47%  177M 0s\n",
      " 14500K .......... .......... .......... .......... .......... 47%  222M 0s\n",
      " 14550K .......... .......... .......... .......... .......... 48%  221M 0s\n",
      " 14600K .......... .......... .......... .......... .......... 48% 67.2M 0s\n",
      " 14650K .......... .......... .......... .......... .......... 48%  217M 0s\n",
      " 14700K .......... .......... .......... .......... .......... 48%  151M 0s\n",
      " 14750K .......... .......... .......... .......... .......... 48%  212M 0s\n",
      " 14800K .......... .......... .......... .......... .......... 48% 53.3M 0s\n",
      " 14850K .......... .......... .......... .......... .......... 49%  179M 0s\n",
      " 14900K .......... .......... .......... .......... .......... 49% 38.0M 0s\n",
      " 14950K .......... .......... .......... .......... .......... 49%  204M 0s\n",
      " 15000K .......... .......... .......... .......... .......... 49% 44.1M 0s\n",
      " 15050K .......... .......... .......... .......... .......... 49%  215M 0s\n",
      " 15100K .......... .......... .......... .......... .......... 49%  195M 0s\n",
      " 15150K .......... .......... .......... .......... .......... 50%  234M 0s\n",
      " 15200K .......... .......... .......... .......... .......... 50%  193M 0s\n",
      " 15250K .......... .......... .......... .......... .......... 50%  213M 0s\n",
      " 15300K .......... .......... .......... .......... .......... 50%  222M 0s\n",
      " 15350K .......... .......... .......... .......... .......... 50%  227M 0s\n",
      " 15400K .......... .......... .......... .......... .......... 50%  184M 0s\n",
      " 15450K .......... .......... .......... .......... .......... 51% 58.1M 0s\n",
      " 15500K .......... .......... .......... .......... .......... 51%  223M 0s\n",
      " 15550K .......... .......... .......... .......... .......... 51%  181M 0s\n",
      " 15600K .......... .......... .......... .......... .......... 51%  213M 0s\n",
      " 15650K .......... .......... .......... .......... .......... 51% 25.3M 0s\n",
      " 15700K .......... .......... .......... .......... .......... 51% 96.7M 0s\n",
      " 15750K .......... .......... .......... .......... .......... 52%  217M 0s\n",
      " 15800K .......... .......... .......... .......... .......... 52%  227M 0s\n",
      " 15850K .......... .......... .......... .......... .......... 52% 70.4M 0s\n",
      " 15900K .......... .......... .......... .......... .......... 52%  180M 0s\n",
      " 15950K .......... .......... .......... .......... .......... 52%  221M 0s\n",
      " 16000K .......... .......... .......... .......... .......... 52%  229M 0s\n",
      " 16050K .......... .......... .......... .......... .......... 53%  176M 0s\n",
      " 16100K .......... .......... .......... .......... .......... 53%  115M 0s\n",
      " 16150K .......... .......... .......... .......... .......... 53%  144M 0s\n",
      " 16200K .......... .......... .......... .......... .......... 53%  208M 0s\n",
      " 16250K .......... .......... .......... .......... .......... 53%  126M 0s\n",
      " 16300K .......... .......... .......... .......... .......... 53%  150M 0s\n",
      " 16350K .......... .......... .......... .......... .......... 54%  136M 0s\n",
      " 16400K .......... .......... .......... .......... .......... 54%  230M 0s\n",
      " 16450K .......... .......... .......... .......... .......... 54%  187M 0s\n",
      " 16500K .......... .......... .......... .......... .......... 54% 63.0M 0s\n",
      " 16550K .......... .......... .......... .......... .......... 54%  136M 0s\n",
      " 16600K .......... .......... .......... .......... .......... 54% 94.7M 0s\n",
      " 16650K .......... .......... .......... .......... .......... 55%  172M 0s\n",
      " 16700K .......... .......... .......... .......... .......... 55%  145M 0s\n",
      " 16750K .......... .......... .......... .......... .......... 55%  139M 0s\n",
      " 16800K .......... .......... .......... .......... .......... 55%  205M 0s\n",
      " 16850K .......... .......... .......... .......... .......... 55%  109M 0s\n",
      " 16900K .......... .......... .......... .......... .......... 55%  187M 0s\n",
      " 16950K .......... .......... .......... .......... .......... 55%  161M 0s\n",
      " 17000K .......... .......... .......... .......... .......... 56%  211M 0s\n",
      " 17050K .......... .......... .......... .......... .......... 56%  190M 0s\n",
      " 17100K .......... .......... .......... .......... .......... 56%  193M 0s\n",
      " 17150K .......... .......... .......... .......... .......... 56%  221M 0s\n",
      " 17200K .......... .......... .......... .......... .......... 56%  220M 0s\n",
      " 17250K .......... .......... .......... .......... .......... 56%  158M 0s\n",
      " 17300K .......... .......... .......... .......... .......... 57%  188M 0s\n",
      " 17350K .......... .......... .......... .......... .......... 57%  211M 0s\n",
      " 17400K .......... .......... .......... .......... .......... 57%  100M 0s\n",
      " 17450K .......... .......... .......... .......... .......... 57% 90.6M 0s\n",
      " 17500K .......... .......... .......... .......... .......... 57%  213M 0s\n",
      " 17550K .......... .......... .......... .......... .......... 57%  220M 0s\n",
      " 17600K .......... .......... .......... .......... .......... 58% 75.0M 0s\n",
      " 17650K .......... .......... .......... .......... .......... 58%  238M 0s\n",
      " 17700K .......... .......... .......... .......... .......... 58%  195M 0s\n",
      " 17750K .......... .......... .......... .......... .......... 58%  199M 0s\n",
      " 17800K .......... .......... .......... .......... .......... 58%  192M 0s\n",
      " 17850K .......... .......... .......... .......... .......... 58%  228M 0s\n",
      " 17900K .......... .......... .......... .......... .......... 59%  216M 0s\n",
      " 17950K .......... .......... .......... .......... .......... 59%  210M 0s\n",
      " 18000K .......... .......... .......... .......... .......... 59%  192M 0s\n",
      " 18050K .......... .......... .......... .......... .......... 59%  217M 0s\n",
      " 18100K .......... .......... .......... .......... .......... 59%  212M 0s\n",
      " 18150K .......... .......... .......... .......... .......... 59%  214M 0s\n",
      " 18200K .......... .......... .......... .......... .......... 60%  187M 0s\n",
      " 18250K .......... .......... .......... .......... .......... 60%  231M 0s\n",
      " 18300K .......... .......... .......... .......... .......... 60%  217M 0s\n",
      " 18350K .......... .......... .......... .......... .......... 60%  199M 0s\n",
      " 18400K .......... .......... .......... .......... .......... 60%  192M 0s\n",
      " 18450K .......... .......... .......... .......... .......... 60%  176M 0s\n",
      " 18500K .......... .......... .......... .......... .......... 61%  161M 0s\n",
      " 18550K .......... .......... .......... .......... .......... 61%  190M 0s\n",
      " 18600K .......... .......... .......... .......... .......... 61%  227M 0s\n",
      " 18650K .......... .......... .......... .......... .......... 61%  182M 0s\n",
      " 18700K .......... .......... .......... .......... .......... 61%  195M 0s\n",
      " 18750K .......... .......... .......... .......... .......... 61%  225M 0s\n",
      " 18800K .......... .......... .......... .......... .......... 62%  220M 0s\n",
      " 18850K .......... .......... .......... .......... .......... 62%  191M 0s\n",
      " 18900K .......... .......... .......... .......... .......... 62%  214M 0s\n",
      " 18950K .......... .......... .......... .......... .......... 62%  236M 0s\n",
      " 19000K .......... .......... .......... .......... .......... 62%  194M 0s\n",
      " 19050K .......... .......... .......... .......... .......... 62%  197M 0s\n",
      " 19100K .......... .......... .......... .......... .......... 63%  222M 0s\n",
      " 19150K .......... .......... .......... .......... .......... 63%  200M 0s\n",
      " 19200K .......... .......... .......... .......... .......... 63%  229M 0s\n",
      " 19250K .......... .......... .......... .......... .......... 63%  174M 0s\n",
      " 19300K .......... .......... .......... .......... .......... 63%  211M 0s\n",
      " 19350K .......... .......... .......... .......... .......... 63%  223M 0s\n",
      " 19400K .......... .......... .......... .......... .......... 64%  201M 0s\n",
      " 19450K .......... .......... .......... .......... .......... 64%  255M 0s\n",
      " 19500K .......... .......... .......... .......... .......... 64%  313M 0s\n",
      " 19550K .......... .......... .......... .......... .......... 64%  343M 0s\n",
      " 19600K .......... .......... .......... .......... .......... 64%  350M 0s\n",
      " 19650K .......... .......... .......... .......... .......... 64%  309M 0s\n",
      " 19700K .......... .......... .......... .......... .......... 65%  352M 0s\n",
      " 19750K .......... .......... .......... .......... .......... 65%  301M 0s\n",
      " 19800K .......... .......... .......... .......... .......... 65%  297M 0s\n",
      " 19850K .......... .......... .......... .......... .......... 65% 68.0M 0s\n",
      " 19900K .......... .......... .......... .......... .......... 65%  267M 0s\n",
      " 19950K .......... .......... .......... .......... .......... 65%  277M 0s\n",
      " 20000K .......... .......... .......... .......... .......... 66%  307M 0s\n",
      " 20050K .......... .......... .......... .......... .......... 66%  247M 0s\n",
      " 20100K .......... .......... .......... .......... .......... 66% 28.1M 0s\n",
      " 20150K .......... .......... .......... .......... .......... 66%  277M 0s\n",
      " 20200K .......... .......... .......... .......... .......... 66%  316M 0s\n",
      " 20250K .......... .......... .......... .......... .......... 66%  229M 0s\n",
      " 20300K .......... .......... .......... .......... .......... 67%  307M 0s\n",
      " 20350K .......... .......... .......... .......... .......... 67%  301M 0s\n",
      " 20400K .......... .......... .......... .......... .......... 67%  288M 0s\n",
      " 20450K .......... .......... .......... .......... .......... 67%  249M 0s\n",
      " 20500K .......... .......... .......... .......... .......... 67%  317M 0s\n",
      " 20550K .......... .......... .......... .......... .......... 67%  264M 0s\n",
      " 20600K .......... .......... .......... .......... .......... 68%  293M 0s\n",
      " 20650K .......... .......... .......... .......... .......... 68%  258M 0s\n",
      " 20700K .......... .......... .......... .......... .......... 68%  299M 0s\n",
      " 20750K .......... .......... .......... .......... .......... 68%  306M 0s\n",
      " 20800K .......... .......... .......... .......... .......... 68%  295M 0s\n",
      " 20850K .......... .......... .......... .......... .......... 68%  235M 0s\n",
      " 20900K .......... .......... .......... .......... .......... 69%  320M 0s\n",
      " 20950K .......... .......... .......... .......... .......... 69% 87.5M 0s\n",
      " 21000K .......... .......... .......... .......... .......... 69%  236M 0s\n",
      " 21050K .......... .......... .......... .......... .......... 69% 77.7M 0s\n",
      " 21100K .......... .......... .......... .......... .......... 69%  285M 0s\n",
      " 21150K .......... .......... .......... .......... .......... 69%  141M 0s\n",
      " 21200K .......... .......... .......... .......... .......... 69%  272M 0s\n",
      " 21250K .......... .......... .......... .......... .......... 70%  265M 0s\n",
      " 21300K .......... .......... .......... .......... .......... 70%  352M 0s\n",
      " 21350K .......... .......... .......... .......... .......... 70%  258M 0s\n",
      " 21400K .......... .......... .......... .......... .......... 70%  227M 0s\n",
      " 21450K .......... .......... .......... .......... .......... 70%  250M 0s\n",
      " 21500K .......... .......... .......... .......... .......... 70%  249M 0s\n",
      " 21550K .......... .......... .......... .......... .......... 71%  301M 0s\n",
      " 21600K .......... .......... .......... .......... .......... 71%  292M 0s\n",
      " 21650K .......... .......... .......... .......... .......... 71%  275M 0s\n",
      " 21700K .......... .......... .......... .......... .......... 71%  123M 0s\n",
      " 21750K .......... .......... .......... .......... .......... 71%  240M 0s\n",
      " 21800K .......... .......... .......... .......... .......... 71%  234M 0s\n",
      " 21850K .......... .......... .......... .......... .......... 72%  286M 0s\n",
      " 21900K .......... .......... .......... .......... .......... 72%  295M 0s\n",
      " 21950K .......... .......... .......... .......... .......... 72% 89.4M 0s\n",
      " 22000K .......... .......... .......... .......... .......... 72%  262M 0s\n",
      " 22050K .......... .......... .......... .......... .......... 72%  239M 0s\n",
      " 22100K .......... .......... .......... .......... .......... 72%  280M 0s\n",
      " 22150K .......... .......... .......... .......... .......... 73%  110M 0s\n",
      " 22200K .......... .......... .......... .......... .......... 73%  242M 0s\n",
      " 22250K .......... .......... .......... .......... .......... 73% 98.4M 0s\n",
      " 22300K .......... .......... .......... .......... .......... 73%  274M 0s\n",
      " 22350K .......... .......... .......... .......... .......... 73%  318M 0s\n",
      " 22400K .......... .......... .......... .......... .......... 73%  296M 0s\n",
      " 22450K .......... .......... .......... .......... .......... 74%  210M 0s\n",
      " 22500K .......... .......... .......... .......... .......... 74%  282M 0s\n",
      " 22550K .......... .......... .......... .......... .......... 74%  270M 0s\n",
      " 22600K .......... .......... .......... .......... .......... 74%  251M 0s\n",
      " 22650K .......... .......... .......... .......... .......... 74% 90.1M 0s\n",
      " 22700K .......... .......... .......... .......... .......... 74%  311M 0s\n",
      " 22750K .......... .......... .......... .......... .......... 75%  149M 0s\n",
      " 22800K .......... .......... .......... .......... .......... 75%  309M 0s\n",
      " 22850K .......... .......... .......... .......... .......... 75%  271M 0s\n",
      " 22900K .......... .......... .......... .......... .......... 75%  281M 0s\n",
      " 22950K .......... .......... .......... .......... .......... 75%  312M 0s\n",
      " 23000K .......... .......... .......... .......... .......... 75%  289M 0s\n",
      " 23050K .......... .......... .......... .......... .......... 76% 58.5M 0s\n",
      " 23100K .......... .......... .......... .......... .......... 76%  283M 0s\n",
      " 23150K .......... .......... .......... .......... .......... 76%  287M 0s\n",
      " 23200K .......... .......... .......... .......... .......... 76%  250M 0s\n",
      " 23250K .......... .......... .......... .......... .......... 76%  302M 0s\n",
      " 23300K .......... .......... .......... .......... .......... 76%  310M 0s\n",
      " 23350K .......... .......... .......... .......... .......... 77%  150M 0s\n",
      " 23400K .......... .......... .......... .......... .......... 77% 74.5M 0s\n",
      " 23450K .......... .......... .......... .......... .......... 77%  236M 0s\n",
      " 23500K .......... .......... .......... .......... .......... 77%  276M 0s\n",
      " 23550K .......... .......... .......... .......... .......... 77%  264M 0s\n",
      " 23600K .......... .......... .......... .......... .......... 77%  269M 0s\n",
      " 23650K .......... .......... .......... .......... .......... 78% 88.5M 0s\n",
      " 23700K .......... .......... .......... .......... .......... 78%  186M 0s\n",
      " 23750K .......... .......... .......... .......... .......... 78%  312M 0s\n",
      " 23800K .......... .......... .......... .......... .......... 78%  308M 0s\n",
      " 23850K .......... .......... .......... .......... .......... 78%  279M 0s\n",
      " 23900K .......... .......... .......... .......... .......... 78%  299M 0s\n",
      " 23950K .......... .......... .......... .......... .......... 79%  309M 0s\n",
      " 24000K .......... .......... .......... .......... .......... 79%  262M 0s\n",
      " 24050K .......... .......... .......... .......... .......... 79%  232M 0s\n",
      " 24100K .......... .......... .......... .......... .......... 79%  324M 0s\n",
      " 24150K .......... .......... .......... .......... .......... 79%  248M 0s\n",
      " 24200K .......... .......... .......... .......... .......... 79%  304M 0s\n",
      " 24250K .......... .......... .......... .......... .......... 80%  269M 0s\n",
      " 24300K .......... .......... .......... .......... .......... 80%  272M 0s\n",
      " 24350K .......... .......... .......... .......... .......... 80%  281M 0s\n",
      " 24400K .......... .......... .......... .......... .......... 80%  318M 0s\n",
      " 24450K .......... .......... .......... .......... .......... 80%  247M 0s\n",
      " 24500K .......... .......... .......... .......... .......... 80%  303M 0s\n",
      " 24550K .......... .......... .......... .......... .......... 81%  363M 0s\n",
      " 24600K .......... .......... .......... .......... .......... 81%  158M 0s\n",
      " 24650K .......... .......... .......... .......... .......... 81%  245M 0s\n",
      " 24700K .......... .......... .......... .......... .......... 81% 87.6M 0s\n",
      " 24750K .......... .......... .......... .......... .......... 81%  176M 0s\n",
      " 24800K .......... .......... .......... .......... .......... 81%  369M 0s\n",
      " 24850K .......... .......... .......... .......... .......... 82%  146M 0s\n",
      " 24900K .......... .......... .......... .......... .......... 82%  136M 0s\n",
      " 24950K .......... .......... .......... .......... .......... 82%  327M 0s\n",
      " 25000K .......... .......... .......... .......... .......... 82%  246M 0s\n",
      " 25050K .......... .......... .......... .......... .......... 82%  304M 0s\n",
      " 25100K .......... .......... .......... .......... .......... 82%  169M 0s\n",
      " 25150K .......... .......... .......... .......... .......... 83%  221M 0s\n",
      " 25200K .......... .......... .......... .......... .......... 83%  146M 0s\n",
      " 25250K .......... .......... .......... .......... .......... 83%  171M 0s\n",
      " 25300K .......... .......... .......... .......... .......... 83%  182M 0s\n",
      " 25350K .......... .......... .......... .......... .......... 83%  236M 0s\n",
      " 25400K .......... .......... .......... .......... .......... 83%  209M 0s\n",
      " 25450K .......... .......... .......... .......... .......... 83%  205M 0s\n",
      " 25500K .......... .......... .......... .......... .......... 84%  180M 0s\n",
      " 25550K .......... .......... .......... .......... .......... 84%  220M 0s\n",
      " 25600K .......... .......... .......... .......... .......... 84%  310M 0s\n",
      " 25650K .......... .......... .......... .......... .......... 84%  331M 0s\n",
      " 25700K .......... .......... .......... .......... .......... 84%  346M 0s\n",
      " 25750K .......... .......... .......... .......... .......... 84%  398M 0s\n",
      " 25800K .......... .......... .......... .......... .......... 85%  363M 0s\n",
      " 25850K .......... .......... .......... .......... .......... 85%  300M 0s\n",
      " 25900K .......... .......... .......... .......... .......... 85%  258M 0s\n",
      " 25950K .......... .......... .......... .......... .......... 85%  201M 0s\n",
      " 26000K .......... .......... .......... .......... .......... 85%  199M 0s\n",
      " 26050K .......... .......... .......... .......... .......... 85%  113M 0s\n",
      " 26100K .......... .......... .......... .......... .......... 86%  152M 0s\n",
      " 26150K .......... .......... .......... .......... .......... 86%  125M 0s\n",
      " 26200K .......... .......... .......... .......... .......... 86%  246M 0s\n",
      " 26250K .......... .......... .......... .......... .......... 86%  257M 0s\n",
      " 26300K .......... .......... .......... .......... .......... 86%  150M 0s\n",
      " 26350K .......... .......... .......... .......... .......... 86%  138M 0s\n",
      " 26400K .......... .......... .......... .......... .......... 87%  194M 0s\n",
      " 26450K .......... .......... .......... .......... .......... 87%  110M 0s\n",
      " 26500K .......... .......... .......... .......... .......... 87%  246M 0s\n",
      " 26550K .......... .......... .......... .......... .......... 87%  337M 0s\n",
      " 26600K .......... .......... .......... .......... .......... 87%  163M 0s\n",
      " 26650K .......... .......... .......... .......... .......... 87%  219M 0s\n",
      " 26700K .......... .......... .......... .......... .......... 88%  354M 0s\n",
      " 26750K .......... .......... .......... .......... .......... 88%  382M 0s\n",
      " 26800K .......... .......... .......... .......... .......... 88%  367M 0s\n",
      " 26850K .......... .......... .......... .......... .......... 88%  331M 0s\n",
      " 26900K .......... .......... .......... .......... .......... 88%  378M 0s\n",
      " 26950K .......... .......... .......... .......... .......... 88%  333M 0s\n",
      " 27000K .......... .......... .......... .......... .......... 89%  342M 0s\n",
      " 27050K .......... .......... .......... .......... .......... 89%  315M 0s\n",
      " 27100K .......... .......... .......... .......... .......... 89%  161M 0s\n",
      " 27150K .......... .......... .......... .......... .......... 89%  230M 0s\n",
      " 27200K .......... .......... .......... .......... .......... 89%  221M 0s\n",
      " 27250K .......... .......... .......... .......... .......... 89%  147M 0s\n",
      " 27300K .......... .......... .......... .......... .......... 90%  209M 0s\n",
      " 27350K .......... .......... .......... .......... .......... 90%  194M 0s\n",
      " 27400K .......... .......... .......... .......... .......... 90%  179M 0s\n",
      " 27450K .......... .......... .......... .......... .......... 90%  176M 0s\n",
      " 27500K .......... .......... .......... .......... .......... 90%  165M 0s\n",
      " 27550K .......... .......... .......... .......... .......... 90%  138M 0s\n",
      " 27600K .......... .......... .......... .......... .......... 91%  187M 0s\n",
      " 27650K .......... .......... .......... .......... .......... 91% 54.4M 0s\n",
      " 27700K .......... .......... .......... .......... .......... 91%  118M 0s\n",
      " 27750K .......... .......... .......... .......... .......... 91%  262M 0s\n",
      " 27800K .......... .......... .......... .......... .......... 91% 95.0M 0s\n",
      " 27850K .......... .......... .......... .......... .......... 91%  130M 0s\n",
      " 27900K .......... .......... .......... .......... .......... 92%  278M 0s\n",
      " 27950K .......... .......... .......... .......... .......... 92%  127M 0s\n",
      " 28000K .......... .......... .......... .......... .......... 92% 80.9M 0s\n",
      " 28050K .......... .......... .......... .......... .......... 92%  157M 0s\n",
      " 28100K .......... .......... .......... .......... .......... 92% 83.7M 0s\n",
      " 28150K .......... .......... .......... .......... .......... 92% 53.6M 0s\n",
      " 28200K .......... .......... .......... .......... .......... 93%  166M 0s\n",
      " 28250K .......... .......... .......... .......... .......... 93%  111M 0s\n",
      " 28300K .......... .......... .......... .......... .......... 93%  142M 0s\n",
      " 28350K .......... .......... .......... .......... .......... 93%  264M 0s\n",
      " 28400K .......... .......... .......... .......... .......... 93%  348M 0s\n",
      " 28450K .......... .......... .......... .......... .......... 93%  195M 0s\n",
      " 28500K .......... .......... .......... .......... .......... 94%  144M 0s\n",
      " 28550K .......... .......... .......... .......... .......... 94%  231M 0s\n",
      " 28600K .......... .......... .......... .......... .......... 94%  302M 0s\n",
      " 28650K .......... .......... .......... .......... .......... 94%  215M 0s\n",
      " 28700K .......... .......... .......... .......... .......... 94%  370M 0s\n",
      " 28750K .......... .......... .......... .......... .......... 94%  366M 0s\n",
      " 28800K .......... .......... .......... .......... .......... 95%  196M 0s\n",
      " 28850K .......... .......... .......... .......... .......... 95%  272M 0s\n",
      " 28900K .......... .......... .......... .......... .......... 95%  357M 0s\n",
      " 28950K .......... .......... .......... .......... .......... 95%  377M 0s\n",
      " 29000K .......... .......... .......... .......... .......... 95%  390M 0s\n",
      " 29050K .......... .......... .......... .......... .......... 95%  304M 0s\n",
      " 29100K .......... .......... .......... .......... .......... 96%  365M 0s\n",
      " 29150K .......... .......... .......... .......... .......... 96%  357M 0s\n",
      " 29200K .......... .......... .......... .......... .......... 96%  376M 0s\n",
      " 29250K .......... .......... .......... .......... .......... 96%  326M 0s\n",
      " 29300K .......... .......... .......... .......... .......... 96%  307M 0s\n",
      " 29350K .......... .......... .......... .......... .......... 96%  377M 0s\n",
      " 29400K .......... .......... .......... .......... .......... 97%  362M 0s\n",
      " 29450K .......... .......... .......... .......... .......... 97%  298M 0s\n",
      " 29500K .......... .......... .......... .......... .......... 97%  365M 0s\n",
      " 29550K .......... .......... .......... .......... .......... 97%  359M 0s\n",
      " 29600K .......... .......... .......... .......... .......... 97%  350M 0s\n",
      " 29650K .......... .......... .......... .......... .......... 97%  320M 0s\n",
      " 29700K .......... .......... .......... .......... .......... 97%  361M 0s\n",
      " 29750K .......... .......... .......... .......... .......... 98%  360M 0s\n",
      " 29800K .......... .......... .......... .......... .......... 98%  381M 0s\n",
      " 29850K .......... .......... .......... .......... .......... 98%  312M 0s\n",
      " 29900K .......... .......... .......... .......... .......... 98%  341M 0s\n",
      " 29950K .......... .......... .......... .......... .......... 98%  364M 0s\n",
      " 30000K .......... .......... .......... .......... .......... 98%  397M 0s\n",
      " 30050K .......... .......... .......... .......... .......... 99%  300M 0s\n",
      " 30100K .......... .......... .......... .......... .......... 99%  340M 0s\n",
      " 30150K .......... .......... .......... .......... .......... 99%  393M 0s\n",
      " 30200K .......... .......... .......... .......... .......... 99%  359M 0s\n",
      " 30250K .......... .......... .......... .......... .......... 99%  308M 0s\n",
      " 30300K .......... .......... .......... .......... .......... 99%  330M 0s\n",
      " 30350K ........                                              100%  484M=0.3s\n",
      "\n",
      "2023-05-24 21:42:29 (94.7 MB/s) - ‘input/go-basic.obo’ saved [31087589/31087589]\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import csv\n",
    "import pandas as pd\n",
    "import json\n",
    "import os\n",
    "from myutils import *\n",
    "os.system('wget -N -P input/ http://purl.obolibrary.org/obo/go/go-basic.obo')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Properties of GO Terms in the original file\n",
    "starts = set()\n",
    "with open('input/go-basic.obo') as fin:\n",
    "    for line in fin:\n",
    "        start = line.split(': ')[0].strip('\\n')\n",
    "        starts.add(start)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'GO:0000001': {'name': ['mitochondrion inheritance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The distribution of mitochondria, including the mitochondrial genome, into daughter cells after mitosis or meiosis, mediated by interactions between mitochondria and the cytoskeleton.\" [GOC:mcc, PMID:10873824, PMID:11389764]'],\n",
       "  'synonym': ['\"mitochondrial inheritance\" EXACT []'],\n",
       "  'is_a': ['GO:0048308 ! organelle inheritance',\n",
       "   'GO:0048311 ! mitochondrion distribution']},\n",
       " 'GO:0000002': {'name': ['mitochondrial genome maintenance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The maintenance of the structure and integrity of the mitochondrial genome; includes replication and segregation of the mitochondrial chromosome.\" [GOC:ai, GOC:vw]'],\n",
       "  'is_a': ['GO:0007005 ! mitochondrion organization']},\n",
       " 'GO:0000003': {'name': ['reproduction'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0019952', 'GO:0050876'],\n",
       "  'def': ['\"The production of new individuals that contain some portion of genetic material inherited from one or more parent organisms.\" [GOC:go_curators, GOC:isa_complete, GOC:jl, ISBN:0198506732]'],\n",
       "  'subset': ['goslim_agr',\n",
       "   'goslim_chembl',\n",
       "   'goslim_flybase_ribbon',\n",
       "   'goslim_pir',\n",
       "   'goslim_plant'],\n",
       "  'synonym': ['\"reproductive physiological process\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Reproduction'],\n",
       "  'is_a': ['GO:0008150 ! biological_process']},\n",
       " 'GO:0000005': {'name': ['obsolete ribosomal chaperone activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Assists in the correct assembly of ribosomes or ribosomal subunits in vivo, but is not a component of the assembled ribosome when performing its normal biological function.\" [GOC:jl, PMID:12150913]'],\n",
       "  'comment': ['This term was made obsolete because it refers to a class of gene products and a biological process rather than a molecular function.'],\n",
       "  'synonym': ['\"ribosomal chaperone activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0042254', 'GO:0044183', 'GO:0051082']},\n",
       " 'GO:0000006': {'name': ['high-affinity zinc transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of zinc ions (Zn2+) from one side of a membrane to the other, probably powered by proton motive force. In high-affinity transport the transporter is able to bind the solute even if it is only present at very low concentrations.\" [TC:2.A.5.1.1]'],\n",
       "  'synonym': ['\"high affinity zinc uptake transmembrane transporter activity\" EXACT []',\n",
       "   '\"high-affinity zinc uptake transmembrane transporter activity\" RELATED []'],\n",
       "  'is_a': ['GO:0005385 ! zinc ion transmembrane transporter activity']},\n",
       " 'GO:0000007': {'name': ['low-affinity zinc ion transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of a solute or solutes from one side of a membrane to the other according to the reaction'],\n",
       "  'is_a': ['GO:0005385 ! zinc ion transmembrane transporter activity']},\n",
       " 'GO:0000008': {'name': ['obsolete thioredoxin'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000013'],\n",
       "  'def': ['\"OBSOLETE. A small disulfide-containing redox protein that serves as a general protein disulfide oxidoreductase. Interacts with a broad range of proteins by a redox mechanism, based on the reversible oxidation of 2 cysteine thiol groups to a disulfide, accompanied by the transfer of 2 electrons and 2 protons. The net result is the covalent interconversion of a disulfide and a dithiol.\" [GOC:kd]'],\n",
       "  'comment': ['This term was made obsolete because it represents gene products.'],\n",
       "  'synonym': ['\"thioredoxin\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0003756', 'GO:0015036']},\n",
       " 'GO:0000009': {'name': ['alpha-1,6-mannosyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the transfer of a mannose residue to an oligosaccharide, forming an alpha-(1->6) linkage.\" [GOC:mcc, PMID:2644248]'],\n",
       "  'synonym': ['\"1,6-alpha-mannosyltransferase activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-449718 \"Addition of a third mannose to the N-glycan precursor by ALG2\"'],\n",
       "  'is_a': ['GO:0000030 ! mannosyltransferase activity']},\n",
       " 'GO:0000010': {'name': ['trans-hexaprenyltranstransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0036422'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"all-trans-heptaprenyl-diphosphate synthase activity\" RELATED [EC:2.5.1.30]',\n",
       "   '\"HepPP synthase activity\" RELATED [EC:2.5.1.30]',\n",
       "   '\"heptaprenyl diphosphate synthase activity\" RELATED []',\n",
       "   '\"heptaprenyl pyrophosphate synthase activity\" RELATED [EC:2.5.1.30]',\n",
       "   '\"heptaprenyl pyrophosphate synthetase activity\" RELATED [EC:2.5.1.30]'],\n",
       "  'xref': ['EC:2.5.1.30',\n",
       "   'MetaCyc:TRANS-HEXAPRENYLTRANSTRANSFERASE-RXN',\n",
       "   'RHEA:27794'],\n",
       "  'is_a': ['GO:0004659 ! prenyltransferase activity']},\n",
       " 'GO:0000011': {'name': ['vacuole inheritance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The distribution of vacuoles into daughter cells after mitosis or meiosis, mediated by interactions between vacuoles and the cytoskeleton.\" [GOC:mcc, PMID:10873824, PMID:14616069]'],\n",
       "  'is_a': ['GO:0007033 ! vacuole organization',\n",
       "   'GO:0048308 ! organelle inheritance']},\n",
       " 'GO:0000012': {'name': ['single strand break repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The repair of single strand breaks in DNA. Repair of such breaks is mediated by the same enzyme systems as are used in base excision repair.\" [PMID:18626472]'],\n",
       "  'is_a': ['GO:0006281 ! DNA repair']},\n",
       " 'GO:0000014': {'name': ['single-stranded DNA endodeoxyribonuclease activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the hydrolysis of ester linkages within a single-stranded deoxyribonucleic acid molecule by creating internal breaks.\" [GOC:mah]'],\n",
       "  'synonym': ['\"single-stranded DNA specific endodeoxyribonuclease activity\" RELATED []',\n",
       "   '\"ssDNA-specific endodeoxyribonuclease activity\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0004520 ! DNA endonuclease activity']},\n",
       " 'GO:0000015': {'name': ['phosphopyruvate hydratase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multimeric enzyme complex, usually a dimer or an octamer, that catalyzes the conversion of 2-phospho-D-glycerate to phosphoenolpyruvate and water.\" [GOC:jl, ISBN:0198506732]'],\n",
       "  'subset': ['goslim_metagenomics'],\n",
       "  'synonym': ['\"enolase complex\" EXACT []'],\n",
       "  'is_a': ['GO:1902494 ! catalytic complex'],\n",
       "  'relationship': ['part_of GO:0005829 ! cytosol']},\n",
       " 'GO:0000016': {'name': ['lactase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"lactose galactohydrolase activity\" EXACT []'],\n",
       "  'xref': ['EC:3.2.1.108',\n",
       "   'MetaCyc:LACTASE-RXN',\n",
       "   'Reactome:R-HSA-189062 \"lactose + H2O => D-glucose + D-galactose\"',\n",
       "   'Reactome:R-HSA-5658001 \"Defective LCT does not hydrolyze Lac\"',\n",
       "   'RHEA:10076'],\n",
       "  'is_a': ['GO:0004553 ! hydrolase activity, hydrolyzing O-glycosyl compounds']},\n",
       " 'GO:0000017': {'name': ['alpha-glucoside transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of alpha-glucosides into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. Alpha-glucosides are glycosides in which the sugar group is a glucose residue, and the anomeric carbon of the bond is in an alpha configuration.\" [GOC:jl, ISBN:0198506732, PMID:9919658]'],\n",
       "  'is_a': ['GO:0042946 ! glucoside transport']},\n",
       " 'GO:0000018': {'name': ['regulation of DNA recombination'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of DNA recombination, a DNA metabolic process in which a new genotype is formed by reassortment of genes resulting in gene combinations different from those that were present in the parents.\" [GOC:go_curators, ISBN:0198506732]'],\n",
       "  'is_a': ['GO:0051052 ! regulation of DNA metabolic process'],\n",
       "  'relationship': ['regulates GO:0006310 ! DNA recombination']},\n",
       " 'GO:0000019': {'name': ['regulation of mitotic recombination'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of DNA recombination during mitosis.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"regulation of recombination within rDNA repeats\" NARROW []'],\n",
       "  'is_a': ['GO:0000018 ! regulation of DNA recombination'],\n",
       "  'relationship': ['regulates GO:0006312 ! mitotic recombination']},\n",
       " 'GO:0000020': {'name': ['obsolete negative regulation of recombination within rDNA repeats'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that stops, prevents, or reduces the frequency, rate or extent of genetic recombination within the DNA of the genes coding for ribosomal RNA.\" [GOC:go_curators, ISBN:0198506732]'],\n",
       "  'comment': ['This term was made obsolete because it describes a substrate-specific process.'],\n",
       "  'synonym': ['\"negative regulation of recombination within rDNA repeats\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0045950']},\n",
       " 'GO:0000022': {'name': ['mitotic spindle elongation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:1905121'],\n",
       "  'def': ['\"The cell cycle process in which the distance is lengthened between poles of the mitotic spindle. Mitotic spindle elongation begins during mitotic prophase and ends during mitotic anaphase B.\" [GOC:mtg_cell_cycle, GOC:vw, PMID:19686686]'],\n",
       "  'synonym': ['\"microtubule sliding involved in mitotic spindle elongation\" RELATED []',\n",
       "   '\"spindle elongation during mitosis\" EXACT []'],\n",
       "  'is_a': ['GO:0051231 ! spindle elongation',\n",
       "   'GO:1903047 ! mitotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0000070 ! mitotic sister chromatid segregation',\n",
       "   'part_of GO:0007052 ! mitotic spindle organization']},\n",
       " 'GO:0000023': {'name': ['maltose metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the disaccharide maltose (4-O-alpha-D-glucopyranosyl-D-glucopyranose), an intermediate in the catabolism of glycogen and starch.\" [GOC:jl, ISBN:0198506732]'],\n",
       "  'synonym': ['\"malt sugar metabolic process\" EXACT []',\n",
       "   '\"malt sugar metabolism\" EXACT []',\n",
       "   '\"maltose metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0005984 ! disaccharide metabolic process']},\n",
       " 'GO:0000024': {'name': ['maltose biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of the disaccharide maltose (4-O-alpha-D-glucopyranosyl-D-glucopyranose).\" [GOC:jl, ISBN:0198506732]'],\n",
       "  'synonym': ['\"malt sugar biosynthesis\" EXACT []',\n",
       "   '\"malt sugar biosynthetic process\" EXACT []',\n",
       "   '\"maltose anabolism\" EXACT []',\n",
       "   '\"maltose biosynthesis\" EXACT []',\n",
       "   '\"maltose formation\" EXACT []',\n",
       "   '\"maltose synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0000023 ! maltose metabolic process',\n",
       "   'GO:0046351 ! disaccharide biosynthetic process']},\n",
       " 'GO:0000025': {'name': ['maltose catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of the disaccharide maltose (4-O-alpha-D-glucopyranosyl-D-glucopyranose).\" [GOC:jl, ISBN:0198506732]'],\n",
       "  'synonym': ['\"malt sugar catabolic process\" EXACT []',\n",
       "   '\"malt sugar catabolism\" EXACT []',\n",
       "   '\"maltose breakdown\" EXACT []',\n",
       "   '\"maltose degradation\" EXACT []',\n",
       "   '\"maltose hydrolysis\" NARROW []'],\n",
       "  'xref': ['MetaCyc:MALTOSECAT-PWY'],\n",
       "  'is_a': ['GO:0000023 ! maltose metabolic process',\n",
       "   'GO:0046352 ! disaccharide catabolic process']},\n",
       " 'GO:0000026': {'name': ['alpha-1,2-mannosyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the transfer of a mannose residue to an oligosaccharide, forming an alpha-(1->2) linkage.\" [GOC:mcc, PMID:10521541]'],\n",
       "  'xref': ['Reactome:R-HSA-446187 \"ALG11 transfers the fourth and fifth Man to the N-glycan precursor\"',\n",
       "   'Reactome:R-HSA-446215 \"ALG9 transfers Man to N-glycan precursor (GlcNAc)2 (Man)6 (PP-Dol)1\"',\n",
       "   'Reactome:R-HSA-446216 \"ALG9 transfers Man to N-glycan precursor (GlcNAc)2 (Man)8 (PP-Dol)1\"',\n",
       "   'Reactome:R-HSA-4551297 \"Defective ALG11 does not transfer Man to the N-glycan precursor\"',\n",
       "   'Reactome:R-HSA-4720478 \"Defective ALG9 does not add the seventh mannose to the N-glycan precursor\"',\n",
       "   'Reactome:R-HSA-9035514 \"Defective ALG9 does not add the last mannose to the N-glycan precursor\"'],\n",
       "  'is_a': ['GO:0000030 ! mannosyltransferase activity']},\n",
       " 'GO:0000027': {'name': ['ribosomal large subunit assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of constituent RNAs and proteins to form the large ribosomal subunit.\" [GOC:jl, PMID:30467428]'],\n",
       "  'synonym': ['\"50S ribosomal subunit assembly\" NARROW [GOC:mah]',\n",
       "   '\"60S ribosomal subunit assembly\" NARROW [GOC:mah]'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0042255 ! ribosome assembly',\n",
       "   'part_of GO:0042273 ! ribosomal large subunit biogenesis']},\n",
       " 'GO:0000028': {'name': ['ribosomal small subunit assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of constituent RNAs and proteins to form the small ribosomal subunit.\" [GOC:jl, PMID:30467428]'],\n",
       "  'synonym': ['\"30S ribosomal subunit assembly\" NARROW [GOC:mah]',\n",
       "   '\"40S ribosomal subunit assembly\" NARROW [GOC:mah]'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0042255 ! ribosome assembly',\n",
       "   'part_of GO:0042274 ! ribosomal small subunit biogenesis']},\n",
       " 'GO:0000030': {'name': ['mannosyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the transfer of a mannosyl group to an acceptor molecule, typically another carbohydrate or a lipid.\" [GOC:ai, GOC:cjm]'],\n",
       "  'xref': ['Reactome:R-HSA-162797 \"mannose (a1-2) mannose (a1-6) (ethanolamineP) mannose (a1-4) glucosaminyl-acyl-PI -> mannose (a1) mannose (a1-2) mannose (a1-6) (ethanolamineP) mannose (a1-4) glucosaminyl-acyl-PI\"',\n",
       "   'Reactome:R-HSA-162830 \"glucosaminyl-acyl-PI + dolichol phosphate D-mannose -> mannose(al1-4)glucosaminyl-acyl-PI + dolichol phosphate\"',\n",
       "   'Reactome:R-HSA-446198 \"ALG12 transfers Man to N-glycan precursor (GlcNAc)2 (Man)7 (PP-Dol)1\"',\n",
       "   'Reactome:R-HSA-4720497 \"Defective ALG12 does not add mannose to the N-glycan precursor\"'],\n",
       "  'is_a': ['GO:0016758 ! hexosyltransferase activity']},\n",
       " 'GO:0000031': {'name': ['mannosylphosphate transferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the transfer of a mannosylphosphate group from one compound to another.\" [GOC:jl]'],\n",
       "  'is_a': ['GO:0016758 ! hexosyltransferase activity']},\n",
       " 'GO:0000032': {'name': ['cell wall mannoprotein biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of cell wall mannoproteins, any cell wall protein that contains covalently bound mannose residues.\" [GOC:ai]'],\n",
       "  'synonym': ['\"cell wall mannoprotein anabolism\" EXACT []',\n",
       "   '\"cell wall mannoprotein biosynthesis\" EXACT []',\n",
       "   '\"cell wall mannoprotein formation\" EXACT []',\n",
       "   '\"cell wall mannoprotein synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0006057 ! mannoprotein biosynthetic process',\n",
       "   'GO:0031506 ! cell wall glycoprotein biosynthetic process']},\n",
       " 'GO:0000033': {'name': ['alpha-1,3-mannosyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the transfer of a mannose residue to an oligosaccharide, forming an alpha-(1->3) linkage.\" [GOC:mcc, PMID:10521541]'],\n",
       "  'xref': ['Reactome:R-HSA-446188 \"ALG3 transfers Man to N-glycan precursor (GlcNAc)2 (Man)5 (PP-Dol)1\"',\n",
       "   'Reactome:R-HSA-446208 \"Addition of a second mannose to the N-glycan precursor by ALG2\"',\n",
       "   'Reactome:R-HSA-4549368 \"Defective ALG2 does not transfer a second Man to N-glycan precursor\"',\n",
       "   'Reactome:R-HSA-4720473 \"Defective ALG3 does not add mannose to the N-glycan precursor\"'],\n",
       "  'is_a': ['GO:0000030 ! mannosyltransferase activity']},\n",
       " 'GO:0000034': {'name': ['adenine deaminase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"ADase activity\" RELATED [EC:3.5.4.2]',\n",
       "   '\"adenase activity\" RELATED [EC:3.5.4.2]',\n",
       "   '\"adenine aminase activity\" RELATED [EC:3.5.4.2]',\n",
       "   '\"adenine aminohydrolase activity\" RELATED [EC:3.5.4.2]'],\n",
       "  'xref': ['EC:3.5.4.2', 'MetaCyc:ADENINE-DEAMINASE-RXN', 'RHEA:23688'],\n",
       "  'is_a': ['GO:0016814 ! hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in cyclic amidines',\n",
       "   'GO:0019239 ! deaminase activity']},\n",
       " 'GO:0000035': {'name': ['acyl binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an acyl group, any group formally derived by removal of the hydroxyl group from the acid function of a carboxylic acid.\" [GOC:curators, ISBN:0198506732]'],\n",
       "  'synonym': ['\"acyl-CoA or acyl binding\" BROAD []'],\n",
       "  'is_a': ['GO:0005488 ! binding']},\n",
       " 'GO:0000036': {'name': ['acyl carrier activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding an acyl group and presenting it for processing or offloading to a cognate enzyme. Covalently binds the acyl group via a phosphopantetheine prosthetic group and mediates protein-protein interactions with the enzyme conferring specificity. The acyl carrier protein (ACP) presents substrates to enzymes involved in fatty acid biosynthesis or in polyketide secondary metabolite biosynthesis.\" [GOC:jl, GOC:vw]'],\n",
       "  'synonym': ['\"ACP phosphopantetheine attachment site binding involved in fatty acid biosynthetic process\" EXACT []'],\n",
       "  'is_a': ['GO:0044620 ! ACP phosphopantetheine attachment site binding',\n",
       "   'GO:0140414 ! phosphopantetheine-dependent carrier activity']},\n",
       " 'GO:0000038': {'name': ['very long-chain fatty acid metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving a very long chain fatty acid, a fatty acid with an aliphatic tail of 22 or more carbons.\" [GOC:hjd]'],\n",
       "  'synonym': ['\"very long chain fatty acid metabolic process\" EXACT [GOC:bf]',\n",
       "   '\"very-long-chain fatty acid metabolic process\" EXACT []',\n",
       "   '\"very-long-chain fatty acid metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006631 ! fatty acid metabolic process']},\n",
       " 'GO:0000039': {'name': ['obsolete plasma membrane long-chain fatty acid transporter'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:ai]'],\n",
       "  'comment': ['This term was made obsolete because it describes a gene product and it contains component information.'],\n",
       "  'synonym': ['\"plasma membrane long-chain fatty acid transporter\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005324', 'GO:0005886']},\n",
       " 'GO:0000041': {'name': ['transition metal ion transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of transition metal ions into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. A transition metal is an element whose atom has an incomplete d-subshell of extranuclear electrons, or which gives rise to a cation or cations with an incomplete d-subshell. Transition metals often have more than one valency state. Biologically relevant transition metals include vanadium, manganese, iron, copper, cobalt, nickel, molybdenum and silver.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"transition metal transport\" EXACT []'],\n",
       "  'is_a': ['GO:0030001 ! metal ion transport']},\n",
       " 'GO:0000044': {'name': ['obsolete ascorbate stabilization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The reduction of the ascorbate free radical to a stable form.\" [GOC:ai, GOC:mtg_electron_transport]'],\n",
       "  'comment': ['This term was made obsolete because it is defined as a function term and is in the process ontology.'],\n",
       "  'synonym': ['\"ascorbate stabilization\" EXACT []',\n",
       "   '\"vitamin C stabilization\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000045': {'name': ['autophagosome assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of a double membrane-bounded structure, the autophagosome, that occurs when a specialized membrane sac, called the isolation membrane, starts to enclose a portion of the cytoplasm.\" [GOC:autophagy, PMID:9412464]'],\n",
       "  'synonym': ['\"autophagic vacuole assembly\" EXACT [GOC:autophagy]',\n",
       "   '\"autophagic vacuole formation\" RELATED [GOC:mah]',\n",
       "   '\"autophagosome biosynthesis\" EXACT []',\n",
       "   '\"autophagosome formation\" EXACT []',\n",
       "   '\"PAS formation\" NARROW []'],\n",
       "  'is_a': ['GO:0070925 ! organelle assembly',\n",
       "   'GO:1905037 ! autophagosome organization']},\n",
       " 'GO:0000047': {'name': ['obsolete Rieske iron-sulfur protein'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:ai]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"Rieske iron-sulfur protein\" EXACT []',\n",
       "   '\"Rieske iron-sulphur protein\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0009055']},\n",
       " 'GO:0000048': {'name': ['peptidyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"peptidyl-tRNA:aminoacyl-tRNA N-peptidyltransferase activity\" RELATED []'],\n",
       "  'xref': ['EC:2.3.2.12',\n",
       "   'MetaCyc:PEPTIDYLTRANSFERASE-RXN',\n",
       "   'Reactome:R-HSA-156912 \"Peptide transfer from P-site tRNA to the A-site tRNA\"'],\n",
       "  'is_a': ['GO:0016755 ! aminoacyltransferase activity']},\n",
       " 'GO:0000049': {'name': ['tRNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000946'],\n",
       "  'def': ['\"Binding to a transfer RNA.\" [GOC:ai]'],\n",
       "  'synonym': ['\"base pairing with tRNA\" NARROW []'],\n",
       "  'is_a': ['GO:0003723 ! RNA binding']},\n",
       " 'GO:0000050': {'name': ['urea cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0006594', 'GO:0006871'],\n",
       "  'def': ['\"The sequence of reactions by which arginine is synthesized from ornithine, then cleaved to yield urea and regenerate ornithine. The overall reaction equation is NH3 + CO2 + aspartate + 3 ATP + 2 H2O = urea + fumarate + 2 ADP + 2 phosphate + AMP + diphosphate.\" [GOC:pde, GOC:vw, ISBN:0198506732]'],\n",
       "  'synonym': ['\"ornithine cycle\" EXACT []',\n",
       "   '\"urea biosynthesis\" EXACT []',\n",
       "   '\"urea biosynthetic process\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Urea_cycle'],\n",
       "  'is_a': ['GO:0019627 ! urea metabolic process',\n",
       "   'GO:0043604 ! amide biosynthetic process',\n",
       "   'GO:1901566 ! organonitrogen compound biosynthetic process']},\n",
       " 'GO:0000051': {'name': ['obsolete urea cycle intermediate metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The chemical reactions and pathways involving any of the intermediate compounds involved in the urea cycle, a cyclic metabolic pathway that converts waste nitrogen in the form of ammonium to urea.\" [GOC:jl, ISBN:0198506732]'],\n",
       "  'comment': ['This term was made obsolete because it is a grouping term that is not useful, but has caused true path violations.'],\n",
       "  'synonym': ['\"urea cycle intermediate metabolic process\" EXACT []',\n",
       "   '\"urea cycle intermediate metabolism\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000052': {'name': ['citrulline metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving citrulline, N5-carbamoyl-L-ornithine, an alpha amino acid not found in proteins.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"citrulline metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:1901605 ! alpha-amino acid metabolic process']},\n",
       " 'GO:0000053': {'name': ['argininosuccinate metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving argininosuccinate, 2-(N(omega)-arginino)succinate, an intermediate in the ornithine-urea cycle, where it is synthesized from citrulline and aspartate.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"argininosuccinate metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006575 ! cellular modified amino acid metabolic process',\n",
       "   'GO:0072350 ! tricarboxylic acid metabolic process',\n",
       "   'GO:1901605 ! alpha-amino acid metabolic process']},\n",
       " 'GO:0000054': {'name': ['ribosomal subunit export from nucleus'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of a ribosomal subunit from the nucleus into the cytoplasm.\" [GOC:ai]'],\n",
       "  'subset': ['goslim_yeast'],\n",
       "  'synonym': ['\"ribosomal subunit export from cell nucleus\" EXACT [GOC:mah]',\n",
       "   '\"ribosomal subunit export out of nucleus\" EXACT [GOC:mah]',\n",
       "   '\"ribosomal subunit transport from nucleus to cytoplasm\" EXACT [GOC:mah]',\n",
       "   '\"ribosomal subunit-nucleus export\" EXACT [GOC:mah]',\n",
       "   '\"ribosome export from nucleus\" RELATED [GOC:mah, GOC:rb]'],\n",
       "  'is_a': ['GO:0031503 ! protein-containing complex localization',\n",
       "   'GO:0033750 ! ribosome localization',\n",
       "   'GO:0051168 ! nuclear export',\n",
       "   'GO:0051656 ! establishment of organelle localization'],\n",
       "  'relationship': ['part_of GO:0042254 ! ribosome biogenesis']},\n",
       " 'GO:0000055': {'name': ['ribosomal large subunit export from nucleus'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000057'],\n",
       "  'def': ['\"The directed movement of a ribosomal large subunit from the nucleus into the cytoplasm.\" [GOC:mah]'],\n",
       "  'synonym': ['\"50S ribosomal subunit export from nucleus\" NARROW [GOC:mah]',\n",
       "   '\"60S ribosomal subunit export from nucleus\" NARROW [GOC:mah]',\n",
       "   '\"ribosomal large subunit export from cell nucleus\" EXACT []',\n",
       "   '\"ribosomal large subunit export out of nucleus\" EXACT []',\n",
       "   '\"ribosomal large subunit transport from nucleus to cytoplasm\" EXACT []',\n",
       "   '\"ribosomal large subunit-nucleus export\" EXACT []'],\n",
       "  'is_a': ['GO:0000054 ! ribosomal subunit export from nucleus']},\n",
       " 'GO:0000056': {'name': ['ribosomal small subunit export from nucleus'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000058'],\n",
       "  'def': ['\"The directed movement of a ribosomal small subunit from the nucleus into the cytoplasm.\" [GOC:mah]'],\n",
       "  'synonym': ['\"30S ribosomal subunit export from nucleus\" NARROW [GOC:mah]',\n",
       "   '\"40S ribosomal subunit export from nucleus\" NARROW [GOC:mah]',\n",
       "   '\"ribosomal small subunit export from cell nucleus\" EXACT []',\n",
       "   '\"ribosomal small subunit export out of nucleus\" EXACT []',\n",
       "   '\"ribosomal small subunit transport from nucleus to cytoplasm\" EXACT []',\n",
       "   '\"ribosomal small subunit-nucleus export\" EXACT []'],\n",
       "  'is_a': ['GO:0000054 ! ribosomal subunit export from nucleus']},\n",
       " 'GO:0000059': {'name': ['obsolete protein import into nucleus, docking'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A protein complex assembly process that contributes to protein import into the nucleus, and that results in the association of a cargo protein, a carrier protein such as an importin alpha/beta heterodimer, and a nucleoporin located at the periphery of the nuclear pore complex.\" [GOC:isa_complete, GOC:mah, PMID:14570049, PMID:7878057, PMID:9126736]'],\n",
       "  'comment': ['This term was made obsolete because the transient assembly is better captured as a protein-protein association, if at all.'],\n",
       "  'synonym': ['\"protein docking during protein import into nucleus\" EXACT []',\n",
       "   '\"protein docking during protein transport from cytoplasm to nucleus\" EXACT []',\n",
       "   '\"protein docking during protein-nucleus import\" EXACT []',\n",
       "   '\"protein transport from cytoplasm to nucleus, docking\" EXACT []',\n",
       "   '\"protein-nucleus import, docking\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000060': {'name': ['obsolete protein import into nucleus, translocation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A protein transport process that contributes to protein import into the nucleus, and that results in the vectorial transfer of a cargo-carrier protein complex through the nuclear pore complex from the cytoplasmic side to the nucleoplasmic side of the nuclear envelope.\" [GOC:curators, ISBN:0198506732, PMID:14570049, PMID:9126736]'],\n",
       "  'comment': ['This term has been obsoleted because it represents a substep of the parent (GO:0006606 protein import into nucleus), has been incorrectly used.'],\n",
       "  'synonym': ['\"protein import into cell nucleus, translocation\" EXACT []',\n",
       "   '\"protein translocation during protein import into nucleus\" EXACT []',\n",
       "   '\"protein translocation during protein transport from cytoplasm to nucleus\" EXACT []',\n",
       "   '\"protein translocation during protein-nucleus import\" EXACT []',\n",
       "   '\"protein transport from cytoplasm to nucleus, translocation\" EXACT []',\n",
       "   '\"protein-nucleus import, translocation\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000061': {'name': ['obsolete protein import into nucleus, substrate release'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A protein complex disassembly process that contributes to protein import into the nucleus, and that results in the dissociation of the cargo protein and the carrier (such as an importin alpha/beta heterodimer) from each other and from the nuclear pore complex.\" [GOC:mah, PMID:14570049, PMID:9126736, PMID:9687515]'],\n",
       "  'comment': ['This term has been obsoleted because it represents a substep of the parent (GO:0006606 protein import into nucleus), has been incorrectly used.'],\n",
       "  'synonym': ['\"protein import into cell nucleus, substrate release\" EXACT []',\n",
       "   '\"protein substrate release during protein import into nucleus\" EXACT []',\n",
       "   '\"protein substrate release during protein transport from cytoplasm to nucleus\" EXACT []',\n",
       "   '\"protein substrate release during protein-nucleus import\" EXACT []',\n",
       "   '\"protein transport from cytoplasm to nucleus, substrate release\" EXACT []',\n",
       "   '\"protein-nucleus import, substrate release\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000062': {'name': ['fatty-acyl-CoA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a fatty-acyl-CoA, any derivative of coenzyme A in which the sulfhydryl group is in thiolester linkage with a fatty acyl group.\" [GOC:jl, GOC:krc, ISBN:0198506732]'],\n",
       "  'synonym': ['\"fatty-acyl binding\" BROAD []',\n",
       "   '\"fatty-acyl-coenzyme A binding\" EXACT []'],\n",
       "  'is_a': ['GO:0120227 ! acyl-CoA binding',\n",
       "   'GO:1901567 ! fatty acid derivative binding']},\n",
       " 'GO:0000064': {'name': ['L-ornithine transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of L-ornithine from one side of a membrane to the other. L-ornithine is 2,5-diaminopentanoic acid.\" [GOC:ai, GOC:mtg_transport, ISBN:0815340729]'],\n",
       "  'synonym': ['\"histidine/arginine/lysine/ornithine porter activity\" RELATED []',\n",
       "   '\"L-ornithine transporter activity\" BROAD []'],\n",
       "  'xref': ['Reactome:R-HSA-70634 \"ornithine (cytosolic) + citrulline (mitochondrial) => ornithine (mitochondrial) + citrulline (cytosolic)\"'],\n",
       "  'is_a': ['GO:0015179 ! L-amino acid transmembrane transporter activity']},\n",
       " 'GO:0000067': {'name': ['obsolete DNA replication and chromosome cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:ai]'],\n",
       "  'comment': ['This term was made obsolete because it has been superseded by more accurate terms to represent the biological processes occurring, and it is not clear that this term represents a useful entity.'],\n",
       "  'synonym': ['\"DNA replication and chromosome cycle\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006260', 'GO:0007059', 'GO:0051276']},\n",
       " 'GO:0000070': {'name': ['mitotic sister chromatid segregation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0016359'],\n",
       "  'def': ['\"The cell cycle process in which replicated homologous chromosomes are organized and then physically separated and apportioned to two sets during the mitotic cell cycle. Each replicated chromosome, composed of two sister chromatids, aligns at the cell equator, paired with its homologous partner. One homolog of each morphologic type goes into each of the resulting chromosome sets.\" [GOC:ai, GOC:jl]'],\n",
       "  'subset': ['goslim_pombe'],\n",
       "  'synonym': ['\"mitotic chromosome segregation\" EXACT []',\n",
       "   '\"mitotic sister-chromatid adhesion release\" NARROW []'],\n",
       "  'is_a': ['GO:0000819 ! sister chromatid segregation',\n",
       "   'GO:1903047 ! mitotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0140014 ! mitotic nuclear division']},\n",
       " 'GO:0000072': {'name': ['obsolete M phase specific microtubule process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A microtubule-based process that occurs only during M phase of the cell cycle.\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because terms already exist for centrosome/spindle organization which would be more suitable for the existing annotations, and the phase could be captured as an annotation extension if necessary.'],\n",
       "  'synonym': ['\"M phase specific microtubule process\" EXACT []',\n",
       "   '\"M-phase specific microtubule process\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0007017']},\n",
       " 'GO:0000073': {'name': ['initial mitotic spindle pole body separation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0030475'],\n",
       "  'def': ['\"The release of duplicated mitotic spindle pole bodies (SPBs) that begins with the nucleation of microtubules from each SPB within the nucleus, leading to V-shaped spindle microtubules. Interpolar microtubules that elongate from each pole are interconnected, forming overlapping microtubules. Capturing and antiparallel sliding apart of microtubules promotes the initial separation of the SPB.\" [GOC:sgd_curators, GOC:vw]'],\n",
       "  'is_a': ['GO:0110100 ! spindle pole body separation',\n",
       "   'GO:1903047 ! mitotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0061804 ! mitotic spindle formation (spindle phase one)',\n",
       "   'part_of GO:1905047 ! mitotic spindle pole body organization']},\n",
       " 'GO:0000075': {'name': ['cell cycle checkpoint signaling'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0031576',\n",
       "   'GO:0071779',\n",
       "   'GO:0072395',\n",
       "   'GO:0072404',\n",
       "   'GO:0072407'],\n",
       "  'def': ['\"A signaling process that controls cell cycle progression by monitoring the integrity of specific cell cycle events. A cell cycle checkpoint begins with detection of deficiencies or defects and ends with signal transduction.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': ['This term should not be used in direct manual annotation as it should always be possible to minimally designate mitotic or meiotic checkpoint, and usually to additionally specify the checkpoint (i.e mitotic spindle assembly checkpoint, mitotic DNA damage checkpoint etc). Note also that the effector processes are not part of the checkpoint but are positively regulated by the checkpoint signaling and should not be annotated here.'],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'synonym': ['\"cell cycle checkpoint\" EXACT []',\n",
       "   '\"G1/S checkpoint\" NARROW []',\n",
       "   '\"G1/S transition checkpoint\" NARROW []',\n",
       "   '\"G2/M checkpoint\" NARROW []',\n",
       "   '\"G2/M transition checkpoint\" NARROW []',\n",
       "   '\"signal transduction involved in cell cycle checkpoint\" EXACT []',\n",
       "   '\"signal transduction involved in G2/M transition checkpoint\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Cell_cycle_checkpoint'],\n",
       "  'is_a': ['GO:0035556 ! intracellular signal transduction',\n",
       "   'GO:1901988 ! negative regulation of cell cycle phase transition']},\n",
       " 'GO:0000076': {'name': ['DNA replication checkpoint signaling'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0072437'],\n",
       "  'def': ['\"A signal transduction process that contributes to a DNA replication checkpoint, that prevents the initiation of nuclear division until DNA replication is complete, thereby ensuring that progeny inherit a full complement of the genome.\" [GOC:curators, GOC:rn, PMID:11728327, PMID:12537518]'],\n",
       "  'synonym': ['\"DNA replication checkpoint\" EXACT []',\n",
       "   '\"signal transduction involved in DNA replication checkpoint\" EXACT []'],\n",
       "  'is_a': ['GO:0031570 ! DNA integrity checkpoint signaling']},\n",
       " 'GO:0000077': {'name': ['DNA damage checkpoint signaling'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0072422'],\n",
       "  'def': ['\"A signal transduction process that contributes to a DNA damage checkpoint.\" [GOC:mah]'],\n",
       "  'synonym': ['\"DNA damage checkpoint\" EXACT []',\n",
       "   '\"DNA damage response, signal transduction resulting in cell cycle arrest\" NARROW []',\n",
       "   '\"signal transduction involved in DNA damage checkpoint\" EXACT []'],\n",
       "  'xref': ['Wikipedia:DNA_damage_checkpoint',\n",
       "   'Wikipedia:Postreplication_checkpoint'],\n",
       "  'is_a': ['GO:0031570 ! DNA integrity checkpoint signaling',\n",
       "   'GO:0042770 ! signal transduction in response to DNA damage']},\n",
       " 'GO:0000078': {'name': ['obsolete cytokinesis after mitosis checkpoint'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A mitotic cell cycle checkpoint that detects whether chromosome segregation is complete and negatively regulates cytokinesis following mitosis.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': ['The reason this term was made obsolete is that the two terms cytokinesis checkpoint (GO:0031565) and cytokinesis after mitosis (GO:0000078) were conflated in meaning.'],\n",
       "  'synonym': ['\"cell morphogenesis checkpoint\" EXACT [GOC:dph, GOC:vw]',\n",
       "   '\"cell shape checkpoint\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044878', 'GO:0044879']},\n",
       " 'GO:0000079': {'name': ['regulation of cyclin-dependent protein serine/threonine kinase activity'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of cyclin-dependent protein serine/threonine kinase activity.\" [GOC:go_curators, GOC:pr]'],\n",
       "  'subset': ['term_scheduled_for_obsoletion'],\n",
       "  'synonym': ['\"regulation of CDK activity\" EXACT []',\n",
       "   '\"regulation of cyclin-dependent protein kinase activity\" BROAD []'],\n",
       "  'is_a': ['GO:0071900 ! regulation of protein serine/threonine kinase activity',\n",
       "   'GO:1904029 ! regulation of cyclin-dependent protein kinase activity']},\n",
       " 'GO:0000080': {'name': ['mitotic G1 phase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle \\'gap\\' phase which is the interval between the completion of DNA segregation by mitosis and the beginning of DNA synthesis.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'synonym': ['\"G1 phase of mitotic cell cycle\" EXACT []'],\n",
       "  'is_a': ['GO:0051318 ! G1 phase'],\n",
       "  'relationship': ['part_of GO:0051329 ! mitotic interphase']},\n",
       " 'GO:0000082': {'name': ['G1/S transition of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The mitotic cell cycle transition by which a cell in G1 commits to S phase. The process begins with the build up of G1 cyclin-dependent kinase (G1 CDK), resulting in the activation of transcription of G1 cyclins. The process ends with the positive feedback of the G1 cyclins on the G1 CDK which commits the cell to S phase, in which DNA replication is initiated.\" [GOC:mtg_cell_cycle]'],\n",
       "  'is_a': ['GO:0044772 ! mitotic cell cycle phase transition',\n",
       "   'GO:0044843 ! cell cycle G1/S phase transition']},\n",
       " 'GO:0000083': {'name': ['regulation of transcription involved in G1/S transition of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that regulates transcription such that the target genes are involved in the transition between G1 and S phase of the mitotic cell cycle.\" [GOC:mtg_cell_cycle]'],\n",
       "  'is_a': ['GO:0006355 ! regulation of DNA-templated transcription',\n",
       "   'GO:1903047 ! mitotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0000082 ! G1/S transition of mitotic cell cycle']},\n",
       " 'GO:0000084': {'name': ['mitotic S phase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase, following G1, during which DNA synthesis takes place as part of a mitotic cell cycle.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'synonym': ['\"S phase of mitotic cell cycle\" EXACT []',\n",
       "   '\"S-phase of mitotic cell cycle\" EXACT []'],\n",
       "  'is_a': ['GO:0051320 ! S phase'],\n",
       "  'relationship': ['part_of GO:0051329 ! mitotic interphase']},\n",
       " 'GO:0000085': {'name': ['mitotic G2 phase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle \\'gap\\' phase which is the interval between the completion of DNA synthesis and the beginning of DNA segregation by mitosis.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'synonym': ['\"G2 phase of mitotic cell cycle\" EXACT []'],\n",
       "  'is_a': ['GO:0051319 ! G2 phase'],\n",
       "  'relationship': ['part_of GO:0051329 ! mitotic interphase']},\n",
       " 'GO:0000086': {'name': ['G2/M transition of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The mitotic cell cycle transition by which a cell in G2 commits to M phase. The process begins when the kinase activity of M cyclin/CDK complex reaches a threshold high enough for the cell cycle to proceed. This is accomplished by activating a positive feedback loop that results in the accumulation of unphosphorylated and active M cyclin/CDK complex.\" [GOC:mtg_cell_cycle]'],\n",
       "  'synonym': ['\"mitotic G2/M transition\" EXACT []'],\n",
       "  'is_a': ['GO:0044772 ! mitotic cell cycle phase transition',\n",
       "   'GO:0044839 ! cell cycle G2/M phase transition']},\n",
       " 'GO:0000087': {'name': ['mitotic M phase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A cell cycle phase during which nuclear division occurs, and which is comprises the phases'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'synonym': ['\"M phase of mitotic cell cycle\" EXACT []',\n",
       "   '\"M-phase of mitotic cell cycle\" EXACT []'],\n",
       "  'is_a': ['GO:0000279 ! M phase', 'GO:0098763 ! mitotic cell cycle phase']},\n",
       " 'GO:0000088': {'name': ['mitotic prophase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which is the first stage of M phase of mitosis and during which chromosomes condense and the two daughter centrioles and their asters migrate toward the poles of the cell.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0051324 ! prophase'],\n",
       "  'relationship': ['part_of GO:0000087 ! mitotic M phase']},\n",
       " 'GO:0000089': {'name': ['mitotic metaphase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase, following prophase, during which chromosomes become aligned on the equatorial plate of the cell as part of a mitotic cell cycle.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0051323 ! metaphase'],\n",
       "  'relationship': ['part_of GO:0000087 ! mitotic M phase']},\n",
       " 'GO:0000090': {'name': ['mitotic anaphase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase during which chromosomes separate and migrate towards the poles of the spindle the as part of a mitotic cell cycle.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0051322 ! anaphase'],\n",
       "  'relationship': ['part_of GO:0000087 ! mitotic M phase']},\n",
       " 'GO:0000091': {'name': ['mitotic anaphase A'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase during which the kinetochore microtubules shorten as chromosomes move toward the spindle poles as part of mitosis.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0000090 ! mitotic anaphase']},\n",
       " 'GO:0000092': {'name': ['mitotic anaphase B'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase during which the polar microtubules elongate and the two poles of the spindle move farther apart as part of mitosis.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0000090 ! mitotic anaphase']},\n",
       " 'GO:0000093': {'name': ['mitotic telophase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which follows anaphase during M phase of mitosis and during which the chromosomes arrive at the poles of the cell and the division of the cytoplasm starts.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0051326 ! telophase'],\n",
       "  'relationship': ['part_of GO:0000087 ! mitotic M phase']},\n",
       " 'GO:0000094': {'name': ['obsolete septin assembly and septum formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:go_curators]'],\n",
       "  'comment': ['This term was made obsolete because it was not defined and the string name implied two separate processes.'],\n",
       "  'synonym': ['\"septin assembly and septum formation\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000917', 'GO:0000918', 'GO:0000921']},\n",
       " 'GO:0000095': {'name': ['S-adenosyl-L-methionine transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0015177'],\n",
       "  'def': ['\"Enables the transfer of S-adenosylmethionine from one side of a membrane to the other. S-adenosylmethionine is S-(5\\'-adenosyl)-L-methionine, an important intermediate in one-carbon metabolism.\" [GOC:ai]'],\n",
       "  'synonym': ['\"S-adenosyl methionine permease activity\" EXACT []',\n",
       "   '\"S-adenosyl methionine transporter activity\" EXACT []',\n",
       "   '\"S-adenosylmethionine permease activity\" EXACT []',\n",
       "   '\"S-adenosylmethionine transmembrane transporter activity\" EXACT []',\n",
       "   '\"S-adenosylmethionine transporter activity\" BROAD []',\n",
       "   '\"SAM transmembrane transporter activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-8855062 \"SLC25A26 exchanges cytosolic AdoMet for mitochondrial AdoHcy\"'],\n",
       "  'is_a': ['GO:0015101 ! organic cation transmembrane transporter activity',\n",
       "   'GO:1901682 ! sulfur compound transmembrane transporter activity']},\n",
       " 'GO:0000096': {'name': ['sulfur amino acid metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving amino acids containing sulfur, comprising cysteine, homocysteine, methionine and selenocysteine.\" [GOC:ai]'],\n",
       "  'synonym': ['\"sulfur amino acid metabolism\" EXACT []',\n",
       "   '\"sulphur amino acid metabolic process\" EXACT []',\n",
       "   '\"sulphur amino acid metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006790 ! sulfur compound metabolic process',\n",
       "   'GO:0019752 ! carboxylic acid metabolic process',\n",
       "   'GO:1901564 ! organonitrogen compound metabolic process']},\n",
       " 'GO:0000097': {'name': ['sulfur amino acid biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of amino acids containing sulfur, comprising cysteine, methionine and selenocysteine.\" [GOC:ai]'],\n",
       "  'synonym': ['\"sulfur amino acid anabolism\" EXACT []',\n",
       "   '\"sulfur amino acid biosynthesis\" EXACT []',\n",
       "   '\"sulfur amino acid formation\" EXACT []',\n",
       "   '\"sulfur amino acid synthesis\" EXACT []',\n",
       "   '\"sulphur amino acid biosynthesis\" EXACT []',\n",
       "   '\"sulphur amino acid biosynthetic process\" EXACT []'],\n",
       "  'is_a': ['GO:0000096 ! sulfur amino acid metabolic process',\n",
       "   'GO:0044272 ! sulfur compound biosynthetic process',\n",
       "   'GO:0046394 ! carboxylic acid biosynthetic process',\n",
       "   'GO:1901566 ! organonitrogen compound biosynthetic process']},\n",
       " 'GO:0000098': {'name': ['sulfur amino acid catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of amino acids containing sulfur, comprising cysteine, methionine and selenocysteine.\" [GOC:ai]'],\n",
       "  'synonym': ['\"sulfur amino acid breakdown\" EXACT []',\n",
       "   '\"sulfur amino acid catabolism\" EXACT []',\n",
       "   '\"sulfur amino acid degradation\" EXACT []',\n",
       "   '\"sulphur amino acid catabolic process\" EXACT []',\n",
       "   '\"sulphur amino acid catabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0000096 ! sulfur amino acid metabolic process',\n",
       "   'GO:0044273 ! sulfur compound catabolic process',\n",
       "   'GO:0046395 ! carboxylic acid catabolic process',\n",
       "   'GO:1901565 ! organonitrogen compound catabolic process']},\n",
       " 'GO:0000099': {'name': ['sulfur amino acid transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of sulfur amino acids from one side of a membrane to the other. Sulphur amino acids contain sulfur in the form of cystine, methionine or their derivatives.\" [GOC:ai, GOC:mtg_transport, ISBN:0815340729]'],\n",
       "  'synonym': ['\"sulfur amino acid transporter activity\" BROAD []',\n",
       "   '\"sulphur amino acid transmembrane transporter activity\" EXACT []',\n",
       "   '\"sulphur amino acid transporter activity\" BROAD []'],\n",
       "  'is_a': ['GO:0015171 ! amino acid transmembrane transporter activity',\n",
       "   'GO:1901682 ! sulfur compound transmembrane transporter activity']},\n",
       " 'GO:0000100': {'name': ['S-methylmethionine transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0015178'],\n",
       "  'def': ['\"Enables the transfer of S-methylmethionine from one side of a membrane to the other.\" [GOC:ai]'],\n",
       "  'synonym': ['\"S-methylmethionine permease activity\" EXACT []',\n",
       "   '\"S-methylmethionine transporter activity\" BROAD []'],\n",
       "  'is_a': ['GO:0072349 ! modified amino acid transmembrane transporter activity',\n",
       "   'GO:1901682 ! sulfur compound transmembrane transporter activity']},\n",
       " 'GO:0000101': {'name': ['sulfur amino acid transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of amino acids containing sulfur (cystine, methionine and their derivatives) into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore.\" [GOC:ai]'],\n",
       "  'synonym': ['\"sulphur amino acid transport\" EXACT []'],\n",
       "  'is_a': ['GO:0006865 ! amino acid transport',\n",
       "   'GO:0072348 ! sulfur compound transport']},\n",
       " 'GO:0000102': {'name': ['L-methionine secondary active transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of L-methionine from one side of a membrane to the other, up its concentration gradient. The transporter binds the solute and undergoes a series of conformational changes. Transport works equally well in either direction and is driven by a chemiosmotic source of energy. Secondary active transporters include symporters and antiporters.\" [GOC:mtg_transport]'],\n",
       "  'synonym': ['\"L-methionine porter activity\" RELATED []'],\n",
       "  'is_a': ['GO:0005294 ! neutral L-amino acid secondary active transmembrane transporter activity',\n",
       "   'GO:0015191 ! L-methionine transmembrane transporter activity',\n",
       "   'GO:1901680 ! sulfur-containing amino acid secondary active transmembrane transporter activity']},\n",
       " 'GO:0000103': {'name': ['sulfate assimilation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0019378'],\n",
       "  'def': ['\"The pathways by which inorganic sulfate is processed and incorporated into sulfated compounds.\" [GOC:jl]'],\n",
       "  'synonym': ['\"sulfate assimilation, phosphoadenylyl sulfate reduction by an oxidoreductase, acting on sulfur group of donors, NAD or NADP as acceptor\" NARROW []',\n",
       "   '\"sulphate assimilation\" EXACT []',\n",
       "   '\"sulphate assimilation, phosphoadenylyl sulphate reduction by an oxidoreductase, acting on sulphur group of donors, NAD or NADP as acceptor\" NARROW []'],\n",
       "  'is_a': ['GO:0006790 ! sulfur compound metabolic process']},\n",
       " 'GO:0000104': {'name': ['succinate dehydrogenase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0019739'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"fumarate dehydrogenase activity\" EXACT []',\n",
       "   '\"fumarate reductase activity\" EXACT []',\n",
       "   '\"fumaric hydrogenase activity\" RELATED [EC:1.3.99.1]',\n",
       "   '\"succinate oxidoreductase activity\" EXACT []',\n",
       "   '\"succinate:(acceptor) oxidoreductase activity\" EXACT []',\n",
       "   '\"succinate:acceptor oxidoreductase activity\" EXACT []',\n",
       "   '\"succinic acid dehydrogenase activity\" EXACT []',\n",
       "   '\"succinic dehydrogenase activity\" BROAD []',\n",
       "   '\"succinodehydrogenase activity\" EXACT []',\n",
       "   '\"succinyl dehydrogenase activity\" EXACT []'],\n",
       "  'xref': ['MetaCyc:SUCC-FUM-OXRED-RXN', 'RHEA:16357'],\n",
       "  'is_a': ['GO:0016627 ! oxidoreductase activity, acting on the CH-CH group of donors']},\n",
       " 'GO:0000105': {'name': ['histidine biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of histidine, 2-amino-3-(1H-imidazol-4-yl)propanoic acid.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"histidine anabolism\" EXACT []',\n",
       "   '\"histidine biosynthesis\" EXACT []',\n",
       "   '\"histidine formation\" EXACT []',\n",
       "   '\"histidine synthesis\" EXACT []'],\n",
       "  'xref': ['MetaCyc:HISTSYN-PWY'],\n",
       "  'is_a': ['GO:0006547 ! histidine metabolic process',\n",
       "   'GO:0008652 ! amino acid biosynthetic process']},\n",
       " 'GO:0000107': {'name': ['imidazoleglycerol-phosphate synthase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"glutamine amidotransferase:cyclase activity\" BROAD []',\n",
       "   '\"imidazole glycerol phosphate synthase activity\" EXACT []',\n",
       "   '\"imidazole-glycerol-phosphate synthase activity\" RELATED []',\n",
       "   '\"imidazoleglycerol phosphate synthase activity\" EXACT []'],\n",
       "  'xref': ['MetaCyc:GLUTAMIDOTRANS-RXN', 'RHEA:24793'],\n",
       "  'is_a': ['GO:0016763 ! pentosyltransferase activity']},\n",
       " 'GO:0000108': {'name': ['obsolete repairosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A stable complex of proteins that carry out the DNA damage recognition and incision reactions characteristic of nucleotide excision repair (NER), such as DNA damage recognition, DNA helix unwinding, and endonucleolytic cleavage at sites flanking damaged DNA; includes TFIIH subunits and additional polypeptides; may form in the absence of DNA damage.\" [PMID:10681587, PMID:9852079]'],\n",
       "  'comment': [\"This term was made obsolete because 'repairosome' has fallen out of use in the literature, and the large complex described in the definition has not been confirmed to exist. The term has also confused annotators.\"],\n",
       "  'synonym': ['\"repairosome\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000109']},\n",
       " 'GO:0000109': {'name': ['nucleotide-excision repair complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Any complex formed of proteins that act in nucleotide-excision repair.\" [PMID:10915862]'],\n",
       "  'comment': ['Note that process information is included in the term and definition for the purpose of describing and distinguishing the complex.'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"UvrB-UvrC complex\" NARROW [PMID:12145219]',\n",
       "   '\"UvrBC complex\" NARROW [GOC:bhm, PMID:12145219]'],\n",
       "  'is_a': ['GO:0140513 ! nuclear protein-containing complex']},\n",
       " 'GO:0000110': {'name': ['nucleotide-excision repair factor 1 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"One of several protein complexes involved in nucleotide-excision repair; possesses DNA damage recognition and endodeoxynuclease activities. In S. cerevisiae, it is composed of Rad1p, Rad10p, and Rad14p; in human the subunits are ERCC4/XPF, ERCC1 and XPA, respectively.\" [PMID:10915862]'],\n",
       "  'comment': ['Note that process and function information are included in the term and definition for the purpose of describing and distinguishing the complex.'],\n",
       "  'synonym': ['\"NEF1 complex\" EXACT []',\n",
       "   '\"XPA-ERCC1-ERCC4 complex\" EXACT [PMID:8197175]'],\n",
       "  'is_a': ['GO:0000109 ! nucleotide-excision repair complex']},\n",
       " 'GO:0000111': {'name': ['nucleotide-excision repair factor 2 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"One of several protein complexes involved in nucleotide-excision repair; possesses damaged DNA binding activity. In S. cerevisiae, it is composed of Rad4p and Rad23p.\" [PMID:10915862]'],\n",
       "  'comment': ['Note that process and function information are included in the term and definition for the purpose of describing and distinguishing the complex.'],\n",
       "  'synonym': ['\"NEF2 complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000109 ! nucleotide-excision repair complex']},\n",
       " 'GO:0000112': {'name': ['nucleotide-excision repair factor 3 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"One of several protein complexes involved in nucleotide-excision repair; possesses endodeoxynuclease and DNA helicase activities. In S. cerevisiae, it is composed of Rad2p and the core TFIIH-Ssl2p complex (core TFIIH is composed of Rad3p, Tfb1p, Tfb2p, Ssl1p, Tfb4p and Tfb5p. Note that Ssl2p is also called Rad25p).\" [GOC:ew, PMID:10915862, PMID:14500720, PMID:7813015]'],\n",
       "  'comment': ['Note that process and function information are included in the term and definition for the purpose of describing and distinguishing the complex.'],\n",
       "  'synonym': ['\"NEF3 complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000109 ! nucleotide-excision repair complex']},\n",
       " 'GO:0000113': {'name': ['nucleotide-excision repair factor 4 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"One of several protein complexes involved in nucleotide-excision repair; possesses DNA damage recognition and DNA-dependent ATPase activities. In S. cerevisiae, it is composed of Rad7p and Rad16p.\" [PMID:10915862]'],\n",
       "  'comment': ['Note that process and function information are included in the term and definition for the purpose of describing and distinguishing the complex.'],\n",
       "  'synonym': ['\"NEF4 complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000109 ! nucleotide-excision repair complex']},\n",
       " 'GO:0000114': {'name': ['obsolete regulation of transcription involved in G1 phase of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that regulates transcription such that the target genes are transcribed as part of the G1 phase of the mitotic cell cycle.\" [GOC:dph, GOC:mah, GOC:tb]'],\n",
       "  'comment': [\"This term was made obsolete because it is unclear exactly what it means. It could mean either 'regulation of transcription during phase X' or 'regulation of transition between phase X and phase Y'.\"],\n",
       "  'synonym': ['\"G1-specific transcription in mitotic cell cycle\" RELATED []',\n",
       "   '\"regulation of transcription from RNA polymerase II promoter during G1 phase of cell cycle\" EXACT []',\n",
       "   '\"regulation of transcription involved in G1 phase of mitotic cell cycle\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000083', 'GO:0006357']},\n",
       " 'GO:0000115': {'name': ['obsolete regulation of transcription involved in S phase of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A cell cycle process that regulates transcription such that the target genes are transcribed as part of the S phase of the mitotic cell cycle.\" [GOC:dph, GOC:mah, GOC:tb]'],\n",
       "  'comment': [\"This term was made obsolete because it is unclear exactly what it means. It could mean either 'regulation of transcription during phase X' or 'regulation of transition between phase X and phase Y'.\"],\n",
       "  'synonym': ['\"regulation of transcription from RNA polymerase II promoter during S-phase of mitotic cell cycle\" RELATED []',\n",
       "   '\"regulation of transcription involved in S phase of mitotic cell cycle\" EXACT []',\n",
       "   '\"regulation of transcription involved in S-phase of mitotic cell cycle\" EXACT []',\n",
       "   '\"S-phase-specific transcription in mitotic cell cycle\" RELATED []',\n",
       "   '\"S-specific transcription in mitotic cell cycle\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000083', 'GO:0006357']},\n",
       " 'GO:0000116': {'name': ['obsolete regulation of transcription involved in G2-phase of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that regulates transcription such that the target genes are transcribed as part of the G2 phase of the mitotic cell cycle.\" [GOC:dph, GOC:mah, GOC:tb]'],\n",
       "  'comment': [\"This term was made obsolete because it is unclear exactly what it means. It could mean either 'regulation of transcription during phase X' or 'regulation of transition between phase X and phase Y'.\"],\n",
       "  'synonym': ['\"G2-specific transcription in mitotic cell cycle\" RELATED []',\n",
       "   '\"regulation of transcription from RNA polymerase II during G2-phase of mitotic cell cycle\" RELATED []',\n",
       "   '\"regulation of transcription involved in G2-phase of mitotic cell cycle\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000117', 'GO:0006357']},\n",
       " 'GO:0000117': {'name': ['regulation of transcription involved in G2/M transition of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that regulates transcription such that the target genes are transcribed as part of the G2/M transition of the mitotic cell cycle.\" [GOC:mtg_cell_cycle]'],\n",
       "  'synonym': ['\"G2/M-specific transcription in mitotic cell cycle\" RELATED []',\n",
       "   '\"regulation of transcription from RNA polymerase II promoter during G2/M transition of mitotic cell cycle\" RELATED []'],\n",
       "  'is_a': ['GO:0006355 ! regulation of DNA-templated transcription',\n",
       "   'GO:1903047 ! mitotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0000086 ! G2/M transition of mitotic cell cycle']},\n",
       " 'GO:0000118': {'name': ['histone deacetylase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that possesses histone deacetylase activity.\" [GOC:mah]'],\n",
       "  'comment': [\"Note that this term represents a location, not a function; the activity possessed by this complex is mentioned in the definition for the purpose of describing and distinguishing the complex. The function of this complex is represented by the molecular function term 'histone deacetylase activity ; GO:0004407'.\"],\n",
       "  'synonym': ['\"HDAC complex\" EXACT []'],\n",
       "  'is_a': ['GO:0140513 ! nuclear protein-containing complex',\n",
       "   'GO:1902494 ! catalytic complex'],\n",
       "  'relationship': ['part_of GO:0005654 ! nucleoplasm']},\n",
       " 'GO:0000120': {'name': ['RNA polymerase I transcription regulator complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A transcription factor complex that acts at a regulatory region of a gene transcribed by RNA polymerase I.\" [GOC:mah]'],\n",
       "  'synonym': ['\"RNA polymerase I transcription factor complex\" NARROW []'],\n",
       "  'is_a': ['GO:0005667 ! transcription regulator complex',\n",
       "   'GO:0140513 ! nuclear protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005730 ! nucleolus']},\n",
       " 'GO:0000121': {'name': ['glycerol-1-phosphatase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"alpha-glycerol phosphatase activity\" RELATED [EC:3.1.3.21]',\n",
       "   '\"alpha-glycerophosphatase activity\" RELATED [EC:3.1.3.21]',\n",
       "   '\"glycerol 3-phosphatase activity\" RELATED [EC:3.1.3.21]',\n",
       "   '\"glycerol 3-phosphate phosphohydrolase activity\" RELATED [EC:3.1.3.21]',\n",
       "   '\"glycerol-1-phosphate phosphohydrolase activity\" RELATED [EC:3.1.3.21]',\n",
       "   '\"glycerol-3-phosphate phosphatase activity\" RELATED [EC:3.1.3.21]'],\n",
       "  'xref': ['EC:3.1.3.21', 'MetaCyc:GLYCEROL-1-PHOSPHATASE-RXN', 'RHEA:11476'],\n",
       "  'is_a': ['GO:0016791 ! phosphatase activity']},\n",
       " 'GO:0000122': {'name': ['negative regulation of transcription by RNA polymerase II'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0010553', 'GO:0045816'],\n",
       "  'def': ['\"Any process that stops, prevents, or reduces the frequency, rate or extent of transcription mediated by RNA polymerase II.\" [GOC:go_curators, GOC:txnOH]'],\n",
       "  'synonym': ['\"down regulation of global transcription from RNA polymerase II promoter\" RELATED []',\n",
       "   '\"down regulation of transcription from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"down-regulation of global transcription from RNA polymerase II promoter\" RELATED []',\n",
       "   '\"down-regulation of transcription from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"downregulation of global transcription from RNA polymerase II promoter\" RELATED []',\n",
       "   '\"downregulation of transcription from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"inhibition of global transcription from RNA polymerase II promoter\" RELATED []',\n",
       "   '\"inhibition of transcription from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"negative regulation of gene-specific transcription from RNA polymerase II promoter\" RELATED []',\n",
       "   '\"negative regulation of global transcription from Pol II promoter\" RELATED []',\n",
       "   '\"negative regulation of transcription from Pol II promoter\" EXACT []',\n",
       "   '\"negative regulation of transcription from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"negative regulation of transcription from RNA polymerase II promoter, global\" RELATED []'],\n",
       "  'is_a': ['GO:0006357 ! regulation of transcription by RNA polymerase II',\n",
       "   'GO:0045892 ! negative regulation of DNA-templated transcription'],\n",
       "  'relationship': ['negatively_regulates GO:0006366 ! transcription by RNA polymerase II']},\n",
       " 'GO:0000123': {'name': ['histone acetyltransferase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that possesses histone acetyltransferase activity.\" [GOC:mah]'],\n",
       "  'comment': [\"Note that this term represents a protein complex, not a function; the activity possessed by this complex is mentioned in the definition for the purpose of describing and distinguishing the complex. The function of this complex is represented by the molecular function term 'histone acetyltransferase activity ; GO:0004402'.\"],\n",
       "  'synonym': ['\"histone acetylase complex\" EXACT []'],\n",
       "  'is_a': ['GO:0031248 ! protein acetyltransferase complex'],\n",
       "  'relationship': ['part_of GO:0000785 ! chromatin']},\n",
       " 'GO:0000124': {'name': ['SAGA complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000125', 'GO:0030914'],\n",
       "  'def': ['\"A SAGA-type histone acetyltransferase complex that deubiquitinates H2A and/or H2B. This complex is organized into several functional submodules'],\n",
       "  'synonym': ['\"PCAF complex\" NARROW []',\n",
       "   '\"PCAF histone acetylase-associated complex\" NARROW []',\n",
       "   '\"Spt-Ada-Gcn5-acetyltransferase complex\" EXACT []',\n",
       "   '\"SPT3-TAF9-GCN5 acetylase complex\" RELATED [GOC:rl, PMID:18838386]',\n",
       "   '\"SPT3-TAF9-PCAF acetylase complex\" NARROW [PMID:18838386]',\n",
       "   '\"STAGA coactivator complex\" RELATED []',\n",
       "   '\"STAGA complex\" RELATED [PMID:32616828, PMID:33004486, PMID:34112237]'],\n",
       "  'is_a': ['GO:0070461 ! SAGA-type complex',\n",
       "   'GO:1905368 ! peptidase complex']},\n",
       " 'GO:0000126': {'name': ['transcription factor TFIIIB complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A transcription factor complex that is involved in regulating transcription from RNA polymerase III (Pol III) promoters. TFIIIB contains the TATA-binding protein (TBP) and two Pol III-specific proteins, B\\'\\' and BRF.\" [GOC:mah, PMID:11433012]'],\n",
       "  'is_a': ['GO:0090576 ! RNA polymerase III transcription regulator complex']},\n",
       " 'GO:0000127': {'name': ['transcription factor TFIIIC complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A heterotrimeric transcription factor complex that is involved in regulating transcription from RNA polymerase III (Pol III) promoters. TFIIIC contains three conserved subunits that associate with the proximal Pol III promoter element, and additional subunits that associate with sequence elements downstream of the promoter and are more diverged among species. It also functions as a boundary element to partition genome content into distinct domains outside Pol III promoter regions.\" [GOC:mah, GOC:vw, PMID:11433012, PMID:16751097]'],\n",
       "  'is_a': ['GO:0090576 ! RNA polymerase III transcription regulator complex']},\n",
       " 'GO:0000128': {'name': ['flocculation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000501',\n",
       "   'GO:0032128',\n",
       "   'GO:0036281',\n",
       "   'GO:0036282',\n",
       "   'GO:0043689',\n",
       "   'GO:0043690'],\n",
       "  'def': ['\"The reversible, non-sexual aggregation of single-celled organisms in suspension to form aggregates of many cells known as flocs.\" [GOC:jl, GOC:vw, PMID:11472912, PMID:21114594, PMID:8740415]'],\n",
       "  'comment': ['The word floc derives from the Latin word floccus, which means a tuft of wool.'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"cell-cell adhesion involved in flocculation\" EXACT []',\n",
       "   '\"cell-cell adhesion involved in flocculation via cell wall protein-carbohydrate interaction\" NARROW []',\n",
       "   '\"co-flocculation\" NARROW [PMID:10689163]',\n",
       "   '\"coflocculation\" NARROW []',\n",
       "   '\"coflocculation via lectin-mannose interaction\" NARROW [PMID:11693916]',\n",
       "   '\"coflocculation via protein-carbohydrate interaction\" NARROW []',\n",
       "   '\"flocculation via cell wall protein-carbohydrate interaction\" RELATED []',\n",
       "   '\"flocculation via extracellular polymer\" RELATED []'],\n",
       "  'xref': ['Wikipedia:Flocculation'],\n",
       "  'is_a': ['GO:0051703 ! biological process involved in intraspecies interaction between organisms',\n",
       "   'GO:0098610 ! adhesion between unicellular organisms']},\n",
       " 'GO:0000131': {'name': ['incipient cellular bud site'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The portion of the budding yeast plasma membrane where a daughter cell will emerge. The yeast marks this spot with bud-site selection proteins before bud emergence occurs. Actin is polarized to this spot just prior to and during bud emergence.\" [GOC:clt]'],\n",
       "  'is_a': ['GO:0140535 ! intracellular protein-containing complex']},\n",
       " 'GO:0000132': {'name': ['establishment of mitotic spindle orientation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0030607', 'GO:0030609'],\n",
       "  'def': ['\"A cell cycle process that sets the alignment of mitotic spindle relative to other cellular structures.\" [GOC:ems]'],\n",
       "  'synonym': ['\"establishment of spindle orientation during mitosis\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"establishment of spindle orientation involved in mitotic cell cycle\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"mitotic spindle orientation\" EXACT []',\n",
       "   '\"orienting of mitotic spindle\" EXACT []'],\n",
       "  'is_a': ['GO:0040001 ! establishment of mitotic spindle localization',\n",
       "   'GO:0051294 ! establishment of spindle orientation']},\n",
       " 'GO:0000133': {'name': ['polarisome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Protein complex that plays a role in determining cell polarity by directing the localized assembly of actin filaments at polarization sites; in Saccharomyces the polarisome includes Bni1p, Spa2p, Pea2p, and Bud6p.\" [PMID:14734532, PMID:14998522, PMID:9632790]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005938 ! cell cortex',\n",
       "   'part_of GO:0030427 ! site of polarized growth']},\n",
       " 'GO:0000136': {'name': ['mannan polymerase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex with alpha-(1->6)-mannosyltransferase activity, located in the cis Golgi membrane; adds mannan to N-linked glycans on proteins.\" [GOC:mcc, PMID:10037752, PMID:11095735, PMID:18083825]'],\n",
       "  'synonym': ['\"alpha-1,6-mannosyltransferase complex\" EXACT []'],\n",
       "  'is_a': ['GO:0031501 ! mannosyltransferase complex',\n",
       "   'GO:0098796 ! membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0000137 ! Golgi cis cisterna',\n",
       "   'part_of GO:0000139 ! Golgi membrane']},\n",
       " 'GO:0000137': {'name': ['Golgi cis cisterna'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The Golgi cisterna closest to the endoplasmic reticulum; the first processing compartment through which proteins pass after export from the ER.\" [ISBN:0815316194]'],\n",
       "  'synonym': ['\"Golgi apparatus cis cisterna\" EXACT []'],\n",
       "  'is_a': ['GO:0031985 ! Golgi cisterna']},\n",
       " 'GO:0000138': {'name': ['Golgi trans cisterna'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The Golgi cisterna farthest from the endoplasmic reticulum; the final processing compartment through which proteins pass before exiting the Golgi apparatus; the compartment in which N-linked protein glycosylation is completed.\" [ISBN:0815316194]'],\n",
       "  'synonym': ['\"late Golgi\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0031985 ! Golgi cisterna']},\n",
       " 'GO:0000139': {'name': ['Golgi membrane'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The lipid bilayer surrounding any of the compartments of the Golgi apparatus.\" [GOC:mah]'],\n",
       "  'synonym': ['\"Golgi apparatus membrane\" EXACT []'],\n",
       "  'is_a': ['GO:0098588 ! bounding membrane of organelle'],\n",
       "  'relationship': ['part_of GO:0005794 ! Golgi apparatus']},\n",
       " 'GO:0000140': {'name': ['acylglycerone-phosphate reductase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"1-acyldihydroxyacetone-phosphate reductase activity\" EXACT []',\n",
       "   '\"1-palmitoylglycerol-3-phosphate:NADP+ oxidoreductase activity\" RELATED [EC:1.1.1.101]',\n",
       "   '\"acyldihydroxyacetone phosphate reductase activity\" RELATED [EC:1.1.1.101]',\n",
       "   '\"palmitoyl dihydroxyacetone phosphate reductase activity\" RELATED [EC:1.1.1.101]',\n",
       "   '\"palmitoyl-dihydroxyacetone-phosphate reductase activity\" RELATED [EC:1.1.1.101]',\n",
       "   '\"palmitoyldihydroxyacetone-phosphate reductase activity\" RELATED [EC:1.1.1.101]'],\n",
       "  'xref': ['EC:1.1.1.101',\n",
       "   'MetaCyc:ACYLGLYCERONE-PHOSPHATE-REDUCTASE-RXN',\n",
       "   'Reactome:R-HSA-75883 \"DHRS7B reduces GO3P to HXDG3P\"',\n",
       "   'RHEA:17341'],\n",
       "  'is_a': ['GO:0016616 ! oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor']},\n",
       " 'GO:0000142': {'name': ['cellular bud neck contractile ring'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A contractile ring, i.e. a cytoskeletal structure composed of actin filaments and myosin, that forms beneath the plasma membrane at the mother-bud neck in mitotic cells that divide by budding in preparation for completing cytokinesis. An example of this structure is found in Saccharomyces cerevisiae.\" [GOC:krc, PMID:16009555]'],\n",
       "  'synonym': ['\"neck ring\" EXACT []'],\n",
       "  'is_a': ['GO:0110085 ! mitotic actomyosin contractile ring'],\n",
       "  'relationship': ['part_of GO:0005935 ! cellular bud neck']},\n",
       " 'GO:0000144': {'name': ['cellular bud neck septin ring'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ring-shaped structure that forms at the site of cytokinesis in the bud neck of a budding cell; composed of members of the conserved family of filament forming proteins called septins as well as septin-associated proteins. In S. cerevisiae, this structure forms at the time of bud emergence and the septins show a high rate of exchange.\" [GOC:krc, PMID:16009555]'],\n",
       "  'is_a': ['GO:0000399 ! cellular bud neck septin structure',\n",
       "   'GO:0005940 ! septin ring',\n",
       "   'GO:0032161 ! cleavage apparatus septin structure']},\n",
       " 'GO:0000145': {'name': ['exocyst'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex peripherally associated with the plasma membrane that determines where vesicles dock and fuse. At least eight complex components are conserved between yeast and mammals.\" [GOC:cilia, PMID:15292201, PMID:27243008, PMID:9700152]'],\n",
       "  'synonym': ['\"exocyst complex\" EXACT []', '\"Sec6/8 complex\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Exocyst'],\n",
       "  'is_a': ['GO:0099023 ! vesicle tethering complex'],\n",
       "  'relationship': ['part_of GO:0005938 ! cell cortex']},\n",
       " 'GO:0000146': {'name': ['microfilament motor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0030898'],\n",
       "  'def': ['\"A motor activity that generates movement along a microfilament, driven by ATP hydrolysis.\" [PMID:29716949]'],\n",
       "  'synonym': ['\"actin filament motor activity\" EXACT [GOC:dph]',\n",
       "   '\"actin-activated ATPase activity\" EXACT []',\n",
       "   '\"actin-dependent ATPase activity\" EXACT []',\n",
       "   '\"actin-filament motor activity\" EXACT []',\n",
       "   '\"muscle motor activity\" NARROW []',\n",
       "   '\"myosin ATPase activity\" RELATED []'],\n",
       "  'xref': ['EC:5.6.1.8',\n",
       "   'Reactome:R-HSA-2316352 \"SLC2A4 (GLUT4) vesicle translocates and docks at the plasma membrane\"',\n",
       "   'Reactome:R-HSA-432237 \"Translocation of Aquaporin-2 from intracellular vesicles to the apical plasma membrane\"',\n",
       "   'Reactome:R-HSA-9023171 \"Insulin secretory granule translocates across the cortical actin network\"'],\n",
       "  'is_a': ['GO:0003774 ! cytoskeletal motor activity',\n",
       "   'GO:0140657 ! ATP-dependent activity']},\n",
       " 'GO:0000147': {'name': ['actin cortical patch assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Assembly of an actin cortical patch, a discrete actin-containing structure found at the plasma membrane of fungal cells.\" [GOC:mah]'],\n",
       "  'is_a': ['GO:0022607 ! cellular component assembly',\n",
       "   'GO:0030866 ! cortical actin cytoskeleton organization',\n",
       "   'GO:0044396 ! actin cortical patch organization']},\n",
       " 'GO:0000148': {'name': ['1,3-beta-D-glucan synthase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that catalyzes the transfer of a glucose group from UDP-glucose to a (1->3)-beta-D-glucan chain.\" [PMID:7983071]'],\n",
       "  'synonym': ['\"(1->3)-beta-glucan synthase complex\" EXACT [GOC:tb]',\n",
       "   '\"1,3-beta-glucan synthase complex\" EXACT [GOC:tb]'],\n",
       "  'is_a': ['GO:0098797 ! plasma membrane protein complex',\n",
       "   'GO:1990234 ! transferase complex']},\n",
       " 'GO:0000149': {'name': ['SNARE binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a SNARE (soluble N-ethylmaleimide-sensitive factor attached protein receptor) protein.\" [PMID:12642621]'],\n",
       "  'subset': ['goslim_chembl'],\n",
       "  'synonym': ['\"SNAP receptor binding\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-210426 \"Glutamate synaptic vesicle docking and priming\"',\n",
       "   'Reactome:R-HSA-210430 \"Release of L-Glutamate at the synapse\"',\n",
       "   'Reactome:R-HSA-265166 \"Exocytosis of Insulin\"',\n",
       "   'Reactome:R-HSA-372505 \"Acetylcholine synaptic vesicle docking and priming\"',\n",
       "   'Reactome:R-HSA-372529 \"Release of acetylcholine at the synapse\"',\n",
       "   'Reactome:R-HSA-374899 \"Release of noradrenaline  at the synapse\"',\n",
       "   'Reactome:R-HSA-374922 \"Noradrenalin synaptic vesicle docking and priming\"',\n",
       "   'Reactome:R-HSA-376357 \"Vamp7 associated Lysosome to Plasma membrane transport\"',\n",
       "   'Reactome:R-HSA-376364 \"Vamp8 associated secretory vesicle to plasma membrane transport\"',\n",
       "   'Reactome:R-HSA-376369 \"Vamp2 associated secretory vesicle to plasma membrane transport\"',\n",
       "   'Reactome:R-HSA-380574 \"Dopamine synaptic vesicle docking and priming\"',\n",
       "   'Reactome:R-HSA-380869 \"Release of docked dopamine loaded synaptic vesicle\"',\n",
       "   'Reactome:R-HSA-380901 \"Release of docked serotonin loaded synaptic vesicle\"',\n",
       "   'Reactome:R-HSA-380905 \"Serotonin loaded synaptic vesicle docking and priming\"',\n",
       "   'Reactome:R-HSA-888589 \"Release of GABA at the synapse\"',\n",
       "   'Reactome:R-HSA-9023173 \"Insulin secretory granule docks at the plasma membrane\"',\n",
       "   'Reactome:R-HSA-917744 \"GABA loaded synaptic vesicle Docking and Priming\"'],\n",
       "  'is_a': ['GO:0005515 ! protein binding']},\n",
       " 'GO:0000150': {'name': ['DNA strand exchange activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the identification and base-pairing of homologous sequences between single-stranded DNA and double-stranded DNA.\" [GOC:elh]'],\n",
       "  'comment': [\"Note that this term represents activities that do not break or form phosphodiester bonds, and is therefore not a parent of 'site-specific recombinase activity ; GO:0009009'.\"],\n",
       "  'subset': ['goslim_metagenomics'],\n",
       "  'synonym': ['\"RecA-family recombinase activity\" RELATED []',\n",
       "   '\"recombinase activity\" EXACT []',\n",
       "   '\"strand exchange activity\" RELATED []',\n",
       "   '\"strand transferase\" NARROW []'],\n",
       "  'xref': ['Reactome:R-HSA-912458 \"Formation of meiotic heteroduplex\"'],\n",
       "  'is_a': ['GO:0140097 ! catalytic activity, acting on DNA']},\n",
       " 'GO:0000151': {'name': ['ubiquitin ligase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that includes a ubiquitin-protein ligase and enables ubiquitin protein ligase activity. The complex also contains other proteins that may confer substrate specificity on the complex.\" [GOC:jh2, PMID:9529603]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'is_a': ['GO:0140535 ! intracellular protein-containing complex',\n",
       "   'GO:1990234 ! transferase complex']},\n",
       " 'GO:0000152': {'name': ['nuclear ubiquitin ligase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ubiquitin ligase complex found in the nucleus.\" [GOC:mah]'],\n",
       "  'is_a': ['GO:0000151 ! ubiquitin ligase complex',\n",
       "   'GO:0140513 ! nuclear protein-containing complex']},\n",
       " 'GO:0000153': {'name': ['cytoplasmic ubiquitin ligase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ubiquitin ligase complex found in the cytoplasm.\" [GOC:mah]'],\n",
       "  'is_a': ['GO:0000151 ! ubiquitin ligase complex'],\n",
       "  'relationship': ['part_of GO:0005737 ! cytoplasm']},\n",
       " 'GO:0000154': {'name': ['rRNA modification'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0016548'],\n",
       "  'def': ['\"The covalent alteration of one or more nucleotides within an rRNA molecule to produce an rRNA molecule with a sequence that differs from that coded genetically.\" [GOC:curators]'],\n",
       "  'comment': [\"The term 'RNA editing' (GO:0016547) was merged into 'RNA modification' (GO:0009451) on the basis of statements in the preface of Modification and Editing of RNA (ISBN:1555811337) that there is no clear distinction between modification and editing. Parallel changes were made for substrate (e.g. tRNA, rRNA, etc.) specific child terms of 'RNA editing'.\"],\n",
       "  'synonym': ['\"rRNA editing\" NARROW [GOC:hjd]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing', 'GO:0009451 ! RNA modification']},\n",
       " 'GO:0000155': {'name': ['phosphorelay sensor kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the phosphorylation of a histidine residue in response to detection of an extracellular signal such as a chemical ligand or change in environment, to initiate a change in cell state or activity. The two-component sensor is a histidine kinase that autophosphorylates a histidine residue in its active site. The phosphate is then transferred to an aspartate residue in a downstream response regulator, to trigger a response.\" [GOC:bf, GOC:mcc, PMID:10966457, PMID:20223701, PMID:9191038]'],\n",
       "  'synonym': ['\"two-component sensor activity\" NARROW []',\n",
       "   '\"two-component sensor molecule\" NARROW []',\n",
       "   '\"two-component system sensor activity\" NARROW []'],\n",
       "  'xref': ['EC:2.7.3.-'],\n",
       "  'is_a': ['GO:0004673 ! protein histidine kinase activity',\n",
       "   'GO:0140299 ! small molecule sensor activity']},\n",
       " 'GO:0000156': {'name': ['phosphorelay response regulator activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Responds to a phosphorelay sensor to initiate a change in cell state or activity. The activity of the response regulator is regulated by transfer of a phosphate from a histidine residue in the sensor, to an aspartate residue in the response regulator. Many but not all response regulators act as transcriptional regulators to elicit a response.\" [GOC:bf, PMID:10966457, PMID:11842140]'],\n",
       "  'synonym': ['\"two-component response regulator activity\" NARROW []'],\n",
       "  'is_a': ['GO:0060089 ! molecular transducer activity']},\n",
       " 'GO:0000159': {'name': ['protein phosphatase type 2A complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that has protein serine/threonine phosphatase activity that is polycation-stimulated (PCS), being directly stimulated by protamine, polylysine, or histone H1; it constitutes a subclass of several enzymes activated by different histones and polylysine, and consists of catalytic, scaffolding, and regulatory subunits. The catalytic and scaffolding subunits form the core enzyme, and the holoenzyme also includes the regulatory subunit.\" [GOC:mah, ISBN:0198547684, PMID:17245430]'],\n",
       "  'synonym': ['\"PP2A complex\" EXACT []',\n",
       "   '\"PP2A-pi\" NARROW []',\n",
       "   '\"PP2a-protector\" NARROW []',\n",
       "   '\"protein phosphatase 2 complex\" RELATED [GOC:dph, GOC:rl]'],\n",
       "  'is_a': ['GO:0008287 ! protein serine/threonine phosphatase complex']},\n",
       " 'GO:0000160': {'name': ['phosphorelay signal transduction system'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A conserved series of molecular signals found in prokaryotes and eukaryotes; involves autophosphorylation of a histidine kinase and the transfer of the phosphate group to an aspartate that then acts as a phospho-donor to response regulator proteins.\" [PMID:9191038]'],\n",
       "  'subset': ['goslim_metagenomics'],\n",
       "  'synonym': ['\"histidyl-aspartyl phosphorelay\" EXACT []'],\n",
       "  'is_a': ['GO:0035556 ! intracellular signal transduction']},\n",
       " 'GO:0000161': {'name': ['osmosensory signaling MAPK cascade'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000167',\n",
       "   'GO:0000168',\n",
       "   'GO:0000169',\n",
       "   'GO:0000173',\n",
       "   'GO:0007233'],\n",
       "  'def': ['\"The series of molecular signals in which a stress-activated protein kinase (SAPK) cascade relays a signal, containing at least a Hog1/Sty1 family MAPK, a Pbs2/Wis1 family MAPKK and a Ssk2/Win1 family MAP3K.\" [PMID:17604854, PMID:9561267]'],\n",
       "  'synonym': ['\"activation of MAPK activity involved in osmosensory signaling pathway\" NARROW []',\n",
       "   '\"activation of MAPKK activity during osmolarity sensing\" NARROW []',\n",
       "   '\"activation of MAPKK activity involved in osmosensory signaling pathway\" NARROW []',\n",
       "   '\"activation of MAPKKK activity during osmolarity sensing\" NARROW []',\n",
       "   '\"activation of MAPKKK activity involved in osmosensory signaling pathway\" NARROW []',\n",
       "   '\"activation of Pbs2 kinase\" NARROW []',\n",
       "   '\"High Osmolarity Glycerol (HOG) MAPK pathway\" NARROW [PMID:20880736]',\n",
       "   '\"Hog1 MAPK pathway\" NARROW [PMID:20880736]',\n",
       "   '\"Hog1/Sty1 stress-activated MAPK cascade\" NARROW []',\n",
       "   '\"inactivation of MAPK activity involved in osmosensory signaling pathway\" NARROW []',\n",
       "   '\"MAPK cascade involved in osmosensory signaling pathway\" EXACT []',\n",
       "   '\"MAPKKK cascade during osmolarity sensing\" EXACT []',\n",
       "   '\"MAPKKK cascade involved in osmosensory signaling pathway\" EXACT [GOC:signaling]',\n",
       "   '\"MAPKKK cascade involved in osmosensory signalling pathway\" EXACT [GOC:mah]',\n",
       "   '\"osmolarity sensing, activation of MAP kinase kinase activity\" NARROW []',\n",
       "   '\"osmolarity sensing, activation of MAP kinase kinase kinase activity\" NARROW []',\n",
       "   '\"osmolarity sensing, activation of MAPK activity\" NARROW []',\n",
       "   '\"osmolarity sensing, activation of MAPKK activity\" NARROW []',\n",
       "   '\"osmolarity sensing, activation of MAPKKK activity\" NARROW []',\n",
       "   '\"osmolarity sensing, MAPKKK cascade\" EXACT []',\n",
       "   '\"termination of MAPK activity during osmolarity sensing\" NARROW []'],\n",
       "  'is_a': ['GO:0007231 ! osmosensory signaling pathway',\n",
       "   'GO:0051403 ! stress-activated MAPK cascade']},\n",
       " 'GO:0000162': {'name': ['tryptophan biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0009096'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of tryptophan, the chiral amino acid 2-amino-3-(1H-indol-3-yl)propanoic acid; tryptophan is synthesized from chorismate via anthranilate.\" [GOC:mah, ISBN:0471331309, MetaCyc:TRPSYN-PWY]'],\n",
       "  'synonym': ['\"aromatic amino acid family biosynthetic process, anthranilate pathway\" EXACT []',\n",
       "   '\"tryptophan anabolism\" EXACT []',\n",
       "   '\"tryptophan biosynthesis\" EXACT []',\n",
       "   '\"tryptophan formation\" EXACT []',\n",
       "   '\"tryptophan synthesis\" EXACT []'],\n",
       "  'xref': ['MetaCyc:TRPSYN-PWY'],\n",
       "  'is_a': ['GO:0006568 ! tryptophan metabolic process',\n",
       "   'GO:0009073 ! aromatic amino acid family biosynthetic process',\n",
       "   'GO:0046219 ! indolalkylamine biosynthetic process',\n",
       "   'GO:1901607 ! alpha-amino acid biosynthetic process']},\n",
       " 'GO:0000164': {'name': ['protein phosphatase type 1 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that possesses magnesium-dependent protein serine/threonine phosphatase (AMD phosphatase) activity, and consists of a catalytic subunit and one or more regulatory subunits that dictates the phosphatase\\'s substrate specificity, function, and activity.\" [GOC:mah, GOC:ssd]'],\n",
       "  'is_a': ['GO:0008287 ! protein serine/threonine phosphatase complex'],\n",
       "  'relationship': ['part_of GO:0005737 ! cytoplasm']},\n",
       " 'GO:0000165': {'name': ['MAPK cascade'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007255'],\n",
       "  'def': ['\"An intracellular protein kinase cascade containing at least a MAPK, a MAPKK and a MAP3K. The cascade can also contain an additional tiers'],\n",
       "  'comment': ['MAPK cascades lie downstream of many cell surface receptors and cooperate in transmitting various extracellular signals to the nucleus. One way by which the specificity of each cascade is regulated is through the existence of several distinct components in each tier of the different cascades. The cascades are typically named according to the component in the MAPK tier.'],\n",
       "  'synonym': ['\"ERK/MAPK cascade\" NARROW []',\n",
       "   '\"MAP kinase cascade\" EXACT []',\n",
       "   '\"MAP kinase kinase kinase cascade\" EXACT []',\n",
       "   '\"MAPK signal transduction\" EXACT [GOC:signaling]',\n",
       "   '\"MAPK signaling\" RELATED []',\n",
       "   '\"MAPK signalling\" RELATED []',\n",
       "   '\"MAPKKK cascade\" EXACT []',\n",
       "   '\"MAPKKK cascade during sporulation\" NARROW []',\n",
       "   '\"mitogen-activated protein kinase cascade\" EXACT [GOC:bf]'],\n",
       "  'xref': ['Wikipedia:MAPK_cascade',\n",
       "   'Wikipedia:Mitogen-activated_protein_kinase'],\n",
       "  'is_a': ['GO:0035556 ! intracellular signal transduction']},\n",
       " 'GO:0000166': {'name': ['nucleotide binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a nucleotide, any compound consisting of a nucleoside that is esterified with (ortho)phosphate or an oligophosphate at any hydroxyl group on the ribose or deoxyribose.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'subset': ['goslim_chembl',\n",
       "   'goslim_metagenomics',\n",
       "   'goslim_pir',\n",
       "   'goslim_plant'],\n",
       "  'is_a': ['GO:0036094 ! small molecule binding',\n",
       "   'GO:1901265 ! nucleoside phosphate binding']},\n",
       " 'GO:0000170': {'name': ['sphingosine hydroxylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the hydroxylation of sphingolipid long chain bases.\" [PMID:9556590]'],\n",
       "  'xref': ['Reactome:R-HSA-428260 \"dihydroceramide + NADPH + H+ + O2 => phytoceramide + NADP+ + H2O\"'],\n",
       "  'is_a': ['GO:0016491 ! oxidoreductase activity']},\n",
       " 'GO:0000171': {'name': ['ribonuclease MRP activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the site-specific cleavage of RNA by a catalytic RNA-mediated mechanism; substrates include the A3 site in the ITS1 of pre-rRNA.\" [PMID:17881380]'],\n",
       "  'synonym': ['\"RNase MRP\" EXACT []'],\n",
       "  'xref': ['Wikipedia:RNase_MRP'],\n",
       "  'is_a': ['GO:0004521 ! RNA endonuclease activity']},\n",
       " 'GO:0000172': {'name': ['ribonuclease MRP complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ribonucleoprotein complex that contains an RNA molecule of the snoRNA family, and cleaves the rRNA precursor as part of rRNA transcript processing. It also has other roles'],\n",
       "  'synonym': ['\"ribonuclease mitochondrial RNA processing complex\" EXACT []',\n",
       "   '\"RNase MRP complex\" EXACT []'],\n",
       "  'is_a': ['GO:0005732 ! sno(s)RNA-containing ribonucleoprotein complex',\n",
       "   'GO:1902555 ! endoribonuclease complex']},\n",
       " 'GO:0000174': {'name': ['obsolete inactivation of MAPK (mating sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Downregulation of MAP kinase activity in the context of transduction of mating pheromone signal, as described for Saccharomyces.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"inactivation of MAPK (mating sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000750']},\n",
       " 'GO:0000175': {'name': [\"3'-5'-RNA exonuclease activity\"],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the sequential cleavage of mononucleotides from a free 3\\' terminus of an RNA molecule.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'synonym': ['\"3\\'-5\\' exoribonuclease activity\" EXACT []',\n",
       "   '\"3\\'-5\\'-exoribonuclease activity\" EXACT []'],\n",
       "  'xref': ['EC:3.1.13.-',\n",
       "   'Reactome:R-HSA-430028 \"Exosome Complex hydrolyzes mRNA by 3\\' to 5\\' exoribonuclease digestion\"',\n",
       "   'Reactome:R-HSA-6791222 \"21S pre-rRNA is nucleolytically processed at site E (site2a) to yield 18SE pre-rRNA\"',\n",
       "   'Reactome:R-HSA-6791227 \"47S pre-rRNA is nucleolytically processed at A\\' (01,A1), site A0, and site 02 (site 6)  to yield 45S pre-rRNA\"',\n",
       "   'Reactome:R-HSA-9682603 \"nsp14 acts as a 3\\'-to-5\\' exonuclease to remove misincorporated nucleotides from nascent RNA\"',\n",
       "   'Reactome:R-HSA-9694632 \"nsp14 acts as a 3\\'-to-5\\' exonuclease to remove misincorporated nucleotides from nascent RNA\"'],\n",
       "  'is_a': [\"GO:0008408 ! 3'-5' exonuclease activity\",\n",
       "   \"GO:0016896 ! RNA exonuclease activity, producing 5'-phosphomonoesters\"]},\n",
       " 'GO:0000176': {'name': ['nuclear exosome (RNase complex)'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ribonuclease complex that has 3-prime to 5-prime processive and distributive hydrolytic exoribonuclease activity and endoribonuclease activity, producing 5-prime-phosphomonoesters. Participates in a multitude of cellular RNA processing and degradation events preventing nuclear export and/or translation of aberrant RNAs. Restricted to processing linear and circular single-stranded RNAs (ssRNA) only. RNAs with complex secondary structures may have to be unwound or pre-processed by co-factors prior to entering the complex, esp if the 3-prime end is structured.\" [PMID:17174896, PMID:20531386, PMID:26726035]'],\n",
       "  'synonym': ['\"eukaryotic exosome multienzyme ribonuclease complex\" EXACT []',\n",
       "   '\"nuclear exosome (ribonuclease complex)\" EXACT []',\n",
       "   '\"nuclear exosome multienzyme ribonuclease complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000178 ! exosome (RNase complex)',\n",
       "   'GO:0140513 ! nuclear protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0031981 ! nuclear lumen']},\n",
       " 'GO:0000177': {'name': ['cytoplasmic exosome (RNase complex)'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ribonuclease complex that has 3-prime to 5-prime processive hydrolytic exoribonuclease activity producing 5-prime-phosphomonoesters. Participates in a multitude of cellular RNA processing and degradation events preventing nuclear export and/or translation of aberrant RNAs. Restricted to processing linear and circular single-stranded RNAs (ssRNA) only. RNAs with complex secondary structures may have to be unwound or pre-processed by co-factors prior to entering the complex, esp if the 3-prime end is structured.\" [PMID:17174896, PMID:20531386, PMID:26726035]'],\n",
       "  'synonym': ['\"cytoplasmic exosome (ribonuclease complex)\" EXACT []',\n",
       "   '\"cytoplasmic exosome multienzyme ribonuclease complex\" EXACT []',\n",
       "   '\"prokaryotic exosome multienzyme ribonuclease complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000178 ! exosome (RNase complex)'],\n",
       "  'relationship': ['part_of GO:0005737 ! cytoplasm']},\n",
       " 'GO:0000178': {'name': ['exosome (RNase complex)'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ribonuclease complex that has 3-prime to 5-prime exoribonuclease activity and possibly endoribonuclease activity, producing 5-prime-phosphomonoesters. Participates in a multitude of cellular RNA processing and degradation events preventing nuclear export and/or translation of aberrant RNAs. Restricted to processing linear and circular single-stranded RNAs (ssRNA) only. RNAs with complex secondary structures may have to be unwound or pre-processed by co-factors prior to entering the complex, esp if the 3-prime end is structured.\" [PMID:17174896, PMID:20531386, PMID:26726035]'],\n",
       "  'comment': [\"Note that this term should not be confused with 'exosome' used in the context of vesicles released from multivesicular bodies.\"],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"exosome (ribonucleasease complex)\" EXACT []',\n",
       "   '\"exosome multienzyme ribonuclease complex\" EXACT []'],\n",
       "  'is_a': ['GO:1905354 ! exoribonuclease complex'],\n",
       "  'relationship': ['part_of GO:0005622 ! intracellular anatomical structure']},\n",
       " 'GO:0000179': {'name': ['rRNA (adenine-N6,N6-)-dimethyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0043790'],\n",
       "  'def': ['\"Catalysis of the dimethylation of two adjacent adenine residues in a rRNA, using S-adenosyl-L-methionine as a methyl donor.\" [ISBN:1555811337, PMID:10690410]'],\n",
       "  'synonym': ['\"18S rRNA dimethylase activity\" EXACT []',\n",
       "   '\"dimethyladenosine transferase activity\" EXACT []',\n",
       "   '\"S-adenosylmethionine-6-N\\', N\\'-adenosyl(rRNA) dimethyltransferase activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-6790994 \"DIMT1 dimethylates adenosine-1850,1851 of 18S rRNA yielding 6-dimethyladenosine-1850,1851\"',\n",
       "   'Reactome:R-HSA-6793066 \"TFB1M dimethylates adenosine-936 and adenosine-937 of 12S rRNA yielding 6-dimethyladenosine-936 and 6-dimethyladenosine-937\"'],\n",
       "  'is_a': ['GO:0008170 ! N-methyltransferase activity',\n",
       "   'GO:0016433 ! rRNA (adenine) methyltransferase activity']},\n",
       " 'GO:0000180': {'name': ['obsolete cytosolic large ribosomal subunit'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:ai]'],\n",
       "  'comment': ['This term was made obsolete because more specific children exist.'],\n",
       "  'synonym': ['\"cytosolic large ribosomal subunit\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0022625']},\n",
       " 'GO:0000181': {'name': ['obsolete cytosolic small ribosomal subunit'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:ai]'],\n",
       "  'comment': ['This term was made obsolete because more specific children exist.'],\n",
       "  'synonym': ['\"cytosolic small ribosomal subunit\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0022627']},\n",
       " 'GO:0000182': {'name': ['rDNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA sequence encoding a ribosomal RNA.\" [GOC:mah]'],\n",
       "  'synonym': ['\"ribosomal DNA binding\" EXACT []'],\n",
       "  'is_a': ['GO:1990837 ! sequence-specific double-stranded DNA binding']},\n",
       " 'GO:0000183': {'name': ['rDNA heterochromatin formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of heterochromatin at ribosomal DNA, characterized by the modified histone H3K9me3.\" [PMID:10219245]'],\n",
       "  'synonym': ['\"chromatin silencing at rDNA\" RELATED []',\n",
       "   '\"chromatin silencing at ribosomal DNA\" BROAD []',\n",
       "   '\"heterochromatic silencing at rDNA\" BROAD []',\n",
       "   '\"rDNA chromatin silencing\" BROAD [GOC:mah]',\n",
       "   '\"rDNA heterochromatin assembly\" EXACT []'],\n",
       "  'is_a': ['GO:0140718 ! facultative heterochromatin formation'],\n",
       "  'relationship': ['part_of GO:1990700 ! nucleolar chromatin organization']},\n",
       " 'GO:0000184': {'name': ['nuclear-transcribed mRNA catabolic process, nonsense-mediated decay'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The nonsense-mediated decay pathway for nuclear-transcribed mRNAs degrades mRNAs in which an amino-acid codon has changed to a nonsense codon; this prevents the translation of such mRNAs into truncated, and potentially harmful, proteins.\" [GOC:krc, GOC:ma, PMID:10025395]'],\n",
       "  'synonym': ['\"mRNA breakdown, nonsense-mediated decay\" EXACT []',\n",
       "   '\"mRNA catabolic process, nonsense-mediated\" EXACT []',\n",
       "   '\"mRNA catabolism, nonsense-mediated\" EXACT []',\n",
       "   '\"mRNA degradation, nonsense-mediated decay\" EXACT []',\n",
       "   '\"nonsense-mediated mRNA decay\" EXACT []',\n",
       "   '\"nuclear mRNA catabolic process, nonsense-mediated decay\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Nonsense-mediated_decay',\n",
       "   'Wikipedia:Nonsense-mediated_mRNA_decay'],\n",
       "  'is_a': ['GO:0000956 ! nuclear-transcribed mRNA catabolic process']},\n",
       " 'GO:0000185': {'name': ['obsolete activation of MAPKKK activity'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that initiates the activity of the inactive enzyme MAP kinase kinase kinase (MAPKKK).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'synonym': ['\"activation of MAP kinase kinase kinase\" EXACT []',\n",
       "   '\"activation of MAP3K activity\" EXACT [GOC:bf]',\n",
       "   '\"activation of MAPKKK activity during sporulation\" NARROW []',\n",
       "   '\"positive regulation of MAP kinase kinase kinase activity\" BROAD []',\n",
       "   '\"positive regulation of MAPKKK activity\" BROAD []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000165', 'GO:0043539']},\n",
       " 'GO:0000186': {'name': ['obsolete activation of MAPKK activity'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The initiation of the activity of the inactive enzyme MAP kinase kinase (MAPKK).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'synonym': ['\"activation of MAP kinase kinase activity\" EXACT []',\n",
       "   '\"activation of MAP/ERK kinase kinase\" EXACT []',\n",
       "   '\"activation of MAP2K activity\" EXACT [GOC:bf]',\n",
       "   '\"activation of MAPKK activity during sporulation\" NARROW []',\n",
       "   '\"positive regulation of MAPKK activity\" BROAD []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000165', 'GO:0043539']},\n",
       " 'GO:0000187': {'name': ['obsolete activation of MAPK activity'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The initiation of the activity of the inactive enzyme MAP kinase (MAPK).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'synonym': ['\"activation of MAP kinase\" EXACT []',\n",
       "   '\"activation of MAPK activity during sporulation\" NARROW []',\n",
       "   '\"MAPK activation\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000165', 'GO:0043539']},\n",
       " 'GO:0000188': {'name': ['obsolete inactivation of MAPK activity'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that terminates the activity of the active enzyme MAP kinase.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'synonym': ['\"inactivation of MAPK during sporulation\" NARROW []',\n",
       "   '\"termination of MAPK activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000165', 'GO:0043539']},\n",
       " 'GO:0000189': {'name': ['obsolete MAPK import into nucleus'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The directed movement of a MAP kinase to the nucleus upon activation.\" [PMID:9561267]'],\n",
       "  'comment': ['The reason for obsoletion is that all proteins are imported into the nucleus via the same mechanism, so the import of individual proteins should be captured with extensions or by GO-CAM models.'],\n",
       "  'synonym': ['\"nuclear translocation of MAPK\" NARROW [GOC:mah]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006606']},\n",
       " 'GO:0000190': {'name': ['obsolete MAPKKK cascade (pseudohyphal growth)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. MAPKKK cascade involved in transduction of signal promoting pseudohyphal or invasive growth.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"MAPKKK cascade (pseudohyphal growth)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000191': {'name': ['obsolete activation of MAPKKK (pseudohyphal growth)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Upregulation of MAPKKK activity in the context of regulating pseudohyphal or invasive growth.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"activation of MAPKKK (pseudohyphal growth)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000192': {'name': ['obsolete activation of MAPKK (pseudohyphal growth)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Upregulation of a MAP kinase kinase in the context of regulating pseudohyphal or invasive growth.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"activation of MAPKK (pseudohyphal growth)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000193': {'name': ['obsolete activation of MAPK (pseudohyphal growth)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Upregulation of MAP kinase activity in the context of regulating pseudohyphal or invasive growth.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"activation of MAPK (pseudohyphal growth)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000194': {'name': ['obsolete inactivation of MAPK (pseudohyphal growth)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Downregulation of MAP kinase activity in the context of regulating pseudohyphal or invasive growth.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"inactivation of MAPK (pseudohyphal growth)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000195': {'name': ['obsolete nuclear translocation of MAPK (pseudohyphal growth)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Movement of a MAP kinase to the nucleus in the context of regulating pseudohyphal or invasive growth.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"nuclear translocation of MAPK (pseudohyphal growth)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000196': {'name': ['cell wall integrity MAPK cascade'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A MAPK cascade that contributes to cell wall organization or biogenesis.\" [PMID:17604854, PMID:9561267]'],\n",
       "  'synonym': ['\"cell integrity MAPK pathway\" EXACT [GOC:vw, PMID:23934882]',\n",
       "   '\"cell wall biogenesis, MAPKKK cascade\" EXACT []',\n",
       "   '\"MAPK cascade involved in cell wall biogenesis\" NARROW [GOC:vw]',\n",
       "   '\"MAPK cascade involved in cell wall organization or biogenesis\" EXACT []',\n",
       "   '\"MAPKKK cascade involved in cell wall biogenesis\" EXACT [GOC:signaling]',\n",
       "   '\"Mpk1 cascade\" NARROW [PMID:10523653]',\n",
       "   '\"Pmk1 MAPK cell integrity signaling\" NARROW [PMID:20032302]',\n",
       "   '\"Pmk1 mitogen-activated protein kinase (MAPK) cell integrity pathway\" NARROW [PMID:23934882]',\n",
       "   '\"PMK1-MAPK signal transduction pathway\" NARROW [PMID:23454094]',\n",
       "   '\"Slt2 cascade\" NARROW [PMID:10523653]'],\n",
       "  'is_a': ['GO:0051403 ! stress-activated MAPK cascade'],\n",
       "  'relationship': ['part_of GO:0071554 ! cell wall organization or biogenesis']},\n",
       " 'GO:0000197': {'name': ['obsolete activation of MAPKKK activity involved in cell wall organization or biogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that initiates the activity of the inactive enzyme MAP kinase kinase kinase in the context of cell wall organization or biogenesis.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"activation of MAP kinase kinase kinase activity during cell wall biogenesis\" EXACT []',\n",
       "   '\"activation of MAPKKK activity involved in cell wall biogenesis\" RELATED []',\n",
       "   '\"activation of MAPKKK activity involved in cell wall integrity\" EXACT [GOC:dgf]',\n",
       "   '\"cell wall biogenesis, activation of MAP kinase kinase kinase activity\" EXACT []',\n",
       "   '\"cell wall biogenesis, activation of MAPKKK activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000165', 'GO:0043539']},\n",
       " 'GO:0000198': {'name': ['obsolete activation of MAPKK activity involved in cell wall organization or biogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that initiates the activity of the inactive enzyme MAP kinase kinase in the context of cell wall organization or biogenesis.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"activation of MAP kinase kinase activity during cell wall biogenesis\" EXACT []',\n",
       "   '\"activation of MAPKK activity involved in cell wall biogenesis\" RELATED []',\n",
       "   '\"activation of MAPKK activity involved in cell wall integrity\" EXACT [GOC:dgf]',\n",
       "   '\"cell wall biogenesis, activation of MAP kinase kinase activity\" EXACT []',\n",
       "   '\"cell wall biogenesis, activation of MAPKK activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000199': {'name': ['obsolete activation of MAPK activity involved in cell wall organization or biogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that initiates the activity of the inactive enzyme MAP kinase in the context of cell wall biogenesis, the assembly and arrangement of the cell wall, the rigid or semi-rigid envelope lying outside the cell membrane of plant, fungal and most prokaryotic cells.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"activation of MAPK activity involved in cell wall biogenesis\" NARROW []',\n",
       "   '\"activation of MAPK activity involved in cell wall integrity\" EXACT [GOC:dgf]',\n",
       "   '\"cell wall biogenesis, activation of MAPK activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000200': {'name': ['obsolete inactivation of MAPK activity involved in cell wall organization or biogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that terminates the activity of the active enzyme MAP kinase in the context of cell wall organization or biogenesis.\" [PMID:9561267]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"cell wall biogenesis, inactivation of MAPK activity\" EXACT []',\n",
       "   '\"cell wall biogenesis, termination of MAPK activity\" EXACT []',\n",
       "   '\"inactivation of MAPK activity involved in cell wall integrity\" EXACT [GOC:dgf]',\n",
       "   '\"inactivation of MAPK activity involved in cell wall organization or biogenesis\" RELATED []',\n",
       "   '\"termination of MAPK activity during cell wall biogenesis\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000201': {'name': ['obsolete MAPK import into nucleus involved in cell wall organization or biogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The directed movement of a MAP kinase to the nucleus that occurs in the context of cell wall organization or biogenesis.\" [PMID:9561267]'],\n",
       "  'comment': ['The reason for obsoletion is that this term represents a GO-CAM model.'],\n",
       "  'synonym': ['\"cell wall biogenesis, MAPK import into nucleus\" EXACT [GOC:mah]',\n",
       "   '\"cell wall biogenesis, nuclear translocation of MAPK\" NARROW [GOC:mah]',\n",
       "   '\"MAPK import into nucleus involved in cell wall biogenesis\" NARROW [GOC:vw]',\n",
       "   '\"MAPK import into nucleus involved in cell wall integrity\" EXACT [GOC:dgf, GOC:mah]',\n",
       "   '\"nuclear translocation of MAPK involved in cell wall biogenesis\" NARROW [GOC:mah]',\n",
       "   '\"nuclear translocation of MAPK involved in cell wall integrity\" NARROW [GOC:dgf, GOC:mah]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006606']},\n",
       " 'GO:0000202': {'name': ['obsolete MAPKKK cascade during sporulation (sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. MAPKKK cascade involved in transduction of signal promoting sporulation. As in, but not restricted to, the taxon Saccharomyces (Saccharomyces, ncbi_taxonomy_id:4930).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"MAPKKK cascade during sporulation (sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000203': {'name': ['obsolete activation of MAPKKK during sporulation (sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Upregulation of MAPKKK activity in the context of sporulation. As in, but not restricted to, the taxon Saccharomyces (Saccharomyces, ncbi_taxonomy_id:4930).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"activation of MAPKKK during sporulation (sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000204': {'name': ['obsolete activation of MAPKK during sporulation (sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Upregulation of a MAP kinase kinase in the context of sporulation. As in, but not restricted to, the taxon Saccharomyces (Saccharomyces, ncbi_taxonomy_id:4930).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"activation of MAPKK during sporulation (sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000205': {'name': ['obsolete activation of MAPK during sporulation (sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Upregulation of MAP kinase activity in the context of sporulation. As in, but not restricted to, the taxon Saccharomyces (Saccharomyces, ncbi_taxonomy_id:4930).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"activation of MAPK during sporulation (sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000206': {'name': ['obsolete inactivation of MAPK during sporulation (sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Downregulation of MAP kinase activity in the context of sporulation. As in, but not restricted to, the taxon Saccharomyces (Saccharomyces, ncbi_taxonomy_id:4930).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"inactivation of MAPK during sporulation (sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000207': {'name': ['obsolete nuclear translocation of MAPK during sporulation (sensu Saccharomyces)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Movement of a MAP kinase to the nucleus in the context of sporulation. As in, but not restricted to, the taxon Saccharomyces (Saccharomyces, ncbi_taxonomy_id:4930).\" [PMID:9561267]'],\n",
       "  'comment': ['This term was made obsolete because it is a gene product specific term.'],\n",
       "  'synonym': ['\"nuclear translocation of MAPK during sporulation (sensu Saccharomyces)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000208': {'name': ['obsolete MAPK import into nucleus involved in osmosensory signaling pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The directed movement of a MAP kinase to the nucleus during osmolarity sensing.\" [PMID:9561267]'],\n",
       "  'comment': ['The reason for obsoletion is that this term represents a GO-CAM model.'],\n",
       "  'synonym': ['\"MAPK import into nucleus during osmolarity sensing\" EXACT [GOC:mah]',\n",
       "   '\"MAPK import into nucleus involved in osmosensory signalling pathway\" EXACT [GOC:mah]',\n",
       "   '\"nuclear translocation of MAPK during osmolarity sensing\" EXACT []',\n",
       "   '\"nuclear translocation of MAPK involved in osmosensory signaling pathway\" NARROW [GOC:mah]',\n",
       "   '\"nuclear translocation of MAPK involved in osmosensory signalling pathway\" NARROW [GOC:mah]',\n",
       "   '\"osmolarity sensing, MAPK import into nucleus\" EXACT [GOC:mah]',\n",
       "   '\"osmolarity sensing, nuclear translocation of MAPK\" NARROW [GOC:mah]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006606']},\n",
       " 'GO:0000209': {'name': ['protein polyubiquitination'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Addition of multiple ubiquitin groups to a protein, forming a ubiquitin chain.\" [ISBN:0815316194]'],\n",
       "  'subset': ['term_scheduled_for_obsoletion'],\n",
       "  'synonym': ['\"polyubiquitin\" RELATED []',\n",
       "   '\"protein polyubiquitinylation\" EXACT []',\n",
       "   '\"protein polyubiquitylation\" EXACT []'],\n",
       "  'is_a': ['GO:0016567 ! protein ubiquitination']},\n",
       " 'GO:0000210': {'name': ['NAD+ diphosphatase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"NAD diphosphatase activity\" BROAD []',\n",
       "   '\"NAD pyrophosphatase activity\" BROAD []',\n",
       "   '\"NAD(+) pyrophosphatase activity\" EXACT []',\n",
       "   '\"NAD+ phosphohydrolase activity\" EXACT []',\n",
       "   '\"NAD+ pyrophosphatase activity\" EXACT []',\n",
       "   '\"NADP pyrophosphatase activity\" RELATED [EC:3.6.1.22]',\n",
       "   '\"nicotinamide adenine dinucleotide pyrophosphatase activity\" RELATED [EC:3.6.1.22]'],\n",
       "  'xref': ['EC:3.6.1.22', 'MetaCyc:NADPYROPHOSPHAT-RXN', 'RHEA:11800'],\n",
       "  'is_a': ['GO:0004551 ! dinucleotide phosphatase activity']},\n",
       " 'GO:0000211': {'name': ['obsolete protein degradation tagging activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Covalent addition of polyubiquitin to another protein, targeting the tagged protein for destruction.\" [GOC:cl, ISBN:0815316194]'],\n",
       "  'comment': ['This term was made obsolete because it represents a biological process and a molecular function.'],\n",
       "  'synonym': ['\"protein degradation tagging activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005515', 'GO:0019941']},\n",
       " 'GO:0000212': {'name': ['meiotic spindle organization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0043147'],\n",
       "  'def': ['\"A process that is carried out at the cellular level which results in the assembly, arrangement of constituent parts, or disassembly of the microtubule spindle during a meiotic cell cycle.\" [GOC:go_curators, GOC:mah]'],\n",
       "  'synonym': ['\"meiotic spindle organisation\" EXACT []',\n",
       "   '\"meiotic spindle organization and biogenesis\" RELATED [GOC:mah]',\n",
       "   '\"meiotic spindle stabilization\" RELATED []',\n",
       "   '\"spindle organization during meiosis\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0007051 ! spindle organization',\n",
       "   'GO:1903046 ! meiotic cell cycle process']},\n",
       " 'GO:0000213': {'name': ['tRNA-intron endonuclease activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the endonucleolytic cleavage of pre-tRNA, producing 5\\'-hydroxyl and 2\\',3\\'-cyclic phosphate termini, and specifically removing the intron.\" [EC:4.6.1.16]'],\n",
       "  'synonym': ['\"tRNA splicing endonuclease activity\" RELATED [EC:4.6.1.16]',\n",
       "   '\"tRNA-intron endoribonuclease activity\" EXACT []'],\n",
       "  'xref': ['EC:4.6.1.16', 'MetaCyc:3.1.27.9-RXN'],\n",
       "  'is_a': ['GO:0004549 ! tRNA-specific ribonuclease activity',\n",
       "   \"GO:0016892 ! RNA endonuclease activity, producing 3'-phosphomonoesters\"]},\n",
       " 'GO:0000214': {'name': ['tRNA-intron endonuclease complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that catalyzes the endonucleolytic cleavage of pre-tRNA, producing 5\\'-hydroxyl and 2\\',3\\'-cyclic phosphate termini, and specifically removing the intron.\" [PMID:22391451]'],\n",
       "  'synonym': ['\"SEN complex\" EXACT [GOC:se, PMID:22391451]',\n",
       "   '\"tRNA splicing endonuclease complex\" EXACT [GOC:se, PMID:22391451]'],\n",
       "  'is_a': ['GO:0140513 ! nuclear protein-containing complex',\n",
       "   'GO:1902555 ! endoribonuclease complex']},\n",
       " 'GO:0000215': {'name': [\"tRNA 2'-phosphotransferase activity\"],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0008665'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'subset': ['goslim_chembl'],\n",
       "  'synonym': ['\"2\\'-phospho-[ligated tRNA]:NAD+ phosphotransferase activity\" RELATED [EC:2.7.1.160]',\n",
       "   '\"2\\'-phospho-tRNA:NAD+ phosphotransferase activity\" RELATED [EC:2.7.1.160]',\n",
       "   '\"2\\'-phosphotransferase activity\" BROAD []',\n",
       "   '\"Tpt1\" RELATED [EC:2.7.1.160]',\n",
       "   '\"Tpt1p\" RELATED [EC:2.7.1.160]',\n",
       "   '\"yeast 2\\'-phosphotransferase activity\" NARROW [EC:2.7.1.160]'],\n",
       "  'xref': ['EC:2.7.1.160', 'MetaCyc:2.7.1.160-RXN', 'RHEA:23324'],\n",
       "  'is_a': ['GO:0016773 ! phosphotransferase activity, alcohol group as acceptor',\n",
       "   'GO:0140101 ! catalytic activity, acting on a tRNA']},\n",
       " 'GO:0000216': {'name': ['obsolete M/G1 transition of mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Progression from M phase to G1 phase of the mitotic cell cycle.\" [GOC:mah, GOC:mtg_cell_cycle]'],\n",
       "  'comment': ['This term was made obsolete because it does not refer to a real biological process.'],\n",
       "  'synonym': ['\"M/G1 transition of mitotic cell cycle\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000217': {'name': ['DNA secondary structure binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA secondary structure element such as a four-way junction, a bubble, a loop, Y-form DNA, or a double-strand/single-strand junction.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0003677 ! DNA binding']},\n",
       " 'GO:0000219': {'name': ['obsolete vacuolar hydrogen-transporting ATPase'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:go_curators]'],\n",
       "  'comment': ['This term was made obsolete because more specific terms were created.'],\n",
       "  'synonym': ['\"V-ATPase\" EXACT []',\n",
       "   '\"vacuolar hydrogen-transporting ATPase\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0016469']},\n",
       " 'GO:0000220': {'name': ['vacuolar proton-transporting V-type ATPase, V0 domain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The V0 domain of a proton-transporting V-type ATPase found in the vacuolar membrane.\" [GOC:mah, PMID:16449553]'],\n",
       "  'comment': ['Note that this domain often consists of five subunits, although in some mammalian tissues it may have an additional subunit.'],\n",
       "  'synonym': ['\"vacuolar hydrogen ion-transporting ATPase V0 domain\" EXACT []'],\n",
       "  'is_a': ['GO:0033179 ! proton-transporting V-type ATPase, V0 domain'],\n",
       "  'relationship': ['part_of GO:0016471 ! vacuolar proton-transporting V-type ATPase complex']},\n",
       " 'GO:0000221': {'name': ['vacuolar proton-transporting V-type ATPase, V1 domain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The V1 domain of a proton-transporting V-type ATPase found in the vacuolar membrane.\" [GOC:mah, PMID:16449553]'],\n",
       "  'comment': ['Note that this domain generally consists of eight subunits.'],\n",
       "  'synonym': ['\"vacuolar hydrogen ion-transporting ATPase V1 domain\" EXACT []'],\n",
       "  'is_a': ['GO:0033180 ! proton-transporting V-type ATPase, V1 domain'],\n",
       "  'relationship': ['part_of GO:0016471 ! vacuolar proton-transporting V-type ATPase complex']},\n",
       " 'GO:0000222': {'name': ['plasma membrane proton-transporting V-type ATPase, V0 domain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The V0 domain of a proton-transporting V-type ATPase found in the plasma membrane.\" [GOC:mah]'],\n",
       "  'synonym': ['\"plasma membrane hydrogen ion-transporting ATPase V0 domain\" EXACT []'],\n",
       "  'is_a': ['GO:0033179 ! proton-transporting V-type ATPase, V0 domain',\n",
       "   'GO:0098797 ! plasma membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0033181 ! plasma membrane proton-transporting V-type ATPase complex']},\n",
       " 'GO:0000223': {'name': ['plasma membrane proton-transporting V-type ATPase, V1 domain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The V1 domain of a proton-transporting V-type ATPase found in the plasma membrane.\" [GOC:mah]'],\n",
       "  'synonym': ['\"plasma membrane hydrogen ion-transporting ATPase V1 domain\" EXACT []'],\n",
       "  'is_a': ['GO:0033180 ! proton-transporting V-type ATPase, V1 domain',\n",
       "   'GO:0098797 ! plasma membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0033181 ! plasma membrane proton-transporting V-type ATPase complex']},\n",
       " 'GO:0000224': {'name': ['peptide-N4-(N-acetyl-beta-glucosaminyl)asparagine amidase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"glycopeptidase activity\" BROAD [EC:3.5.1.52]',\n",
       "   '\"glycopeptide N-glycosidase activity\" BROAD [EC:3.5.1.52]',\n",
       "   '\"jack-bean glycopeptidase\" NARROW [EC:3.5.1.52]',\n",
       "   '\"N-glycanase activity\" RELATED [EC:3.5.1.52]',\n",
       "   '\"N-linked-glycopeptide-(N-acetyl-beta-D-glucosaminyl)-L-asparagine amidohydrolase activity\" RELATED [EC:3.5.1.52]',\n",
       "   '\"N-oligosaccharide glycopeptidase activity\" RELATED [EC:3.5.1.52]',\n",
       "   '\"peptide:N-glycanase\" RELATED []',\n",
       "   '\"PNGase\" EXACT []',\n",
       "   '\"PNGase A\" RELATED [EC:3.5.1.52]',\n",
       "   '\"PNGase F\" RELATED [EC:3.5.1.52]'],\n",
       "  'xref': ['EC:3.5.1.52',\n",
       "   'MetaCyc:3.5.1.52-RXN',\n",
       "   'Reactome:R-HSA-8850594 \"Deglycosylation complex hydrolyses N-glycans from unfolded glycoproteins\"'],\n",
       "  'is_a': ['GO:0016811 ! hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in linear amides']},\n",
       " 'GO:0000225': {'name': ['N-acetylglucosaminylphosphatidylinositol deacetylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'comment': ['Note that this function was formerly EC:3.1.1.69.'],\n",
       "  'synonym': ['\"6-(N-acetyl-alpha-D-glucosaminyl)-1-phosphatidyl-1D-myo-inositol acetylhydrolase activity\" RELATED [EC:3.5.1.89]',\n",
       "   '\"acetylglucosaminylphosphatidylinositol deacetylase activity\" RELATED [EC:3.5.1.89]',\n",
       "   '\"GlcNAc-PI de-N-acetylase activity\" RELATED [EC:3.5.1.89]',\n",
       "   '\"GlcNAc-PI deacetylase activity\" RELATED [EC:3.5.1.89]',\n",
       "   '\"N-acetyl-D-glucosaminylphosphatidylinositol acetylhydrolase activity\" RELATED [EC:3.5.1.89]',\n",
       "   '\"N-acetylglucosaminylphosphatidylinositol de-N-acetylase activity\" RELATED [EC:3.5.1.89]'],\n",
       "  'xref': ['EC:3.5.1.89',\n",
       "   'MetaCyc:3.1.1.69-RXN',\n",
       "   'Reactome:R-HSA-162857 \"N-acetylglucosaminyl-PI + H2O -> glucosaminyl-PI + acetate\"',\n",
       "   'RHEA:11660'],\n",
       "  'is_a': ['GO:0016811 ! hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in linear amides',\n",
       "   'GO:0019213 ! deacetylase activity']},\n",
       " 'GO:0000226': {'name': ['microtubule cytoskeleton organization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A process that is carried out at the cellular level which results in the assembly, arrangement of constituent parts, or disassembly of cytoskeletal structures comprising microtubules and their associated proteins.\" [GOC:mah]'],\n",
       "  'subset': ['goslim_pombe'],\n",
       "  'synonym': ['\"microtubule cytoskeleton organisation\" EXACT [GOC:mah]',\n",
       "   '\"microtubule cytoskeleton organization and biogenesis\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0007010 ! cytoskeleton organization',\n",
       "   'GO:0007017 ! microtubule-based process']},\n",
       " 'GO:0000227': {'name': ['oxaloacetate secondary active transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of oxaloacetate from one side of a membrane to the other, up its concentration gradient. The transporter binds the solute and undergoes a series of conformational changes. Transport works equally well in either direction and is driven by a chemiosmotic source of energy. Secondary active transporters include symporters and antiporters.\" [GOC:mtg_transport]'],\n",
       "  'synonym': ['\"oxaloacetate carrier activity\" RELATED []'],\n",
       "  'is_a': ['GO:0015131 ! oxaloacetate transmembrane transporter activity',\n",
       "   'GO:0015291 ! secondary active transmembrane transporter activity']},\n",
       " 'GO:0000228': {'name': ['nuclear chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A chromosome that encodes the nuclear genome and is found in the nucleus of a eukaryotic cell during the cell cycle phases when the nucleus is intact.\" [GOC:dph, GOC:mah]'],\n",
       "  'subset': ['goslim_chembl', 'goslim_generic'],\n",
       "  'synonym': ['\"nuclear interphase chromosome\" NARROW []'],\n",
       "  'is_a': ['GO:0005694 ! chromosome'],\n",
       "  'relationship': ['part_of GO:0031981 ! nuclear lumen']},\n",
       " 'GO:0000229': {'name': ['obsolete cytoplasmic chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A chromosome found in the cytoplasm.\" [GOC:mah]'],\n",
       "  'comment': ['This term was obsoleted because it is an unnecessary grouping class.'],\n",
       "  'synonym': ['\"cytoplasmic interphase chromosome\" NARROW []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005694']},\n",
       " 'GO:0000230': {'name': ['obsolete nuclear mitotic chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A chromosome found in the nucleus during mitosis.\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it is based on a temporal division of the cell cycle.'],\n",
       "  'synonym': ['\"nuclear mitotic chromosome\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000794']},\n",
       " 'GO:0000231': {'name': ['obsolete cytoplasmic mitotic chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A chromosome found in the cytoplasm during mitosis.\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it is based on a temporal division of the cell cycle.'],\n",
       "  'synonym': ['\"cytoplasmic mitotic chromosome\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000793']},\n",
       " 'GO:0000232': {'name': ['obsolete nuclear interphase chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A chromosome found in the nucleus during interphase.\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it is based on a temporal division of the cell cycle.'],\n",
       "  'synonym': ['\"nuclear interphase chromosome\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000228']},\n",
       " 'GO:0000233': {'name': ['obsolete cytoplasmic interphase chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A chromosome found in the cytoplasm during interphase.\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it is based on a temporal division of the cell cycle.'],\n",
       "  'synonym': ['\"cytoplasmic interphase chromosome\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005694']},\n",
       " 'GO:0000234': {'name': ['phosphoethanolamine N-methyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"phosphoethanolamine methyltransferase activity\" RELATED [EC:2.1.1.103]',\n",
       "   '\"S-adenosyl-L-methionine:ethanolamine-phosphate N-methyltransferase activity\" RELATED [EC:2.1.1.103]'],\n",
       "  'xref': ['EC:2.1.1.103',\n",
       "   'KEGG_REACTION:R02037',\n",
       "   'MetaCyc:2.1.1.103-RXN',\n",
       "   'RHEA:20365'],\n",
       "  'is_a': ['GO:0008170 ! N-methyltransferase activity',\n",
       "   'GO:0008757 ! S-adenosylmethionine-dependent methyltransferase activity']},\n",
       " 'GO:0000235': {'name': ['astral microtubule'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Any of the spindle microtubules that radiate in all directions from the spindle poles and are thought to contribute to the forces that separate the poles and position them in relation to the rest of the cell.\" [ISBN:0815316194]'],\n",
       "  'is_a': ['GO:0005876 ! spindle microtubule',\n",
       "   'GO:0005881 ! cytoplasmic microtubule'],\n",
       "  'relationship': ['part_of GO:0005818 ! aster']},\n",
       " 'GO:0000236': {'name': ['mitotic prometaphase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase in higher eukaryotes which follows mitotic prophase and during which the nuclear envelope is disrupted and breaks into membrane vesicles, and the spindle microtubules enter the nuclear region. Kinetochores mature on each centromere and attach to some of the spindle microtubules. Kinetochore microtubules begin the process of aligning chromosomes in one plane halfway between the poles.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'is_a': ['GO:0000087 ! mitotic M phase']},\n",
       " 'GO:0000237': {'name': ['leptotene'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which is the first stage of prophase I in meiosis, and during which the chromosomes first become visible.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'xref': ['Wikipedia:Leptotene', 'Wikipedia:Meiosis#Leptotene'],\n",
       "  'is_a': ['GO:0098764 ! meiosis I cell cycle phase'],\n",
       "  'relationship': ['part_of GO:0007128 ! meiotic prophase I']},\n",
       " 'GO:0000238': {'name': ['zygotene'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which follows leptotene during prophase I of meiosis, and during which each chromosome pairs with its homolog; the two become aligned and crossing over may occur.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'xref': ['Wikipedia:Meiosis#Zygotene', 'Wikipedia:Zygotene'],\n",
       "  'is_a': ['GO:0098764 ! meiosis I cell cycle phase'],\n",
       "  'relationship': ['part_of GO:0007128 ! meiotic prophase I']},\n",
       " 'GO:0000239': {'name': ['pachytene'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which follows zygotene during prophase I of meiosis, and during which crossing over occurs between a chromatid in one partner and another chromatid in the homologous chromosome.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'xref': ['Wikipedia:Meiosis#Pachytene', 'Wikipedia:Pachytene'],\n",
       "  'is_a': ['GO:0098764 ! meiosis I cell cycle phase'],\n",
       "  'relationship': ['part_of GO:0007128 ! meiotic prophase I']},\n",
       " 'GO:0000240': {'name': ['diplotene'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which follows pachytene during prophase I of meiosis, during which the homologous chromosomes begin to separate and the synaptonemal complex dissolves.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'xref': ['Wikipedia:Diplotene', 'Wikipedia:Meiosis#Diplotene'],\n",
       "  'is_a': ['GO:0098764 ! meiosis I cell cycle phase'],\n",
       "  'relationship': ['part_of GO:0007128 ! meiotic prophase I']},\n",
       " 'GO:0000241': {'name': ['diakinesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle phase which follows diplotene during prophase I of meiosis, the separation of homologous chromosomes is complete and crossing over has occurred.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'xref': ['Wikipedia:Diakinesis', 'Wikipedia:Meiosis#Diakinesis'],\n",
       "  'is_a': ['GO:0098764 ! meiosis I cell cycle phase'],\n",
       "  'relationship': ['part_of GO:0007128 ! meiotic prophase I']},\n",
       " 'GO:0000242': {'name': ['pericentriolar material'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A network of small fibers that surrounds the centrioles in cells; contains the microtubule nucleating activity of the centrosome.\" [GOC:clt, ISBN:0815316194]'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005813 ! centrosome']},\n",
       " 'GO:0000243': {'name': ['commitment complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A spliceosomal complex that is formed by association of the U1 snRNP with the 5\\' splice site of an unspliced intron in an RNA transcript.\" [GOC:krc, ISBN:0879695897, PMID:9150140]'],\n",
       "  'synonym': ['\"mammalian spliceosomal complex E\" NARROW [GOC:krc, GOC:mah, ISBN:0879695897, ISBN:0879697393]',\n",
       "   '\"mammalian spliceosomal E complex\" NARROW [GOC:mah]',\n",
       "   '\"yeast spliceosomal complex CC\" NARROW [GOC:krc, GOC:mah, ISBN:0879695897, ISBN:0879697393]'],\n",
       "  'is_a': ['GO:0005684 ! U2-type spliceosomal complex']},\n",
       " 'GO:0000244': {'name': ['spliceosomal tri-snRNP complex assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000351', 'GO:0000355'],\n",
       "  'def': ['\"The formation of a tri-snRNP complex containing U4 and U6 (or U4atac and U6atac) snRNAs and U5 snRNAs and associated proteins. This includes reannealing of U4 and U6 (or U4atac and U6atac) snRNAs released from previous rounds of splicing to reform the U4/U6 snRNP (or U4atac/U6atac snRNP) as well as the subsequent association of the U5 snRNP with the U4/U6 snRNP (or U4atac/U6atac snRNP) to form a tri-snRNP that is ready to reassemble into another spliceosome complex.\" [ISBN:0879695897, PMID:9452384]'],\n",
       "  'synonym': ['\"assembly of spliceosomal tri-snRNP\" EXACT []',\n",
       "   '\"assembly of spliceosomal tri-snRNP U4/U6.U5\" NARROW []',\n",
       "   '\"assembly of spliceosomal tri-snRNP U4atac/U6atac.U5\" NARROW []',\n",
       "   '\"snRNP recycling\" BROAD []',\n",
       "   '\"spliceosomal tri-snRNP assembly\" EXACT []',\n",
       "   '\"spliceosomal tri-snRNP U4/U6.U5 assembly\" NARROW []',\n",
       "   '\"spliceosomal tri-snRNP U4atac/U6atac.U5 assembly\" NARROW []'],\n",
       "  'is_a': ['GO:0000387 ! spliceosomal snRNP assembly']},\n",
       " 'GO:0000245': {'name': ['spliceosomal complex assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of a spliceosomal complex, a ribonucleoprotein apparatus that catalyzes nuclear mRNA splicing via transesterification reactions.\" [PMID:9476892]'],\n",
       "  'synonym': ['\"spliceosome assembly\" BROAD []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000398 ! mRNA splicing, via spliceosome']},\n",
       " 'GO:0000246': {'name': ['delta24(24-1) sterol reductase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"C-24(28) sterol reductase activity\" RELATED [EC:1.3.1.71]',\n",
       "   '\"D24(24-1)-sterol reductase activity\" EXACT []',\n",
       "   '\"delta24(241)-sterol reductase activity\" RELATED [EC:1.3.1.71]',\n",
       "   '\"ergosterol:NADP+ delta24(241)-oxidoreductase activity\" RELATED [EC:1.3.1.71]',\n",
       "   '\"sterol Delta(24(28))-methylene reductase activity\" RELATED [EC:1.3.1.71]',\n",
       "   '\"sterol Delta(24(28))-reductase activity\" RELATED [EC:1.3.1.71]',\n",
       "   '\"sterol delta-24(28) methylene reductase activity\" EXACT []',\n",
       "   '\"sterol delta-24(28) reductase activity\" EXACT []',\n",
       "   '\"sterol delta24(28)-methylene reductase activity\" RELATED [EC:1.3.1.71]',\n",
       "   '\"sterol delta24(28)-reductase activity\" RELATED [EC:1.3.1.71]'],\n",
       "  'xref': ['EC:1.3.1.71',\n",
       "   'KEGG_REACTION:R05641',\n",
       "   'MetaCyc:1.3.1.71-RXN',\n",
       "   'RHEA:18501'],\n",
       "  'is_a': ['GO:0016628 ! oxidoreductase activity, acting on the CH-CH group of donors, NAD or NADP as acceptor']},\n",
       " 'GO:0000247': {'name': ['C-8 sterol isomerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction which results in unsaturation at C-7 in the B ring of sterols.\" [MetaCyc:RXN3O-203, PMID:8988026]'],\n",
       "  'synonym': ['\"delta-8-delta-7 sterol isomerase activity\" EXACT []'],\n",
       "  'is_a': ['GO:0016863 ! intramolecular oxidoreductase activity, transposing C=C bonds']},\n",
       " 'GO:0000248': {'name': ['C-5 sterol desaturase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"sterol-C5-desaturase activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-195664 \"Cholesta-7,24-dien-3beta-ol is desaturated to form cholesta-5,7,24-trien-3beta-ol\"',\n",
       "   'Reactome:R-HSA-6807053 \"SC5D desaturates LTHSOL to 7-dehydroCHOL\"'],\n",
       "  'is_a': ['GO:0016717 ! oxidoreductase activity, acting on paired donors, with oxidation of a pair of donors resulting in the reduction of molecular oxygen to two molecules of water',\n",
       "   'GO:0070704 ! sterol desaturase activity']},\n",
       " 'GO:0000249': {'name': ['C-22 sterol desaturase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the formation of the C-22(23) double bond in the sterol side chain. An example reaction'],\n",
       "  'is_a': ['GO:0016717 ! oxidoreductase activity, acting on paired donors, with oxidation of a pair of donors resulting in the reduction of molecular oxygen to two molecules of water',\n",
       "   'GO:0070704 ! sterol desaturase activity']},\n",
       " 'GO:0000250': {'name': ['lanosterol synthase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"(S)-2,3-epoxysqualene mutase (cyclizing, lanosterol-forming)\" RELATED [EC:5.4.99.7]',\n",
       "   '\"2,3-epoxysqualene lanosterol cyclase activity\" RELATED [EC:5.4.99.7]',\n",
       "   '\"2,3-epoxysqualene--lanosterol cyclase activity\" RELATED [EC:5.4.99.7]',\n",
       "   '\"2,3-epoxysqualene-lanosterol cyclase activity\" EXACT []',\n",
       "   '\"2,3-oxidosqualene sterol cyclase activity\" BROAD [EC:5.4.99.7]',\n",
       "   '\"2,3-oxidosqualene-lanosterol cyclase activity\" RELATED [EC:5.4.99.7]',\n",
       "   '\"lanosterol 2,3-oxidosqualene cyclase activity\" RELATED [EC:5.4.99.7]',\n",
       "   '\"OSC\" EXACT []',\n",
       "   '\"oxidosqualene--lanosterol cyclase activity\" RELATED [EC:5.4.99.7]',\n",
       "   '\"oxidosqualene-lanosterol cyclase activity\" EXACT []',\n",
       "   '\"oxidosqualene:lanosterol cyclase activity\" EXACT [PMID:18033581]',\n",
       "   '\"squalene 2,3-epoxide:lanosterol cyclase activity\" RELATED [EC:5.4.99.7]',\n",
       "   '\"squalene epoxidase-cyclase activity\" BROAD [EC:5.4.99.7]',\n",
       "   '\"squalene-2,3-oxide-lanosterol cyclase activity\" RELATED [EC:5.4.99.7]'],\n",
       "  'xref': ['EC:5.4.99.7',\n",
       "   'KEGG_REACTION:R03199',\n",
       "   'MetaCyc:LANOSTEROL-SYNTHASE-RXN',\n",
       "   'Reactome:R-HSA-191366 \"Squalene 2,3-epoxide cyclizes, forming lanosterol\"',\n",
       "   'RHEA:14621'],\n",
       "  'is_a': ['GO:0031559 ! oxidosqualene cyclase activity']},\n",
       " 'GO:0000252': {'name': ['C-3 sterol dehydrogenase (C-4 sterol decarboxylase) activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'comment': ['Note that zymosterol is cholesta-8,24-dien-3-ol.'],\n",
       "  'synonym': ['\"3beta-hydroxy-4alpha-methylcholestenecarboxylate 3-dehydrogenase (decarboxylating)\" RELATED [EC:1.1.1.170]',\n",
       "   '\"3beta-hydroxy-4beta-methyl-5alpha-cholest-7-ene-4alpha-carboxylate:NAD(P)+ 3-oxidoreductase (decarboxylating)\" EXACT []',\n",
       "   '\"3beta-hydroxy-4beta-methylcholestenecarboxylate 3-dehydrogenase (decarboxylating)\" RELATED []',\n",
       "   '\"3beta-hydroxy-4beta-methylcholestenoate dehydrogenase activity\" RELATED []',\n",
       "   '\"C-3 sterol dehydrogenase (C-4 decarboxylase) activity\" RELATED []',\n",
       "   '\"C-3 sterol dehydrogenase activity\" RELATED []',\n",
       "   '\"sterol 4alpha-carboxylic decarboxylase activity\" RELATED []',\n",
       "   '\"sterol-4-carboxylate 3-dehydrogenase (decarboxylating) activity\" RELATED []',\n",
       "   '\"sterol-4alpha-carboxylate 3-dehydrogenase (decarboxylating)\" RELATED []'],\n",
       "  'xref': ['EC:1.1.1.170',\n",
       "   'Reactome:R-HSA-194642 \"4-methyl,4-carboxycholesta-8(9),24-dien-3beta-ol is decarboxylated and oxidized to form 4-methylcholesta-8(9),24-dien-3-one\"',\n",
       "   'Reactome:R-HSA-194718 \"4-carboxycholesta-8(9),24-dien-3beta-ol is decarboxylated and oxidized to form cholesta-8(9),24-dien-3-one (zymosterone)\"'],\n",
       "  'is_a': ['GO:0016616 ! oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor']},\n",
       " 'GO:0000253': {'name': ['3-keto sterol reductase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0050576'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'comment': ['Note that zymosterol is cholesta-8,24-dien-3-ol.'],\n",
       "  'synonym': ['\"3-keto-steroid reductase activity\" EXACT []',\n",
       "   '\"3-KSR activity\" RELATED [EC:1.1.1.270]',\n",
       "   '\"3beta-hydroxy-steroid:NADP+ 3-oxidoreductase\" RELATED [EC:1.1.1.270]'],\n",
       "  'xref': ['EC:1.1.1.270',\n",
       "   'MetaCyc:1.1.1.270-RXN',\n",
       "   'MetaCyc:RXN3O-4110',\n",
       "   'MetaCyc:RXN66-19',\n",
       "   'MetaCyc:RXN66-24',\n",
       "   'MetaCyc:RXN66-314',\n",
       "   'MetaCyc:RXN66-319',\n",
       "   'Reactome:R-HSA-194632 \"Zymosterone (cholesta-8(9),24-dien-3-one) is reduced to zymosterol (cholesta-8(9),24-dien-3beta-ol)\"',\n",
       "   'Reactome:R-HSA-194689 \"4-methylcholesta-8(9),24-dien-3-one is reduced to 4-methylcholesta-8(9),24-dien-3beta-ol\"',\n",
       "   'RHEA:18409'],\n",
       "  'is_a': ['GO:0016616 ! oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor']},\n",
       " 'GO:0000254': {'name': ['C-4 methylsterol oxidase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'comment': ['Note that zymosterol is cholesta-8,24-dien-3-ol.'],\n",
       "  'synonym': ['\"4-methylsterol oxidase activity\" EXACT [EC:1.14.18.9]',\n",
       "   '\"methylsterol hydroxylase activity\" RELATED [EC:1.14.18.9]',\n",
       "   '\"methylsterol monooxygenase activity\" RELATED [EC:1.14.18.9]'],\n",
       "  'xref': ['EC:1.14.18.9',\n",
       "   'MetaCyc:1.14.13.72-RXN',\n",
       "   'Reactome:R-HSA-194641 \"4,4-dimethylcholesta-8(9),24-dien-3beta-ol is oxidized to 4-methyl,4-carboxycholesta-8(9),24-dien-3beta-ol\"',\n",
       "   'Reactome:R-HSA-194669 \"4-methylcholesta-8(9),24-dien-3beta-ol is oxidized to 4-carboxycholesta-8(9),24-dien-3beta-ol\"',\n",
       "   'RHEA:55220'],\n",
       "  'is_a': ['GO:0016716 ! oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen, another compound as one donor, and incorporation of one atom of oxygen']},\n",
       " 'GO:0000255': {'name': ['allantoin metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving allantoin, (2,5-dioxo-4-imidazolidinyl)urea, an intermediate or end product of purine catabolism.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'synonym': ['\"allantoin metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0043603 ! amide metabolic process',\n",
       "   'GO:0046483 ! heterocycle metabolic process',\n",
       "   'GO:1901360 ! organic cyclic compound metabolic process',\n",
       "   'GO:1901564 ! organonitrogen compound metabolic process']},\n",
       " 'GO:0000256': {'name': ['allantoin catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of allantoin, (2,5-dioxo-4-imidazolidinyl)urea.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'synonym': ['\"allantoin breakdown\" EXACT []',\n",
       "   '\"allantoin catabolism\" EXACT []',\n",
       "   '\"allantoin degradation\" EXACT []'],\n",
       "  'is_a': ['GO:0000255 ! allantoin metabolic process',\n",
       "   'GO:0043605 ! amide catabolic process',\n",
       "   'GO:0046700 ! heterocycle catabolic process',\n",
       "   'GO:1901361 ! organic cyclic compound catabolic process',\n",
       "   'GO:1901565 ! organonitrogen compound catabolic process']},\n",
       " 'GO:0000257': {'name': ['nitrilase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"acetonitrilase activity\" RELATED [EC:3.5.5.1]',\n",
       "   '\"benzonitrilase activity\" RELATED [EC:3.5.5.1]',\n",
       "   '\"nitrile aminohydrolase activity\" RELATED [EC:3.5.5.1]'],\n",
       "  'xref': ['EC:3.5.5.1',\n",
       "   'KEGG_REACTION:R00540',\n",
       "   'MetaCyc:3.5.5.1-RXN',\n",
       "   'RHEA:21724',\n",
       "   'UM-BBD_enzymeID:e0283'],\n",
       "  'is_a': ['GO:0016815 ! hydrolase activity, acting on carbon-nitrogen (but not peptide) bonds, in nitriles']},\n",
       " 'GO:0000258': {'name': ['obsolete isoleucine/valine:sodium symporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the reaction'],\n",
       "  'comment': ['This term was made obsolete because it represents a multifunctional gene product.'],\n",
       "  'synonym': ['\"isoleucine/valine:sodium symporter activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005283', 'GO:0005304', 'GO:0015188']},\n",
       " 'GO:0000259': {'name': ['obsolete intracellular nucleoside transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Enables the directed movement of a nucleoside, a nucleobase linked to either beta-D-ribofuranose (ribonucleoside) or 2-deoxy-beta-D-ribofuranose (a deoxyribonucleotide) within a cell.\" [GOC:ai]'],\n",
       "  'comment': ['This term was made obsolete because it contains component and function information.'],\n",
       "  'synonym': ['\"intracellular nucleoside transmembrane transporter activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005337', 'GO:0005622']},\n",
       " 'GO:0000260': {'name': ['obsolete hydrogen-translocating V-type ATPase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the reaction'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"hydrogen-translocating V-type ATPase activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0046961']},\n",
       " 'GO:0000261': {'name': ['obsolete sodium-translocating V-type ATPase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the reaction'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"sodium-translocating V-type ATPase activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0046962']},\n",
       " 'GO:0000262': {'name': ['mitochondrial chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A chromosome found in the mitochondrion of a eukaryotic cell.\" [GOC:mah]'],\n",
       "  'synonym': ['\"mitochondrial DNA\" NARROW []',\n",
       "   '\"mitochondrial genome\" RELATED []',\n",
       "   '\"mtDNA\" NARROW []'],\n",
       "  'xref': ['NIF_Subcellular:sao1186327184'],\n",
       "  'is_a': ['GO:0005694 ! chromosome'],\n",
       "  'relationship': ['part_of GO:0042645 ! mitochondrial nucleoid']},\n",
       " 'GO:0000263': {'name': ['obsolete heterotrimeric G-protein GTPase, alpha-subunit'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Subunit of a heterotrimeric G-protein GTPase that contains the guanine nucleotide binding site and possesses GTPase activity.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'comment': ['This term was made obsolete because it does not represent a molecular function.'],\n",
       "  'synonym': ['\"heterotrimeric G-protein GTPase, alpha-subunit\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0003924', 'GO:0005834']},\n",
       " 'GO:0000264': {'name': ['obsolete heterotrimeric G-protein GTPase, beta-subunit'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Subunit of a heterotrimeric G-protein GTPase; associates tightly with the gamma subunit.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'comment': ['This term was made obsolete because it does not represent a molecular function.'],\n",
       "  'synonym': ['\"heterotrimeric G-protein GTPase, beta-subunit\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0003924', 'GO:0005834']},\n",
       " 'GO:0000265': {'name': ['obsolete heterotrimeric G-protein GTPase, gamma-subunit'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Smallest subunit of a heterotrimeric G-protein GTPase; associates tightly with the beta subunit.\" [GOC:mah, ISBN:0198547684]'],\n",
       "  'comment': ['This term was made obsolete because it does not represent a molecular function.'],\n",
       "  'synonym': ['\"heterotrimeric G-protein GTPase, gamma-subunit\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0003924', 'GO:0005834']},\n",
       " 'GO:0000266': {'name': ['mitochondrial fission'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The division of a mitochondrion within a cell to form two or more separate mitochondrial compartments.\" [PMID:11038192]'],\n",
       "  'synonym': ['\"mitochondrial division\" EXACT []',\n",
       "   '\"mitochondrial proliferation\" RELATED []'],\n",
       "  'is_a': ['GO:0007005 ! mitochondrion organization',\n",
       "   'GO:0048285 ! organelle fission']},\n",
       " 'GO:0000267': {'name': ['obsolete cell fraction'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A generic term for parts of cells prepared by disruptive biochemical techniques.\" [GOC:ma]'],\n",
       "  'comment': ['This term was made obsolete because it refers to a cell fractionation experimental result and not a bona fide cellular component.'],\n",
       "  'synonym': ['\"cell fraction\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005575']},\n",
       " 'GO:0000268': {'name': ['peroxisome targeting sequence binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0005051'],\n",
       "  'def': ['\"Binding to a peroxisomal targeting sequence, a sequence of amino acids within a protein that acts as a signal for the localization of a protein into the peroxisome.\" [GOC:mah, ISBN:0879693568]'],\n",
       "  'synonym': ['\"peroxisome targeting signal receptor\" NARROW []',\n",
       "   '\"PTS binding\" EXACT []',\n",
       "   '\"PTS receptor\" NARROW []'],\n",
       "  'is_a': ['GO:0005048 ! signal sequence binding']},\n",
       " 'GO:0000269': {'name': ['toxin export channel activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the energy independent passage of toxins, sized less than 1000 Da, across a membrane towards the outside of the cell. The transmembrane portions of porins consist exclusively of beta-strands which form a beta-barrel. They are found in the outer membranes of Gram-negative bacteria, mitochondria, plastids and possibly acid-fast Gram-positive bacteria.\" [GOC:mtg_transport]'],\n",
       "  'is_a': ['GO:0015288 ! porin activity',\n",
       "   'GO:0019534 ! toxin transmembrane transporter activity']},\n",
       " 'GO:0000270': {'name': ['peptidoglycan metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0009284'],\n",
       "  'def': ['\"The chemical reactions and pathways involving peptidoglycans, any of a class of glycoconjugates found only in bacterial cell walls and consisting of strands of glycosaminoglycan cross-linked by oligopeptides to form a huge and rigid network.\" [ISBN:0198506732, PMID:33139480]'],\n",
       "  'synonym': ['\"murein metabolic process\" EXACT []',\n",
       "   '\"murein metabolism\" EXACT []',\n",
       "   '\"peptidoglycan metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0030203 ! glycosaminoglycan metabolic process']},\n",
       " 'GO:0000271': {'name': ['polysaccharide biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of a polysaccharide, a polymer of many (typically more than 10) monosaccharide residues linked glycosidically.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"glycan biosynthesis\" EXACT []',\n",
       "   '\"glycan biosynthetic process\" EXACT []',\n",
       "   '\"polysaccharide anabolism\" EXACT []',\n",
       "   '\"polysaccharide biosynthesis\" EXACT []',\n",
       "   '\"polysaccharide formation\" EXACT []',\n",
       "   '\"polysaccharide synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0005976 ! polysaccharide metabolic process',\n",
       "   'GO:0009059 ! macromolecule biosynthetic process',\n",
       "   'GO:0016051 ! carbohydrate biosynthetic process']},\n",
       " 'GO:0000272': {'name': ['polysaccharide catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0044244'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of a polysaccharide, a polymer of many (typically more than 10) monosaccharide residues linked glycosidically.\" [PMID:33139480]'],\n",
       "  'synonym': ['\"multicellular organismal polysaccharide catabolic process\" NARROW []',\n",
       "   '\"polysaccharide breakdown\" EXACT []',\n",
       "   '\"polysaccharide catabolism\" EXACT []',\n",
       "   '\"polysaccharide degradation\" EXACT []'],\n",
       "  'is_a': ['GO:0005976 ! polysaccharide metabolic process',\n",
       "   'GO:0009057 ! macromolecule catabolic process',\n",
       "   'GO:0016052 ! carbohydrate catabolic process']},\n",
       " 'GO:0000274': {'name': ['mitochondrial proton-transporting ATP synthase, stator stalk'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"One of two stalks that connect the catalytic core of the hydrogen-transporting ATP synthase to the mitochondrial membrane-associated F0 proteins; is thought to prevent futile rotation of the catalytic core.\" [GOC:mtg_sensu, PMID:10838056]'],\n",
       "  'synonym': ['\"mitochondrial proton-transporting ATP synthase, peripheral stalk\" EXACT []'],\n",
       "  'is_a': ['GO:0045265 ! proton-transporting ATP synthase, stator stalk',\n",
       "   'GO:0098800 ! inner mitochondrial membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0000276 ! mitochondrial proton-transporting ATP synthase complex, coupling factor F(o)']},\n",
       " 'GO:0000275': {'name': ['mitochondrial proton-transporting ATP synthase complex, catalytic sector F(1)'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The catalytic sector of the mitochondrial hydrogen-transporting ATP synthase; it comprises the catalytic core and central stalk, and is peripherally associated with the mitochondrial inner membrane when the entire ATP synthase is assembled.\" [GOC:mtg_sensu, PMID:10838056]'],\n",
       "  'comment': [\"See also the cellular component term 'mitochondrial inner membrane ; GO:0005743'.\"],\n",
       "  'synonym': ['\"hydrogen-transporting ATP synthase, F1 sector\" BROAD []',\n",
       "   '\"proton-transporting ATP synthase complex, catalytic core F(1)\" BROAD []'],\n",
       "  'is_a': ['GO:0045261 ! proton-transporting ATP synthase complex, catalytic core F(1)',\n",
       "   'GO:0098800 ! inner mitochondrial membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0005753 ! mitochondrial proton-transporting ATP synthase complex']},\n",
       " 'GO:0000276': {'name': ['mitochondrial proton-transporting ATP synthase complex, coupling factor F(o)'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"All non-F1 subunits of the mitochondrial hydrogen-transporting ATP synthase, including integral and peripheral mitochondrial inner membrane proteins.\" [GOC:mtg_sensu, PMID:10838056]'],\n",
       "  'is_a': ['GO:0045263 ! proton-transporting ATP synthase complex, coupling factor F(o)',\n",
       "   'GO:0098800 ! inner mitochondrial membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0005753 ! mitochondrial proton-transporting ATP synthase complex']},\n",
       " 'GO:0000277': {'name': ['[cytochrome c]-lysine N-methyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"cytochrome c (lysine) methyltransferase activity\" RELATED [EC:2.1.1.59]',\n",
       "   '\"cytochrome c methyltransferase activity\" RELATED [EC:2.1.1.59]',\n",
       "   '\"cytochrome c-lysine N-methyltransferase activity\" RELATED [EC:2.1.1.59]',\n",
       "   '\"cytochrome c-specific protein methylase III activity\" RELATED [EC:2.1.1.59]',\n",
       "   '\"cytochrome c-specific protein-lysine methyltransferase activity\" RELATED [EC:2.1.1.59]',\n",
       "   '\"S-adenosyl-L-methionine:cytochrome c-L-lysine 6-N-methyltransferase activity\" RELATED [EC:2.1.1.59]',\n",
       "   '\"S-adenosyl-L-methionine:cytochrome c-L-lysine N6-methyltransferase activity\" RELATED [EC:2.1.1.59]'],\n",
       "  'xref': ['EC:2.1.1.59', 'MetaCyc:2.1.1.59-RXN', 'RHEA:24312'],\n",
       "  'is_a': ['GO:0016279 ! protein-lysine N-methyltransferase activity']},\n",
       " 'GO:0000278': {'name': ['mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007067'],\n",
       "  'def': ['\"Progression through the phases of the mitotic cell cycle, the most common eukaryotic cell cycle, which canonically comprises four successive phases called G1, S, G2, and M and includes replication of the genome and the subsequent segregation of chromosomes into daughter cells. In some variant cell cycles nuclear replication or nuclear division may not be followed by cell division, or G1 and G2 phases may be absent.\" [GOC:mah, ISBN:0815316194, Reactome:69278]'],\n",
       "  'comment': [\"Note that this term should not be confused with 'GO:0140014 ; mitotic nuclear division'. 'GO:0000278 ; mitotic cell cycle represents the entire mitotic cell cycle, while 'GO:0140014 ; mitotic nuclear division' specifically represents the actual nuclear division step of the mitotic cell cycle.\"],\n",
       "  'subset': ['goslim_chembl',\n",
       "   'goslim_drosophila',\n",
       "   'goslim_generic',\n",
       "   'goslim_yeast'],\n",
       "  'synonym': ['\"mitosis\" RELATED []'],\n",
       "  'xref': ['Wikipedia:Mitosis'],\n",
       "  'is_a': ['GO:0007049 ! cell cycle']},\n",
       " 'GO:0000279': {'name': ['M phase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A cell cycle phase during which nuclear division occurs, and which is comprises the phases'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. If you are trying to make an annotation to x phase, it is likely that the correct annotation is 'regulation of x/y phase transition' or to a process which occurs during the reported phase (i.e mitotic DNA replication for mitotic S-phase). To capture the phase when a specific location or process is observed, the phase term can be used in an annotation extension (PMID:24885854) applied to a cellular component term (with the relation exists_during) or a biological process term (with the relation happens_during).\"],\n",
       "  'subset': ['gocheck_do_not_annotate'],\n",
       "  'synonym': ['\"M-phase\" EXACT []'],\n",
       "  'xref': ['Wikipedia:M_phase'],\n",
       "  'is_a': ['GO:0022403 ! cell cycle phase']},\n",
       " 'GO:0000280': {'name': ['nuclear division'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The division of a cell nucleus into two nuclei, with DNA and other nuclear contents distributed between the daughter nuclei.\" [GOC:mah]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"karyokinesis\" RELATED []'],\n",
       "  'is_a': ['GO:0048285 ! organelle fission']},\n",
       " 'GO:0000281': {'name': ['mitotic cytokinesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A cell cycle process that results in the division of the cytoplasm of a cell after mitosis, resulting in the separation of the original cell into two daughter cells.\" [GOC:mtg_cell_cycle]'],\n",
       "  'subset': ['goslim_pombe'],\n",
       "  'synonym': ['\"cytokinesis after mitosis\" EXACT []'],\n",
       "  'is_a': ['GO:0061640 ! cytoskeleton-dependent cytokinesis',\n",
       "   'GO:1903047 ! mitotic cell cycle process']},\n",
       " 'GO:0000282': {'name': ['cellular bud site selection'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The specification of the site where a daughter cell will form, in organisms that reproduce by budding. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:mah]'],\n",
       "  'comment': [\"Note that this term was split from 'bud site selection/establishment of cell polarity (sensu Saccharomyces) ; GO:0007115' (a sibling term, 'establishment of cell polarity (sensu Saccharomyces) ; GO:0000283', was created but has since been merged with 'establishment of cell polarity' ; GO:0030010).\"],\n",
       "  'synonym': ['\"bud site selection/establishment of cell polarity\" BROAD []'],\n",
       "  'is_a': ['GO:0030010 ! establishment of cell polarity',\n",
       "   'GO:1903047 ! mitotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0000281 ! mitotic cytokinesis']},\n",
       " 'GO:0000284': {'name': ['obsolete shmoo orientation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:elh]'],\n",
       "  'comment': ['This term was made obsolete because it is an organism specific term. The entire branch of the ontology was reorganized to be more generally applicable.'],\n",
       "  'synonym': ['\"shmoo orientation\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000753']},\n",
       " 'GO:0000285': {'name': ['1-phosphatidylinositol-3-phosphate 5-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"ATP:1-phosphatidyl-1D-myo-inositol-3-phosphate 5-phosphotransferase activity\" RELATED [EC:2.7.1.150]',\n",
       "   '\"phosphatidylinositol 3-phosphate 5-kinase activity\" RELATED [EC:2.7.1.150]',\n",
       "   '\"phosphatidylinositol-3-phosphate 5-kinase activity\" EXACT []',\n",
       "   '\"type III PIP kinase activity\" RELATED [EC:2.7.1.150]'],\n",
       "  'xref': ['EC:2.7.1.150',\n",
       "   'KEGG_REACTION:R05802',\n",
       "   'MetaCyc:2.7.1.150-RXN',\n",
       "   'Reactome:R-HSA-1675910 \"PI3P is phosphorylated to PI(3,5)P2 by PIKFYVE at the late endosome membrane\"',\n",
       "   'Reactome:R-HSA-1675921 \"PI3P is phosphorylated to PI(3,5)P2 by PIKFYVE at the Golgi membrane\"',\n",
       "   'Reactome:R-HSA-1676134 \"PI3P is phosphorylated to PI(3,5)P2 by PIP5K1A/B at the plasma membrane\"',\n",
       "   'Reactome:R-HSA-1676168 \"PI3P is phosphorylated to PI(3,5)P2 by PIKFYVE at the early endosome membrane\"',\n",
       "   'RHEA:13609'],\n",
       "  'is_a': ['GO:0016307 ! phosphatidylinositol phosphate kinase activity']},\n",
       " 'GO:0000286': {'name': ['alanine dehydrogenase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"AlaDH\" RELATED [EC:1.4.1.1]',\n",
       "   '\"alanine oxidoreductase activity\" RELATED [EC:1.4.1.1]',\n",
       "   '\"alpha-alanine dehydrogenase activity\" RELATED [EC:1.4.1.1]',\n",
       "   '\"L-alanine dehydrogenase activity\" RELATED [EC:1.4.1.1]',\n",
       "   '\"L-alanine:NAD+ oxidoreductase (deaminating)\" RELATED [EC:1.4.1.1]',\n",
       "   '\"NAD-dependent alanine dehydrogenase activity\" RELATED [EC:1.4.1.1]',\n",
       "   '\"NAD-linked alanine dehydrogenase activity\" RELATED [EC:1.4.1.1]',\n",
       "   '\"NADH-dependent alanine dehydrogenase activity\" RELATED [EC:1.4.1.1]'],\n",
       "  'xref': ['EC:1.4.1.1', 'MetaCyc:ALANINE-DEHYDROGENASE-RXN', 'RHEA:18405'],\n",
       "  'is_a': ['GO:0016638 ! oxidoreductase activity, acting on the CH-NH2 group of donors']},\n",
       " 'GO:0000287': {'name': ['magnesium ion binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a magnesium (Mg) ion.\" [GOC:ai]'],\n",
       "  'synonym': ['\"magnesium binding\" EXACT []', '\"Mg binding\" EXACT []'],\n",
       "  'is_a': ['GO:0046872 ! metal ion binding']},\n",
       " 'GO:0000288': {'name': ['nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A major pathway of degradation of nuclear-transcribed mRNAs that proceeds through a series of ordered steps that includes poly(A) tail shortening and that can regulate mRNA stability.\" [GOC:jp, GOC:krc]'],\n",
       "  'synonym': ['\"deadenylation-dependent mRNA decay\" EXACT []',\n",
       "   '\"mRNA breakdown, deadenylation-dependent decay\" EXACT []',\n",
       "   '\"mRNA catabolic process, deadenylation-dependent\" EXACT []',\n",
       "   '\"mRNA catabolic process, deadenylylation-dependent\" EXACT []',\n",
       "   '\"mRNA catabolism, deadenylation-dependent\" EXACT []',\n",
       "   '\"mRNA catabolism, deadenylylation-dependent\" EXACT []',\n",
       "   '\"mRNA degradation, deadenylation-dependent decay\" EXACT []',\n",
       "   '\"nuclear mRNA catabolic process, deadenylation-dependent decay\" EXACT []'],\n",
       "  'is_a': ['GO:0000956 ! nuclear-transcribed mRNA catabolic process',\n",
       "   'GO:0061157 ! mRNA destabilization']},\n",
       " 'GO:0000289': {'name': ['nuclear-transcribed mRNA poly(A) tail shortening'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Shortening of the poly(A) tail of a nuclear-transcribed mRNA from full length to an oligo(A) length.\" [GOC:krc]'],\n",
       "  'synonym': ['\"3\\' to 5\\' mRNA deadenylation\" RELATED []',\n",
       "   '\"mRNA deadenylation\" RELATED []',\n",
       "   '\"nuclear mRNA poly(A) tail shortening\" RELATED []'],\n",
       "  'is_a': ['GO:0000956 ! nuclear-transcribed mRNA catabolic process'],\n",
       "  'relationship': ['part_of GO:0000288 ! nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay']},\n",
       " 'GO:0000290': {'name': ['deadenylation-dependent decapping of nuclear-transcribed mRNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Cleavage of the 5\\'-cap of a nuclear mRNA triggered by shortening of the poly(A) tail to below a minimum functional length.\" [GOC:krc]'],\n",
       "  'synonym': ['\"deadenylation-dependent decapping of nuclear mRNA\" EXACT []',\n",
       "   '\"deadenylylation-dependent decapping\" EXACT []'],\n",
       "  'is_a': ['GO:0000956 ! nuclear-transcribed mRNA catabolic process',\n",
       "   'GO:0110156 ! methylguanosine-cap decapping'],\n",
       "  'relationship': ['part_of GO:0000288 ! nuclear-transcribed mRNA catabolic process, deadenylation-dependent decay']},\n",
       " 'GO:0000291': {'name': ['nuclear-transcribed mRNA catabolic process, exonucleolytic'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of the transcript body of a nuclear-transcribed mRNA that occurs when the ends are not protected by the 5\\'-cap or the 3\\'-poly(A) tail.\" [GOC:krc]'],\n",
       "  'synonym': ['\"exonucleolytic degradation of mRNA\" EXACT []',\n",
       "   '\"mRNA breakdown, exonucleolytic\" EXACT []',\n",
       "   '\"mRNA degradation, exonucleolytic\" EXACT []',\n",
       "   '\"nuclear mRNA catabolic process, exonucleolytic\" EXACT []'],\n",
       "  'is_a': ['GO:0000956 ! nuclear-transcribed mRNA catabolic process']},\n",
       " 'GO:0000292': {'name': ['RNA fragment catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0030452'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of a fragment of RNA, such as excised introns or sequences removed from ribosomal RNA during processing.\" [GOC:mah]'],\n",
       "  'synonym': ['\"group I intron catabolic process\" NARROW []',\n",
       "   '\"RNA fragment breakdown\" EXACT []',\n",
       "   '\"RNA fragment catabolism\" EXACT [GOC:mah]',\n",
       "   '\"RNA fragment degradation\" EXACT []'],\n",
       "  'is_a': ['GO:0006401 ! RNA catabolic process']},\n",
       " 'GO:0000293': {'name': ['ferric-chelate reductase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"ferric chelate reductase activity\" EXACT []',\n",
       "   '\"iron chelate reductase activity\" EXACT []'],\n",
       "  'is_a': ['GO:0016722 ! oxidoreductase activity, acting on metal ions']},\n",
       " 'GO:0000294': {'name': ['nuclear-transcribed mRNA catabolic process, RNase MRP-dependent'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A minor MRP-dependent nuclear-transcribed mRNA degradation pathway that begins with an endonucleolytic cleavage to generate unprotected ends.\" [PMID:14729943]'],\n",
       "  'synonym': ['\"nuclear mRNA catabolic process, endonucleolytic cleavage-dependent decay\" EXACT []',\n",
       "   '\"nuclear-transcribed mRNA catabolic process, endonucleolytic cleavage-dependent decay\" RELATED []'],\n",
       "  'is_a': ['GO:0000956 ! nuclear-transcribed mRNA catabolic process']},\n",
       " 'GO:0000295': {'name': ['adenine nucleotide transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of adenine nucleotides (AMP, ADP, and ATP) from one side of a membrane to the other.\" [PMID:11566870]'],\n",
       "  'xref': ['Reactome:R-HSA-389652 \"PMP34-mediated exchange of cytosolic ATP for peroxisomal AMP\"'],\n",
       "  'is_a': ['GO:0015216 ! purine nucleotide transmembrane transporter activity']},\n",
       " 'GO:0000296': {'name': ['spermine transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of spermine, N,N-bis(3-aminopropyl)-1,4-diaminobutane, a polyamine formed by the transfer of a propylamine group from decarboxylated S-adenosylmethionine to spermidine, into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore.\" [GOC:krc, ISBN:0198506732]'],\n",
       "  'is_a': ['GO:0015846 ! polyamine transport']},\n",
       " 'GO:0000297': {'name': ['spermine transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of spermine from one side of a membrane to the other. Spermine is a polybasic amine found in human sperm, in ribosomes and in some viruses, which is involved in nucleic acid packaging. Synthesis is regulated by ornithine decarboxylase which plays a key role in control of DNA replication.\" [GOC:ai]'],\n",
       "  'is_a': ['GO:0015203 ! polyamine transmembrane transporter activity']},\n",
       " 'GO:0000298': {'name': ['endopolyphosphatase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"metaphosphatase activity\" BROAD [EC:3.6.1.10]',\n",
       "   '\"polymetaphosphatase activity\" RELATED [EC:3.6.1.10]',\n",
       "   '\"polyphosphatase activity\" BROAD [EC:3.6.1.10]',\n",
       "   '\"polyphosphate depolymerase activity\" BROAD [EC:3.6.1.10]',\n",
       "   '\"polyphosphate polyphosphohydrolase activity\" RELATED [EC:3.6.1.10]'],\n",
       "  'xref': ['EC:3.6.1.10', 'MetaCyc:ENDOPOLYPHOSPHATASE-RXN', 'RHEA:22452'],\n",
       "  'is_a': ['GO:0016462 ! pyrophosphatase activity']},\n",
       " 'GO:0000299': {'name': ['obsolete integral to membrane of membrane fraction'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. Integral to that fraction of cells, prepared by disruptive biochemical methods, that includes the plasma and other membranes; require detergents, such as Triton X-100, to be released from membranes.\" [PMID:10512869]'],\n",
       "  'comment': ['This term was made obsolete because it refers to a cell fractionation experimental result and not a bona fide cellular component.'],\n",
       "  'synonym': ['\"integral to membrane of membrane fraction\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000300': {'name': ['obsolete peripheral to membrane of membrane fraction'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. Peripheral to that fraction of cells, prepared by disruptive biochemical methods, that includes the plasma and other membranes; can be extracted from membrane fraction with high concentrations of salt or high pH.\" [PMID:10512869]'],\n",
       "  'comment': ['This term was made obsolete because it refers to a cell fractionation experimental result and not a bona fide cellular component.'],\n",
       "  'synonym': ['\"peripheral to membrane of membrane fraction\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0019898']},\n",
       " 'GO:0000301': {'name': ['retrograde transport, vesicle recycling within Golgi'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The retrograde movement of substances within the Golgi, mediated by COP I vesicles. Cis-Golgi vesicles are constantly moving forward through the Golgi stack by cisternal progression, eventually becoming trans-Golgi vesicles. They then selectively transport membrane and luminal proteins from the trans- to the medial-Golgi while leaving others behind in the trans-Golgi cisternae; similarly, they selectively move proteins from the medial- to the cis-Golgi.\" [ISBN:0716731363]'],\n",
       "  'synonym': ['\"retrograde (vesicle recycling within Golgi) transport\" EXACT []'],\n",
       "  'is_a': ['GO:0006891 ! intra-Golgi vesicle-mediated transport']},\n",
       " 'GO:0000302': {'name': ['response to reactive oxygen species'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a reactive oxygen species stimulus. Reactive oxygen species include singlet oxygen, superoxide, and oxygen free radicals.\" [GOC:krc]'],\n",
       "  'synonym': ['\"response to active oxygen species\" EXACT []',\n",
       "   '\"response to AOS\" EXACT []',\n",
       "   '\"response to reactive oxidative species\" EXACT []',\n",
       "   '\"response to reactive oxygen intermediate\" EXACT []',\n",
       "   '\"response to ROI\" EXACT []',\n",
       "   '\"response to ROS\" EXACT []'],\n",
       "  'is_a': ['GO:0006979 ! response to oxidative stress',\n",
       "   'GO:1901700 ! response to oxygen-containing compound']},\n",
       " 'GO:0000303': {'name': ['response to superoxide'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a superoxide stimulus. Superoxide is the anion, oxygen-, formed by addition of one electron to dioxygen (O2) or any compound containing the superoxide anion.\" [GOC:krc, ISBN:0198506732]'],\n",
       "  'is_a': ['GO:0000305 ! response to oxygen radical']},\n",
       " 'GO:0000304': {'name': ['response to singlet oxygen'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a singlet oxygen stimulus. Singlet oxygen is a dioxygen (O2) molecule in which two 2p electrons have similar spin. Singlet oxygen is more highly reactive than the form in which these electrons are of opposite spin, and it is produced in mutant chloroplasts lacking carotenoids and by leukocytes during metabolic burst.\" [GOC:krc, ISBN:0124325653, ISBN:0198506732]'],\n",
       "  'is_a': ['GO:0000302 ! response to reactive oxygen species',\n",
       "   'GO:0010035 ! response to inorganic substance']},\n",
       " 'GO:0000305': {'name': ['response to oxygen radical'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of an oxygen radical stimulus. An oxygen radical is any oxygen species that carries a free electron; examples include hydroxyl radicals and the superoxide anion.\" [GOC:krc, ISBN:0124325653]'],\n",
       "  'is_a': ['GO:0000302 ! response to reactive oxygen species',\n",
       "   'GO:0010035 ! response to inorganic substance']},\n",
       " 'GO:0000306': {'name': ['obsolete extrinsic component of vacuolar membrane'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. The component of a vacuolar membrane consisting of gene products and protein complexes that are loosely bound to one of its surfaces, but not integrated into the hydrophobic region.\" [GOC:dos, GOC:jl, GOC:mah]'],\n",
       "  'comment': ['This term was obsoleted because it represents protein topology, not a cellular component.'],\n",
       "  'synonym': ['\"extrinsic to vacuolar membrane\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005774']},\n",
       " 'GO:0000307': {'name': ['cyclin-dependent protein kinase holoenzyme complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Cyclin-dependent protein kinases (CDKs) are enzyme complexes that contain a kinase catalytic subunit associated with a regulatory cyclin partner.\" [GOC:krc, PMID:11602261]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"CDK holoenzyme\" EXACT []'],\n",
       "  'is_a': ['GO:1902554 ! serine/threonine protein kinase complex']},\n",
       " 'GO:0000308': {'name': ['cytoplasmic cyclin-dependent protein kinase holoenzyme complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Cyclin-dependent protein kinase (CDK) complex found in the cytoplasm.\" [GOC:krc]'],\n",
       "  'synonym': ['\"CDK holoenzyme\" BROAD []'],\n",
       "  'is_a': ['GO:0000307 ! cyclin-dependent protein kinase holoenzyme complex'],\n",
       "  'relationship': ['part_of GO:0005737 ! cytoplasm']},\n",
       " 'GO:0000309': {'name': ['nicotinamide-nucleotide adenylyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"adenosine triphosphate-nicotinamide mononucleotide transadenylase activity\" RELATED [EC:2.7.7.1]',\n",
       "   '\"ATP:nicotinamide-nucleotide adenylyltransferase activity\" EXACT []',\n",
       "   '\"ATP:NMN adenylyltransferase activity\" RELATED [EC:2.7.7.1]',\n",
       "   '\"diphosphopyridine nucleotide pyrophosphorylase activity\" RELATED [EC:2.7.7.1]',\n",
       "   '\"NAD(+) diphosphorylase activity\" NARROW [EC:2.7.7.1]',\n",
       "   '\"NAD(+) pyrophosphorylase activity\" NARROW [EC:2.7.7.1]',\n",
       "   '\"NAD+ diphosphorylase activity\" RELATED [EC:2.7.7.1]',\n",
       "   '\"NAD+ pyrophosphorylase activity\" RELATED [EC:2.7.7.1]',\n",
       "   '\"nicotinamide adenine dinucleotide pyrophosphorylase activity\" NARROW [EC:2.7.7.1]',\n",
       "   '\"nicotinamide mononucleotide adenylyltransferase activity\" NARROW [EC:2.7.7.1]',\n",
       "   '\"NMN adenylyltransferase activity\" NARROW [EC:2.7.7.1]',\n",
       "   '\"NMNAT activity\" NARROW [EC:2.7.7.1]'],\n",
       "  'xref': ['EC:2.7.7.1',\n",
       "   'MetaCyc:2.7.7.1-RXN',\n",
       "   'Reactome:R-HSA-8939959 \"NMNAT2 transfers an adenylyl group from ATP to NMN to yield NAD+\"',\n",
       "   'RHEA:21360'],\n",
       "  'is_a': ['GO:0070566 ! adenylyltransferase activity']},\n",
       " 'GO:0000310': {'name': ['xanthine phosphoribosyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0009043'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"5-phospho-alpha-D-ribose-1-diphosphate:xanthine phospho-D-ribosyltransferase activity\" RELATED [EC:2.4.2.22]',\n",
       "   '\"Xan phosphoribosyltransferase activity\" RELATED [EC:2.4.2.22]',\n",
       "   '\"xanthine-guanine phosphoribosyltransferase activity\" EXACT []',\n",
       "   '\"xanthosine 5\\'-phosphate pyrophosphorylase activity\" RELATED [EC:2.4.2.22]',\n",
       "   '\"xanthylate pyrophosphorylase activity\" RELATED [EC:2.4.2.22]',\n",
       "   '\"xanthylic pyrophosphorylase activity\" RELATED [EC:2.4.2.22]',\n",
       "   '\"XMP pyrophosphorylase activity\" RELATED [EC:2.4.2.22]',\n",
       "   '\"XMP:diphosphate 5-phospho-alpha-D-ribosyltransferase activity\" RELATED [EC:2.4.2.22]'],\n",
       "  'xref': ['EC:2.4.2.22', 'MetaCyc:XANPRIBOSYLTRAN-RXN', 'RHEA:10800'],\n",
       "  'is_a': ['GO:0106130 ! purine phosphoribosyltransferase activity']},\n",
       " 'GO:0000311': {'name': ['plastid large ribosomal subunit'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The larger of the two subunits of a plastid ribosome. Two sites on the ribosomal large subunit are involved in translation'],\n",
       "  'is_a': ['GO:0000315 ! organellar large ribosomal subunit'],\n",
       "  'relationship': ['part_of GO:0009547 ! plastid ribosome']},\n",
       " 'GO:0000312': {'name': ['plastid small ribosomal subunit'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The smaller of the two subunits of a plastid ribosome.\" [GOC:mcc]'],\n",
       "  'is_a': ['GO:0000314 ! organellar small ribosomal subunit'],\n",
       "  'relationship': ['part_of GO:0009547 ! plastid ribosome']},\n",
       " 'GO:0000313': {'name': ['organellar ribosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ribosome contained within a subcellular membrane-bounded organelle.\" [GOC:mah, GOC:mcc]'],\n",
       "  'is_a': ['GO:0005840 ! ribosome'],\n",
       "  'relationship': ['part_of GO:0043226 ! organelle']},\n",
       " 'GO:0000314': {'name': ['organellar small ribosomal subunit'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The smaller of the two subunits of an organellar ribosome.\" [GOC:mcc]'],\n",
       "  'is_a': ['GO:0015935 ! small ribosomal subunit'],\n",
       "  'relationship': ['part_of GO:0000313 ! organellar ribosome']},\n",
       " 'GO:0000315': {'name': ['organellar large ribosomal subunit'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The larger of the two subunits of an organellar ribosome. Two sites on the ribosomal large subunit are involved in translation'],\n",
       "  'is_a': ['GO:0015934 ! large ribosomal subunit'],\n",
       "  'relationship': ['part_of GO:0000313 ! organellar ribosome']},\n",
       " 'GO:0000316': {'name': ['sulfite transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of sulfite into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore.\" [GOC:krc]'],\n",
       "  'synonym': ['\"sulphite transport\" EXACT []'],\n",
       "  'is_a': ['GO:0015698 ! inorganic anion transport',\n",
       "   'GO:0072348 ! sulfur compound transport']},\n",
       " 'GO:0000319': {'name': ['sulfite transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of sulfite ions from one side of a membrane to the other.\" [GOC:as]'],\n",
       "  'synonym': ['\"sulphite transporter activity\" EXACT []'],\n",
       "  'is_a': ['GO:0015103 ! inorganic anion transmembrane transporter activity',\n",
       "   'GO:1901682 ! sulfur compound transmembrane transporter activity',\n",
       "   'GO:1901702 ! salt transmembrane transporter activity']},\n",
       " 'GO:0000320': {'name': ['re-entry into mitotic cell cycle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The resumption of the mitotic cell division cycle by cells that were in a quiescent or other non-dividing state.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0022402 ! cell cycle process']},\n",
       " 'GO:0000321': {'name': ['re-entry into mitotic cell cycle after pheromone arrest'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The resumption of the mitotic cell division cycle by pheromone-arrested cells that have not mated. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:krc, PMID:9927449]'],\n",
       "  'is_a': ['GO:0000320 ! re-entry into mitotic cell cycle',\n",
       "   'GO:0000754 ! adaptation of signaling pathway by response to pheromone involved in conjugation with cellular fusion']},\n",
       " 'GO:0000322': {'name': ['storage vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A vacuole that functions primarily in the storage of materials, including nutrients, pigments, waste products, and small molecules.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0005773 ! vacuole']},\n",
       " 'GO:0000323': {'name': ['lytic vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A vacuole that is maintained at an acidic pH and which contains degradative enzymes, including a wide variety of acid hydrolases.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0005773 ! vacuole']},\n",
       " 'GO:0000324': {'name': ['fungal-type vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A vacuole that has both lytic and storage functions. The fungal vacuole is a large, membrane-bounded organelle that functions as a reservoir for the storage of small molecules (including polyphosphate, amino acids, several divalent cations (e.g. calcium), other ions, and other small molecules) as well as being the primary compartment for degradation. It is an acidic compartment, containing an ensemble of acid hydrolases. At least in S. cerevisiae, there are indications that the morphology of the vacuole is variable and correlated with the cell cycle, with logarithmically growing cells having a multilobed, reticulated vacuole, while stationary phase cells contain a single large structure.\" [GOC:mah, GOC:mtg_sensu, ISBN:0879693649]'],\n",
       "  'synonym': ['\"vacuole, cell cycle-correlated morphology\" EXACT []'],\n",
       "  'is_a': ['GO:0000322 ! storage vacuole', 'GO:0000323 ! lytic vacuole']},\n",
       " 'GO:0000325': {'name': ['plant-type vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A closed structure that is completely surrounded by a unit membrane, contains liquid, and retains the same shape regardless of cell cycle phase. An example of this structure is found in Arabidopsis thaliana.\" [GOC:mtg_sensu, ISBN:0815316208]'],\n",
       "  'synonym': ['\"vacuole, cell cycle-independent morphology\" EXACT []'],\n",
       "  'is_a': ['GO:0005773 ! vacuole']},\n",
       " 'GO:0000326': {'name': ['protein storage vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A storage vacuole that contains a lytic vacuole; identified in plants.\" [PMID:11739409]'],\n",
       "  'is_a': ['GO:0000322 ! storage vacuole', 'GO:0000325 ! plant-type vacuole']},\n",
       " 'GO:0000327': {'name': ['lytic vacuole within protein storage vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A membrane-bounded compartment containing crystals of phytic acid and proteins characteristic of a lytic vacuole, found within a storage vacuole.\" [PMID:11739490]'],\n",
       "  'is_a': ['GO:0000323 ! lytic vacuole'],\n",
       "  'relationship': ['part_of GO:0000326 ! protein storage vacuole']},\n",
       " 'GO:0000328': {'name': ['fungal-type vacuole lumen'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The volume enclosed within the vacuolar membrane of a vacuole, the shape of which correlates with cell cycle phase. An example of this structure is found in Saccharomyces cerevisiae.\" [GOC:krc, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"lumen of vacuole with cell cycle-correlated morphology\" EXACT []'],\n",
       "  'is_a': ['GO:0005775 ! vacuolar lumen'],\n",
       "  'relationship': ['part_of GO:0000324 ! fungal-type vacuole']},\n",
       " 'GO:0000329': {'name': ['fungal-type vacuole membrane'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The lipid bilayer surrounding a vacuole, the shape of which correlates with cell cycle phase. The membrane separates its contents from the cytoplasm of the cell. An example of this structure is found in Saccharomyces cerevisiae.\" [GOC:krc, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"fungal-type vacuolar membrane\" RELATED []',\n",
       "   '\"membrane of vacuole with cell cycle-correlated morphology\" EXACT []'],\n",
       "  'is_a': ['GO:0098852 ! lytic vacuole membrane'],\n",
       "  'relationship': ['part_of GO:0000324 ! fungal-type vacuole']},\n",
       " 'GO:0000330': {'name': ['plant-type vacuole lumen'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The volume enclosed within the vacuolar membrane of a vacuole that retains the same shape regardless of cell cycle phase. An example of this is found in Arabidopsis thaliana.\" [GOC:krc, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"lumen of vacuole with cell cycle-independent morphology\" EXACT []'],\n",
       "  'is_a': ['GO:0005775 ! vacuolar lumen'],\n",
       "  'relationship': ['part_of GO:0000325 ! plant-type vacuole']},\n",
       " 'GO:0000331': {'name': ['contractile vacuole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A specialized vacuole of eukaryotic cells, especially Protozoa, that fills with water from the cytoplasm and then discharges this externally by the opening of contractile vacuole pores. One of its functions is osmoregulatory.\" [GOC:jl, PMID:10503189, PMID:23890380]'],\n",
       "  'synonym': ['\"central bladder\" RELATED [PMID:23890380]',\n",
       "   '\"central vacuole\" RELATED [PMID:23890380]'],\n",
       "  'xref': ['Wikipedia:Contractile_vacuole'],\n",
       "  'is_a': ['GO:0005773 ! vacuole', 'GO:0031410 ! cytoplasmic vesicle'],\n",
       "  'relationship': ['part_of GO:0062159 ! contractile vacuole complex']},\n",
       " 'GO:0000332': {'name': ['template for synthesis of G-rich strand of telomere DNA activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Provision of the template used by reverse transcriptase to synthesize the G-rich strand of telomeric DNA.\" [PMID:11812242, PMID:7958872]'],\n",
       "  'comment': ['Note that this term describes the activity of an RNA gene product that interacts with other nucleic acid molecules via base pairing; it should not be used to annotate proteins.'],\n",
       "  'synonym': ['\"telomerase RNA\" EXACT []', '\"telomerase, template\" EXACT []'],\n",
       "  'is_a': ['GO:0000497 ! DNA template activity'],\n",
       "  'relationship': ['part_of GO:0003720 ! telomerase activity']},\n",
       " 'GO:0000333': {'name': ['telomerase catalytic core complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The minimal catalytic core of telomerase is a ribonucleoprotein complex composed of a catalytic reverse transcriptase subunit and an RNA subunit that provides the template for telomeric DNA addition.\" [GOC:BHF-UCL, PMID:11884619, PMID:1808260]'],\n",
       "  'synonym': ['\"TERT-TERC complex\" RELATED [GOC:nc, PMID:19701182]'],\n",
       "  'is_a': ['GO:0005697 ! telomerase holoenzyme complex',\n",
       "   'GO:0061695 ! transferase complex, transferring phosphorus-containing groups']},\n",
       " 'GO:0000334': {'name': ['3-hydroxyanthranilate 3,4-dioxygenase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"3-hydroxyanthranilate oxygenase activity\" RELATED [EC:1.13.11.6]',\n",
       "   '\"3-hydroxyanthranilate:oxygen 3,4-oxidoreductase (decyclizing)\" RELATED [EC:1.13.11.6]',\n",
       "   '\"3-hydroxyanthranilic acid dioxygenase activity\" RELATED [EC:1.13.11.6]',\n",
       "   '\"3-hydroxyanthranilic acid oxygenase activity\" RELATED [EC:1.13.11.6]',\n",
       "   '\"3-hydroxyanthranilic oxygenase activity\" RELATED [EC:1.13.11.6]',\n",
       "   '\"3HAO\" RELATED [EC:1.13.11.6]'],\n",
       "  'xref': ['EC:1.13.11.6',\n",
       "   'KEGG_REACTION:R02665',\n",
       "   'MetaCyc:1.13.11.6-RXN',\n",
       "   'Reactome:R-HSA-71218 \"3-hydroxyanthranilate + O2 => 2-amino-3-carboxymuconate semialdehyde\"',\n",
       "   'RHEA:17953',\n",
       "   'UM-BBD_reactionID:r1027'],\n",
       "  'is_a': ['GO:0016702 ! oxidoreductase activity, acting on single donors with incorporation of molecular oxygen, incorporation of two atoms of oxygen']},\n",
       " 'GO:0000335': {'name': ['obsolete negative regulation of transposition, DNA-mediated'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that stops, prevents, or reduces the frequency, rate or extent of DNA transposition.\" [GOC:dph, GOC:krc, GOC:tb]'],\n",
       "  'comment': ['This term was obsoleted because there is no evidence that this process exists.'],\n",
       "  'synonym': ['\"down regulation of DNA transposition\" EXACT []',\n",
       "   '\"down-regulation of DNA transposition\" EXACT []',\n",
       "   '\"downregulation of DNA transposition\" EXACT []',\n",
       "   '\"inhibition of DNA transposition\" NARROW []',\n",
       "   '\"negative regulation of DNA transposition\" EXACT [GOC:dph]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006313']},\n",
       " 'GO:0000336': {'name': ['obsolete positive regulation of transposition, DNA-mediated'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that activates or increases the frequency, rate or extent of DNA transposition.\" [GOC:dph, GOC:krc]'],\n",
       "  'comment': ['This term was obsoleted because there is no evidence that this process exists.'],\n",
       "  'synonym': ['\"activation of DNA transposition\" NARROW []',\n",
       "   '\"positive regulation of DNA transposition\" EXACT [GOC:dph]',\n",
       "   '\"stimulation of DNA transposition\" NARROW []',\n",
       "   '\"up regulation of DNA transposition\" EXACT []',\n",
       "   '\"up-regulation of DNA transposition\" EXACT []',\n",
       "   '\"upregulation of DNA transposition\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006313']},\n",
       " 'GO:0000337': {'name': ['obsolete regulation of transposition, DNA-mediated'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that modulates the frequency, rate or extent of DNA transposition, the process of transposing (moving to a different location) a segment of a chromosome or a piece of a DNA molecule.\" [GOC:dph, GOC:krc]'],\n",
       "  'comment': ['This term was obsoleted because there is no evidence that this process exists.'],\n",
       "  'synonym': ['\"regulation of DNA transposition\" EXACT [GOC:dph]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006313']},\n",
       " 'GO:0000338': {'name': ['protein deneddylation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0010388'],\n",
       "  'def': ['\"The removal of a ubiquitin-like protein of the NEDD8 type from a protein.\" [GOC:krc]'],\n",
       "  'subset': ['term_scheduled_for_obsoletion'],\n",
       "  'synonym': ['\"cullin deneddylation\" NARROW []'],\n",
       "  'is_a': ['GO:0070646 ! protein modification by small protein removal']},\n",
       " 'GO:0000339': {'name': ['RNA cap binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a 7-methylguanosine (m7G) group or derivative located at the 5\\' end of an RNA molecule.\" [GOC:krc]'],\n",
       "  'synonym': ['\"binding to mRNA cap\" NARROW []',\n",
       "   '\"mRNA cap binding\" NARROW []',\n",
       "   '\"snRNA cap binding\" NARROW []'],\n",
       "  'is_a': ['GO:0003723 ! RNA binding']},\n",
       " 'GO:0000340': {'name': ['RNA 7-methylguanosine cap binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a 7-methylguanosine group added cotranscriptionally to the 5\\' end of RNA molecules transcribed by polymerase II.\" [GOC:krc]'],\n",
       "  'synonym': ['\"RNA m7G cap binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000339 ! RNA cap binding']},\n",
       " 'GO:0000341': {'name': ['RNA trimethylguanosine cap binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to the trimethylguanosine (m(3)(2,2,7)-GTP) group located at the 5\\' end of some RNA molecules. Such trimethylated cap structures, generally produced by posttranscriptional modification of a 7-methylguanosine cap, are often found on snRNAs and snoRNAs transcribed by RNA polymerase II, but have also be found on snRNAs transcribed by RNA polymerase III. They have also been found on a subset of the mRNA population in some species, e.g. C. elegans.\" [GOC:krc]'],\n",
       "  'synonym': ['\"RNA m2,2,7G cap binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000339 ! RNA cap binding']},\n",
       " 'GO:0000342': {'name': ['RNA cap 4 binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a hypermethylated cap structure consisting of 7-methylguanosine (m(7)G) followed by four methylated nucleotides (cap 4)'],\n",
       "  'is_a': ['GO:0000339 ! RNA cap binding']},\n",
       " 'GO:0000343': {'name': ['plastid-encoded plastid RNA polymerase complex A'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A plastid-encoded DNA-directed RNA polymerase complex that resembles eubacterial multisubunit RNA polymerases, with a core composed of alpha, beta, and beta-prime subunits. An additional subunit, a sigma factor, is required for promoter recognition. PEP-A is generated from the PEP-B form during chloroplast maturation to generate a complex composed of at least thirteen polypeptides that is not sensitive to the antibiotic rifampicin, like its precursor form the PEP-B complex.\" [PMID:10946105]'],\n",
       "  'synonym': ['\"PEP-A\" EXACT []'],\n",
       "  'is_a': ['GO:0000427 ! plastid-encoded plastid RNA polymerase complex']},\n",
       " 'GO:0000344': {'name': ['plastid-encoded plastid RNA polymerase complex B'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A plastid-encoded DNA-directed RNA polymerase complex that resembles eubacterial multisubunit RNA polymerases with a core composed of alpha, beta, and beta-prime subunits. An additional subunit, a sigma factor, is required for promoter recognition. PEP-B is distinguished from PEP-A by its sensitivity to the antibiotic rifampicin. PEP-B is found in both etioplasts and chloroplasts, but is the predominate form in etioplasts. It forms the core of the PEP-A form; the conversion from PEP-B to PEP-A occurs during chloroplast maturation.\" [PMID:10946105]'],\n",
       "  'synonym': ['\"PEP-B\" EXACT []'],\n",
       "  'is_a': ['GO:0000427 ! plastid-encoded plastid RNA polymerase complex']},\n",
       " 'GO:0000345': {'name': ['cytosolic DNA-directed RNA polymerase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The eubacterial DNA-directed RNA polymerase is a multisubunit complex with a core composed of the essential subunits beta-prime, beta, and two copies of alpha and a fifth nonessential subunit called omega. An additional subunit, a sigma factor, is required for promoter recognition and specificity.\" [PMID:11158566]'],\n",
       "  'is_a': ['GO:0000428 ! DNA-directed RNA polymerase complex'],\n",
       "  'relationship': ['part_of GO:0005829 ! cytosol']},\n",
       " 'GO:0000346': {'name': ['transcription export complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The transcription export (TREX) complex couples transcription elongation by RNA polymerase II to mRNA export. The complex associates with the polymerase and travels with it along the length of the transcribed gene. TREX is composed of the THO transcription elongation complex as well as other proteins that couple THO to mRNA export proteins. The TREX complex is known to be found in a wide range of eukaryotes, including S. cerevisiae and metazoans.\" [GOC:krc, PMID:11979277]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"TREX complex\" EXACT []'],\n",
       "  'is_a': ['GO:0140513 ! nuclear protein-containing complex']},\n",
       " 'GO:0000347': {'name': ['THO complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The THO complex is a nuclear complex that is required for transcription elongation through genes containing tandemly repeated DNA sequences. The THO complex is also part of the TREX (TRanscription EXport) complex that is involved in coupling transcription to export of mRNAs to the cytoplasm. In S. cerevisiae, it is composed of four subunits'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'is_a': ['GO:0140513 ! nuclear protein-containing complex']},\n",
       " 'GO:0000348': {'name': ['mRNA branch site recognition'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000370', 'GO:0000371'],\n",
       "  'def': ['\"Recognition of the pre-mRNA branch site sequence by components of the assembling spliceosome.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this step represents the formation of the B complex (yeast) or the A complex (mammalian).'],\n",
       "  'synonym': ['\"nuclear mRNA branch site recognition\" EXACT [GOC:vw]',\n",
       "   '\"spliceosomal A complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal A complex formation\" NARROW []',\n",
       "   '\"spliceosomal B complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal B complex formation\" NARROW []',\n",
       "   '\"U12-type nuclear mRNA branch site recognition\" NARROW []',\n",
       "   '\"U2-type nuclear mRNA branch site recognition\" NARROW []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000245 ! spliceosomal complex assembly']},\n",
       " 'GO:0000349': {'name': ['generation of catalytic spliceosome for first transesterification step'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000356', 'GO:0000357'],\n",
       "  'def': ['\"Formation of a catalytic spliceosome complex ready to perform the first splicing reaction. This occurs by an ATP-dependent conformational change of the pre-catalytic spliceosome.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this step represents the formation of the A2-2 complex (yeast) or the C1 complex (mammalian).'],\n",
       "  'synonym': ['\"catalytic spliceosome assembly for first transesterification step\" RELATED []',\n",
       "   '\"formation of catalytic spliceosome for first transesterification step\" EXACT []',\n",
       "   '\"spliceosomal A2-2 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal A2-2 complex formation\" NARROW []',\n",
       "   '\"spliceosomal C1 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal C1 complex formation\" NARROW []',\n",
       "   '\"U12-type catalytic spliceosome formation for first transesterification step\" NARROW []',\n",
       "   '\"U2-type catalytic spliceosome formation for first transesterification step\" NARROW []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000393 ! spliceosomal conformational changes to generate catalytic conformation']},\n",
       " 'GO:0000350': {'name': ['generation of catalytic spliceosome for second transesterification step'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000358', 'GO:0000359'],\n",
       "  'def': ['\"Conformational rearrangement of the spliceosomal complex containing the RNA products from the 1st step of splicing to form the catalytic site for the second step of splicing.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this step represents formation of the A2-3 complex (yeast) or the C2 complex (mammalian).'],\n",
       "  'synonym': ['\"catalytic spliceosome assembly for second transesterification step\" RELATED []',\n",
       "   '\"formation of catalytic spliceosome for second transesterification step\" EXACT []',\n",
       "   '\"formation of catalytic U12-type spliceosome for second transesterification step\" NARROW []',\n",
       "   '\"formation of catalytic U2-type spliceosome for second transesterification step\" NARROW []',\n",
       "   '\"formation of spliceosomal A2-2 complex\" NARROW []',\n",
       "   '\"formation of spliceosomal C1 complex\" NARROW []',\n",
       "   '\"lariat formation, 5\\'-splice site cleavage\" RELATED []',\n",
       "   '\"spliceosomal A2-3 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal A2-3 complex formation\" NARROW []',\n",
       "   '\"spliceosomal C2 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal C2 complex formation\" NARROW []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000393 ! spliceosomal conformational changes to generate catalytic conformation']},\n",
       " 'GO:0000352': {'name': ['trans assembly of SL-containing precatalytic spliceosome'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Assembly of a spliceosomal complex containing the SL RNA and the pre-mRNA to be joined, as well as all the spliceosomal snRNPs involved in trans leader splicing. Formation of the trans leader spliceosome brings together the quadruple SL/U4/U5/U6 snRNP and the complex of the U2 snRNP with the splice site of the pre-mRNA.\" [GOC:krc, GOC:mtg_mpo, ISBN:0879695897]'],\n",
       "  'synonym': ['\"trans assembly of spliced leader-containing precatalytic spliceosome\" EXACT []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000245 ! spliceosomal complex assembly',\n",
       "   'part_of GO:0045291 ! mRNA trans splicing, SL addition']},\n",
       " 'GO:0000353': {'name': ['formation of quadruple SL/U4/U5/U6 snRNP'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Formation of a quadruple snRNP complex composed of the spliced leader (SL) RNA along with the U4/U6-U5 tri-snRNP complex. Interactions that may facilitate this include a duplex between the SL and U6 RNAs and interactions between the U5 RNA and the exon sequence at the 5\\' splice site within the SL RNA.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': [\"Note that this step is analogous to 5' splice site selection in cis-splicing.\"],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0045291 ! mRNA trans splicing, SL addition']},\n",
       " 'GO:0000354': {'name': ['cis assembly of pre-catalytic spliceosome'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000360', 'GO:0000361'],\n",
       "  'def': ['\"Assembly of a spliceosomal complex containing the intact pre-mRNA and all of the spliceosomal snRNPs. This occurs when the tri-snRNP associates with the pre-mRNA and associated snRNPs in an ATP-dependent manner.\" [GOC:krc, GOC:mtg_mpo, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this step represents formation of the A2-1 complex (yeast) or the B1 complex (mammals).'],\n",
       "  'synonym': ['\"cis assembly of U12-type pre-catalytic spliceosome\" NARROW []',\n",
       "   '\"cis assembly of U2-type pre-catalytic spliceosome\" NARROW []',\n",
       "   '\"formation of spliceosomal A2-1 complex\" NARROW []',\n",
       "   '\"formation of spliceosomal B1 complex\" NARROW []',\n",
       "   '\"spliceosomal A2-1 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal A2-1 complex formation\" NARROW []',\n",
       "   '\"spliceosomal B1 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal B1 complex formation\" NARROW []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000245 ! spliceosomal complex assembly',\n",
       "   'part_of GO:0045292 ! mRNA cis splicing, via spliceosome']},\n",
       " 'GO:0000362': {'name': ['obsolete first U2-type spliceosomal transesterification activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the first transesterification reaction of U2-type spliceosomal mRNA splicing. The intron branch site adenosine is the nucleophile attacking the 5\\' splice site, resulting in cleavage at this position. In cis splicing, this is the step that forms a lariat structure of the intron RNA, which is still joined to the 3\\' exon. The catalytic site is thought to be formed by U6 and/or U2 snRNA, and/or associated proteins.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['This term was made obsolete because the activities used in the U2- and U12-type complexes are the same. The cellular component ontology should (and does) represent the two different complexes, but the molecular function ontology should not.'],\n",
       "  'synonym': ['\"first U2-type spliceosomal transesterification activity\" EXACT []',\n",
       "   '\"lariat formation, 5\\'-splice site cleavage\" BROAD []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000384']},\n",
       " 'GO:0000363': {'name': ['obsolete first U12-type spliceosomal transesterification activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the first transesterification reaction of U12-type spliceosomal mRNA splicing. The intron branch site adenosine is the nucleophile attacking the 5\\' splice site, resulting in cleavage at this position. In cis splicing, this is the step that forms a lariat structure of the intron RNA, which is still joined to the 3\\' exon. The catalytic site is thought to be formed by U6atac snRNA and/or U2atac snRNA, and/or associated proteins.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['This term was made obsolete because the activities used in the U2- and U12-type complexes are the same. The cellular component ontology should (and does) represent the two different complexes, but the molecular function ontology should not.'],\n",
       "  'synonym': ['\"first U12-type spliceosomal transesterification activity\" EXACT []',\n",
       "   '\"lariat formation, 5\\'-splice site cleavage\" BROAD []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000384']},\n",
       " 'GO:0000364': {'name': ['obsolete second U2-type spliceosomal transesterification activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the second transesterification reaction of U2-type spliceosomal mRNA splicing. Ligation of the two exons occurs via a transesterification reaction where the free 3\\'-hydroxyl group of the 5\\' exon is the nucleophile attacking the 3\\' splice site. Non-expressed sequences are now detached from the exons. In cis splicing, the intron is in a lariat structure. The catalytic site is thought to be formed by U6 and/or U2 snRNAs and/or associated proteins.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['This term was made obsolete because the activities used in the U2- and U12-type complexes are the same. The cellular component ontology should (and does) represent the two different complexes, but the molecular function ontology should not.'],\n",
       "  'synonym': ['\"3\\'-splice site cleavage, exon ligation\" BROAD []',\n",
       "   '\"second U2-type spliceosomal transesterification activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000386']},\n",
       " 'GO:0000365': {'name': ['mRNA trans splicing, via spliceosome'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The joining together of exons from two different primary transcripts of messenger RNA (mRNA) via a spliceosomal mechanism, so that mRNA consisting only of the joined exons is produced.\" [GOC:krc, ISBN:0879695897, PMID:18458335]'],\n",
       "  'synonym': ['\"nuclear mRNA trans splicing, via spliceosome\" EXACT [GOC:vw]',\n",
       "   '\"nuclear mRNA trans splicing, via U2-type spliceosome\" NARROW []'],\n",
       "  'is_a': ['GO:0000398 ! mRNA splicing, via spliceosome']},\n",
       " 'GO:0000366': {'name': ['intergenic mRNA trans splicing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The joining together of two independently transcribed RNAs from two different genes, each of which also produces mRNA(s) via cis-splicing.\" [GOC:krc, PMID:11726664, PMID:12110900]'],\n",
       "  'synonym': ['\"intergenic nuclear mRNA trans splicing\" EXACT [GOC:vw]'],\n",
       "  'is_a': ['GO:0000365 ! mRNA trans splicing, via spliceosome',\n",
       "   'GO:0000380 ! alternative mRNA splicing, via spliceosome']},\n",
       " 'GO:0000367': {'name': ['obsolete second U12-type spliceosomal transesterification activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the second transesterification reaction of U12-type spliceosomal mRNA splicing. Ligation of the two exons occurs via a transesterification reaction where the free 3\\'-hydroxyl group of the 5\\' exon is the nucleophile attacking the 3\\' splice site. Non-expressed sequences are now detached from the exons. In cis splicing, the intron is in a lariat structure. The catalytic site is thought to be formed by U6 and/or U2 snRNAs and/or associated proteins.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['This term was made obsolete because the activities used in the U2- and U12-type complexes are the same. The cellular component ontology should (and does) represent the two different complexes, but the molecular function ontology should not.'],\n",
       "  'synonym': ['\"3\\'-splice site cleavage, exon ligation\" BROAD []',\n",
       "   '\"second U12-type spliceosomal transesterification activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0000386']},\n",
       " 'GO:0000372': {'name': ['Group I intron splicing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The splicing of Group I introns. This occurs by a ribozymic mechanism where the intron sequence forms a distinct 3D structure, characteristic of Group I introns and involved in determining the locations of the splice sites (there do not appear to be consensus splice site sequences) as well as having a role in catalyzing the splicing reactions, though protein factors are also required in vivo. Splicing occurs by a series of two transesterification reactions, generally with exogenous guanosine as the initiating nucleophile. The intron is excised as a linear piece (though it may subsequently circularize).\" [GOC:krc, PMID:11377794]'],\n",
       "  'comment': ['Note that Group I introns are known to be found in a number of places'],\n",
       "  'synonym': ['\"mRNA splicing\" RELATED []'],\n",
       "  'is_a': ['GO:0000376 ! RNA splicing, via transesterification reactions with guanosine as nucleophile']},\n",
       " 'GO:0000373': {'name': ['Group II intron splicing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The splicing of Group II introns. This occurs by a ribozymic mechanism where the intron sequence forms a distinct 3D structure, characteristic of Group II introns and containing splice site consensus sequences, that is involved in catalyzing the splicing reactions, though protein factors are also required in vivo. Splicing occurs by a series of two transesterification reactions (mechanistically similar to those for splicing of nuclear mRNAs) initiated by a bulged adenosine residue within the intron sequence as the initiating nucleophile. The intron is excised as a lariat.\" [GOC:krc, PMID:11377794]'],\n",
       "  'comment': ['Note that Group II introns are known to be found in a number of places'],\n",
       "  'synonym': ['\"mRNA splicing\" RELATED []'],\n",
       "  'is_a': ['GO:0000377 ! RNA splicing, via transesterification reactions with bulged adenosine as nucleophile']},\n",
       " 'GO:0000374': {'name': ['Group III intron splicing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The splicing of Group III introns. This occurs by a ribozymic mechanism where the intron sequence forms a distinct 3D structure, characteristic of Group III introns, that is involved in catalyzing the splicing reactions, though protein factors are also required in vivo. Splicing occurs by a series of two transesterification reactions begun by a bulged adenosine residue within the intron sequence as the initiating nucleophile. The intron is excised as a lariat. Though very similar in structure and mechanism to Group II introns, Group III introns are smaller and more streamlined and the splice site consensus sequences are not as well conserved.\" [GOC:krc, PMID:11377794]'],\n",
       "  'comment': ['Note that Group III introns are known to be found in mRNA of plastids of euglenoid protists.'],\n",
       "  'synonym': ['\"mRNA splicing\" RELATED []'],\n",
       "  'is_a': ['GO:0000377 ! RNA splicing, via transesterification reactions with bulged adenosine as nucleophile']},\n",
       " 'GO:0000375': {'name': ['RNA splicing, via transesterification reactions'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000385', 'GO:0031202'],\n",
       "  'def': ['\"Splicing of RNA via a series of two transesterification reactions.\" [GOC:krc]'],\n",
       "  'comment': ['Note that nuclear mRNA, Group I, Group II, and Group III introns are all spliced by a series of two transesterification reactions that occur within the RNA itself, or between two RNAs in trans splicing. Some of these require one or more proteins to stabilize the catalytic conformation, while others are autocatalytic. Note that tRNA introns are spliced by a different catalytic mechanism.'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"pre-mRNA splicing factor activity\" RELATED [GOC:krc, GOC:mah]',\n",
       "   '\"RNA splicing factor activity, transesterification mechanism\" RELATED [GOC:krc, GOC:mah]',\n",
       "   '\"spliceosomal catalysis\" RELATED []'],\n",
       "  'is_a': ['GO:0008380 ! RNA splicing']},\n",
       " 'GO:0000376': {'name': ['RNA splicing, via transesterification reactions with guanosine as nucleophile'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Splicing of RNA via a series of two transesterification reactions with exogenous guanosine as the initiating nucleophile.\" [GOC:krc, PMID:11377794]'],\n",
       "  'is_a': ['GO:0000375 ! RNA splicing, via transesterification reactions']},\n",
       " 'GO:0000377': {'name': ['RNA splicing, via transesterification reactions with bulged adenosine as nucleophile'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Splicing of RNA via a series of two transesterification reactions with a bulged adenosine residue from the intron branch point as the initiating nucleophile. When the initial RNA for the splicing reaction is a single molecule (cis splicing), the excised intron is released in a lariat structure.\" [GOC:krc, PMID:11377794]'],\n",
       "  'synonym': ['\"lariat RNA biosynthesis\" RELATED []',\n",
       "   '\"lariat RNA formation\" RELATED []'],\n",
       "  'is_a': ['GO:0000375 ! RNA splicing, via transesterification reactions']},\n",
       " 'GO:0000378': {'name': ['RNA exon ligation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The RNA metabolic process that joins two exons, each of which has free ends that were generated by endonucleolytic cleavages, by a ligation reaction.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this is not a part of spliceosomal RNA splicing.'],\n",
       "  'is_a': ['GO:0016070 ! RNA metabolic process'],\n",
       "  'relationship': ['part_of GO:0000394 ! RNA splicing, via endonucleolytic cleavage and ligation']},\n",
       " 'GO:0000379': {'name': ['tRNA-type intron splice site recognition and cleavage'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"RNA processing that begins when the tertiary structure of a tRNA type intron is recognized, and ends when the endonucleolytic cleavage of the RNA at both the 5\\' and 3\\' splice sites occurs.\" [GOC:krc, GOC:mah, ISBN:0879695897]'],\n",
       "  'is_a': ['GO:0034470 ! ncRNA processing'],\n",
       "  'relationship': ['part_of GO:0006388 ! tRNA splicing, via endonucleolytic cleavage and ligation']},\n",
       " 'GO:0000380': {'name': ['alternative mRNA splicing, via spliceosome'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process of generating multiple mRNA molecules from a given set of exons by differential use of exons from the primary transcript(s) to form multiple mature mRNAs that vary in their exon composition.\" [GOC:krc, PMID:12110900]'],\n",
       "  'comment': ['Note that this process most commonly occurs in cis, selecting or skipping exons from the same primary transcript, but it has also been observed to occur in trans at low frequency, at least in some mammals.'],\n",
       "  'synonym': ['\"alternative nuclear mRNA splicing, via spliceosome\" EXACT [GOC:vw]',\n",
       "   '\"splice site selection\" BROAD []'],\n",
       "  'is_a': ['GO:0000398 ! mRNA splicing, via spliceosome']},\n",
       " 'GO:0000381': {'name': ['regulation of alternative mRNA splicing, via spliceosome'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of alternative splicing of nuclear mRNAs.\" [GOC:krc]'],\n",
       "  'synonym': ['\"regulation of alternative nuclear mRNA splicing, via spliceosome\" EXACT [GOC:vw]',\n",
       "   '\"splice site selection\" BROAD []'],\n",
       "  'is_a': ['GO:0048024 ! regulation of mRNA splicing, via spliceosome'],\n",
       "  'relationship': ['regulates GO:0000380 ! alternative mRNA splicing, via spliceosome']},\n",
       " 'GO:0000384': {'name': ['first spliceosomal transesterification activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the first transesterification reaction of spliceosomal mRNA splicing. The intron branch site adenosine is the nucleophile attacking the 5\\' splice site, resulting in cleavage at this position. In cis splicing, this is the step that forms a lariat structure of the intron RNA, while it is still joined to the 3\\' exon.\" [GOC:krc, ISBN:0879695897, PMID:19239890]'],\n",
       "  'synonym': ['\"lariat formation, 5\\'-splice site cleavage\" RELATED []'],\n",
       "  'is_a': ['GO:0140098 ! catalytic activity, acting on RNA']},\n",
       " 'GO:0000386': {'name': ['second spliceosomal transesterification activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the second transesterification reaction of spliceosomal mRNA splicing. Ligation of the two exons occurs via a transesterification reaction where the free 3\\'-hydroxyl group of the 5\\' exon is the nucleophile attacking the 3\\' splice site. Non-expressed sequences are now detached from the exons. In cis splicing, the intron is in a lariat structure.\" [GOC:krc, ISBN:0879695897, PMID:19239890]'],\n",
       "  'synonym': ['\"3\\'-splice site cleavage, exon ligation\" BROAD []',\n",
       "   '\"lariat formation, 5\\'-splice site cleavage\" RELATED []'],\n",
       "  'xref': ['Reactome:R-HSA-9770236 \"Formation of the Spliceosomal P complex and exon ligation (new)\"'],\n",
       "  'is_a': ['GO:0140098 ! catalytic activity, acting on RNA']},\n",
       " 'GO:0000387': {'name': ['spliceosomal snRNP assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of one or more snRNA and multiple protein components to form a ribonucleoprotein complex that is involved in formation of the spliceosome.\" [GOC:krc, GOC:mah, ISBN:0879695897]'],\n",
       "  'synonym': ['\"spliceosomal snRNP biogenesis\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000398 ! mRNA splicing, via spliceosome']},\n",
       " 'GO:0000388': {'name': ['spliceosome conformational change to release U4 (or U4atac) and U1 (or U11)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000396', 'GO:0000397'],\n",
       "  'def': ['\"Rearrangement of the pre-catalytic spliceosome containing U4 (or U4atac) and U1 (or U11) snRNPs to unpair U4 (or U4atac) from U6 (or U6atac) and release it from the spliceosomal complex along with U1 (or U11).\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this step represents formation of the A1 complex (yeast) or the B2 complex (mammalian).'],\n",
       "  'synonym': ['\"3\\'-splice site cleavage, exon ligation\" BROAD []',\n",
       "   '\"spliceosomal A1 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal A1 complex formation\" NARROW []',\n",
       "   '\"spliceosomal B2 complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal B2 complex formation\" NARROW []',\n",
       "   '\"U12-type spliceosome conformational change to release U4atac and U11\" NARROW []',\n",
       "   '\"U2-type spliceosome conformational change to release U4 and U1\" NARROW []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000393 ! spliceosomal conformational changes to generate catalytic conformation']},\n",
       " 'GO:0000389': {'name': [\"mRNA 3'-splice site recognition\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000382', 'GO:0000383'],\n",
       "  'def': ['\"Recognition of the intron 3\\'-splice site by components of the assembling U2- or U12-type spliceosome.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'synonym': ['\"nuclear mRNA 3\\'-splice site recognition\" EXACT [GOC:vw]',\n",
       "   '\"U12-type nuclear mRNA 3\\'-splice site recognition\" NARROW []',\n",
       "   '\"U2-type nuclear mRNA 3\\'-splice site recognition\" NARROW []'],\n",
       "  'is_a': ['GO:0006376 ! mRNA splice site recognition']},\n",
       " 'GO:0000390': {'name': ['spliceosomal complex disassembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000391', 'GO:0000392'],\n",
       "  'def': ['\"Disassembly of a spliceosomal complex with the ATP-dependent release of the product RNAs, one of which is composed of the joined exons. In cis splicing, the other product is the excised sequence, often a single intron, in a lariat structure.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'synonym': ['\"spliceosome complex disassembly\" EXACT []',\n",
       "   '\"spliceosome disassembly\" BROAD []',\n",
       "   '\"U12-type spliceosome disassembly\" NARROW []',\n",
       "   '\"U2-type spliceosome disassembly\" NARROW []'],\n",
       "  'is_a': ['GO:0032988 ! protein-RNA complex disassembly'],\n",
       "  'relationship': ['part_of GO:0000398 ! mRNA splicing, via spliceosome']},\n",
       " 'GO:0000393': {'name': ['spliceosomal conformational changes to generate catalytic conformation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Structural rearrangements of the spliceosome complex, containing RNA to be spliced, to generate a catalytic conformation.\" [GOC:krc]'],\n",
       "  'synonym': ['\"3\\'-splice site cleavage, exon ligation\" BROAD []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0000398 ! mRNA splicing, via spliceosome']},\n",
       " 'GO:0000394': {'name': ['RNA splicing, via endonucleolytic cleavage and ligation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Splicing of RNA via recognition of the folded RNA structure that brings the 5\\' and 3\\' splice sites into proximity and cleavage of the RNA at both the 3\\' and 5\\' splice sites by an endonucleolytic mechanism, followed by ligation of the exons.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that while typically associated with tRNA splicing, this mechanism of splicing is known to be used for some non-tRNA substrates, e.g. HAC1 (YFL031W) in S. cerevisiae and an intron in the 23S rRNA of the Archaeal species Desulfurococcus mobilis.'],\n",
       "  'synonym': ['\"mRNA splicing\" BROAD []'],\n",
       "  'is_a': ['GO:0008380 ! RNA splicing']},\n",
       " 'GO:0000395': {'name': [\"mRNA 5'-splice site recognition\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000368', 'GO:0000369'],\n",
       "  'def': ['\"Recognition of the intron 5\\'-splice site by components of the assembling spliceosome.\" [GOC:krc, ISBN:0879695897]'],\n",
       "  'comment': ['Note that this step represents formation of the Commitment Complex (CC, in yeast) or the E complex (mammalian).'],\n",
       "  'synonym': ['\"nuclear mRNA 5\\' splice site recognition\" EXACT []',\n",
       "   '\"nuclear mRNA 5\\'-splice site recognition\" EXACT [GOC:vw]',\n",
       "   '\"spliceosomal CC complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal CC complex formation\" NARROW []',\n",
       "   '\"spliceosomal commitment complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal commitment complex formation\" NARROW []',\n",
       "   '\"spliceosomal E complex biosynthesis\" NARROW []',\n",
       "   '\"spliceosomal E complex formation\" NARROW []',\n",
       "   '\"U12-type nuclear mRNA 5\\' splice site recognition\" NARROW []',\n",
       "   '\"U12-type nuclear mRNA 5\\'-splice site recognition\" NARROW []',\n",
       "   '\"U2-type nuclear mRNA 5\\' splice site recognition\" NARROW []',\n",
       "   '\"U2-type nuclear mRNA 5\\'-splice site recognition\" NARROW []'],\n",
       "  'is_a': ['GO:0006376 ! mRNA splice site recognition'],\n",
       "  'relationship': ['part_of GO:0045292 ! mRNA cis splicing, via spliceosome']},\n",
       " 'GO:0000398': {'name': ['mRNA splicing, via spliceosome'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0006374', 'GO:0006375'],\n",
       "  'def': ['\"The joining together of exons from one or more primary transcripts of messenger RNA (mRNA) and the excision of intron sequences, via a spliceosomal mechanism, so that mRNA consisting only of the joined exons is produced.\" [GOC:krc, ISBN:0198506732, ISBN:0879695897]'],\n",
       "  'comment': [\"Note that although the many U12-type introns have the sequence AT-AC at the intron termini, some introns with these terminal sequences are spliced by the U2-type spliceosome. The distinguishing characteristics are sequences near the 5' splice site and the branch point sequences of the intron. Note that although the majority of U2-type introns have the sequence GU-AG at the intron termini, some introns with these terminal sequences are spliced by the U12-type spliceosome. The distinguishing characteristics are sequences near the 5' splice site and the branch point sequences of the intron.\"],\n",
       "  'synonym': ['\"mRNA splicing\" BROAD []',\n",
       "   '\"nuclear mRNA splicing via U12-type spliceosome\" NARROW []',\n",
       "   '\"nuclear mRNA splicing via U2-type spliceosome\" NARROW []',\n",
       "   '\"nuclear mRNA splicing, via spliceosome\" EXACT [GOC:krc, GOC:vw]',\n",
       "   '\"pre-mRNA splicing\" BROAD []',\n",
       "   '\"splicing AT-AC intron\" RELATED []',\n",
       "   '\"splicing GT-AG intron\" RELATED []'],\n",
       "  'is_a': ['GO:0000377 ! RNA splicing, via transesterification reactions with bulged adenosine as nucleophile',\n",
       "   'GO:0006397 ! mRNA processing']},\n",
       " 'GO:0000399': {'name': ['cellular bud neck septin structure'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Any of a series of septin structures that are localized in the bud neck of a budding fungal cell during the cell cycle.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005935 ! cellular bud neck',\n",
       "   'part_of GO:0032153 ! cell division site']},\n",
       " 'GO:0000400': {'name': ['four-way junction DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA segment containing four-way junctions, also known as Holliday junctions, a structure where two DNA double strands are held together by reciprocal exchange of two of the four strands, one strand each from the two original helices.\" [GOC:krc, ISBN:0815332181, PMID:15563464]'],\n",
       "  'synonym': ['\"forked DNA binding\" BROAD []',\n",
       "   '\"Holliday junction binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000217 ! DNA secondary structure binding']},\n",
       " 'GO:0000401': {'name': ['open form four-way junction DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA segment containing the open form of a four-way junction, also known as a Holliday junction, a structure where two DNA double strands are held together by reciprocal exchange of two of the four strands, one strand each from the two original helices. The open form of a four-way junction can be diagrammed without any of the strands crossing over.\" [GOC:krc, ISBN:0815332181, PMID:15563464]'],\n",
       "  'synonym': ['\"open form Holliday junction binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000400 ! four-way junction DNA binding']},\n",
       " 'GO:0000402': {'name': ['crossed form four-way junction DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA segment containing the crossed form of a four-way junction, also known as a Holliday junction, a structure where two DNA double strands are held together by reciprocal exchange of two of the four strands, one strand each from the two original helices. The crossed form of a four-way junction cannot be diagrammed without any of the strands crossing over, and instead contains a single crossover between two of the strands.\" [GOC:krc, ISBN:0815332181, PMID:15563464]'],\n",
       "  'synonym': ['\"crossed form Holliday junction binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000400 ! four-way junction DNA binding']},\n",
       " 'GO:0000403': {'name': ['Y-form DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA segment shaped like a Y. This shape occurs when DNA contains a region of paired double-stranded DNA on one end and a region of unpaired DNA strands on the opposite end.\" [GOC:elh, PMID:16781730]'],\n",
       "  'synonym': ['\"forked DNA binding\" BROAD []',\n",
       "   '\"splayed Y-form DNA binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000217 ! DNA secondary structure binding']},\n",
       " 'GO:0000404': {'name': ['heteroduplex DNA loop binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA segment containing a loop. A loop occurs when DNA contains a large insertion or deletion that causes a region of unpaired single-stranded DNA to loop out, while the rest of the DNA is in a paired double-stranded configuration.\" [GOC:elh, PMID:16781730]'],\n",
       "  'synonym': ['\"loop DNA binding\" RELATED [GOC:jh]'],\n",
       "  'is_a': ['GO:0000217 ! DNA secondary structure binding',\n",
       "   'GO:0032135 ! DNA insertion or deletion binding']},\n",
       " 'GO:0000405': {'name': ['bubble DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to DNA segment that contains a bubble. A bubble occurs when DNA contains a region of unpaired, single-stranded DNA flanked on both sides by regions of paired, double-stranded DNA.\" [GOC:elh, GOC:vw, PMID:16781730]'],\n",
       "  'is_a': ['GO:0000217 ! DNA secondary structure binding']},\n",
       " 'GO:0000406': {'name': ['double-strand/single-strand DNA junction binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA segment that contains double-stranded DNA flanked by a region of single-stranded DNA.\" [GOC:elh, PMID:16781730]'],\n",
       "  'is_a': ['GO:0000217 ! DNA secondary structure binding']},\n",
       " 'GO:0000407': {'name': ['phagophore assembly site'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Punctate structures proximal to the endoplasmic reticulum which are the sites where the Atg machinery assembles upon autophagy induction.\" [GOC:elh, PMID:11689437, PMID:12048214, PMID:12554655]'],\n",
       "  'synonym': ['\"PAS\" EXACT []',\n",
       "   '\"perivacuolar space\" NARROW []',\n",
       "   '\"pre-autophagosomal structure\" NARROW []'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005737 ! cytoplasm']},\n",
       " 'GO:0000408': {'name': ['EKC/KEOPS complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex involved in t6A tRNA modification. For example, in Saccharomyces cerevisiae the complex contains Bud32p, Kae1p, Gon7p, Cgi121p, and Pcc1p.\" [GOC:elh, GOC:vw, PMID:16564010, PMID:16874308, PMID:21183954, PMID:23945934]'],\n",
       "  'comment': ['Originally proposed to be involved in transcription as well as promoting telomere uncapping and telomere elongation.'],\n",
       "  'synonym': ['\"endopeptidase-like kinase chromatin-associated protein complex\" EXACT []',\n",
       "   '\"KEOPS/EKC complex\" EXACT [PMID:21183954, PMID:23945934]',\n",
       "   '\"kinase, putative endopeptidase and other proteins of small size protein complex\" EXACT []',\n",
       "   '\"TCTC\" RELATED [PMID:25629598]',\n",
       "   '\"threonyl-carbamoly transferase complex\" RELATED [PMID:25629598]'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex']},\n",
       " 'GO:0000409': {'name': ['regulation of transcription by galactose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving galactose that modulates the frequency, rate or extent or transcription.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0006355 ! regulation of DNA-templated transcription']},\n",
       " 'GO:0000410': {'name': ['carbon catabolite repression of transcription by galactose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of galactose that leads to a decrease in the frequency, rate, or extent of transcription of specific genes involved in the metabolism of other carbon sources. Carbon catabolite repression is a mechanism of genetic regulation which the accumulation of catabolites of one substance in the cell represses the formation of enzymes that contribute to the catabolism of other substances.\" [GOC:mah]'],\n",
       "  'synonym': ['\"down regulation of transcription by galactose\" EXACT []',\n",
       "   '\"down-regulation of transcription by galactose\" EXACT []',\n",
       "   '\"downregulation of transcription by galactose\" EXACT []',\n",
       "   '\"inhibition of transcription by galactose\" NARROW []'],\n",
       "  'is_a': ['GO:0000409 ! regulation of transcription by galactose',\n",
       "   'GO:0045013 ! carbon catabolite repression of transcription']},\n",
       " 'GO:0000411': {'name': ['positive regulation of transcription by galactose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving galactose that activates or increases the rate of transcription.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"activation of transcription by galactose\" NARROW []',\n",
       "   '\"stimulation of transcription by galactose\" NARROW []',\n",
       "   '\"up regulation of transcription by galactose\" EXACT []',\n",
       "   '\"up-regulation of transcription by galactose\" EXACT []',\n",
       "   '\"upregulation of transcription by galactose\" EXACT []'],\n",
       "  'is_a': ['GO:0000409 ! regulation of transcription by galactose',\n",
       "   'GO:0045991 ! carbon catabolite activation of transcription']},\n",
       " 'GO:0000412': {'name': ['histone peptidyl-prolyl isomerization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The modification of a histone by cis-trans isomerization of a proline residue.\" [GOC:krc]'],\n",
       "  'subset': ['term_scheduled_for_obsoletion'],\n",
       "  'synonym': ['\"histone proline isomerization\" EXACT []'],\n",
       "  'is_a': ['GO:0000413 ! protein peptidyl-prolyl isomerization',\n",
       "   'GO:0016570 ! histone modification']},\n",
       " 'GO:0000413': {'name': ['protein peptidyl-prolyl isomerization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The modification of a protein by cis-trans isomerization of a proline residue.\" [GOC:krc, PMID:16959570]'],\n",
       "  'subset': ['term_scheduled_for_obsoletion'],\n",
       "  'synonym': ['\"protein proline isomerization\" EXACT []'],\n",
       "  'is_a': ['GO:0018208 ! peptidyl-proline modification']},\n",
       " 'GO:0000414': {'name': ['obsolete regulation of histone H3-K36 methylation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that modulates the frequency, rate or extent of the covalent addition of a methyl group to the lysine at position 36 of histone H3.\" [GOC:krc]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0140673']},\n",
       " 'GO:0000415': {'name': ['obsolete negative regulation of histone H3-K36 methylation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that stops, prevents, or reduces the frequency, rate or extent of the covalent addition of a methyl group to the lysine at position 36 of histone H3.\" [GOC:krc]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'synonym': ['\"down regulation of histone H3-K36 methylation\" EXACT []',\n",
       "   '\"down-regulation of histone H3-K36 methylation\" EXACT []',\n",
       "   '\"downregulation of histone H3-K36 methylation\" EXACT []',\n",
       "   '\"inhibition of histone H3-K36 methylation\" NARROW []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006338']},\n",
       " 'GO:0000416': {'name': ['obsolete positive regulation of histone H3-K36 methylation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that activates or increases the frequency, rate or extent of the covalent addition of a methyl group to the lysine at position 36 of histone H3.\" [GOC:krc]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'synonym': ['\"activation of histone H3-K36 methylation\" NARROW []',\n",
       "   '\"stimulation of histone H3-K36 methylation\" NARROW []',\n",
       "   '\"up regulation of histone H3-K36 methylation\" EXACT []',\n",
       "   '\"up-regulation of histone H3-K36 methylation\" EXACT []',\n",
       "   '\"upregulation of histone H3-K36 methylation\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0046975']},\n",
       " 'GO:0000417': {'name': ['HIR complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex proposed to be involved in replication-independent nucleosome assembly, by promoting histone deposition onto DNA. For example, in Saccharomyces, the complex contains Hir1p, Hir2p, Hir3p, and Hpc2p.\" [GOC:elh, GOC:mah, PMID:16303565, PMID:17180700]'],\n",
       "  'synonym': ['\"HIRA complex\" EXACT [PMID:19620282, PMID:20976105]'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex']},\n",
       " 'GO:0000418': {'name': ['RNA polymerase IV complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000420'],\n",
       "  'def': ['\"RNA polymerase IV is a multisubunit RNA polymerase complex found in the nucleus of plants and involved in accumulation of siRNAs and in DNA methylation-dependent silencing of endogenous repeated sequences. Pol IV is composed of subunits that are paralogous or identical to the 12 subunits of Pol II. The largest and second-largest subunits of Pol IV are the catalytic subunits and share similarity with the corresponding subunits of other eukaryotic and bacterial multisubunit RNA polymerases. The second largest subunit is also found in RNA polymerase V, while the largest subunit is found only in RNAP IV complex.\" [GOC:krc, GOC:mtg_sensu, PMID:15692015, PMID:15766525, PMID:16140984, PMID:19110459]'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase IV complex\" EXACT []',\n",
       "   '\"DNA-directed RNA polymerase IVa complex\" EXACT []'],\n",
       "  'is_a': ['GO:0055029 ! nuclear DNA-directed RNA polymerase complex']},\n",
       " 'GO:0000419': {'name': ['RNA polymerase V complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0080137'],\n",
       "  'def': ['\"RNA polymerase V is a multisubunit RNA polymerase complex found in the nucleus of plants and involved in accumulation of siRNAs and in DNA methylation-dependent silencing of endogenous repeated sequences. Pol V is composed of subunits that are paralogous or identical to the 12 subunits of Pol II. Two large subunits comprise the most conserved portion including the catalytic site and share similarity with other eukaryotic and bacterial multisubunit RNA polymerases. The second largest subunit is also found in RNA polymerase IVa, while the largest subunit is found only in the IVa complex and contains an extended C-terminal domain (CTD) that includes multiple repeats of a 16 amino-acid consensus sequence as well as other sequences. The remainder of the complex is composed of smaller subunits.\" [GOC:krc, GOC:mtg_sensu, PMID:16140984, PMID:19110459]'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase IVb complex\" EXACT []',\n",
       "   '\"DNA-directed RNA polymerase V complex\" EXACT []'],\n",
       "  'is_a': ['GO:0055029 ! nuclear DNA-directed RNA polymerase complex']},\n",
       " 'GO:0000421': {'name': ['autophagosome membrane'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The lipid bilayer surrounding an autophagosome, a double-membrane-bounded vesicle in which endogenous cellular material is sequestered.\" [GOC:autophagy, GOC:isa_complete]'],\n",
       "  'synonym': ['\"autophagic vacuole membrane\" EXACT [GOC:autophagy]'],\n",
       "  'is_a': ['GO:0005774 ! vacuolar membrane'],\n",
       "  'relationship': ['part_of GO:0005776 ! autophagosome']},\n",
       " 'GO:0000422': {'name': ['autophagy of mitochondrion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The autophagic process in which mitochondria are delivered to a type of vacuole and degraded in response to changing cellular conditions.\" [GOC:autophagy, PMID:15798367, PMID:19289147, PMID:23065344]'],\n",
       "  'synonym': ['\"mitochondrion autophagy\" EXACT [GOC:autophagy]',\n",
       "   '\"mitophagy\" RELATED []'],\n",
       "  'xref': ['Wikipedia:Autophagy_(cellular)#Selective_autophagy',\n",
       "   'Wikipedia:Mitophagy'],\n",
       "  'is_a': ['GO:0006914 ! autophagy',\n",
       "   'GO:0061726 ! mitochondrion disassembly']},\n",
       " 'GO:0000423': {'name': ['mitophagy'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The selective autophagy process in which a mitochondrion is degraded by macroautophagy.\" [PMID:15798367]'],\n",
       "  'comment': ['Note that this terms refers to the macroautophagy process and is named by common usage. Be aware that there are a separate micromitophagy and mitophagy by induced vacuole formation processes.'],\n",
       "  'synonym': ['\"macromitophagy\" EXACT []'],\n",
       "  'is_a': ['GO:0000422 ! autophagy of mitochondrion',\n",
       "   'GO:0061912 ! selective autophagy']},\n",
       " 'GO:0000424': {'name': ['micromitophagy'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Degradation of a mitochondrion by lysosomal microautophagy.\" [PMID:15798367, PMID:27003723]'],\n",
       "  'comment': ['Note that this term is not a child of mitophagy because the community genrally uses that latter to refer to the macroautophagy of mitochondria.'],\n",
       "  'is_a': ['GO:0000422 ! autophagy of mitochondrion',\n",
       "   'GO:0016237 ! lysosomal microautophagy']},\n",
       " 'GO:0000425': {'name': ['pexophagy'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The selective autophagy process in which a peroxisome is degraded by macroautophagy.\" [GOC:autophagy, PMID:12914914, PMID:16973210]'],\n",
       "  'synonym': ['\"macropexophagy\" EXACT []'],\n",
       "  'is_a': ['GO:0030242 ! autophagy of peroxisome',\n",
       "   'GO:0061912 ! selective autophagy']},\n",
       " 'GO:0000426': {'name': ['micropexophagy'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Degradation of a peroxisome by lysosomal microautophagy.\" [GOC:autophagy, GOC:pad, PMID:12914914, PMID:15350980, PMID:16973210]'],\n",
       "  'comment': ['Note that this term is not a child of pexophagy because the community usually uses the latter to refer to the macroautohagy process.'],\n",
       "  'is_a': ['GO:0016237 ! lysosomal microautophagy',\n",
       "   'GO:0030242 ! autophagy of peroxisome']},\n",
       " 'GO:0000427': {'name': ['plastid-encoded plastid RNA polymerase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"An RNA polymerase complex containing polypeptides encoded by the plastid genome. Plastid-encoded DNA-directed RNA polymerases resemble eubacterial multisubunit RNA polymerases, with a core composed of alpha, beta, and beta-prime subunits. Some forms contain multiple additional subunits. An additional sigma factor subunit is required for promoter recognition.\" [GOC:krc, GOC:mah, GOC:pj]'],\n",
       "  'is_a': ['GO:0000428 ! DNA-directed RNA polymerase complex'],\n",
       "  'relationship': ['part_of GO:0009536 ! plastid']},\n",
       " 'GO:0000428': {'name': ['DNA-directed RNA polymerase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex that possesses DNA-directed RNA polymerase activity.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0030880 ! RNA polymerase complex']},\n",
       " 'GO:0000429': {'name': ['carbon catabolite regulation of transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of one carbon source leads to the modulation of the frequency, rate, or extent of transcription, from an RNA polymerase II promoter, of specific genes involved in the metabolism of other carbon sources.\" [GOC:krc, GOC:mah]'],\n",
       "  'synonym': ['\"regulation of transcription from RNA polymerase II promoter by carbon catabolites\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0006357 ! regulation of transcription by RNA polymerase II',\n",
       "   'GO:0045990 ! carbon catabolite regulation of transcription']},\n",
       " 'GO:0000430': {'name': ['regulation of transcription from RNA polymerase II promoter by glucose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving glucose that modulates the frequency, rate or extent of transcription from an RNA polymerase II promoter.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0000429 ! carbon catabolite regulation of transcription from RNA polymerase II promoter',\n",
       "   'GO:0046015 ! regulation of transcription by glucose']},\n",
       " 'GO:0000431': {'name': ['regulation of transcription from RNA polymerase II promoter by galactose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving galactose that modulates the frequency, rate or extent of transcription from an RNA polymerase II promoter.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0000409 ! regulation of transcription by galactose',\n",
       "   'GO:0000429 ! carbon catabolite regulation of transcription from RNA polymerase II promoter']},\n",
       " 'GO:0000432': {'name': ['positive regulation of transcription from RNA polymerase II promoter by glucose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving glucose that activates or increases the rate of transcription from an RNA polymerase II promoter.\" [GOC:krc]'],\n",
       "  'synonym': ['\"activation of transcription from RNA polymerase II promoter by glucose\" NARROW []',\n",
       "   '\"stimulation of transcription from RNA polymerase II promoter by glucose\" NARROW []',\n",
       "   '\"up regulation of transcription from RNA polymerase II promoter by glucose\" EXACT []',\n",
       "   '\"up-regulation of transcription from RNA polymerase II promoter by glucose\" EXACT []',\n",
       "   '\"upregulation of transcription from RNA polymerase II promoter by glucose\" EXACT []'],\n",
       "  'is_a': ['GO:0000430 ! regulation of transcription from RNA polymerase II promoter by glucose',\n",
       "   'GO:0000436 ! carbon catabolite activation of transcription from RNA polymerase II promoter',\n",
       "   'GO:0046016 ! positive regulation of transcription by glucose']},\n",
       " 'GO:0000433': {'name': ['carbon catabolite repression of transcription from RNA polymerase II promoter by glucose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of glucose leads to a decrease in the frequency, rate, or extent of transcription of specific RNA polymerase II-transcribed genes involved in the metabolism of other carbon sources. Carbon catabolite repression is a mechanism of genetic regulation which the accumulation of catabolites of one substance in the cell represses the formation of enzymes that contribute to the catabolism of other substances.\" [GOC:krc]'],\n",
       "  'synonym': ['\"down regulation of transcription from RNA polymerase II promoter by glucose\" EXACT []',\n",
       "   '\"down-regulation of transcription from RNA polymerase II promoter by glucose\" EXACT []',\n",
       "   '\"downregulation of transcription from RNA polymerase II promoter by glucose\" EXACT []',\n",
       "   '\"inhibition of transcription from RNA polymerase II promoter by glucose\" NARROW []'],\n",
       "  'is_a': ['GO:0045014 ! carbon catabolite repression of transcription by glucose',\n",
       "   'GO:0061987 ! negative regulation of transcription from RNA polymerase II promoter by glucose']},\n",
       " 'GO:0000434': {'name': ['carbon catabolite repression of transcription from RNA polymerase II promoter by galactose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving galactose that stops, prevents or reduces the rate of transcription from an RNA polymerase II promoter.\" [GOC:krc]'],\n",
       "  'synonym': ['\"down regulation of transcription from RNA polymerase II promoter by galactose\" EXACT []',\n",
       "   '\"down-regulation of transcription from RNA polymerase II promoter by galactose\" EXACT []',\n",
       "   '\"downregulation of transcription from RNA polymerase II promoter by galactose\" EXACT []',\n",
       "   '\"inhibition of transcription from RNA polymerase II promoter by galactose\" NARROW []'],\n",
       "  'is_a': ['GO:0000410 ! carbon catabolite repression of transcription by galactose',\n",
       "   'GO:0000431 ! regulation of transcription from RNA polymerase II promoter by galactose',\n",
       "   'GO:0000437 ! carbon catabolite repression of transcription from RNA polymerase II promoter']},\n",
       " 'GO:0000435': {'name': ['positive regulation of transcription from RNA polymerase II promoter by galactose'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving galactose that activates or increases the rate of transcription from an RNA polymerase II promoter.\" [GOC:krc]'],\n",
       "  'synonym': ['\"activation of transcription from RNA polymerase II promoter by galactose\" NARROW []',\n",
       "   '\"stimulation of transcription from RNA polymerase II promoter by galactose\" NARROW []',\n",
       "   '\"up regulation of transcription from RNA polymerase II promoter by galactose\" EXACT []',\n",
       "   '\"up-regulation of transcription from RNA polymerase II promoter by galactose\" EXACT []',\n",
       "   '\"upregulation of transcription from RNA polymerase II promoter by galactose\" EXACT []'],\n",
       "  'is_a': ['GO:0000411 ! positive regulation of transcription by galactose',\n",
       "   'GO:0000431 ! regulation of transcription from RNA polymerase II promoter by galactose',\n",
       "   'GO:0000436 ! carbon catabolite activation of transcription from RNA polymerase II promoter']},\n",
       " 'GO:0000436': {'name': ['carbon catabolite activation of transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involving carbon catabolites that activates or increases the frequency, rate or extent of transcription from an RNA polymerase II promoter.\" [GOC:krc]'],\n",
       "  'synonym': ['\"positive regulation of transcription from RNA polymerase II promoter by carbon catabolites\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0000429 ! carbon catabolite regulation of transcription from RNA polymerase II promoter',\n",
       "   'GO:0045944 ! positive regulation of transcription by RNA polymerase II',\n",
       "   'GO:0045991 ! carbon catabolite activation of transcription']},\n",
       " 'GO:0000437': {'name': ['carbon catabolite repression of transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of one carbon source leads to a decrease in the frequency, rate, or extent of transcription, from an RNA polymerase II promoter, of specific genes involved in the metabolism of other carbon sources.\" [GOC:krc]'],\n",
       "  'synonym': ['\"negative regulation of transcription from RNA polymerase II promoter by carbon catabolites\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0000122 ! negative regulation of transcription by RNA polymerase II',\n",
       "   'GO:0000429 ! carbon catabolite regulation of transcription from RNA polymerase II promoter',\n",
       "   'GO:0045013 ! carbon catabolite repression of transcription']},\n",
       " 'GO:0000438': {'name': ['core TFIIH complex portion of holo TFIIH complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000443'],\n",
       "  'def': ['\"The core TFIIH complex when it is part of the general transcription factor TFIIH.\" [GOC:ew, GOC:krc, PMID:14500720, PMID:22308316, PMID:22572993, PMID:7813015]'],\n",
       "  'synonym': ['\"SSL2-core TFIIH complex portion of holo TFIIH complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000439 ! transcription factor TFIIH core complex'],\n",
       "  'relationship': ['part_of GO:0005675 ! transcription factor TFIIH holo complex']},\n",
       " 'GO:0000439': {'name': ['transcription factor TFIIH core complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000441'],\n",
       "  'def': ['\"The 7 subunit core of TFIIH that is a part of either the general transcription factor holo-TFIIH or the nucleotide-excision repair factor 3 complex. In S. cerevisiae/humans the complex is composed of'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"core TFIIH complex\" EXACT []',\n",
       "   '\"SSL2-core TFIIH complex\" EXACT []'],\n",
       "  'is_a': ['GO:0090575 ! RNA polymerase II transcription regulator complex']},\n",
       " 'GO:0000440': {'name': ['core TFIIH complex portion of NEF3 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000442'],\n",
       "  'def': ['\"The core TFIIH complex when it is part of the nucleotide-excision repair factor 3 (NEF3).\" [GOC:ew, GOC:krc, PMID:14500720, PMID:22308316, PMID:22572993, PMID:7813015]'],\n",
       "  'synonym': ['\"SSL2-core TFIIH complex portion of NEF3 complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000439 ! transcription factor TFIIH core complex'],\n",
       "  'relationship': ['part_of GO:0000112 ! nucleotide-excision repair factor 3 complex']},\n",
       " 'GO:0000444': {'name': ['MIS12/MIND type complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000818'],\n",
       "  'def': ['\"A multiprotein kinetochore subcomplex that binds to centromeric chromatin and forms part of the inner kinetochore. It helps to recruit outer kinetochore subunits that will bind to microtubules. In humans, it consists of MIS12, DSN1, NSL1 and PMF1.\" [GOC:krc, PMID:14633972, PMID:16585270]'],\n",
       "  'synonym': ['\"MIND complex\" EXACT []',\n",
       "   '\"Mis12 complex\" NARROW []',\n",
       "   '\"Mtw1p Including Nnf1p-Nsl1p-Dsn1p complex\" RELATED []',\n",
       "   '\"nuclear MIS12/MIND complex\" RELATED []'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0000939 ! inner kinetochore']},\n",
       " 'GO:0000445': {'name': ['THO complex part of transcription export complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The THO complex when it is part of the TREX (TRanscription EXport) complex that is involved in coupling transcription to export of mRNAs to the cytoplasm. In S. cerevisiae, it is composed of four subunits'],\n",
       "  'synonym': ['\"THO complex part of TREX complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000347 ! THO complex'],\n",
       "  'relationship': ['part_of GO:0000346 ! transcription export complex']},\n",
       " 'GO:0000446': {'name': ['nucleoplasmic THO complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The THO complex when it is acting as a nuclear complex that is required for transcription elongation through genes containing tandemly repeated DNA sequences. In S. cerevisiae, it is composed of four subunits'],\n",
       "  'is_a': ['GO:0000347 ! THO complex',\n",
       "   'GO:0008023 ! transcription elongation factor complex']},\n",
       " 'GO:0000447': {'name': ['endonucleolytic cleavage in ITS1 to separate SSU-rRNA from 5.8S rRNA and LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage between the SSU-rRNA and the 5.8S rRNA of an rRNA molecule originally produced as a tricistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators, PMID:10690410]'],\n",
       "  'synonym': ['\"endonucleolytic cleavage at A2\" NARROW []'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000462 ! maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)',\n",
       "   'part_of GO:0000466 ! maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000448': {'name': ['cleavage in ITS2 between 5.8S rRNA and LSU-rRNA of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000490'],\n",
       "  'def': ['\"Endonucleolytic cleavage within ITS2 between the 5.8S rRNA and the LSU-rRNA of an rRNA molecule originally produced as a tricistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators, PMID:10690410]'],\n",
       "  'synonym': ['\"cleavage at C2\" RELATED []',\n",
       "   '\"cleavage in ITS2 of tricistronic rRNA transcript to separate 5.8S and LSU rRNAs (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\" EXACT []'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000463 ! maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)',\n",
       "   'part_of GO:0000466 ! maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000449': {'name': ['endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 5S)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage of a pre-rRNA molecule originally produced as a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the Large Subunit (LSU) rRNA, and the 5S rRNA, in that order, from 5\\' to 3\\' along the primary transcript. For example, primary ribosomal RNA transcripts containing three genes, in this order, are produced in E. coli and other prokaryotic species. Note that the use of the word tricistronic refers only to the number of mature rRNA molecules which will be produced from the primary transcript and ignores tRNAs that may also be present within the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000450': {'name': ['cleavage of bicistronic rRNA transcript (SSU-rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage of pre-rRNAs originally produced as a bicistronic rRNA transcript that contains the SSU-rRNA and the LSU-rRNA in that order from 5\\' to 3\\' along the primary transcript. Primary ribosomal RNA transcripts with two genes in this order are produced in Archaeal species.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000451': {'name': [\"rRNA 2'-O-methylation\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The addition of a methyl group to the 2\\'-oxygen atom of a nucleotide residue in an rRNA molecule during ribosome biogenesis.\" [GOC:curators, ISBN:1555811337]'],\n",
       "  'is_a': ['GO:0031167 ! rRNA methylation']},\n",
       " 'GO:0000452': {'name': [\"snoRNA guided rRNA 2'-O-methylation\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The posttranscriptional addition of methyl groups to the 2\\'-oxygen atom of nucleotide residues in an rRNA molecule during ribosome biogenesis using a snoRNA guide that targets the position of methylation.\" [GOC:curators, ISBN:1555811337]'],\n",
       "  'synonym': ['\"snoRNA guided rRNA 2\\'-O-ribose methylation\" RELATED [GOC:ascb_2009, GOC:dph, GOC:tb]'],\n",
       "  'is_a': [\"GO:0000451 ! rRNA 2'-O-methylation\"]},\n",
       " 'GO:0000453': {'name': [\"obsolete enzyme-directed rRNA 2'-O-methylation\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The addition of methyl groups to the 2\\'-oxygen atom of nucleotide residues in an rRNA molecule during ribosome biogenesis where the methylase specifies the site that becomes methylated without using a guide RNA.\" [GOC:curators, ISBN:1555811337]'],\n",
       "  'comment': ['This term  was obsoleted because it represents a molecular function.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006364']},\n",
       " 'GO:0000454': {'name': ['snoRNA guided rRNA pseudouridine synthesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The intramolecular conversion of uridine to pseudouridine in an rRNA molecule during ribosome biogenesis using a snoRNA guide that targets the position of pseudouridylation.\" [GOC:curators, ISBN:1555811337]'],\n",
       "  'is_a': ['GO:0031118 ! rRNA pseudouridine synthesis']},\n",
       " 'GO:0000455': {'name': ['enzyme-directed rRNA pseudouridine synthesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The intramolecular conversion of uridine to pseudouridine during ribosome biogenesis where the enzyme specifies the site that becomes pseudouridylated without using a guide RNA.\" [GOC:curators, ISBN:1555811337]'],\n",
       "  'is_a': ['GO:0031118 ! rRNA pseudouridine synthesis']},\n",
       " 'GO:0000456': {'name': ['obsolete dimethylation involved in SSU-rRNA maturation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Dimethylation of the N6 amino groups of two consecutive adenosine residues near the 3\\'-end of the SSU rRNA. This process has been conserved from bacteria to eukaryotes.\" [GOC:curators, GOC:dph, GOC:tb, ISBN:1555811337]'],\n",
       "  'comment': ['The reason for obsoletion is that this term can be captured as a GO-CAM model.'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000457': {'name': ['endonucleolytic cleavage between SSU-rRNA and LSU-rRNA of tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 5S)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage to separate a pre-SSU-rRNA from a pre-LSU-rRNA originally produced as a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the Large Subunit (LSU) rRNA, and the 5S rRNA, in that order, from 5\\' to 3\\' along the primary transcript. Note that the use of the word tricistronic refers only to the number of mature rRNA molecules which will be produced from the primary transcript and ignores tRNAs that may also be present within the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000449 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 5S)'],\n",
       "  'relationship': ['part_of GO:0002108 ! maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA,5S)',\n",
       "   'part_of GO:0002109 ! maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA,5S)']},\n",
       " 'GO:0000458': {'name': ['endonucleolytic cleavage between LSU-rRNA and 5S rRNA of tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 5S)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage to separate a pre-LSU-rRNA from a pre-5S rRNA originally produced as a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the Large Subunit (LSU) rRNA, and the 5S rRNA, in that order, from 5\\' to 3\\' along the primary transcript. Note that the use of the word tricistronic refers only to the number of mature rRNA molecules which will be produced from the primary transcript and ignores tRNAs that may also be present within the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000449 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 5S)'],\n",
       "  'relationship': ['part_of GO:0002108 ! maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, LSU-rRNA,5S)']},\n",
       " 'GO:0000459': {'name': ['obsolete exonucleolytic trimming involved in rRNA processing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Exonucleolytic digestion of a pre-rRNA molecule in the process to generate a mature rRNA molecule.\" [GOC:curators]'],\n",
       "  'comment': ['The reason for obsoletion is that this term is pre-composed and should be represented as a GO-CAM model.'],\n",
       "  'synonym': ['\"exonucleolytic trimming during rRNA processing\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0006364']},\n",
       " 'GO:0000460': {'name': ['maturation of 5.8S rRNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor 5.8S ribosomal RNA (rRNA) molecule into a mature 5.8S rRNA molecule.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000461': {'name': [\"endonucleolytic cleavage to generate mature 3'-end of SSU-rRNA from (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage at the 3\\'-end of the SSU-rRNA from an originally tricistronic rRNA transcript that contained the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript, to produce the mature end of the SSU-rRNA.\" [GOC:krc, PMID:10690410]'],\n",
       "  'synonym': ['\"endonucleolytic cleavage at site D\" NARROW []'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)',\n",
       "   \"GO:0031125 ! rRNA 3'-end processing\"],\n",
       "  'relationship': ['part_of GO:0000462 ! maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000462': {'name': ['maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:1990041'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Small SubUnit (SSU) ribosomal RNA (rRNA) molecule into a mature SSU-rRNA molecule from the pre-rRNA molecule originally produced as a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, 5.8S rRNA, and the Large Subunit (LSU) in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'synonym': ['\"maturation of 18S rRNA\" NARROW []'],\n",
       "  'is_a': ['GO:0030490 ! maturation of SSU-rRNA']},\n",
       " 'GO:0000463': {'name': ['maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Large SubUnit (LSU) ribosomal RNA (rRNA) molecule into a mature LSU-rRNA molecule from the pre-rRNA molecule originally produced as a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, 5.8S rRNA, and Large Subunit (LSU) in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000470 ! maturation of LSU-rRNA']},\n",
       " 'GO:0000464': {'name': ['endonucleolytic cleavage in ITS1 upstream of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage within Internal Transcribed Spacer 1 (ITS1) upstream of the 5.8S rRNA derived from an originally tricistronic rRNA transcript that contained the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript. In S. cerevisiae, this endonucleolytic cleavage within ITS1 initiates the maturation of the LSU and the 5.8S rRNAs.\" [GOC:krc, PMID:10690410]'],\n",
       "  'synonym': ['\"endonucleolytic cleavage at A3\" NARROW []'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000463 ! maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)',\n",
       "   'part_of GO:0000466 ! maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000465': {'name': [\"exonucleolytic trimming to generate mature 5'-end of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Exonucleolytic digestion of a pre-rRNA molecule to generate the mature 5\\'-end of a 5.8S rRNA molecule derived from an originally tricistronic pre-rRNA transcript that contained the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:krc, PMID:10690410]'],\n",
       "  'is_a': [\"GO:0000967 ! rRNA 5'-end processing\"],\n",
       "  'relationship': ['part_of GO:0000466 ! maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000466': {'name': ['maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of an rRNA molecule originally produced as part of a tricistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators, PMID:10690410]'],\n",
       "  'is_a': ['GO:0000460 ! maturation of 5.8S rRNA']},\n",
       " 'GO:0000467': {'name': [\"exonucleolytic trimming to generate mature 3'-end of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Exonucleolytic digestion of a pre-rRNA molecule to generate the mature 3\\'-end of a 5.8S rRNA molecule derived from an originally tricistronic pre-rRNA transcript that contained the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:krc, PMID:10690410]'],\n",
       "  'is_a': [\"GO:0031125 ! rRNA 3'-end processing\"],\n",
       "  'relationship': ['part_of GO:0000466 ! maturation of 5.8S rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000468': {'name': [\"generation of mature 3'-end of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in generating the mature 3\\'-end of an LSU-rRNA derived from a tricistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:krc, PMID:10690410]'],\n",
       "  'synonym': ['\"processing at B2\" NARROW []'],\n",
       "  'is_a': [\"GO:0031125 ! rRNA 3'-end processing\"],\n",
       "  'relationship': ['part_of GO:0000463 ! maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000469': {'name': ['obsolete cleavage involved in rRNA processing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any phosphodiester bond hydrolysis involved in the conversion of a primary ribosomal RNA (rRNA) transcript into a mature rRNA molecule.\" [GOC:curators]'],\n",
       "  'comment': ['The reason for obsoletion is that this term is pre-composed and should be represented as a GO-CAM model.'],\n",
       "  'synonym': ['\"cleavage during rRNA processing\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0006364']},\n",
       " 'GO:0000470': {'name': ['maturation of LSU-rRNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Large SubUnit (LSU) ribosomal RNA (rRNA) molecule into a mature LSU-rRNA molecule.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing'],\n",
       "  'relationship': ['part_of GO:0042273 ! ribosomal large subunit biogenesis']},\n",
       " 'GO:0000471': {'name': [\"endonucleolytic cleavage in 3'-ETS of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage within the 3\\'-External Transcribed Spacer (ETS) of a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript. In S. cerevisiae, endonucleolytic cleavage within the 3\\'-ETS of the pre-RNA, which may occur cotranscriptionally, is the first step in rRNA processing, and initiates a cascade of subsequent processing and modification events.\" [GOC:krc, PMID:10690410]'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000472': {'name': [\"endonucleolytic cleavage to generate mature 5'-end of SSU-rRNA from (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage between the 5\\'-External Transcribed Spacer (5\\'-ETS) and the 5\\' end of the SSU-rRNA of a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript, to produce the mature end of the SSU-rRNA.\" [GOC:curators, PMID:10690410]'],\n",
       "  'synonym': ['\"endonucleolytic cleavage at A1\" NARROW []'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)',\n",
       "   \"GO:0000967 ! rRNA 5'-end processing\"],\n",
       "  'relationship': ['part_of GO:0000462 ! maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000473': {'name': ['maturation of LSU-rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Large SubUnit (LSU) ribosomal RNA (rRNA) molecule into a mature LSU-rRNA molecule from the pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, 5.8 S rRNA, 2S rRNA, and Large Subunit (LSU) in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000470 ! maturation of LSU-rRNA']},\n",
       " 'GO:0000474': {'name': ['maturation of SSU-rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Small SubUnit (SSU) ribosomal RNA (rRNA) molecule into a mature SSU-rRNA molecule from the pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, 5.8 S rRNA, 2S rRNA, and Large Subunit (LSU) in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0030490 ! maturation of SSU-rRNA']},\n",
       " 'GO:0000475': {'name': ['maturation of 2S rRNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor 2S ribosomal RNA (rRNA) molecule into a mature 2S rRNA molecule.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000476': {'name': ['maturation of 4.5S rRNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor 4.5S ribosomal RNA (rRNA) molecule into a mature 4.5S rRNA molecule.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000477': {'name': [\"generation of mature 5'-end of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Cleavage within ITS2 to generate the mature 5\\'-end of an LSU-rRNA derived from a tricistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators, PMID:10690410]'],\n",
       "  'synonym': ['\"processing at C1\" NARROW []'],\n",
       "  'is_a': [\"GO:0000967 ! rRNA 5'-end processing\"],\n",
       "  'relationship': ['part_of GO:0000463 ! maturation of LSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000478': {'name': ['obsolete endonucleolytic cleavage involved in rRNA processing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any endonucleolytic cleavage involved in the conversion of a primary ribosomal RNA (rRNA) transcript into a mature rRNA molecule. Some endonucleolytic cleavages produce the mature end, while others are a step in the process of generating the mature end from the pre-rRNA.\" [GOC:krc, PMID:10690410]'],\n",
       "  'comment': ['The reason for obsoletion is that this term is pre-composed and should be represented as a GO-CAM model.'],\n",
       "  'synonym': ['\"endonucleolytic cleavage during rRNA processing\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000479': {'name': ['endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage of a pre-rRNA molecule originally produced as a tricistronic rRNA transcript that contains the Small SubUnit (SSU) rRNA, the 5.8S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript. Primary ribosomal RNA transcripts with three genes, in this order, are produced in the nuclei of many eukaryotic species, including S. cerevisiae.\" [GOC:curators, PMID:10690410]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000480': {'name': [\"endonucleolytic cleavage in 5'-ETS of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage within the 5\\'-External Transcribed Spacer (ETS) of a tricistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the 5.8S rRNA, and the Large Subunit (LSU) rRNA in that order from 5\\' to 3\\' along the primary transcript. Endonucleolytic cleavage within the 5\\'-ETS of the pre-RNA is conserved as one of the early steps of rRNA processing in all eukaryotes, but the specific position of cleavage is variable.\" [GOC:curators, PMID:10690410, PMID:15282326]'],\n",
       "  'synonym': ['\"endonucleolytic cleavage at A-prime\" NARROW [PMID:15282326]',\n",
       "   '\"endonucleolytic cleavage at A0\" NARROW []'],\n",
       "  'is_a': ['GO:0000479 ! endonucleolytic cleavage of tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000462 ! maturation of SSU-rRNA from tricistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000481': {'name': ['maturation of 5S rRNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor 5S ribosomal RNA (rRNA) molecule into a mature 5S rRNA molecule.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000482': {'name': ['maturation of 5S rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor 5S ribosomal RNA (rRNA) molecule into a mature 5S rRNA molecule from the pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, Large Subunit (LSU) the 4.5S rRNA, and the 5S rRNA in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000481 ! maturation of 5S rRNA']},\n",
       " 'GO:0000483': {'name': ['endonucleolytic cleavage of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage of a pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small SubUnit (SSU) rRNA, the 5.8S rRNA, 2S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript. Primary ribosomal RNA transcripts with four genes, in this order, are produced in the nuclei of D. melanogaster as well as in those of other dipteran species.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing']},\n",
       " 'GO:0000484': {'name': ['cleavage between SSU-rRNA and 5.8S rRNA of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage between the SSU-rRNA and the 5.8S rRNA of an rRNA molecule originally produced as a tetracistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, 2S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000483 ! endonucleolytic cleavage of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000474 ! maturation of SSU-rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)',\n",
       "   'part_of GO:0000487 ! maturation of 5.8S rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000485': {'name': ['cleavage between 2S rRNA and LSU-rRNA of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage between the LSU-rRNA and the 2S rRNA of an rRNA molecule originally produced as a tetracistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, 2S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators, PMID:768488]'],\n",
       "  'is_a': ['GO:0000483 ! endonucleolytic cleavage of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000473 ! maturation of LSU-rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)',\n",
       "   'part_of GO:0000475 ! maturation of 2S rRNA']},\n",
       " 'GO:0000486': {'name': ['cleavage between 5.8S rRNA and 2S rRNA of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Endonucleolytic cleavage between the 5.8S rRNA and the 2S rRNA of an rRNA molecule originally produced as a tetracistronic rRNA transcript that contained the Small SubUnit (SSU) rRNA, the 5.8S rRNA, 2S rRNA, and the Large SubUnit (LSU) rRNA, in that order, from 5\\' to 3\\' along the primary transcript.\" [GOC:curators, PMID:768488]'],\n",
       "  'is_a': ['GO:0000483 ! endonucleolytic cleavage of tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'relationship': ['part_of GO:0000475 ! maturation of 2S rRNA',\n",
       "   'part_of GO:0000487 ! maturation of 5.8S rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)']},\n",
       " 'GO:0000487': {'name': ['maturation of 5.8S rRNA from tetracistronic rRNA transcript (SSU-rRNA, 5.8S rRNA, 2S rRNA, LSU-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor 5.8S ribosomal RNA (rRNA) molecule into a mature 5.8S rRNA molecule from the pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, the 8.8S rRNA, the 2S rRNA, and the Large Subunit (LSU) in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000460 ! maturation of 5.8S rRNA']},\n",
       " 'GO:0000488': {'name': ['maturation of LSU-rRNA from tetracistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 4.5S-rRNA, 5S-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Large SubUnit (LSU) ribosomal RNA (rRNA) molecule into a mature LSU-rRNA molecule from the pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, Large Subunit (LSU) the 4.5S rRNA, and the 5S rRNA in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0000470 ! maturation of LSU-rRNA']},\n",
       " 'GO:0000489': {'name': ['maturation of SSU-rRNA from tetracistronic rRNA transcript (SSU-rRNA, LSU-rRNA, 4.5S-rRNA, 5S-rRNA)'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the maturation of a precursor Small SubUnit (SSU) ribosomal RNA (rRNA) molecule into a mature SSU-rRNA molecule from the pre-rRNA molecule originally produced as a tetracistronic rRNA transcript that contains the Small Subunit (SSU) rRNA, Large Subunit (LSU) the 4.5S rRNA, and the 5S rRNA in that order from 5\\' to 3\\' along the primary transcript.\" [GOC:curators]'],\n",
       "  'is_a': ['GO:0030490 ! maturation of SSU-rRNA']},\n",
       " 'GO:0000491': {'name': ['small nucleolar ribonucleoprotein complex assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of proteins and a snoRNA to form a small nucleolar ribonucleoprotein (snoRNP) complex.\" [GOC:krc]'],\n",
       "  'synonym': ['\"snoRNP assembly\" EXACT []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly']},\n",
       " 'GO:0000492': {'name': ['box C/D snoRNP assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of proteins and a box C/D snoRNA to form a box C/D small nucleolar ribonucleoprotein (snoRNP) complex.\" [GOC:krc]'],\n",
       "  'synonym': ['\"box C/D small nucleolar ribonucleoprotein complex assembly\" EXACT []'],\n",
       "  'is_a': ['GO:0000491 ! small nucleolar ribonucleoprotein complex assembly']},\n",
       " 'GO:0000493': {'name': ['box H/ACA snoRNP assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of proteins and a box H/ACA snoRNA to form a box H/ACA small nucleolar ribonucleoprotein (snoRNP) complex.\" [GOC:krc, PMID:12515383]'],\n",
       "  'synonym': ['\"box H/ACA small nucleolar ribonucleoprotein complex assembly\" EXACT []'],\n",
       "  'is_a': ['GO:0000491 ! small nucleolar ribonucleoprotein complex assembly']},\n",
       " 'GO:0000494': {'name': [\"box C/D RNA 3'-end processing\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in forming the mature 3\\' end of a box C/D RNA molecule.\" [GOC:krc]'],\n",
       "  'synonym': ['\"box C/D RNA 3\\' end processing\" EXACT []',\n",
       "   '\"box C/D snoRNA 3\\'-end processing\" NARROW []',\n",
       "   '\"box C/D sRNA 3\\'-end processing\" NARROW []'],\n",
       "  'is_a': [\"GO:0031126 ! sno(s)RNA 3'-end processing\",\n",
       "   'GO:0034963 ! box C/D RNA processing']},\n",
       " 'GO:0000495': {'name': [\"box H/ACA RNA 3'-end processing\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in forming the mature 3\\' end of a box H/ACA RNA molecule.\" [GOC:krc]'],\n",
       "  'synonym': ['\"box H/ACA RNA 3\\' end processing\" EXACT []',\n",
       "   '\"box H/ACA snoRNA 3\\'-end processing\" NARROW []',\n",
       "   '\"box H/ACA sRNA 3\\'-end processing\" NARROW []'],\n",
       "  'is_a': [\"GO:0031126 ! sno(s)RNA 3'-end processing\",\n",
       "   'GO:0034964 ! box H/ACA RNA processing']},\n",
       " 'GO:0000497': {'name': ['DNA template activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to nucleic acid via hydrogen bonds between the bases of a gene product molecule and the bases of a target DNA molecule.\" [GOC:krc]'],\n",
       "  'comment': ['Note that with respect to annotation, \"base pairing\" and its child terms are intended to be used to annotate the activity of gene products composed of nucleic acid, presumably RNA, to interact with DNA molecules via base pairing. Internal base pairing with itself is considered part of the secondary structure of the molecule and is not within the scope of GO function.'],\n",
       "  'synonym': ['\"base pairing with DNA\" RELATED []'],\n",
       "  'is_a': ['GO:0003677 ! DNA binding',\n",
       "   'GO:0140489 ! molecular template activity']},\n",
       " 'GO:0000500': {'name': ['RNA polymerase I upstream activating factor complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A complex required for the transcription of rDNA by RNA polymerase I. In yeast the complex consists of Rrrn5p, Rrn9p, Rrn10p, histones H3 and H4, and Uaf30p.\" [PMID:11500378]'],\n",
       "  'synonym': ['\"RNA polymerase I upstream activation factor complex\" EXACT []',\n",
       "   '\"UAF\" EXACT []'],\n",
       "  'is_a': ['GO:0000120 ! RNA polymerase I transcription regulator complex']},\n",
       " 'GO:0000502': {'name': ['proteasome complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A large multisubunit complex which catalyzes protein degradation, found in eukaryotes, archaea and some bacteria. In eukaryotes, this complex consists of the barrel shaped proteasome core complex and one or two associated proteins or complexes that act in regulating entry into or exit from the core.\" [GOC:rb, Wikipedia:Proteasome]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"26S proteasome\" NARROW []', '\"proteasome\" EXACT [GOC:cjm]'],\n",
       "  'is_a': ['GO:0140535 ! intracellular protein-containing complex',\n",
       "   'GO:1905369 ! endopeptidase complex']},\n",
       " 'GO:0000504': {'name': ['obsolete proteasome regulatory particle (sensu Bacteria)'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. A multisubunit complex that recognizes and unfolds ubiquitinated proteins, and translocates them to the core complex in an ATP dependent manner. As in, but not restricted to, the taxon Bacteria (Bacteria, ncbi_taxonomy_id:2).\" [GOC:rb]'],\n",
       "  'comment': ['This term was made obsolete because there is no ubiquitin in bacteria and they do not have proteasome regulatory particles. Instead they have proteasome-activating nucleotidase.'],\n",
       "  'synonym': ['\"26S proteasome\" NARROW []',\n",
       "   '\"proteasome regulatory particle (sensu Bacteria)\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000506': {'name': ['glycosylphosphatidylinositol-N-acetylglucosaminyltransferase (GPI-GnT) complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"An enzyme complex that catalyzes the transfer of GlcNAc from UDP-GlcNAc to an acceptor phosphatidylinositol, the first step in the production of GPI anchors for cell surface proteins. The complex contains PIG-A, PIG-C, PIG-H, PIG-Q, PIG-P, and DPM2 in human, and Eri1p, Gpi1p, Gpi2p, Gpi15p, Gpi19p, and Spt14p in budding yeast.\" [GOC:kp, GOC:rb, PMID:10944123, PMID:15163411]'],\n",
       "  'comment': [\"Note that this term should not be confused with 'GPI-anchor transamidase complex ; GO:0042765', which represents a distinct complex with a different catalytic activity.\"],\n",
       "  'synonym': ['\"GPI-GlcNAc transferase complex\" EXACT []',\n",
       "   '\"GPI-GnT complex\" EXACT []',\n",
       "   '\"GPI-N-acetylglucosaminyltransferase complex\" EXACT []'],\n",
       "  'is_a': ['GO:0098796 ! membrane protein complex',\n",
       "   'GO:0140534 ! endoplasmic reticulum protein-containing complex',\n",
       "   'GO:1990234 ! transferase complex'],\n",
       "  'relationship': ['part_of GO:0005789 ! endoplasmic reticulum membrane']},\n",
       " 'GO:0000510': {'name': ['H3-H4 histone complex chaperone activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A histone chaperone that carries a H3-H4 histone complex.\" [PMID:28053344]'],\n",
       "  'comment': ['Mammalian cells have seven known sequence variants of histone H3. These are denoted as Histone H3.1, Histone H3.2, Histone H3.3, Histone H3.4 (H3T), Histone H3.5, Histone H3.X and Histone H3.Y (https://en.wikipedia.org/wiki/Histone_H3).'],\n",
       "  'synonym': ['\"H3-H4 histone carrier activity\" EXACT []',\n",
       "   '\"H3.1-H4 histone complex chaperone activity\" NARROW []',\n",
       "   '\"H3.2-H4 histone complex chaperone activity\" NARROW []',\n",
       "   '\"H3.3-H4 histone complex chaperone activity\" NARROW []'],\n",
       "  'is_a': ['GO:0140713 ! histone chaperone activity']},\n",
       " 'GO:0000511': {'name': ['H2A-H2B histone complex chaperone activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A histone chaperone that carries a H2A-H2B histone complex.\" [PMID:28053344]'],\n",
       "  'synonym': ['\"H2A-H2B histone carrier activity\" EXACT []'],\n",
       "  'is_a': ['GO:0140713 ! histone chaperone activity']},\n",
       " 'GO:0000512': {'name': ['lncRNA-mediated post-transcriptional gene silencing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A post-transcriptional gene silencing pathway in which regulatory long noncoding RNAs (lncRNAs) elicit silencing of specific target genes, often miRNAs or mRNAs.\" [PMID:34454184, PMID:35055152]'],\n",
       "  'comment': ['This term should be applied to lncRNAs and proteins associated with this process and not to the target (miRNAs or mRNAs) of the process.'],\n",
       "  'synonym': ['\"lncRNA-mediated gene silencing\" BROAD []'],\n",
       "  'is_a': ['GO:0035194 ! ncRNA-mediated post-transcriptional gene silencing']},\n",
       " 'GO:0000513': {'name': ['actin severing activator activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binds to and increases the activity of a actin severing protein.\" [PMID:25451933]'],\n",
       "  'is_a': ['GO:0140677 ! molecular function activator activity']},\n",
       " 'GO:0000514': {'name': ['3-sulfino-L-alanine'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of a solute or solutes from one side of a membrane to the other according to the reaction'],\n",
       "  'xref': ['RHEA:70967'],\n",
       "  'is_a': ['GO:0015078 ! proton transmembrane transporter activity',\n",
       "   'GO:0015175 ! neutral L-amino acid transmembrane transporter activity',\n",
       "   'GO:0015179 ! L-amino acid transmembrane transporter activity',\n",
       "   'GO:0042887 ! amide transmembrane transporter activity',\n",
       "   'GO:0072349 ! modified amino acid transmembrane transporter activity',\n",
       "   'GO:0140848 ! amino acid:monoatomic cation antiporter activity',\n",
       "   'GO:1901680 ! sulfur-containing amino acid secondary active transmembrane transporter activity',\n",
       "   'GO:1901702 ! salt transmembrane transporter activity']},\n",
       " 'GO:0000515': {'name': ['aspartate:glutamate, proton antiporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of a solute or solutes from one side of a membrane to the other according to the reaction'],\n",
       "  'xref': ['RHEA:70783'],\n",
       "  'is_a': ['GO:0005310 ! dicarboxylic acid transmembrane transporter activity',\n",
       "   'GO:0015078 ! proton transmembrane transporter activity',\n",
       "   'GO:0015556 ! C4-dicarboxylate transmembrane transporter activity',\n",
       "   'GO:0042887 ! amide transmembrane transporter activity',\n",
       "   'GO:0140848 ! amino acid:monoatomic cation antiporter activity',\n",
       "   'GO:1901702 ! salt transmembrane transporter activity']},\n",
       " 'GO:0000578': {'name': ['embryonic axis specification'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The establishment, maintenance and elaboration of a pattern along a line or a point in an embryo.\" [GOC:dph, GOC:go_curators, GOC:sdb_2009, GOC:tb]'],\n",
       "  'synonym': ['\"embryonic axis determination\" RELATED []'],\n",
       "  'is_a': ['GO:0009798 ! axis specification',\n",
       "   'GO:0009880 ! embryonic pattern specification']},\n",
       " 'GO:0000700': {'name': ['mismatch base pair DNA N-glycosylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the removal of single bases present in mismatches by the cleavage the N-C1\\' glycosidic bond between the target damaged DNA base and the deoxyribose sugar. The reaction releases a free base and leaves an apurinic/apyrimidinic (AP) site.\" [GOC:elh, PMID:9224623]'],\n",
       "  'is_a': ['GO:0019104 ! DNA N-glycosylase activity']},\n",
       " 'GO:0000701': {'name': ['purine-specific mismatch base pair DNA N-glycosylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the removal of purines present in mismatches, especially opposite oxidized purines, by cleaving the N-C1\\' glycosidic bond between the target damaged DNA base and the deoxyribose sugar. The reaction releases a free base and leaves an apurinic (AP) site.\" [GOC:elh, PMID:9224623]'],\n",
       "  'synonym': ['\"A/G-specific adenine DNA glycosylase activity\" NARROW []'],\n",
       "  'xref': ['EC:3.2.2.31'],\n",
       "  'is_a': ['GO:0000700 ! mismatch base pair DNA N-glycosylase activity']},\n",
       " 'GO:0000702': {'name': ['oxidized base lesion DNA N-glycosylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the removal of oxidized bases by cleaving the N-C1\\' glycosidic bond between the target damaged DNA base and the deoxyribose sugar. The reaction releases a free base and leaves an apurinic/apyrimidinic (AP) site.\" [GOC:elh, PMID:11554296]'],\n",
       "  'is_a': ['GO:0019104 ! DNA N-glycosylase activity']},\n",
       " 'GO:0000703': {'name': ['oxidized pyrimidine nucleobase lesion DNA N-glycosylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0019004'],\n",
       "  'def': ['\"Catalysis of the removal oxidized pyrimidine bases by cleaving the N-C1\\' glycosidic bond between the oxidized pyrimidine and the deoxyribose sugar. The reaction involves formation of a covalent enzyme-pyrimidine base intermediate. Release of the enzyme and free base by a beta-elimination or a beta, gamma-elimination mechanism results in the cleavage of the DNA backbone 3\\' of the apyrimidinic (AP) site.\" [GOC:elh, PMID:11554296]'],\n",
       "  'comment': [\"Consider also annotating to the molecular function term 'DNA-(apurinic or apyrimidinic site) lyase activity ; GO:0003906'.\"],\n",
       "  'synonym': ['\"bifunctional DNA glycosylase\" BROAD []',\n",
       "   '\"DNA glycosylase/AP-lyase\" BROAD []',\n",
       "   '\"DNA glycosylase/beta-lyase\" BROAD []',\n",
       "   '\"endodeoxyribonuclease III\" RELATED []',\n",
       "   '\"endonuclease III\" RELATED []',\n",
       "   '\"endonuclease VIII activity\" RELATED []',\n",
       "   '\"oxidized pyrimidine base lesion DNA N-glycosylase activity\" EXACT [GOC:go_curators]',\n",
       "   '\"pyrimidine-specific oxidized base lesion DNA N-glycosylase activity\" RELATED []'],\n",
       "  'xref': ['Reactome:R-HSA-110224 \"Cleavage of thymine glycol by NTHL1 glycosylase\"',\n",
       "   'Reactome:R-HSA-110226 \"Cleavage of cytosine glycol by NTHL1 glycosylase\"',\n",
       "   'Reactome:R-HSA-110227 \"Cleavage of dihydrouracil by NTHL1 glycosylase\"'],\n",
       "  'is_a': ['GO:0000702 ! oxidized base lesion DNA N-glycosylase activity']},\n",
       " 'GO:0000704': {'name': ['pyrimidine dimer DNA N-glycosylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the removal of pyrimidine dimers by removing the 5\\' pyrimidine of the dimer by cleaving the N-C1\\' glycosidic bond between the 5\\' pyrimidine of the dimer and the deoxyribose sugar. The reaction releases the 5\\' pyrimidine of the dimer and leaves an apurinic (AP) site. The reaction involves the formation of a covalent enzyme substrate intermediate. Release of the enzyme and free base by a beta-elimination or a beta, gamma-elimination mechanism results in the cleavage of the DNA backbone 3\\' of the apyrimidinic (AP) site.\" [GOC:elh, PMID:9224623]'],\n",
       "  'is_a': ['GO:0019104 ! DNA N-glycosylase activity']},\n",
       " 'GO:0000705': {'name': ['achiasmate meiosis I'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The first division of meiosis in which homologous chromosomes are paired and segregated from each other, occurring in the constitutive absence of chiasmata.\" [GOC:elh, GOC:sart, PMID:10690419]'],\n",
       "  'synonym': ['\"achiasmate meiosis I nuclear division\" EXACT []'],\n",
       "  'is_a': ['GO:0007127 ! meiosis I']},\n",
       " 'GO:0000706': {'name': ['meiotic DNA double-strand break processing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle process in which the 5\\' to 3\\' exonucleolytic resection of the DNA at the site of the break to form a 3\\' single-strand DNA overhang occurs. This takes place during meiosis.\" [GOC:elh, PMID:9334324]'],\n",
       "  'is_a': ['GO:0000729 ! DNA double-strand break processing',\n",
       "   'GO:1903046 ! meiotic cell cycle process']},\n",
       " 'GO:0000707': {'name': ['meiotic DNA recombinase assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During meiosis, the aggregation, arrangement and bonding together of strand exchange proteins (recombinases) to form higher order oligomers on single-stranded DNA.\" [GOC:elh, PMID:11459983]'],\n",
       "  'is_a': ['GO:0000730 ! DNA recombinase assembly',\n",
       "   'GO:0061982 ! meiosis I cell cycle process']},\n",
       " 'GO:0000708': {'name': ['meiotic strand invasion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle process in which the nucleoprotein complex (composed of the broken single-strand DNA and the recombinase) searches and identifies a region of homology in intact duplex DNA. The broken single-strand DNA displaces the like strand and forms Watson-Crick base pairs with its complement, forming a duplex in which each strand is from one of the two recombining DNA molecules. This occurs during meiosis.\" [GOC:elh, PMID:10915877]'],\n",
       "  'synonym': ['\"meiotic D-loop biosynthesis\" RELATED []',\n",
       "   '\"meiotic D-loop formation\" RELATED []',\n",
       "   '\"meiotic displacement loop biosynthesis\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"meiotic displacement loop formation\" RELATED [GOC:mah, GOC:vw]'],\n",
       "  'is_a': ['GO:0042148 ! strand invasion',\n",
       "   'GO:0061982 ! meiosis I cell cycle process']},\n",
       " 'GO:0000709': {'name': ['meiotic joint molecule formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The conversion of the paired broken DNA and homologous duplex DNA into a four-stranded branched intermediate, known as a joint molecule, formed during meiotic recombination. These joint molecules contain Holliday junctions on either side of heteroduplex DNA.\" [GOC:elh, PMID:8521495]'],\n",
       "  'is_a': ['GO:1903046 ! meiotic cell cycle process'],\n",
       "  'relationship': ['part_of GO:0007131 ! reciprocal meiotic recombination']},\n",
       " 'GO:0000710': {'name': ['meiotic mismatch repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A system for the identification and correction of base-base mismatches, small insertion-deletion loops, and regions of heterology that are present in duplex DNA formed with strands from two recombining molecules. Correction of the mismatch can result in non-Mendelian segregation of alleles following meiosis.\" [GOC:elh, PMID:10357855]'],\n",
       "  'is_a': ['GO:0006298 ! mismatch repair',\n",
       "   'GO:0061982 ! meiosis I cell cycle process']},\n",
       " 'GO:0000711': {'name': ['meiotic DNA repair synthesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During meiosis, the synthesis of DNA proceeding from the broken 3\\' single-strand DNA end that uses the homologous intact duplex as the template.\" [GOC:elh, PMID:9334324]'],\n",
       "  'is_a': ['GO:0000731 ! DNA synthesis involved in DNA repair',\n",
       "   'GO:0061982 ! meiosis I cell cycle process']},\n",
       " 'GO:0000712': {'name': ['resolution of meiotic recombination intermediates'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cleavage and rejoining of intermediates, such as Holliday junctions, formed during meiotic recombination to produce two intact molecules in which genetic material has been exchanged.\" [GOC:elh, PMID:11733053]'],\n",
       "  'synonym': ['\"resolution of meiotic joint molecules as recombinants\" NARROW [GOC:elh, GOC:mah]'],\n",
       "  'is_a': ['GO:0061982 ! meiosis I cell cycle process'],\n",
       "  'relationship': ['part_of GO:0007131 ! reciprocal meiotic recombination',\n",
       "   'part_of GO:0045132 ! meiotic chromosome segregation']},\n",
       " 'GO:0000713': {'name': ['meiotic heteroduplex formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During meiosis, the formation of a stable duplex DNA that contains one strand from each of the two recombining DNA molecules.\" [GOC:elh, PMID:9334324]'],\n",
       "  'is_a': ['GO:0061982 ! meiosis I cell cycle process'],\n",
       "  'relationship': ['part_of GO:0006310 ! DNA recombination']},\n",
       " 'GO:0000714': {'name': ['meiotic strand displacement'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle process in which the broken 3\\' single-strand DNA molecule that formed heteroduplex DNA with its complement in an intact duplex DNA is rejected. The Watson-Crick base pairing in the original duplex is restored. The rejected 3\\' single-strand DNA molecule reanneals with its original complement to reform two intact duplex molecules. This occurs during meiosis.\" [GOC:elh, PMID:10357855]'],\n",
       "  'synonym': ['\"meiotic D-loop dissociation\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"meiotic D-loop processing\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"meiotic displacement loop dissociation\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"meiotic displacement loop processing\" RELATED [GOC:mah, GOC:vw]'],\n",
       "  'is_a': ['GO:0000732 ! strand displacement',\n",
       "   'GO:0061982 ! meiosis I cell cycle process']},\n",
       " 'GO:0000715': {'name': ['nucleotide-excision repair, DNA damage recognition'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The identification of lesions in DNA, such as pyrimidine-dimers, intrastrand cross-links, and bulky adducts. The wide range of substrate specificity suggests the repair complex recognizes distortions in the DNA helix.\" [GOC:elh, PMID:10197977]'],\n",
       "  'synonym': ['\"pyrimidine-dimer repair, DNA damage recognition\" RELATED []'],\n",
       "  'is_a': ['GO:0051276 ! chromosome organization'],\n",
       "  'relationship': ['part_of GO:0006289 ! nucleotide-excision repair']},\n",
       " 'GO:0000716': {'name': ['transcription-coupled nucleotide-excision repair, DNA damage recognition'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The identification of lesions on the actively transcribed strand of the DNA duplex as well as a small subset of lesions not recognized by the general nucleotide-excision repair pathway.\" [GOC:elh, PMID:10197977]'],\n",
       "  'synonym': ['\"pyrimidine-dimer repair, DNA damage recognition\" RELATED []'],\n",
       "  'is_a': ['GO:0000715 ! nucleotide-excision repair, DNA damage recognition'],\n",
       "  'relationship': ['part_of GO:0006283 ! transcription-coupled nucleotide-excision repair']},\n",
       " 'GO:0000717': {'name': ['nucleotide-excision repair, DNA duplex unwinding'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The unwinding, or local denaturation, of the DNA duplex to create a bubble around the site of the DNA damage.\" [GOC:elh, PMID:10197977]'],\n",
       "  'is_a': ['GO:0032508 ! DNA duplex unwinding'],\n",
       "  'relationship': ['part_of GO:0006289 ! nucleotide-excision repair']},\n",
       " 'GO:0000718': {'name': ['nucleotide-excision repair, DNA damage removal'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The removal of the oligonucleotide that contains the DNA damage. The oligonucleotide is formed by dual incisions that flank the site of DNA damage.\" [GOC:elh, PMID:10197977]'],\n",
       "  'is_a': ['GO:0044349 ! DNA excision'],\n",
       "  'relationship': ['part_of GO:0006289 ! nucleotide-excision repair']},\n",
       " 'GO:0000719': {'name': ['photoreactive repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The repair of UV-induced T-T, C-T and C-C dimers by directly reversing the damage to restore the original pyrimidines.\" [GOC:elh, PMID:10915863]'],\n",
       "  'synonym': ['\"pyrimidine-dimer repair by photolyase\" NARROW []'],\n",
       "  'is_a': ['GO:0006290 ! pyrimidine dimer repair']},\n",
       " 'GO:0000720': {'name': ['pyrimidine dimer repair by nucleotide-excision repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The repair of UV-induced T-T, C-T, and C-C dimers by the recognition and removal of the damaged DNA strand from the DNA helix as an oligonucleotide. The small gap left in the DNA helix is filled in by the sequential action of DNA polymerase and DNA ligase.\" [GOC:elh]'],\n",
       "  'comment': [\"Note that the repair of pyrimidine dimers by nucleotide excision repair involves the same gene products that are involved in general nucleotide excision repair. Consider also annotating to other children of the biological process term 'nucleotide-excision repair ; GO:0006289'.\"],\n",
       "  'is_a': ['GO:0006289 ! nucleotide-excision repair',\n",
       "   'GO:0006290 ! pyrimidine dimer repair']},\n",
       " 'GO:0000721': {'name': ['(R,R)-butanediol dehydrogenase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reversible reaction'],\n",
       "  'synonym': ['\"(R)-2,3-butanediol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"(R)-diacetyl reductase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"1-amino-2-propanol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"1-amino-2-propanol oxidoreductase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"2,3-butanediol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"aminopropanol oxidoreductase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"butylene glycol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"butyleneglycol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"D-(-)-butanediol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"D-1-amino-2-propanol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"D-1-amino-2-propanol:NAD(2) oxidoreductase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"D-aminopropanol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"D-butanediol dehydrogenase activity\" RELATED [EC:1.1.1.4]',\n",
       "   '\"diacetyl (acetoin) reductase activity\" RELATED [EC:1.1.1.4]'],\n",
       "  'xref': ['EC:1.1.1.4',\n",
       "   'MetaCyc:RR-BUTANEDIOL-DEHYDROGENASE-RXN',\n",
       "   'RHEA:24340'],\n",
       "  'is_a': ['GO:0016616 ! oxidoreductase activity, acting on the CH-OH group of donors, NAD or NADP as acceptor']},\n",
       " 'GO:0000722': {'name': ['telomere maintenance via recombination'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any recombinational process that contributes to the maintenance of proper telomeric length.\" [GOC:elh, PMID:11850777]'],\n",
       "  'synonym': ['\"telomerase-independent telomere maintenance\" RELATED []'],\n",
       "  'is_a': ['GO:0000723 ! telomere maintenance',\n",
       "   'GO:0006312 ! mitotic recombination']},\n",
       " 'GO:0000723': {'name': ['telomere maintenance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that contributes to the maintenance of proper telomeric length and structure by affecting and monitoring the activity of telomeric proteins, the length of telomeric DNA and the replication and repair of the DNA. These processes includes those that shorten, lengthen, replicate and repair the telomeric DNA sequences.\" [GOC:BHF, GOC:BHF_telomere, GOC:elh, GOC:rl, PMID:11092831]'],\n",
       "  'synonym': ['\"regulation of telomere length\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0006259 ! DNA metabolic process',\n",
       "   'GO:0032200 ! telomere organization']},\n",
       " 'GO:0000724': {'name': ['double-strand break repair via homologous recombination'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0016924'],\n",
       "  'def': ['\"The error-free repair of a double-strand break in DNA in which the broken DNA molecule is repaired using homologous sequences. A strand in the broken DNA searches for a homologous region in an intact chromosome to serve as the template for DNA synthesis. The restoration of two intact DNA molecules results in the exchange, reciprocal or nonreciprocal, of genetic material between the intact DNA molecule and the broken DNA molecule.\" [GOC:elh, PMID:10357855]'],\n",
       "  'synonym': ['\"HDR\" EXACT [GOC:vk]',\n",
       "   '\"homologous recombinational repair\" EXACT []',\n",
       "   '\"homology-directed repair\" EXACT [GOC:vk]',\n",
       "   '\"HRR\" EXACT []',\n",
       "   '\"Rad51-dependent recombinational repair\" EXACT [GOC:mah]',\n",
       "   '\"Rhp51-dependent recombinational repair\" EXACT []'],\n",
       "  'is_a': ['GO:0000725 ! recombinational repair',\n",
       "   'GO:0006302 ! double-strand break repair']},\n",
       " 'GO:0000725': {'name': ['recombinational repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A DNA repair process that involves the exchange, reciprocal or nonreciprocal, of genetic material between the broken DNA molecule and a homologous DNA region.\" [GOC:elh]'],\n",
       "  'xref': ['Wikipedia:Recombinational_repair'],\n",
       "  'is_a': ['GO:0006281 ! DNA repair', 'GO:0006310 ! DNA recombination']},\n",
       " 'GO:0000726': {'name': ['obsolete non-recombinational repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A DNA repair process in which that does not require the exchange of genetic material between the broken DNA molecule and a homologous DNA region.\" [GOC:elh]'],\n",
       "  'comment': ['This term was obsoleted because it was defined negatively, and it represents an unnecessary grouping class.'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000727': {'name': ['double-strand break repair via break-induced replication'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The error-free repair of a double-strand break in DNA in which the centromere-proximal end of a broken chromosome searches for a homologous region in an intact chromosome. DNA synthesis initiates from the 3\\' end of the invading DNA strand, using the intact chromosome as the template, and progresses to the end of the chromosome.\" [GOC:elh, PMID:10357855]'],\n",
       "  'is_a': ['GO:0000724 ! double-strand break repair via homologous recombination']},\n",
       " 'GO:0000729': {'name': ['DNA double-strand break processing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The 5\\' to 3\\' exonucleolytic resection of the DNA at the site of the break to form a 3\\' single-strand DNA overhang.\" [PMID:10357855]'],\n",
       "  'is_a': ['GO:0006259 ! DNA metabolic process'],\n",
       "  'relationship': ['part_of GO:0006302 ! double-strand break repair']},\n",
       " 'GO:0000730': {'name': ['DNA recombinase assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of strand exchange proteins (recombinases) into higher order oligomers on single-stranded DNA.\" [PMID:10357855]'],\n",
       "  'synonym': ['\"Rad51 nucleoprotein filament formation\" EXACT [GOC:elh, GOC:vw]'],\n",
       "  'is_a': ['GO:0065004 ! protein-DNA complex assembly',\n",
       "   'GO:0090735 ! DNA repair complex assembly'],\n",
       "  'relationship': ['part_of GO:0045003 ! double-strand break repair via synthesis-dependent strand annealing']},\n",
       " 'GO:0000731': {'name': ['DNA synthesis involved in DNA repair'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Synthesis of DNA that proceeds from the broken 3\\' single-strand DNA end and uses the homologous intact duplex as the template.\" [PMID:10357855]'],\n",
       "  'synonym': ['\"DNA repair synthesis\" BROAD []',\n",
       "   '\"DNA synthesis during DNA repair\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"mitotic DNA repair synthesis\" NARROW [GOC:mah]'],\n",
       "  'is_a': ['GO:0071897 ! DNA biosynthetic process'],\n",
       "  'relationship': ['part_of GO:0006281 ! DNA repair']},\n",
       " 'GO:0000732': {'name': ['strand displacement'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The rejection of the broken 3\\' single-strand DNA molecule that formed heteroduplex DNA with its complement in an intact duplex DNA. The Watson-Crick base pairing in the original duplex is restored. The rejected 3\\' single-strand DNA molecule reanneals with its original complement to reform two intact duplex molecules.\" [PMID:10357855]'],\n",
       "  'synonym': ['\"D-loop dissociation\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"D-loop processing\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"displacement loop dissociation\" RELATED [GOC:mah, GOC:vw]',\n",
       "   '\"displacement loop processing\" RELATED [GOC:mah, GOC:vw]'],\n",
       "  'is_a': ['GO:0006259 ! DNA metabolic process'],\n",
       "  'relationship': ['part_of GO:0006310 ! DNA recombination']},\n",
       " 'GO:0000733': {'name': ['obsolete DNA strand renaturation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The identification and annealing of complementary base pairs in single-strand DNA.\" [GOC:elh]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0036310']},\n",
       " 'GO:0000735': {'name': ['removal of nonhomologous ends'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The removal of nonhomologous sequences at the broken 3\\' single-strand DNA end before DNA repair synthesis can occur.\" [PMID:10357855]'],\n",
       "  'is_a': ['GO:0006259 ! DNA metabolic process']},\n",
       " 'GO:0000736': {'name': ['double-strand break repair via single-strand annealing, removal of nonhomologous ends'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During DSBR via single-strand annealing, the removal of nonhomologous sequences at the broken 3\\' single-strand DNA end before DNA repair synthesis can occur.\" [PMID:10357855]'],\n",
       "  'is_a': ['GO:0000735 ! removal of nonhomologous ends'],\n",
       "  'relationship': ['part_of GO:0045002 ! double-strand break repair via single-strand annealing']},\n",
       " 'GO:0000737': {'name': ['obsolete DNA catabolic process, endonucleolytic'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The chemical reactions and pathways resulting in the breakdown of DNA, involving the hydrolysis of internal 3\\',5\\'-phosphodiester bonds in one or two strands of deoxyribonucleotides.\" [GOC:elh, GOC:mah]'],\n",
       "  'comment': ['The reason is that these terms represent Molecular functions in the Biological Process ontology.'],\n",
       "  'synonym': ['\"DNA breakdown, endonucleolytic\" EXACT []',\n",
       "   '\"DNA degradation, endonucleolytic\" EXACT []',\n",
       "   '\"endonucleolytic degradation of DNA\" EXACT []',\n",
       "   '\"endonucleolytic DNA catabolism\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0006308']},\n",
       " 'GO:0000738': {'name': ['obsolete DNA catabolic process, exonucleolytic'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The chemical reactions and pathways resulting in the breakdown of DNA, involving the hydrolysis of terminal 3\\',5\\'-phosphodiester bonds in one or two strands of deoxyribonucleotides.\" [GOC:elh, GOC:mah]'],\n",
       "  'comment': ['The reason is that these terms represent Molecular functions in the Biological Process ontology.'],\n",
       "  'synonym': ['\"DNA breakdown, exonucleolytic\" EXACT []',\n",
       "   '\"DNA degradation, exonucleolytic\" EXACT []',\n",
       "   '\"exonucleolytic degradation of DNA\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0006308']},\n",
       " 'GO:0000739': {'name': ['obsolete DNA strand annealing activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Facilitates the base-pairing of complementary single-stranded DNA.\" [GOC:elh]'],\n",
       "  'comment': ['This term was made obsolete because it describes a process and not an activity.'],\n",
       "  'synonym': ['\"DNA strand annealing activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0036292', 'GO:0036310']},\n",
       " 'GO:0000740': {'name': ['nuclear membrane fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The joining of 2 or more lipid bilayer membranes that surround the nucleus.\" [GOC:elh]'],\n",
       "  'is_a': ['GO:0071763 ! nuclear membrane organization',\n",
       "   'GO:0090174 ! organelle membrane fusion']},\n",
       " 'GO:0000741': {'name': ['karyogamy'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007335'],\n",
       "  'def': ['\"The creation of a single nucleus from multiple nuclei as a result of fusing the lipid bilayers that surround each nuclei.\" [GOC:elh]'],\n",
       "  'synonym': ['\"nuclear fusion\" EXACT []',\n",
       "   '\"nuclear fusion during karyogamy\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Karyogamy'],\n",
       "  'is_a': ['GO:0006997 ! nucleus organization',\n",
       "   'GO:0048284 ! organelle fusion']},\n",
       " 'GO:0000742': {'name': ['karyogamy involved in conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During sexual reproduction, the creation of a single nucleus from multiple nuclei as a result of fusing the lipid bilayers that surround each nuclei. This occurs after cytogamy.\" [GOC:elh]'],\n",
       "  'synonym': ['\"karyogamy during conjugation with cellular fusion\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0000741 ! karyogamy', 'GO:0022414 ! reproductive process'],\n",
       "  'relationship': ['part_of GO:0000747 ! conjugation with cellular fusion']},\n",
       " 'GO:0000743': {'name': ['nuclear migration involved in conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0006946'],\n",
       "  'def': ['\"The microtubule-based movement of nuclei towards one another as a prelude to karyogamy in organisms undergoing conjugation with cellular fusion.\" [GOC:clt, GOC:vw, PMID:16380440]'],\n",
       "  'synonym': ['\"nuclear congression\" EXACT [GOC:vw]',\n",
       "   '\"nuclear migration during conjugation with cellular fusion\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0022414 ! reproductive process',\n",
       "   'GO:0030473 ! nuclear migration along microtubule'],\n",
       "  'relationship': ['part_of GO:0000747 ! conjugation with cellular fusion']},\n",
       " 'GO:0000744': {'name': ['karyogamy involved in conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During sexual reproduction, the creation of a single nucleus from two nuclei as a result of fusing the nuclear envelopes that surround each nuclei. This takes place following the mutual exchange of one of the two nuclei produced by the mitosis that follows the second meiotic nuclear division. This occurs in ciliated protozoans such as Tetrahymena.\" [GOC:mah, GOC:pg]'],\n",
       "  'synonym': ['\"karyogamy involved in conjugation without cellular fusion\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0000741 ! karyogamy', 'GO:0022414 ! reproductive process'],\n",
       "  'relationship': ['part_of GO:0000748 ! conjugation with mutual genetic exchange']},\n",
       " 'GO:0000745': {'name': ['nuclear migration involved in conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The net movement of nuclei towards one another, leading to the bilateral transfer of genetic material in organisms undergoing conjugation without cellular fusion.\" [GOC:clt, GOC:mah]'],\n",
       "  'synonym': ['\"nuclear exchange during conjugation without cellular fusion\" RELATED []',\n",
       "   '\"nuclear migration involved in conjugation without cellular fusion\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0007097 ! nuclear migration',\n",
       "   'GO:0022414 ! reproductive process'],\n",
       "  'relationship': ['part_of GO:0000744 ! karyogamy involved in conjugation with mutual genetic exchange']},\n",
       " 'GO:0000746': {'name': ['obsolete conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The union or introduction of genetic information from compatible mating types that results in a genetically different individual. Conjugation requires direct cellular contact between the organisms.\" [GOC:elh]'],\n",
       "  'comment': ['This term was obsoleted because it is an unnecessary grouping class.'],\n",
       "  'xref': ['Wikipedia:Conjugation'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000747': {'name': ['conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007322', 'GO:0007333', 'GO:0030461', 'GO:0030477'],\n",
       "  'def': ['\"A conjugation process that results in the union of cellular and genetic information from compatible mating types. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:elh]'],\n",
       "  'subset': ['goslim_candida', 'goslim_pir', 'goslim_pombe', 'goslim_yeast'],\n",
       "  'synonym': ['\"cell fusion\" RELATED []', '\"mating\" RELATED []'],\n",
       "  'xref': ['Wikipedia:Conjugation'],\n",
       "  'is_a': ['GO:0019953 ! sexual reproduction']},\n",
       " 'GO:0000748': {'name': ['conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A conjugation process that results in the mutual exchange and union of only genetic information between compatible mating types. Conjugation without cellular fusion requires direct cellular contact between the organisms without plasma membrane fusion. The organisms involved in conjugation without cellular fusion separate after nuclear exchange.\" [GOC:elh, PMID:22444146]'],\n",
       "  'synonym': ['\"conjugation without cellular fusion\" EXACT []'],\n",
       "  'is_a': ['GO:0019953 ! sexual reproduction']},\n",
       " 'GO:0000749': {'name': ['response to pheromone triggering conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007328', 'GO:0030434'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a pheromone stimulus that positively regulates the process of conjugation with cellular fusion. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:clt]'],\n",
       "  'synonym': ['\"response to pheromone during conjugation with cellular fusion\" RELATED [GOC:dph]'],\n",
       "  'is_a': ['GO:0071444 ! cellular response to pheromone'],\n",
       "  'relationship': ['positively_regulates GO:0000747 ! conjugation with cellular fusion']},\n",
       " 'GO:0000750': {'name': ['pheromone-dependent signal transduction involved in conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007330', 'GO:0030454'],\n",
       "  'def': ['\"A signal transduction process resulting in the relay, amplification or dampening of a signal generated in response to pheromone exposure in organisms that undergo conjugation with cellular fusion. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:clt]'],\n",
       "  'synonym': ['\"transduction of mating signal\" BROAD []'],\n",
       "  'is_a': ['GO:0032005 ! signal transduction involved in positive regulation of conjugation with cellular fusion'],\n",
       "  'relationship': ['part_of GO:0000749 ! response to pheromone triggering conjugation with cellular fusion']},\n",
       " 'GO:0000751': {'name': ['mitotic cell cycle G1 arrest in response to pheromone'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0030571'],\n",
       "  'def': ['\"The cell cycle regulatory process in which the mitotic cell cycle is halted during G1 as a result of a pheromone stimulus. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:clt, GOC:dph, GOC:mah, GOC:tb]'],\n",
       "  'synonym': ['\"cell cycle arrest in response to pheromone\" BROAD []'],\n",
       "  'is_a': ['GO:0045930 ! negative regulation of mitotic cell cycle'],\n",
       "  'relationship': ['part_of GO:0000749 ! response to pheromone triggering conjugation with cellular fusion']},\n",
       " 'GO:0000752': {'name': ['agglutination involved in conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007334'],\n",
       "  'def': ['\"The aggregation or adhesion of compatible mating types via complementary cell-cell interactions during conjugation with cellular fusion of a unicellular organism. An example of this process is agglutination in Saccharomyces cerevisiae.\" [GOC:elh]'],\n",
       "  'synonym': ['\"agglutination\" BROAD []',\n",
       "   '\"cell-cell adhesion during conjugation with cellular fusion\" EXACT []',\n",
       "   '\"cell-cell adhesion during mating\" EXACT []'],\n",
       "  'is_a': ['GO:0022414 ! reproductive process',\n",
       "   'GO:0051703 ! biological process involved in intraspecies interaction between organisms',\n",
       "   'GO:0098609 ! cell-cell adhesion'],\n",
       "  'relationship': ['part_of GO:0000747 ! conjugation with cellular fusion',\n",
       "   'part_of GO:0000749 ! response to pheromone triggering conjugation with cellular fusion']},\n",
       " 'GO:0000753': {'name': ['cell morphogenesis involved in conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007332'],\n",
       "  'def': ['\"The change in form (cell shape and size) that occurs during sexual reproduction in order to facilitate direct contact between the compatible mating types in organisms that undergo conjugation cellular fusion.\" [GOC:clt]'],\n",
       "  'synonym': ['\"shmoo orientation\" NARROW []', '\"shmooing\" NARROW []'],\n",
       "  'is_a': ['GO:0000902 ! cell morphogenesis',\n",
       "   'GO:0003006 ! developmental process involved in reproduction'],\n",
       "  'relationship': ['part_of GO:0000747 ! conjugation with cellular fusion']},\n",
       " 'GO:0000754': {'name': ['adaptation of signaling pathway by response to pheromone involved in conjugation with cellular fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007331', 'GO:0030453'],\n",
       "  'def': ['\"In organisms that undergo conjugation with cellular fusion, the process resulting in desensitization following exposure to pheromone stimulus that act to down-regulate further stimulation or block initial conjugation responses. An example of this is the adaptation to pheromone during conjugation with cellular fusion in Saccharomyces cerevisiae.\" [GOC:clt]'],\n",
       "  'synonym': ['\"adaptation of signalling pathway by response to pheromone involved in conjugation with cellular fusion\" EXACT [GOC:mah]',\n",
       "   '\"adaptation to pheromone during conjugation with cellular fusion\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"desensitization to pheromone during conjugation with cellular fusion\" EXACT []'],\n",
       "  'is_a': ['GO:0023058 ! adaptation of signaling pathway'],\n",
       "  'relationship': ['part_of GO:0000749 ! response to pheromone triggering conjugation with cellular fusion']},\n",
       " 'GO:0000755': {'name': ['cytogamy'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0000218', 'GO:0030462'],\n",
       "  'def': ['\"During conjugation with cellular fusion, the process resulting in creating a single cell from complementary mating types. The localized remodeling and dissolution of external protective structures allow the fusion of the plasma membranes and cytoplasmic mixing. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:elh]'],\n",
       "  'synonym': ['\"zygote formation\" RELATED []'],\n",
       "  'is_a': ['GO:0022413 ! reproductive process in single-celled organism'],\n",
       "  'relationship': ['part_of GO:0000747 ! conjugation with cellular fusion']},\n",
       " 'GO:0000756': {'name': ['response to pheromone regulating conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a pheromone stimulus regulating the process of conjugation without cellular fusion.\" [GOC:clt]'],\n",
       "  'synonym': ['\"response to pheromone triggering conjugation without cellular fusion\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0007154 ! cell communication',\n",
       "   'GO:0071444 ! cellular response to pheromone'],\n",
       "  'relationship': ['regulates GO:0000748 ! conjugation with mutual genetic exchange']},\n",
       " 'GO:0000757': {'name': ['obsolete signal transduction involved in regulation of conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A signal transduction process resulting in the relay, amplification or dampening of a signal generated in response to pheromone exposure in organisms that undergo conjugation without cellular fusion.\" [GOC:clt]'],\n",
       "  'comment': ['This term was obsoleted because its label and definitions were not clear. It had never been used for annotation.'],\n",
       "  'synonym': ['\"signal transduction involved in conjugation without cellular fusion\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000758': {'name': ['agglutination involved in conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation or adhesion of compatible mating types via complementary cell-cell interactions during conjugation without cellular fusion of a unicellular organism.\" [GOC:elh]'],\n",
       "  'synonym': ['\"agglutination involved in conjugation without cellular fusion\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"sexual flocculation\" EXACT []'],\n",
       "  'is_a': ['GO:0051703 ! biological process involved in intraspecies interaction between organisms',\n",
       "   'GO:0098609 ! cell-cell adhesion'],\n",
       "  'relationship': ['part_of GO:0000756 ! response to pheromone regulating conjugation with mutual genetic exchange']},\n",
       " 'GO:0000759': {'name': ['obsolete cell morphogenesis involved in conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The change in form (cell shape and size) that occurs during sexual reproduction in order to facilitate direct contact between the compatible mating types in organisms that undergo conjugation without cellular fusion.\" [GOC:clt]'],\n",
       "  'comment': ['This term was obsoleted because there is no evidence that this process exists.'],\n",
       "  'synonym': ['\"cellular morphogenesis involved in conjugation without cellular fusion\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000760': {'name': ['adaptation to pheromone regulating conjugation with mutual genetic exchange'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"In organisms that undergo conjugation without cellular fusion, the process resulting in desensitization following exposure to pheromone stimulus that act to down-regulate further stimulation or block initial conjugation responses.\" [GOC:clt]'],\n",
       "  'synonym': ['\"adaptation to pheromone involved conjugation without cellular fusion\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"desensitization to pheromone during conjugation without cellular fusion\" EXACT []'],\n",
       "  'is_a': ['GO:0000756 ! response to pheromone regulating conjugation with mutual genetic exchange',\n",
       "   'GO:0022401 ! negative adaptation of signaling pathway']},\n",
       " 'GO:0000761': {'name': ['conjugant formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"During conjugation without cellular fusion, the process that results in pairing complementary mating types. Localized morphological, cytological, and cytoskeletal changes connect the mating types without cytoplasmic mixing.\" [GOC:elh]'],\n",
       "  'is_a': ['GO:0022414 ! reproductive process',\n",
       "   'GO:0098610 ! adhesion between unicellular organisms'],\n",
       "  'relationship': ['part_of GO:0000748 ! conjugation with mutual genetic exchange']},\n",
       " 'GO:0000762': {'name': ['pheromone-induced unidirectional conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Unidirectional transfer of genetic information triggered by to a pheromone signal.\" [GOC:elh, PMID:17360276, PMID:27021562, PMID:31191478]'],\n",
       "  'is_a': ['GO:0009291 ! unidirectional conjugation']},\n",
       " 'GO:0000763': {'name': ['obsolete cell morphogenesis involved in unidirectional conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The change in form (cell shape and size) that occurs during sexual reproduction in order to facilitate direct contact between the compatible mating types in organisms that undergo unidirectional conjugation.\" [GOC:clt]'],\n",
       "  'comment': ['This term was made obsolete because unidirectional conjugation occurs only in prokaryotes, which do not undergo morphogenetic changes associated with conjugation. In short, the process described by this term does not occur.'],\n",
       "  'synonym': ['\"cell morphogenesis involved in unidirectional conjugation\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000764': {'name': ['obsolete cellular morphogenesis involved in pheromone-induced unidirectional conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The change in form (cell shape and size) that contributes to sexual reproduction in order to facilitate direct contact between the compatible mating types in organisms that undergo pheromone-induced unidirectional conjugation.\" [GOC:clt]'],\n",
       "  'comment': ['This term was made obsolete because unidirectional conjugation occurs only in prokaryotes, which do not undergo morphogenetic changes associated with conjugation. In short, the process described by this term does not occur.'],\n",
       "  'synonym': ['\"cellular morphogenesis during pheromone-induced unidirectional\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"cellular morphogenesis involved in pheromone-induced unidirectional conjugation\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000765': {'name': ['response to pheromone regulating pheromone-induced unidirectional conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a pheromone stimulus that regulates the process of pheromone-induced unidirectional conjugation.\" [GOC:clt]'],\n",
       "  'synonym': ['\"response to pheromone during pheromone-induced unidirectional\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0019236 ! response to pheromone'],\n",
       "  'relationship': ['regulates GO:0000762 ! pheromone-induced unidirectional conjugation']},\n",
       " 'GO:0000766': {'name': ['negative adaptation of signaling pathway by response to pheromone involved in pheromone-induced unidirectional conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"In organisms that undergo pheromone-induced unidirectional conjugation, the process involved in desensitization following exposure to pheromone stimulus that acts to down-regulate further stimulation or block initial conjugation responses.\" [GOC:clt]'],\n",
       "  'synonym': ['\"adaptation to pheromone during pheromone-induced unidirectional conjugation\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"desensitization to pheromone during pheromone-induced unidirectional conjugation\" EXACT []',\n",
       "   '\"negative adaptation of signalling pathway by response to pheromone involved in pheromone-induced unidirectional conjugation\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0022401 ! negative adaptation of signaling pathway'],\n",
       "  'relationship': ['part_of GO:0000765 ! response to pheromone regulating pheromone-induced unidirectional conjugation']},\n",
       " 'GO:0000767': {'name': ['obsolete cell morphogenesis involved in conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The change in form (cell shape and size) that occurs during sexual reproduction in order to facilitate direct contact between the compatible mating types.\" [GOC:elh]'],\n",
       "  'comment': ['This term was obsoleted because it is an unnecessary grouping class.'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000768': {'name': ['syncytium formation by plasma membrane fusion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of a syncytium, a mass of cytoplasm containing several nuclei enclosed within a single plasma membrane, by the fusion of the plasma membranes of two or more individual cells.\" [GOC:mtg_muscle, GOC:tb]'],\n",
       "  'synonym': ['\"cell fusion\" BROAD []'],\n",
       "  'is_a': ['GO:0006949 ! syncytium formation',\n",
       "   'GO:0140253 ! cell-cell fusion']},\n",
       " 'GO:0000769': {'name': ['syncytium formation by mitosis without cytokinesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of a syncytium, a mass of cytoplasm containing several nuclei enclosed within a single plasma membrane, by one or more rounds of nuclear division without cytokinesis.\" [GOC:mah, GOC:tb]'],\n",
       "  'synonym': ['\"syncytium formation by mitosis without cell division\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0006949 ! syncytium formation']},\n",
       " 'GO:0000770': {'name': ['peptide pheromone export'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007325'],\n",
       "  'def': ['\"The directed movement of a peptide pheromone out of a cell by a secretion or export pathway used solely for the export of peptide pheromones.\" [GOC:elh]'],\n",
       "  'synonym': ['\"a-factor export\" NARROW []'],\n",
       "  'is_a': ['GO:0015833 ! peptide transport']},\n",
       " 'GO:0000771': {'name': ['obsolete agglutination involved in conjugation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The aggregation or adhesion of compatible mating types via complementary cell-cell interactions prior to the formation of irreversible cellular contacts during conjugation.\" [GOC:elh]'],\n",
       "  'comment': ['This term was obsoleted because it is an unnecessary grouping class.'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000772': {'name': ['mating pheromone activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"The activity of binding to and activating specific cell surface receptors, thereby inducing a behavioral or physiological response(s) from a responding organism or cell that leads to the transfer or union of genetic material between organisms or cells. The mating pheromone can either be retained on the cell surface or secreted.\" [GOC:clt, GOC:elh]'],\n",
       "  'is_a': ['GO:0005186 ! pheromone activity']},\n",
       " 'GO:0000773': {'name': ['phosphatidyl-N-methylethanolamine N-methyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"methyltransferase II\" RELATED [EC:2.1.1.71]',\n",
       "   '\"phosphatidyl-N-methylethanolamine methyltransferase activity\" RELATED [EC:2.1.1.71]',\n",
       "   '\"phosphatidyl-N-monomethylethanolamine methyltransferase activity\" RELATED [EC:2.1.1.71]',\n",
       "   '\"phosphatidylethanolamine methyltransferase I\" RELATED [EC:2.1.1.71]',\n",
       "   '\"phosphatidylmonomethylethanolamine methyltransferase activity\" RELATED [EC:2.1.1.71]',\n",
       "   '\"phospholipid methyltransferase activity\" RELATED [EC:2.1.1.71]',\n",
       "   '\"PLMT\" EXACT []',\n",
       "   '\"S-adenosyl-L-methionine:phosphatidyl-N-methylethanolamine N-methyltransferase activity\" RELATED [EC:2.1.1.71]'],\n",
       "  'xref': ['EC:2.1.1.71', 'MetaCyc:2.1.1.71-RXN', 'RHEA:32735'],\n",
       "  'is_a': ['GO:0008170 ! N-methyltransferase activity',\n",
       "   'GO:0008757 ! S-adenosylmethionine-dependent methyltransferase activity']},\n",
       " 'GO:0000774': {'name': ['adenyl-nucleotide exchange factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binds to and stimulates the hydrolysis and exchange of adenyl nucleotides by other proteins.\" [GOC:kd]'],\n",
       "  'xref': ['Reactome:R-HSA-5252079 \"HSP110s exchange ATP for ADP on HSP70s:ADP\"'],\n",
       "  'is_a': ['GO:0030554 ! adenyl nucleotide binding',\n",
       "   'GO:0060590 ! ATPase regulator activity']},\n",
       " 'GO:0000775': {'name': ['chromosome, centromeric region'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0097521'],\n",
       "  'def': ['\"The region of a chromosome that includes the centromeric DNA and associated proteins. In monocentric chromosomes, this region corresponds to a single area of the chromosome, whereas in holocentric chromosomes, it is evenly distributed along the chromosome.\" [GOC:cjm, GOC:elh, GOC:kmv, GOC:pr]'],\n",
       "  'comment': [\"Note that this term can be used in place of the obsolete cellular component term 'centromere ; GO:0005698'.\"],\n",
       "  'synonym': ['\"centromere\" RELATED []',\n",
       "   '\"centromere complex\" EXACT []',\n",
       "   '\"chromosome, centric region\" EXACT []',\n",
       "   '\"chromosome, pericentric region\" RELATED []'],\n",
       "  'xref': ['SO:0000577'],\n",
       "  'is_a': ['GO:0098687 ! chromosomal region']},\n",
       " 'GO:0000776': {'name': ['kinetochore'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000777', 'GO:0000778', 'GO:0005699', 'GO:0031617'],\n",
       "  'def': ['\"A multisubunit complex that is located at the centromeric region of DNA and provides an attachment point for the spindle microtubules.\" [GOC:elh]'],\n",
       "  'comment': ['Note that the kinetochore overlaps the centromeric DNA, but centromeric DNA is not part of the kinetochore.'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"condensed chromosome kinetochore\" EXACT []',\n",
       "   '\"condensed nuclear chromosome kinetochore\" EXACT []',\n",
       "   '\"NMS complex\" NARROW []'],\n",
       "  'xref': ['Wikipedia:Kinetochore'],\n",
       "  'is_a': ['GO:0043232 ! intracellular non-membrane-bounded organelle',\n",
       "   'GO:0099080 ! supramolecular complex'],\n",
       "  'relationship': ['part_of GO:0000779 ! condensed chromosome, centromeric region']},\n",
       " 'GO:0000779': {'name': ['condensed chromosome, centromeric region'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000780'],\n",
       "  'def': ['\"The region of a condensed chromosome that includes the centromere and associated proteins, including the kinetochore. In monocentric chromosomes, this region corresponds to a single area of the chromosome, whereas in holocentric chromosomes, it is evenly distributed along the chromosome.\" [GOC:elh, GOC:kmv]'],\n",
       "  'comment': [\"Note that this term can be used in place of the obsolete cellular component term 'centromere ; GO:0005698'. Use with caution because this term refers to a specific region of the chromosome and not a protein complex.\"],\n",
       "  'synonym': ['\"condensed chromosome, centric region\" EXACT []',\n",
       "   '\"condensed chromosome, centromere\" RELATED []',\n",
       "   '\"condensed chromosome, pericentric region\" RELATED []',\n",
       "   '\"condensed nuclear chromosome, centromeric region\" NARROW []'],\n",
       "  'is_a': ['GO:0000775 ! chromosome, centromeric region'],\n",
       "  'relationship': ['part_of GO:0000793 ! condensed chromosome']},\n",
       " 'GO:0000781': {'name': ['chromosome, telomeric region'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000784'],\n",
       "  'def': ['\"The end of a linear chromosome, required for the integrity and maintenance of the end. A chromosome telomere usually includes a region of telomerase-encoded repeats the length of which rarely exceeds 20 bp each and that permits the formation of a telomeric loop (T-loop). The telomeric repeat region is usually preceded by a sub-telomeric region that is gene-poor but rich in repetitive elements. Some telomeres only consist of the latter part (for eg. D. melanogaster telomeres).\" [GOC:elh]'],\n",
       "  'comment': [\"Note that this term can be used in place of the obsolete cellular component term 'telomere ; GO:0005696'. Use with caution because this term refers to a specific region of the chromosome and not a protein complex.\"],\n",
       "  'synonym': ['\"nuclear chromosome, telomere\" NARROW []',\n",
       "   '\"nuclear chromosome, telomeric region\" NARROW []',\n",
       "   '\"telomere\" RELATED []'],\n",
       "  'xref': ['SO:0000624'],\n",
       "  'is_a': ['GO:0098687 ! chromosomal region']},\n",
       " 'GO:0000782': {'name': ['telomere cap complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A complex of DNA and protein located at the end of a linear chromosome that protects and stabilizes a linear chromosome.\" [GOC:elh]'],\n",
       "  'comment': [\"Note that this term can be used in place of the obsolete cellular component term 'telomere ; GO:0005696'. Use with caution because this term refers to a specific protein complex and not a region of the chromosome.\"],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'is_a': ['GO:0032993 ! protein-DNA complex'],\n",
       "  'relationship': ['part_of GO:0140445 ! chromosome, telomeric repeat region']},\n",
       " 'GO:0000783': {'name': ['nuclear telomere cap complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A complex of DNA and protein located at the end of a linear chromosome in the nucleus that protects and stabilizes a linear chromosome.\" [GOC:elh]'],\n",
       "  'comment': [\"Note that this term can be used in place of the obsolete cellular component term 'telomere ; GO:0005696'. Use with caution because this term refers to a specific protein complex and not a region of the chromosome.\"],\n",
       "  'is_a': ['GO:0000782 ! telomere cap complex',\n",
       "   'GO:0140513 ! nuclear protein-containing complex']},\n",
       " 'GO:0000785': {'name': ['chromatin'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000789', 'GO:0000790', 'GO:0005717'],\n",
       "  'def': ['\"The ordered and organized complex of DNA, protein, and sometimes RNA, that forms the chromosome.\" [GOC:elh, PMID:20404130]'],\n",
       "  'comment': ['Chromosomes include parts that are not part of the chromatin. Examples include the kinetochore.'],\n",
       "  'synonym': ['\"chromosome scaffold\" RELATED []',\n",
       "   '\"cytoplasmic chromatin\" NARROW []',\n",
       "   '\"nuclear chromatin\" NARROW []'],\n",
       "  'xref': ['NIF_Subcellular:sao1615953555'],\n",
       "  'is_a': ['GO:0032993 ! protein-DNA complex'],\n",
       "  'relationship': ['part_of GO:0005694 ! chromosome']},\n",
       " 'GO:0000786': {'name': ['nucleosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000787', 'GO:0000788', 'GO:0005718'],\n",
       "  'def': ['\"A complex comprised of DNA wound around a multisubunit core and associated proteins, which forms the primary packing unit of DNA into higher order structures.\" [GOC:elh]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"cytoplasmic nucleosome\" NARROW []',\n",
       "   '\"nuclear nucleosome\" NARROW []'],\n",
       "  'xref': ['Wikipedia:Nucleosome'],\n",
       "  'is_a': ['GO:0032993 ! protein-DNA complex'],\n",
       "  'relationship': ['part_of GO:0000785 ! chromatin']},\n",
       " 'GO:0000791': {'name': ['euchromatin'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0005719', 'GO:0035327'],\n",
       "  'def': ['\"A dispersed and relatively uncompacted form of chromatin that is in a transcription-competent conformation.\" [PMID:32017156]'],\n",
       "  'synonym': ['\"nuclear euchromatin\" RELATED []',\n",
       "   '\"transcriptionally active chromatin\" EXACT []'],\n",
       "  'xref': ['NIF_Subcellular:sao445485807', 'Wikipedia:Euchromatin'],\n",
       "  'is_a': ['GO:0000785 ! chromatin']},\n",
       " 'GO:0000792': {'name': ['heterochromatin'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0005720', 'GO:0035328'],\n",
       "  'def': ['\"A compact and highly condensed form of chromatin that is refractory to transcription.\" [PMID:32017156]'],\n",
       "  'synonym': ['\"nuclear heterochromatin\" NARROW []',\n",
       "   '\"transcriptionally inactive chromatin\" EXACT []',\n",
       "   '\"transcriptionally silent chromatin\" EXACT []'],\n",
       "  'xref': ['NIF_Subcellular:sao581845896', 'Wikipedia:Heterochromatin'],\n",
       "  'is_a': ['GO:0000785 ! chromatin']},\n",
       " 'GO:0000793': {'name': ['condensed chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A highly compacted molecule of DNA and associated proteins resulting in a cytologically distinct structure.\" [GOC:elh]'],\n",
       "  'comment': [\"Note that this term can be used to annotate gene products that localize to a mitotic chromosome in an organism that undergoes an 'open mitosis' in which the nuclear envelope breaks down during mitosis.\"],\n",
       "  'synonym': ['\"cytoplasmic mitotic chromosome\" RELATED []',\n",
       "   '\"metaphase chromosome\" RELATED []',\n",
       "   '\"mitotic chromosome\" RELATED []'],\n",
       "  'is_a': ['GO:0005694 ! chromosome']},\n",
       " 'GO:0000794': {'name': ['condensed nuclear chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A highly compacted molecule of DNA and associated proteins resulting in a cytologically distinct nuclear chromosome.\" [GOC:elh]'],\n",
       "  'comment': [\"Note that this term and its children can be used to annotate gene products that localize to a mitotic chromosome in an organism that undergoes a 'closed mitosis' in which the nuclear envelope does not break down during mitosis and for gene products that localize to a meiotic chromosome.\"],\n",
       "  'synonym': ['\"meiotic chromosome\" RELATED []',\n",
       "   '\"nuclear mitotic chromosome\" RELATED []'],\n",
       "  'is_a': ['GO:0000228 ! nuclear chromosome',\n",
       "   'GO:0000793 ! condensed chromosome']},\n",
       " 'GO:0000795': {'name': ['synaptonemal complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0005716'],\n",
       "  'def': ['\"A proteinaceous scaffold found between homologous chromosomes during meiosis. It consists of 2 lateral elements and a central element, all running parallel to each other. Transverse filaments connect the lateral elements to the central element.\" [DOI:10.5772/29752, GOC:elh]'],\n",
       "  'xref': ['Wikipedia:Synaptonemal_complex'],\n",
       "  'is_a': ['GO:0099086 ! synaptonemal structure']},\n",
       " 'GO:0000796': {'name': ['condensin complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000797',\n",
       "   'GO:0000799',\n",
       "   'GO:0005676',\n",
       "   'GO:0008620',\n",
       "   'GO:0008621',\n",
       "   'GO:0061814'],\n",
       "  'def': ['\"A multisubunit protein complex that plays a central role in chromosome condensation in meiosis and mitosis.\" [GOC:elh, PMID:17268547, PMID:21795393]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"13S condensin complex\" NARROW []',\n",
       "   '\"8S condensin complex\" NARROW []',\n",
       "   '\"condensin core heterodimer\" NARROW []',\n",
       "   '\"condensin I complex\" NARROW []',\n",
       "   '\"nuclear condensin complex\" NARROW []',\n",
       "   '\"SMC complex\" RELATED []',\n",
       "   '\"Smc2-Smc4 complex\" NARROW []'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005694 ! chromosome']},\n",
       " 'GO:0000800': {'name': ['lateral element'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A proteinaceous core found between sister chromatids during meiotic prophase.\" [GOC:elh]'],\n",
       "  'comment': ['In species that have a synaptonemal complex, the lateral elements are part of this complex. S.pombe is an example of a species that lacks a (canonical) synaptonemal complex, but still has lateral elements.'],\n",
       "  'synonym': ['\"axial element\" EXACT [GOC:ascb_2009, GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0000795 ! synaptonemal complex']},\n",
       " 'GO:0000801': {'name': ['central element'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A structural unit of the synaptonemal complex found between the lateral elements.\" [GOC:elh]'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0000795 ! synaptonemal complex']},\n",
       " 'GO:0000802': {'name': ['transverse filament'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A structural unit of the synaptonemal complex that spans the regions between the lateral elements and connects them.\" [GOC:elh]'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0000795 ! synaptonemal complex']},\n",
       " 'GO:0000803': {'name': ['sex chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A chromosome involved in sex determination.\" [GOC:elh]'],\n",
       "  'is_a': ['GO:0005694 ! chromosome']},\n",
       " 'GO:0000804': {'name': ['W chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The sex chromosome present in females of species in which the female is the heterogametic sex; generally, the sex chromosome that pairs with the Z chromosome in the heterogametic sex. The W chromosome is absent from the cells of males and present in one copy in the somatic cells of females.\" [GOC:mah, GOC:mr, ISBN:0321000382, PMID:20622855]'],\n",
       "  'xref': ['Wikipedia:ZW_sex-determination_system'],\n",
       "  'is_a': ['GO:0000803 ! sex chromosome']},\n",
       " 'GO:0000805': {'name': ['X chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The sex chromosome present in both sexes of species in which the male is the heterogametic sex. Two copies of the X chromosome are present in each somatic cell of females and one copy is present in males.\" [GOC:mah, GOC:mr, ISBN:0582227089, PMID:20622855, Wikipedia:XY_sex-determination_system]'],\n",
       "  'xref': ['Wikipedia:X_chromosome'],\n",
       "  'is_a': ['GO:0000803 ! sex chromosome']},\n",
       " 'GO:0000806': {'name': ['Y chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The sex chromosome present in males of species in which the male is the heterogametic sex; generally, the sex chromosome that pairs with the X chromosome in the heterogametic sex. The Y chromosome is absent from the cells of females and present in one copy in the somatic cells of males.\" [GOC:mah, GOC:mr, ISBN:0582227089, PMID:20622855, Wikipedia:XY_sex-determination_system]'],\n",
       "  'xref': ['Wikipedia:Y_chromosome'],\n",
       "  'is_a': ['GO:0000803 ! sex chromosome']},\n",
       " 'GO:0000807': {'name': ['Z chromosome'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The sex chromosome present in both sexes of species in which the female is the heterogametic sex. Two copies of the Z chromosome are present in each somatic cell of males and one copy is present in females.\" [GOC:mah, GOC:mr, ISBN:0321000382, PMID:20622855]'],\n",
       "  'xref': ['Wikipedia:ZW_sex-determination_system'],\n",
       "  'is_a': ['GO:0000803 ! sex chromosome']},\n",
       " 'GO:0000808': {'name': ['origin recognition complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multisubunit complex that is located at the replication origins of a chromosome.\" [GOC:elh]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'synonym': ['\"ORC\" EXACT []',\n",
       "   '\"origin of replication recognition complex\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Origin_recognition_complex'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005694 ! chromosome']},\n",
       " 'GO:0000809': {'name': ['cytoplasmic origin of replication recognition complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multisubunit complex that is located at the replication origins of a chromosome in the cytoplasm.\" [GOC:elh]'],\n",
       "  'synonym': ['\"cytoplasmic ORC\" EXACT []', '\"prokaryotic ORC\" RELATED []'],\n",
       "  'is_a': ['GO:0000808 ! origin recognition complex'],\n",
       "  'relationship': ['part_of GO:0005737 ! cytoplasm']},\n",
       " 'GO:0000810': {'name': ['diacylglycerol diphosphate phosphatase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"DGPP phosphatase activity\" EXACT []',\n",
       "   '\"DGPP phosphohydrolase activity\" EXACT []',\n",
       "   '\"diacylglycerol pyrophosphate phosphatase activity\" EXACT []'],\n",
       "  'xref': ['EC:3.6.1.75', 'MetaCyc:RXN-11277', 'RHEA:27449'],\n",
       "  'is_a': ['GO:0016462 ! pyrophosphatase activity']},\n",
       " 'GO:0000811': {'name': ['GINS complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A heterotetrameric protein complex that associates with replication origins, where it is required for the initiation of DNA replication, and with replication forks.\" [GOC:rb, GOC:rn, PMID:12730134, PMID:16990792, PMID:17467990]'],\n",
       "  'synonym': ['\"Go, Ichi, Ni and San complex\" EXACT []'],\n",
       "  'is_a': ['GO:0031261 ! DNA replication preinitiation complex'],\n",
       "  'relationship': ['part_of GO:0000228 ! nuclear chromosome']},\n",
       " 'GO:0000812': {'name': ['Swr1 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multisubunit protein complex that is involved in chromatin remodeling. It is required for the incorporation of the histone variant H2AZ into chromatin. In S. cerevisiae, the complex contains Swr1p, a Swi2/Snf2-related ATPase, and 12 additional subunits.\" [GOC:rb, PMID:14645854, PMID:14690608, PMID:19355820]'],\n",
       "  'synonym': ['\"SWR-C\" EXACT []'],\n",
       "  'is_a': ['GO:0000118 ! histone deacetylase complex',\n",
       "   'GO:0097346 ! INO80-type complex'],\n",
       "  'relationship': ['part_of GO:0000228 ! nuclear chromosome']},\n",
       " 'GO:0000813': {'name': ['ESCRT I complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"An endosomal sorting complex required for transport. It consists of the class E vacuolar protein sorting (Vps) proteins and interacts with ubiquitinated cargoes.\" [GOC:rb, PMID:12892785, PMID:12900393]'],\n",
       "  'synonym': ['\"endosomal sorting complex required for transport\" BROAD []'],\n",
       "  'is_a': ['GO:0036452 ! ESCRT complex',\n",
       "   'GO:0098796 ! membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0010008 ! endosome membrane']},\n",
       " 'GO:0000814': {'name': ['ESCRT II complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"An endosomal sorting complex required for transport and functions downstream of ESCRT I complex. It consists of the class E vacuolar protein sorting (Vps) proteins and is required for the membrane recruitment of ESCRT III complex and binds to ubiquitinated cargoes.\" [GOC:rb, PMID:12892785, PMID:12900393]'],\n",
       "  'synonym': ['\"endosomal sorting complex required for transport\" BROAD []'],\n",
       "  'is_a': ['GO:0036452 ! ESCRT complex',\n",
       "   'GO:0098796 ! membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0010008 ! endosome membrane']},\n",
       " 'GO:0000815': {'name': ['ESCRT III complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A complex with membrane scission activity that plays a major role in many processes where membranes are remodelled - including endosomal transport (vesicle budding), nuclear envelope organisation (membrane closure, mitotic bridge cleavage), and cytokinesis (abscission).\" [PMID:17556548, PMID:22361144, PMID:28242692, PMID:31132588, PMID:32243490, PMID:34449766]'],\n",
       "  'synonym': ['\"endosomal sorting complex required for transport\" BROAD []'],\n",
       "  'is_a': ['GO:0036452 ! ESCRT complex',\n",
       "   'GO:0098796 ! membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0010008 ! endosome membrane']},\n",
       " 'GO:0000817': {'name': ['COMA complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A kinetochore multiprotein complex that bridges the subunits that are in contact with centromeric DNA and the subunits bound to microtubules during kinetochore assembly. In yeast, consists of Ctf19p, Okp1p, Mcm21p, and Ame1p.\" [GOC:se, PMID:14633972]'],\n",
       "  'synonym': ['\"Ctf19p-Okp1p-Mcm1p-Ame1p complex\" EXACT []'],\n",
       "  'is_a': ['GO:0140513 ! nuclear protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0000228 ! nuclear chromosome',\n",
       "   'part_of GO:0000776 ! kinetochore']},\n",
       " 'GO:0000819': {'name': ['sister chromatid segregation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell cycle process in which sister chromatids are organized and then physically separated and apportioned to two or more sets.\" [GOC:ai, GOC:elh]'],\n",
       "  'is_a': ['GO:0051276 ! chromosome organization',\n",
       "   'GO:0098813 ! nuclear chromosome segregation']},\n",
       " 'GO:0000820': {'name': ['regulation of glutamine family amino acid metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of the chemical reactions and pathways involving amino acids of the glutamine family, comprising arginine, glutamate, glutamine and proline.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"regulation of glutamine family amino acid metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006521 ! regulation of cellular amino acid metabolic process'],\n",
       "  'relationship': ['regulates GO:0009064 ! glutamine family amino acid metabolic process']},\n",
       " 'GO:0000821': {'name': ['regulation of arginine metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of the chemical reactions and pathways involving arginine, 2-amino-5-(carbamimidamido)pentanoic acid.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"regulation of arginine metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0000820 ! regulation of glutamine family amino acid metabolic process'],\n",
       "  'relationship': ['regulates GO:0006525 ! arginine metabolic process']},\n",
       " 'GO:0000822': {'name': ['inositol hexakisphosphate binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to inositol hexakisphosphate.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"InsP6 binding\" EXACT []', '\"IP6 binding\" EXACT []'],\n",
       "  'is_a': ['GO:0043168 ! anion binding', 'GO:0043178 ! alcohol binding']},\n",
       " 'GO:0000823': {'name': ['inositol-1,4,5-trisphosphate 6-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"ATP:1D-myo-inositol-1,4,5-trisphosphate 6-phosphotransferase activity\" EXACT []',\n",
       "   '\"inositol polyphosphate multikinase activity\" BROAD []',\n",
       "   '\"inositol trisphosphate 6-kinase activity\" BROAD []',\n",
       "   '\"IpmK\" BROAD []'],\n",
       "  'xref': ['EC:2.7.1.151', 'KEGG_REACTION:R05800', 'MetaCyc:2.7.1.151-RXN'],\n",
       "  'is_a': ['GO:0051766 ! inositol trisphosphate kinase activity']},\n",
       " 'GO:0000824': {'name': ['inositol tetrakisphosphate 3-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"1D-myo-inositol-tetrakisphosphate 3-kinase activity\" EXACT []',\n",
       "   '\"EC:2.7.1.151\" BROAD []',\n",
       "   '\"inositol 1,4,5,6-tetrakisphosphate 3-kinase activity\" EXACT []',\n",
       "   '\"inositol polyphosphate multikinase activity\" BROAD []',\n",
       "   '\"IpmK\" BROAD []'],\n",
       "  'xref': ['Reactome:R-HSA-1855185 \"I(1,4,5,6)P4 is phosphorylated to I(1,3,4,5,6)P5 by IPMK in the nucleus\"',\n",
       "   'RHEA:11856'],\n",
       "  'is_a': ['GO:0051765 ! inositol tetrakisphosphate kinase activity']},\n",
       " 'GO:0000825': {'name': ['inositol tetrakisphosphate 6-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"1D-myo-inositol-tetrakisphosphate 6-kinase activity\" EXACT []',\n",
       "   '\"inositol 1,3,4,5-tetrakisphosphate 6-kinase activity\" EXACT []'],\n",
       "  'xref': ['MetaCyc:RXN-7184',\n",
       "   'Reactome:R-HSA-1855206 \"I(1,3,4,5)P4 is phosphorylated to I(1,3,4,5,6)P5 by IPMK in the nucleus\"'],\n",
       "  'is_a': ['GO:0051765 ! inositol tetrakisphosphate kinase activity']},\n",
       " 'GO:0000826': {'name': ['obsolete inositol pyrophosphate synthase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the phosphorylation of inositol phosphates which possess diphosphate bonds.\" [GOC:elh, PMID:16429326]'],\n",
       "  'comment': [\"This term was made obsolete because it is incorrectly defined; 'inositol pyrophosphate' is a inositol derivative to which one (or more) pyrophosphate moieties are attached. 'Inositol pyrophosphate synthase' refers to a class of enzymes that catalyze various different reactions that result in the formation of an IP, so one cannot formulate a general reaction for them.\"],\n",
       "  'synonym': ['\"inositol diphosphate synthase activity\" EXACT []',\n",
       "   '\"inositol pyrophosphate synthase activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000827': {'name': ['inositol-1,3,4,5,6-pentakisphosphate kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'xref': ['EC:2.7.4.21',\n",
       "   'Reactome:R-HSA-1855181 \"I(1,3,4,5,6)P5 is phosphorylated to 5-PP-IP4 by IP6K1/2 in the nucleus\"',\n",
       "   'Reactome:R-HSA-1855223 \"I(1,3,4,5,6)P5 is phosphorylated to 5-PP-IP4 by IP6K1/3 in the cytosol\"'],\n",
       "  'is_a': ['GO:0016301 ! kinase activity',\n",
       "   'GO:0016776 ! phosphotransferase activity, phosphate group as acceptor']},\n",
       " 'GO:0000828': {'name': ['inositol hexakisphosphate kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"IP6 kinase\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-1855207 \"IP6 is phosphorylated to 5-PP-IP5 by IP6K1/2 in the nucleus\"',\n",
       "   'Reactome:R-HSA-1855216 \"IP6 is phosphorylated to 1-PP-IP5 by PPIP5K1/2 in the cytosol\"',\n",
       "   'Reactome:R-HSA-1855227 \"IP6 is phosphorylated to 5-PP-IP5 by IP6K1/3 in the cytosol\"'],\n",
       "  'is_a': ['GO:0016301 ! kinase activity',\n",
       "   'GO:0016776 ! phosphotransferase activity, phosphate group as acceptor']},\n",
       " 'GO:0000829': {'name': ['inositol heptakisphosphate kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'xref': ['Reactome:R-HSA-1855157 \"1-PP-IP5 is phosphorylated to 1,5-(PP)2-IP4 by IP6K1/2 in the nucleus\"',\n",
       "   'Reactome:R-HSA-1855182 \"5-PP-IP5 is phosphorylated to 1,5-(PP)2-IP4 by PPIP5K1/2 in the cytosol\"',\n",
       "   'Reactome:R-HSA-1855194 \"1-PP-IP5 is phosphorylated to 1,5-(PP)2-IP4 by IP6K1/3 in the cytosol\"'],\n",
       "  'is_a': ['GO:0016301 ! kinase activity',\n",
       "   'GO:0016776 ! phosphotransferase activity, phosphate group as acceptor']},\n",
       " 'GO:0000830': {'name': ['inositol hexakisphosphate 4-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'is_a': ['GO:0000828 ! inositol hexakisphosphate kinase activity']},\n",
       " 'GO:0000831': {'name': ['inositol hexakisphosphate 6-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'is_a': ['GO:0000828 ! inositol hexakisphosphate kinase activity']},\n",
       " 'GO:0000832': {'name': ['inositol hexakisphosphate 5-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"ATP:1D-myo-inositol-hexakisphosphate 5-phosphotransferase activity\" EXACT []'],\n",
       "  'xref': ['EC:2.7.4.21',\n",
       "   'KEGG_REACTION:R09087',\n",
       "   'MetaCyc:2.7.1.152-RXN',\n",
       "   'RHEA:12793'],\n",
       "  'is_a': ['GO:0000828 ! inositol hexakisphosphate kinase activity']},\n",
       " 'GO:0000833': {'name': ['inositol heptakisphosphate 4-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"diphosphoinositol-pentakisphosphate 4-kinase activity\" EXACT []',\n",
       "   '\"IP7 4-kinase activity\" EXACT []'],\n",
       "  'is_a': ['GO:0000829 ! inositol heptakisphosphate kinase activity']},\n",
       " 'GO:0000834': {'name': ['inositol heptakisphosphate 6-kinase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'is_a': ['GO:0000829 ! inositol heptakisphosphate kinase activity']},\n",
       " 'GO:0000835': {'name': ['ER ubiquitin ligase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A ubiquitin ligase complex found in the ER.\" [GOC:elh]'],\n",
       "  'is_a': ['GO:0000153 ! cytoplasmic ubiquitin ligase complex',\n",
       "   'GO:0098796 ! membrane protein complex',\n",
       "   'GO:0140534 ! endoplasmic reticulum protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005789 ! endoplasmic reticulum membrane']},\n",
       " 'GO:0000836': {'name': ['Hrd1p ubiquitin ligase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multiprotein complex that recognizes and ubiquitinates proteins with misfolded luminal and membrane domains during ER-associated protein degradation (ERAD). In S. cerevisiae, this complex contains the ubiquitin ligase Hrd1p. In mammals, this complex contains the ubiquitin ligase HRD1 (Synoviolin) or AMFR (gp78).\" [GOC:bf, GOC:elh, PMID:16619026, PMID:16873066, PMID:21454652]'],\n",
       "  'synonym': ['\"HRD1 ubiquitin ligase complex\" RELATED [GOC:bf]'],\n",
       "  'is_a': ['GO:0000835 ! ER ubiquitin ligase complex']},\n",
       " 'GO:0000837': {'name': ['Doa10p ubiquitin ligase complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multiprotein complex that recognizes and ubiquitinates membrane proteins with misfolded cytosolic domains during ER-associated protein degradation (ERAD). In S. cerevisiae, this complex contains the ubiquitin ligase Ssm4p/Doa10p.\" [GOC:elh, PMID:16873066]'],\n",
       "  'synonym': ['\"Ssm4p ubiquitin ligase complex\" EXACT []'],\n",
       "  'is_a': ['GO:0000835 ! ER ubiquitin ligase complex']},\n",
       " 'GO:0000838': {'name': ['Hrd1p ubiquitin ligase ERAD-M complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multiprotein complex that recognizes and ubiquitinates proteins with misfolded membrane domains during ER-associated protein degradation (ERAD). In S. cerevisiae, this complex contains the ubiquitin ligase Hrd1p.\" [GOC:elh, PMID:16873066]'],\n",
       "  'is_a': ['GO:0000836 ! Hrd1p ubiquitin ligase complex']},\n",
       " 'GO:0000839': {'name': ['Hrd1p ubiquitin ligase ERAD-L complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multiprotein complex that recognizes and ubiquitinates proteins with misfolded luminal domains during ER-associated protein degradation (ERAD). In S. cerevisiae, this complex contains the ubiquitin ligase Hrd1p.\" [GOC:elh, PMID:16873065, PMID:16873066]'],\n",
       "  'is_a': ['GO:0000836 ! Hrd1p ubiquitin ligase complex']},\n",
       " 'GO:0000900': {'name': ['mRNA regulatory element binding translation repressor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Antagonizes the ribosome-mediated translation of mRNA into a polypeptide via direct binding (through a selective and non-covalent interaction) to nucleic acid.\" [GOC:clt, GOC:vw, PMID:29061112, PMID:7523370]'],\n",
       "  'synonym': ['\"translation repressor activity, mRNA regulatory element binding\" EXACT []',\n",
       "   '\"translation repressor activity, nucleic acid binding\" BROAD []'],\n",
       "  'is_a': ['GO:0030371 ! translation repressor activity',\n",
       "   'GO:0090079 ! translation regulator activity, nucleic acid binding']},\n",
       " 'GO:0000901': {'name': ['translation repressor activity, non-nucleic acid binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Antagonizes the ribosome-mediated translation of mRNA into a polypeptide but does not bind directly to nucleic acid.\" [GOC:clt]'],\n",
       "  'is_a': ['GO:0030371 ! translation repressor activity',\n",
       "   'GO:0045183 ! translation factor activity, non-nucleic acid binding']},\n",
       " 'GO:0000902': {'name': ['cell morphogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007148', 'GO:0045790', 'GO:0045791'],\n",
       "  'def': ['\"The developmental process in which the size or shape of a cell is generated and organized.\" [GOC:clt, GOC:dph, GOC:go_curators, GOC:tb]'],\n",
       "  'subset': ['goslim_chembl', 'goslim_drosophila', 'goslim_yeast'],\n",
       "  'synonym': ['\"cellular morphogenesis\" EXACT []'],\n",
       "  'is_a': ['GO:0009653 ! anatomical structure morphogenesis']},\n",
       " 'GO:0000903': {'name': ['obsolete regulation of cell shape during vegetative growth phase'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0090061'],\n",
       "  'def': ['\"OBSOLETE. Any process that modulates the surface configuration of a cell during the vegetative growth phase. The vegetative growth phase is the growth phase during which single celled organisms reproduce by budding or other asexual methods.\" [GOC:clt, GOC:go_curators, GOC:vw]'],\n",
       "  'comment': ['The reason for obsoletion is that biological phases should be captured as extensions.'],\n",
       "  'synonym': ['\"cell morphogenesis during vegetative growth phase\" RELATED [GOC:vw]',\n",
       "   '\"cellular morphogenesis during vegetative growth\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"regulation of cell morphogenesis during vegetative growth phase\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0000904': {'name': ['cell morphogenesis involved in differentiation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The change in form (cell shape and size) that occurs when relatively unspecialized cells, e.g. embryonic or regenerative cells, acquire specialized structural and/or functional features that characterize the cells, tissues, or organs of the mature organism or some other relatively stable phase of the organism\\'s life history.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0000902 ! cell morphogenesis'],\n",
       "  'relationship': ['part_of GO:0048468 ! cell development']},\n",
       " 'GO:0000905': {'name': ['sporocarp development involved in asexual reproduction'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of a spore-bearing structure by fungus where spores will arise from asexual reproduction.\" [GOC:clt, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"conidium development\" NARROW []',\n",
       "   '\"fruiting body formation involved in asexual reproduction\" BROAD []',\n",
       "   '\"haploid fruiting\" RELATED []',\n",
       "   '\"homokaryotic fruiting\" RELATED []',\n",
       "   '\"imperfect stage fruiting body development\" NARROW []',\n",
       "   '\"monokaryotic fruiting\" RELATED []'],\n",
       "  'is_a': ['GO:0030584 ! sporocarp development'],\n",
       "  'relationship': ['part_of GO:0019954 ! asexual reproduction']},\n",
       " 'GO:0000906': {'name': ['6,7-dimethyl-8-ribityllumazine synthase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"lumazine synthase activity\" BROAD [MetaCyc:LUMAZINESYN-CPLX]'],\n",
       "  'xref': ['EC:2.5.1.78', 'MetaCyc:LUMAZINESYN-RXN', 'RHEA:26152'],\n",
       "  'is_a': ['GO:0016765 ! transferase activity, transferring alkyl or aryl (other than methyl) groups']},\n",
       " 'GO:0000907': {'name': ['sulfonate dioxygenase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"sulfonate/alpha-ketoglutarate dioxygenase activity\" EXACT []',\n",
       "   '\"sulphonate dioxygenase activity\" EXACT []'],\n",
       "  'is_a': ['GO:0016706 ! 2-oxoglutarate-dependent dioxygenase activity']},\n",
       " 'GO:0000908': {'name': ['taurine dioxygenase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"2-aminoethanesulfonate dioxygenase activity\" RELATED [EC:1.14.11.17]',\n",
       "   '\"alpha-ketoglutarate-dependent taurine dioxygenase activity\" RELATED [EC:1.14.11.17]',\n",
       "   '\"taurine, 2-oxoglutarate:O2 oxidoreductase (sulfite-forming)\" RELATED [EC:1.14.11.17]'],\n",
       "  'xref': ['EC:1.14.11.17',\n",
       "   'KEGG_REACTION:R05320',\n",
       "   'MetaCyc:RXN0-299',\n",
       "   'RHEA:15909'],\n",
       "  'is_a': ['GO:0016706 ! 2-oxoglutarate-dependent dioxygenase activity']},\n",
       " 'GO:0000909': {'name': ['sporocarp development involved in sexual reproduction'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of a fruiting body organ over time, from its formation to the mature structure. The fruiting body is a spore bearing structure. In fungi, the sporocarp (also known as fruiting body) is a multicellular structure on which spore-producing structures, such as basidia or asci, are borne. The fruiting body is part of the sexual phase of a fungal life cycle, with the rest of the life cycle being characterized by vegetative mycelial growth. The sporocarp of a basidiomycete is known as a basidiocarp, while the fruiting body of an ascomycete is known as an ascocarp. A significant range of different shapes and morphologies is found in both basidiocarps and ascocarps; these features play an important role in the identification and taxonomy of fungi.\" [GOC:clt, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"ascus development\" NARROW []',\n",
       "   '\"fruiting body development involved in sexual reproduction\" BROAD []',\n",
       "   '\"fruiting body formation involved in sexual reproduction\" BROAD []',\n",
       "   '\"perfect stage fruiting body development\" NARROW []'],\n",
       "  'is_a': ['GO:0030584 ! sporocarp development'],\n",
       "  'relationship': ['part_of GO:0019953 ! sexual reproduction']},\n",
       " 'GO:0000910': {'name': ['cytokinesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0007104', 'GO:0016288', 'GO:0033205'],\n",
       "  'def': ['\"The division of the cytoplasm and the plasma membrane of a cell and its partitioning into two daughter cells.\" [GOC:mtg_cell_cycle]'],\n",
       "  'comment': [\"Note that this term should not be used for direct annotation. When annotating eukaryotic species, mitotic or meiotic cytokinesis should always be specified for manual annotation and for prokaryotic species use 'FtsZ-dependent cytokinesis ; GO:0043093' or Cdv-dependent cytokinesis ; GO:0061639. Also, note that cytokinesis does not necessarily result in physical separation and detachment of the two daughter cells from each other.\"],\n",
       "  'subset': ['gocheck_do_not_manually_annotate',\n",
       "   'goslim_candida',\n",
       "   'goslim_generic',\n",
       "   'goslim_yeast',\n",
       "   'prokaryote_subset'],\n",
       "  'synonym': ['\"cell cycle cytokinesis\" EXACT []',\n",
       "   '\"cytokinesis involved in cell cycle\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'xref': ['Wikipedia:Cytokinesis'],\n",
       "  'is_a': ['GO:0022402 ! cell cycle process'],\n",
       "  'relationship': ['part_of GO:0051301 ! cell division']},\n",
       " 'GO:0000911': {'name': ['cytokinesis by cell plate formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process of dividing the cytoplasm of a parent cell where a structure forms in the cytoplasm and grows until reaching the plasma membrane, thereby completely separating the cytoplasms of adjacent progeny cells. An example of this is found in Arabidopsis thaliana.\" [GOC:clt]'],\n",
       "  'is_a': ['GO:0000910 ! cytokinesis']},\n",
       " 'GO:0000912': {'name': ['assembly of actomyosin apparatus involved in cytokinesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The assembly and arrangement of an apparatus composed of actin, myosin, and associated proteins that will function in cytokinesis.\" [GOC:mtg_cell_cycle]'],\n",
       "  'synonym': ['\"actomyosin apparatus assembly involved in cytokinesis\" EXACT [GOC:mah]',\n",
       "   '\"cytokinesis, formation of actomyosin apparatus\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"formation of actomyosin apparatus involved in cytokinesis\" RELATED [GOC:dph]'],\n",
       "  'is_a': ['GO:0022607 ! cellular component assembly',\n",
       "   'GO:0032506 ! cytokinetic process'],\n",
       "  'relationship': ['part_of GO:0061640 ! cytoskeleton-dependent cytokinesis']},\n",
       " 'GO:0000913': {'name': ['preprophase band assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of a set of components to form the preprophase band, a dense band of microtubules that marks the position in the cell where cytokinesis will occur in cells that perform cytokinesis by cell plate formation.\" [GOC:clt, GOC:mah]'],\n",
       "  'synonym': ['\"preprophase band formation\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0022402 ! cell cycle process',\n",
       "   'GO:0022607 ! cellular component assembly'],\n",
       "  'relationship': ['part_of GO:0000911 ! cytokinesis by cell plate formation']},\n",
       " 'GO:0000914': {'name': ['phragmoplast assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of a structure composed of actin, myosin, and associated proteins that will function in cytokinesis in cells that perform cytokinesis by cell plate formation. The structure usually contains antiparallel microtubules and membrane (often visible as vesicles).\" [GOC:clt]'],\n",
       "  'synonym': ['\"phragmoplast formation\" RELATED [GOC:dph]'],\n",
       "  'is_a': ['GO:1902407 ! assembly of actomyosin apparatus involved in mitotic cytokinesis'],\n",
       "  'relationship': ['part_of GO:0000911 ! cytokinesis by cell plate formation']},\n",
       " 'GO:0000915': {'name': ['actomyosin contractile ring assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0045573', 'GO:2000708'],\n",
       "  'def': ['\"The process of assembly of a ring composed of actin, myosin, and associated proteins that will function in cytokinesis.\" [GOC:clt, GOC:dph, GOC:tb]'],\n",
       "  'synonym': ['\"constriction ring assembly\" NARROW []',\n",
       "   '\"contractile ring assembly\" EXACT [GOC:ascb_2009, GOC:dph, GOC:tb]',\n",
       "   '\"cytokinesis, actomyosin contractile ring assembly\" EXACT []',\n",
       "   '\"cytokinesis, actomyosin contractile ring formation\" RELATED [GOC:ascb_2009, GOC:dph, GOC:tb]',\n",
       "   '\"cytokinesis, actomyosin ring biosynthesis\" RELATED []',\n",
       "   '\"cytokinesis, actomyosin ring formation\" RELATED []',\n",
       "   '\"cytokinesis, contractile ring assembly\" RELATED [GOC:mah]',\n",
       "   '\"myosin filament organisation involved in cytokinetic actomyosin contractile ring assembly\" RELATED [GOC:mah]',\n",
       "   '\"myosin filament organization involved in cytokinetic actomyosin contractile ring assembly\" RELATED []',\n",
       "   '\"myosin filament organization of constriction ring assembly\" RELATED [GOC:obol]',\n",
       "   '\"myosin filament organization of contractile ring assembly\" RELATED [GOC:obol]'],\n",
       "  'is_a': ['GO:0000912 ! assembly of actomyosin apparatus involved in cytokinesis',\n",
       "   'GO:0044837 ! actomyosin contractile ring organization']},\n",
       " 'GO:0000916': {'name': ['actomyosin contractile ring contraction'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process of an actomyosin ring getting smaller in diameter, in the context of cytokinesis that takes place as part of a cell cycle.\" [GOC:clt, GOC:dph, GOC:mah, GOC:tb]'],\n",
       "  'synonym': ['\"actomyosin contractile ring constriction\" EXACT [GOC:vw]',\n",
       "   '\"contractile ring contraction involved in cell cycle cytokinesis\" EXACT [GOC:mah, GOC:vw]',\n",
       "   '\"cytokinesis, actomyosin ring contraction\" EXACT []',\n",
       "   '\"cytokinesis, contractile ring contraction\" BROAD [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0036213 ! contractile ring contraction'],\n",
       "  'relationship': ['part_of GO:0044837 ! actomyosin contractile ring organization']},\n",
       " 'GO:0000917': {'name': ['division septum assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0071937', 'GO:1902411'],\n",
       "  'def': ['\"The assembly and arrangement of a septum that spans the plasma membrane interface between progeny cells following cytokinesis. The progeny cells that form a division septum are not able to exchange intracellular material.\" [GOC:mtg_cell_cycle]'],\n",
       "  'synonym': ['\"division septum assembly involved in cell cycle cytokinesis\" NARROW []',\n",
       "   '\"division septum assembly involved in cell cycle cytokinesis involved in mitotic cell cycle\" NARROW [GOC:TermGenie]',\n",
       "   '\"division septum formation\" EXACT [GOC:mah]',\n",
       "   '\"division septum formation involved in cell cycle cytokinesis\" NARROW []',\n",
       "   '\"division septum formation involved in cell cycle cytokinesis involved in mitotic cell cycle\" NARROW [GOC:TermGenie]',\n",
       "   '\"division septum formation involved in mitotic cell cycle\" NARROW [GOC:TermGenie]',\n",
       "   '\"formation of division septum\" EXACT []',\n",
       "   '\"formation of division septum involved in mitotic cell cycle\" NARROW [GOC:TermGenie]',\n",
       "   '\"mitotic division septum assembly\" NARROW []',\n",
       "   '\"septation\" BROAD []',\n",
       "   '\"septin assembly and septum biosynthesis\" RELATED []',\n",
       "   '\"septin assembly and septum biosynthesis involved in mitotic cell cycle\" NARROW [GOC:TermGenie]',\n",
       "   '\"septin assembly and septum formation\" RELATED []',\n",
       "   '\"septin assembly and septum formation involved in mitotic cell cycle\" NARROW [GOC:TermGenie]'],\n",
       "  'is_a': ['GO:0090529 ! cell septum assembly']},\n",
       " 'GO:0000918': {'name': ['division septum site selection'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process of marking the site where a division septum will form.\" [GOC:clt]'],\n",
       "  'synonym': ['\"selection of site for barrier cell septum biosynthesis\" EXACT []',\n",
       "   '\"selection of site for barrier cell septum formation\" EXACT []',\n",
       "   '\"selection of site for division septum formation\" EXACT []',\n",
       "   '\"septin assembly and septum biosynthesis\" BROAD []',\n",
       "   '\"septin assembly and septum formation\" BROAD []',\n",
       "   '\"septum positioning\" EXACT []'],\n",
       "  'is_a': ['GO:0032506 ! cytokinetic process'],\n",
       "  'relationship': ['part_of GO:0000917 ! division septum assembly']},\n",
       " 'GO:0000919': {'name': ['cell plate assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process of assembly, maturation, and growth of the cell plate to the cell periphery in cells that divide by cell plate formation; often involves deposition of cell wall material in and around the phragmoplast.\" [GOC:clt]'],\n",
       "  'synonym': ['\"cell plate formation\" RELATED [GOC:mah]'],\n",
       "  'is_a': ['GO:0022402 ! cell cycle process',\n",
       "   'GO:0022607 ! cellular component assembly'],\n",
       "  'relationship': ['part_of GO:0000911 ! cytokinesis by cell plate formation']},\n",
       " 'GO:0000920': {'name': ['septum digestion after cytokinesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:1902409', 'GO:2000695'],\n",
       "  'def': ['\"The process of physically separating the septal cell wall material by enzymatic digestion, that occurs after daughter cells are separated by cytokinesis.\" [GOC:mtg_cell_cycle, GOC:vw]'],\n",
       "  'comment': [\"This term should not be used to describe the last step of cytokinesis in organisms without a cell wall, ie, the cell resealing of the plasma membrane via abscission. Consider annotating to 'GO:0061952 midbody abscission' to capture this process.\"],\n",
       "  'synonym': ['\"cell separation after cytokinesis\" RELATED []',\n",
       "   '\"cell separation following cytokinesis\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"cytokinetic cell separation\" RELATED []',\n",
       "   '\"daughter cell separation\" RELATED []',\n",
       "   '\"mitotic cytokinetic cell separation\" BROAD []'],\n",
       "  'is_a': ['GO:0009987 ! cellular process'],\n",
       "  'relationship': ['part_of GO:0051301 ! cell division']},\n",
       " 'GO:0000921': {'name': ['septin ring assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of septins and associated proteins to form an organized structure resembling a ring at the cell cortex.\" [GOC:clt]'],\n",
       "  'synonym': ['\"septin assembly and septum biosynthesis\" BROAD []',\n",
       "   '\"septin assembly and septum formation\" BROAD []'],\n",
       "  'is_a': ['GO:0031106 ! septin ring organization',\n",
       "   'GO:0065003 ! protein-containing complex assembly',\n",
       "   'GO:0140694 ! non-membrane-bounded organelle assembly']},\n",
       " 'GO:0000922': {'name': ['spindle pole'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0030615'],\n",
       "  'def': ['\"Either of the ends of a spindle, where spindle microtubules are organized; usually contains a microtubule organizing center and accessory molecules, spindle microtubules and astral microtubules.\" [GOC:clt]'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005819 ! spindle']},\n",
       " 'GO:0000923': {'name': ['equatorial microtubule organizing center'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A microtubule organizing center formed by a band of gamma-tubulin that is recruited to a circumferential band of F-actin at the midpoint of a cell and which nucleates microtubules from the cell division site at the end of mitosis.\" [PMID:11792817]'],\n",
       "  'synonym': ['\"EMTOC\" EXACT []',\n",
       "   '\"equatorial microtubule organising centre\" EXACT []'],\n",
       "  'is_a': ['GO:0005815 ! microtubule organizing center'],\n",
       "  'relationship': ['part_of GO:0032153 ! cell division site']},\n",
       " 'GO:0000930': {'name': ['gamma-tubulin complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A multiprotein complex composed of gamma-tubulin and other non-tubulin proteins. Gamma-tubulin complexes are localized to microtubule organizing centers, and play an important role in the nucleation of microtubules. The number and complexity of non-tubulin proteins associated with these complexes varies between species.\" [GOC:clt, PMID:12134075]'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005815 ! microtubule organizing center']},\n",
       " 'GO:0000931': {'name': ['gamma-tubulin ring complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000924',\n",
       "   'GO:0000925',\n",
       "   'GO:0000926',\n",
       "   'GO:0000929',\n",
       "   'GO:0008274',\n",
       "   'GO:0055031',\n",
       "   'GO:0055032',\n",
       "   'GO:0055033',\n",
       "   'GO:0061494'],\n",
       "  'def': ['\"A complex of gamma tubulin and associated proteins thought to be formed by multimerization of gamma-tubulin small complexes. An example of this structure is found in Schizosaccharomyces pombe.\" [GOC:mtg_sensu, PMID:12134075, PMID:17021256]'],\n",
       "  'synonym': ['\"gamma-tubulin large complex\" RELATED []',\n",
       "   '\"gamma-tubulin large complex, centrosomal\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, eMTOC\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, equatorial microtubule organizing center\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, equatorial microtubule organizing centre\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, iMTOC\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, interphase microtubule organizing center\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, interphase microtubule organizing centre\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, mitotic spindle pole body\" NARROW []',\n",
       "   '\"gamma-tubulin large complex, spindle pole body\" NARROW []',\n",
       "   '\"gamma-tubulin ring complex, centrosomal\" NARROW []',\n",
       "   '\"gamma-TuRC\" EXACT []'],\n",
       "  'is_a': ['GO:0000930 ! gamma-tubulin complex']},\n",
       " 'GO:0000932': {'name': ['P-body'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A focus in the cytoplasm where mRNAs may become inactivated by decapping or some other mechanism. Protein and RNA localized to these foci are involved in mRNA degradation, nonsense-mediated mRNA decay (NMD), translational repression, and RNA-mediated gene silencing.\" [GOC:clt, PMID:12730603]'],\n",
       "  'synonym': ['\"cytoplasmic foci\" RELATED []',\n",
       "   '\"cytoplasmic mRNA processing body\" EXACT []',\n",
       "   '\"P body\" EXACT []'],\n",
       "  'xref': ['Wikipedia:P_body'],\n",
       "  'is_a': ['GO:0036464 ! cytoplasmic ribonucleoprotein granule']},\n",
       " 'GO:0000933': {'name': ['adventitious septum'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A cell septum whose formation is independent of nuclear division.\" [GOC:clt, ISBN:0471940526]'],\n",
       "  'is_a': ['GO:0030428 ! cell septum']},\n",
       " 'GO:0000934': {'name': ['porous cell septum'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A septum or cross wall which does not entirely span the space between two portions of cell wall and may contain a specialized central pore structure. A porous septum allows the movement of organelles and/or cytoplasm between compartments.\" [GOC:clt]'],\n",
       "  'synonym': ['\"porous septum\" BROAD []'],\n",
       "  'is_a': ['GO:0030428 ! cell septum']},\n",
       " 'GO:0000935': {'name': ['division septum'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0043187'],\n",
       "  'def': ['\"A cell septum which forms as part of the division site and functions in the compartmentalization of a cell into two daughter cells at division. A division septum spans a cell and does not allow exchange of organelles or cytoplasm between compartments.\" [GOC:clt, GOC:vw]'],\n",
       "  'synonym': ['\"cell septum surface\" RELATED []',\n",
       "   '\"complete septum\" EXACT []',\n",
       "   '\"divison septum\" EXACT []',\n",
       "   '\"septum surface\" BROAD []'],\n",
       "  'is_a': ['GO:0030428 ! cell septum'],\n",
       "  'relationship': ['part_of GO:0032153 ! cell division site']},\n",
       " 'GO:0000936': {'name': ['primary cell septum'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A cell septum that forms following nuclear division.\" [GOC:clt, ISBN:0471940526]'],\n",
       "  'synonym': ['\"primary septum\" BROAD []'],\n",
       "  'is_a': ['GO:0000935 ! division septum']},\n",
       " 'GO:0000937': {'name': ['dolipore septum'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A septum, or cross-wall, between two portions of a cell or hypha; contains a central pore around which the septum is swollen to form a barrel-shaped structure; pore is covered on each side of the septum by a septal pore cap (parenthosome).\" [GOC:clt]'],\n",
       "  'is_a': ['GO:0000934 ! porous cell septum']},\n",
       " 'GO:0000938': {'name': ['GARP complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A quatrefoil tethering complex required for retrograde traffic from the early endosome back to the late Golgi and biogenesis of cytoplasmic vesicles.\" [GOC:clt, GOC:rn, PMID:10637310, PMID:12077354, PMID:12446664]'],\n",
       "  'synonym': ['\"Golgi associated retrograde protein complex\" EXACT []',\n",
       "   '\"VFT tethering complex\" EXACT []',\n",
       "   '\"Vps fifty three tethering complex\" EXACT []'],\n",
       "  'is_a': ['GO:0099023 ! vesicle tethering complex'],\n",
       "  'relationship': ['part_of GO:0005794 ! Golgi apparatus',\n",
       "   'part_of GO:0031410 ! cytoplasmic vesicle']},\n",
       " 'GO:0000939': {'name': ['inner kinetochore'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000941'],\n",
       "  'def': ['\"The region of a kinetochore closest to centromeric DNA which contains many CENP proteins organized in various subcomplexes including CENP-C, CENP-LN, CENP-HIKM, CENP-OPQUR and CENP-TWSX, but excluding the CENP-A containing heterochromatin.\" [GOC:clt, PMID:10619130, PMID:11483983, PMID:32516549]'],\n",
       "  'synonym': ['\"condensed chromosome inner kinetochore\" EXACT []',\n",
       "   '\"condensed nuclear chromosome inner kinetochore\" EXACT []',\n",
       "   '\"inner centromere core complex\" RELATED []',\n",
       "   '\"inner kinetochore of condensed chromosome\" EXACT []',\n",
       "   '\"inner kinetochore of condensed nuclear chromosome\" EXACT []',\n",
       "   '\"inner kinetochore plate\" NARROW []'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0000776 ! kinetochore']},\n",
       " 'GO:0000940': {'name': ['outer kinetochore'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0000942'],\n",
       "  'def': ['\"The region of a kinetochore most external to centromeric DNA; this outer region mediates kinetochore-microtubule interactions.\" [GOC:clt, PMID:11483983]'],\n",
       "  'synonym': ['\"condensed chromosome outer kinetochore\" EXACT []',\n",
       "   '\"condensed nuclear chromosome outer kinetochore\" EXACT []',\n",
       "   '\"outer kinetochore of condensed chromosome\" EXACT []',\n",
       "   '\"outer kinetochore of condensed nuclear chromosome\" EXACT []',\n",
       "   '\"outer kinetochore plate\" NARROW []'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0000776 ! kinetochore']},\n",
       " 'GO:0000943': {'name': ['retrotransposon nucleocapsid'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A complex of the retrotransposon RNA genome, reverse transcriptase, integrase, and associated molecules required for reproduction and integration of the retrotransposon into the host genome; the main structural molecule of the nucleocapsid is often a gag protein homolog.\" [GOC:clt, PMID:10861903]'],\n",
       "  'synonym': ['\"Virus-like particle\" EXACT []', '\"VLP\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Virus-like_particle'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005634 ! nucleus']},\n",
       " 'GO:0000947': {'name': ['amino acid catabolic process to alcohol via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of amino acids to produce alcohols with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of alcohols, often collectively referred to as fusel alcohols. Depending on the redox state of the cells, carboxylic acid derivatives may be produced instead of alcohols.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0000955 ! amino acid catabolic process via Ehrlich pathway',\n",
       "   'GO:0046165 ! alcohol biosynthetic process']},\n",
       " 'GO:0000948': {'name': ['amino acid catabolic process to carboxylic acid via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of amino acids to produce carboxylic acids with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of carboxylic acids, sometimes collectively referred to as fusel acids. Depending on the redox state of the cells, alcohol derivatives may be produced instead of carboxylic acids.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0000955 ! amino acid catabolic process via Ehrlich pathway',\n",
       "   'GO:0046394 ! carboxylic acid biosynthetic process']},\n",
       " 'GO:0000949': {'name': ['aromatic amino acid family catabolic process to alcohol via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of aromatic amino acids to produce aromatic alcohols with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. When an aromatic family amino acid, phenylalanine, tyrosine, or tryptophan, is used as the substrate, 2-phenylethanol, 4-hydroxyphenylethanol, or tryptophol, respectively, is produced. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of alcohols, often collectively referred to as fusel alcohols. Depending on the redox state of the cells, carboxylic acid derivatives may be produced instead of alcohols.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0000947 ! amino acid catabolic process to alcohol via Ehrlich pathway',\n",
       "   'GO:0009074 ! aromatic amino acid family catabolic process']},\n",
       " 'GO:0000950': {'name': ['branched-chain amino acid catabolic process to alcohol via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of branched chain amino acids to produce branched chain alcohols with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. When a branched chain family amino acid, leucine, isoleucine, or valine, is used as the substrate, 3-methylbutanol, 2-methylbutanol, or 2-methylpropanol, respectively, is produced. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of alcohols, often collectively referred to as fusel alcohols. Depending on the redox state of the cells, carboxylic acid derivatives may be produced instead of alcohols.\" [GOC:krc, PMID:18281432]'],\n",
       "  'synonym': ['\"branched chain family amino acid catabolic process to alcohol via Ehrlich pathway\" EXACT [GOC:krc]'],\n",
       "  'is_a': ['GO:0000947 ! amino acid catabolic process to alcohol via Ehrlich pathway',\n",
       "   'GO:0009083 ! branched-chain amino acid catabolic process']},\n",
       " 'GO:0000951': {'name': ['methionine catabolic process to 3-methylthiopropanol'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of branched chain amino acids to produce branched chain alcohols with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. When methionine is used as the substrate, 3-methylthiopropanol is produced. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of alcohols, often collectively referred to as fusel alcohols. Depending on the redox state of the cells, carboxylic acid derivatives may be produced instead of alcohols.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0000947 ! amino acid catabolic process to alcohol via Ehrlich pathway',\n",
       "   'GO:0009087 ! methionine catabolic process',\n",
       "   'GO:0044272 ! sulfur compound biosynthetic process']},\n",
       " 'GO:0000952': {'name': ['aromatic amino acid family catabolic process to carboxylic acid via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of amino acids to produce carboxylic acids with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. When an aromatic family amino acid, phenylalanine, tyrosine, or tryptophan, is used as the substrate, 2-phenylethanoate, 4-hydroxyphenylethanoate, or 2-(Indol-3-yl)-ethanoate, respectively, is produced. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of carboxylic acids, sometimes collectively referred to as fusel acids. Depending on the redox state of the cells, alcohol derivatives may be produced instead of carboxylic acids.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0000948 ! amino acid catabolic process to carboxylic acid via Ehrlich pathway',\n",
       "   'GO:0009074 ! aromatic amino acid family catabolic process']},\n",
       " 'GO:0000953': {'name': ['branched-chain amino acid catabolic process to carboxylic acid via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of amino acids to produce carboxylic acids with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. When a branched chain family amino acid, leucine, isoleucine, or valine, is used as the substrate, 3-methylbutanoate, 2-methylbutanoate, or 2-methylpropanoate, respectively, is produced. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of carboxylic acids, sometimes collectively referred to as fusel acids. Depending on the redox state of the cells, alcohol derivatives may be produced instead of carboxylic acids.\" [GOC:krc, PMID:18281432]'],\n",
       "  'synonym': ['\"branched chain family amino acid catabolic process to carboxylic acid via Ehrlich pathway\" EXACT [GOC:krc]'],\n",
       "  'is_a': ['GO:0000948 ! amino acid catabolic process to carboxylic acid via Ehrlich pathway',\n",
       "   'GO:0009083 ! branched-chain amino acid catabolic process']},\n",
       " 'GO:0000954': {'name': ['methionine catabolic process to 3-methylthiopropanoate'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of amino acids to produce carboxylic acids with one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. When methionine is used as the substrate, 3-methylthiopropanoate is produced. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of carboxylic acids, sometimes collectively referred to as fusel acids. Depending on the redox state of the cells, alcohol derivatives may be produced instead of carboxylic acids.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0000948 ! amino acid catabolic process to carboxylic acid via Ehrlich pathway',\n",
       "   'GO:0006633 ! fatty acid biosynthetic process',\n",
       "   'GO:0009087 ! methionine catabolic process',\n",
       "   'GO:0044272 ! sulfur compound biosynthetic process']},\n",
       " 'GO:0000955': {'name': ['amino acid catabolic process via Ehrlich pathway'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving the catabolism of amino acids to produce alcohols or carboxylic acids containing one carbon less than the starting amino acid. In S. cerevisiae, this is known to occur for leucine, isoleucine, valine, methionine, phenylalanine, tyrosine, or tryptophan. Often referred to as the Ehrlich pathway, these reactions generally occur during fermentation to produce a variety of alcohols, often collectively referred to as fusel alcohols. Depending on the redox state of the cells, carboxylic acid derivatives, sometimes referred to as fusel acids, may be produced instead of alcohols.\" [GOC:krc, PMID:18281432]'],\n",
       "  'is_a': ['GO:0009063 ! amino acid catabolic process']},\n",
       " 'GO:0000956': {'name': ['nuclear-transcribed mRNA catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of nuclear-transcribed mRNAs in eukaryotic cells.\" [GOC:krc]'],\n",
       "  'synonym': ['\"nuclear mRNA breakdown\" EXACT []',\n",
       "   '\"nuclear mRNA catabolism\" EXACT []',\n",
       "   '\"nuclear mRNA degradation\" EXACT []'],\n",
       "  'is_a': ['GO:0006402 ! mRNA catabolic process']},\n",
       " 'GO:0000957': {'name': ['mitochondrial RNA catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of RNA transcribed from the mitochondrial genome and occurring in the mitochondrion.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0000959 ! mitochondrial RNA metabolic process',\n",
       "   'GO:0006401 ! RNA catabolic process']},\n",
       " 'GO:0000958': {'name': ['mitochondrial mRNA catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of mRNA transcribed from the mitochondrial genome and occurring in the mitochondrion.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0000957 ! mitochondrial RNA catabolic process',\n",
       "   'GO:0006402 ! mRNA catabolic process']},\n",
       " 'GO:0000959': {'name': ['mitochondrial RNA metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving RNA transcribed from the mitochondrial genome and occurring in the mitochondrion.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0016070 ! RNA metabolic process']},\n",
       " 'GO:0000960': {'name': ['regulation of mitochondrial RNA catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate or extent of the chemical reactions and pathways involving catabolism in the mitochondrion of RNA transcribed from the mitochondrial genome.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0031329 ! regulation of cellular catabolic process',\n",
       "   'GO:0051252 ! regulation of RNA metabolic process'],\n",
       "  'relationship': ['regulates GO:0000957 ! mitochondrial RNA catabolic process']},\n",
       " 'GO:0000961': {'name': ['negative regulation of mitochondrial RNA catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that stops, prevents, or reduces the frequency, rate or extent of the chemical reactions and pathways involving catabolism in the mitochondrion of RNA transcribed from the mitochondrial genome.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0000960 ! regulation of mitochondrial RNA catabolic process',\n",
       "   'GO:1902369 ! negative regulation of RNA catabolic process'],\n",
       "  'relationship': ['negatively_regulates GO:0000957 ! mitochondrial RNA catabolic process']},\n",
       " 'GO:0000962': {'name': ['positive regulation of mitochondrial RNA catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that activates or increases the frequency, rate or extent of the chemical reactions and pathways involving catabolism in the mitochondrion of RNA transcribed from the mitochondrial genome.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0000960 ! regulation of mitochondrial RNA catabolic process',\n",
       "   'GO:0031331 ! positive regulation of cellular catabolic process',\n",
       "   'GO:0051254 ! positive regulation of RNA metabolic process'],\n",
       "  'relationship': ['positively_regulates GO:0000957 ! mitochondrial RNA catabolic process']},\n",
       " 'GO:0000963': {'name': ['mitochondrial RNA processing'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The conversion of a primary RNA molecule transcribed from a mitochondrial genome into one or more mature RNA molecules; occurs in the mitochondrion.\" [GOC:krc, GOC:mah]'],\n",
       "  'is_a': ['GO:0000959 ! mitochondrial RNA metabolic process',\n",
       "   'GO:0006396 ! RNA processing',\n",
       "   'GO:0140053 ! mitochondrial gene expression']},\n",
       " 'GO:0000964': {'name': [\"mitochondrial RNA 5'-end processing\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in forming the mature 5\\' end of an RNA molecule transcribed from a mitochondrial genome; occurs in the mitochondrion.\" [GOC:krc, GOC:mah]'],\n",
       "  'synonym': ['\"mitochondrial RNA 5\\' end processing\" RELATED []'],\n",
       "  'is_a': ['GO:0000963 ! mitochondrial RNA processing',\n",
       "   \"GO:0000966 ! RNA 5'-end processing\"]},\n",
       " 'GO:0000965': {'name': [\"mitochondrial RNA 3'-end processing\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in forming the mature 3\\' end of an RNA molecule transcribed from a mitochondrial genome; occurs in the mitochondrion.\" [GOC:krc, GOC:mah]'],\n",
       "  'synonym': ['\"mitochondrial RNA 3\\' end processing\" RELATED []'],\n",
       "  'is_a': ['GO:0000963 ! mitochondrial RNA processing',\n",
       "   \"GO:0031123 ! RNA 3'-end processing\"]},\n",
       " 'GO:0000966': {'name': [\"RNA 5'-end processing\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in forming the mature 5\\' end of an RNA molecule.\" [GOC:krc]'],\n",
       "  'synonym': ['\"RNA 5\\' end processing\" EXACT []'],\n",
       "  'is_a': ['GO:0006396 ! RNA processing']},\n",
       " 'GO:0000967': {'name': [\"rRNA 5'-end processing\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in forming the mature 5\\' end of an rRNA molecule.\" [GOC:krc]'],\n",
       "  'synonym': ['\"rRNA 5\\' end processing\" EXACT []'],\n",
       "  'is_a': ['GO:0006364 ! rRNA processing',\n",
       "   \"GO:0034471 ! ncRNA 5'-end processing\"]},\n",
       " 'GO:0000968': {'name': ['tRNA exon ligation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"An RNA exon ligation process that rejoins two exons of a pre-tRNA which has had the intron removed.\" [GOC:krc]'],\n",
       "  'is_a': ['GO:0000378 ! RNA exon ligation'],\n",
       "  'relationship': ['part_of GO:0006388 ! tRNA splicing, via endonucleolytic cleavage and ligation']},\n",
       " 'GO:0000969': {'name': ['tRNA exon ligation utilizing ATP as source of linkage phosphate'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A tRNA exon ligation process in which the splice junction phosphate is derived from exogenous ATP. This type of ligation to rejoin the 5\\' and 3\\' exons of a tRNA is observed in vertebrate species.\" [GOC:krc, PMID:17786051]'],\n",
       "  'is_a': ['GO:0000968 ! tRNA exon ligation']},\n",
       " 'GO:0000970': {'name': ['tRNA exon ligation utilizing GTP as source of linkage phosphate'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A tRNA exon ligation process in which the splice junction phosphate is derived from exogenous GTP. This type of ligation to rejoin the 5\\' and 3\\' exons of a tRNA is observed in the yeast Saccharomyces cerevisiae where the ligation reaction also produces a 2\\'-phosphate at the splice junction which is subsequently removed as part of the ligation process.\" [GOC:krc, PMID:18217203, PMID:9299409]'],\n",
       "  'is_a': ['GO:0000968 ! tRNA exon ligation']},\n",
       " 'GO:0000971': {'name': [\"tRNA exon ligation utilizing 2',3' cyclic phosphate of 5'-exon as source of linkage phosphate\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A tRNA exon ligation process in which the splice junction phosphate is derived from the 2\\',3\\' cyclic phosphate at the 3\\'-end of the 5\\'-exon. This type of ligation to rejoin the 5\\' and 3\\' exons of a tRNA is observed in wheat, Chlamydomonas, and vertebrate species including humans.\" [GOC:krc, PMID:17786051, PMID:18217203, PMID:9299409]'],\n",
       "  'is_a': ['GO:0000968 ! tRNA exon ligation']},\n",
       " 'GO:0000972': {'name': ['transcription-dependent tethering of RNA polymerase II gene DNA at nuclear periphery'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chromosome organization process in which the DNA sequence containing a gene transcribed by RNA polymerase II is maintained in a specific location at the nuclear periphery. In S. cerevisiae, this process involves cis-acting DNA sequences such as the TATA box and upstream activating sequence (UAS) elements, trans-acting transcriptional activators, and also the 3\\'-UTR of the transcript.\" [GOC:krc, PMID:18614049]'],\n",
       "  'is_a': ['GO:0051276 ! chromosome organization']},\n",
       " 'GO:0000973': {'name': ['post-transcriptional tethering of RNA polymerase II gene DNA at nuclear periphery'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chromosome organization process in which the DNA sequence containing a gene transcribed by RNA polymerase II is maintained in a specific location at the nuclear periphery even after transcription has been repressed.\" [GOC:krc, PMID:17373856, PMID:18614049]'],\n",
       "  'synonym': ['\"posttranscriptional tethering of RNA polymerase II gene DNA at nuclear periphery\" EXACT []'],\n",
       "  'is_a': ['GO:0000972 ! transcription-dependent tethering of RNA polymerase II gene DNA at nuclear periphery']},\n",
       " 'GO:0000974': {'name': ['Prp19 complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A protein complex consisting of Prp19 and associated proteins that is involved in the transition from the precatalytic spliceosome to the activated form that catalyzes step 1 of splicing, and which remains associated with the spliceosome through the second catalytic step. It is widely conserved, found in both yeast and mammals, though the exact composition varies. In S. cerevisiae, it contains Prp19p, Ntc20p, Snt309p, Isy1p, Syf2p, Cwc2p, Prp46p, Clf1p, Cef1p, and Syf1p.\" [GOC:krc, PMID:16540691, PMID:19239890]'],\n",
       "  'synonym': ['\"nineteen complex\" NARROW []',\n",
       "   '\"NTC\" RELATED []',\n",
       "   '\"Prp19/CDC5 complex\" NARROW []'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex']},\n",
       " 'GO:0000976': {'name': ['transcription cis-regulatory region binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000975', 'GO:0000984', 'GO:0001017', 'GO:0044212'],\n",
       "  'def': ['\"Binding to a specific sequence of DNA that is part of a regulatory region that controls transcription of that section of the DNA. The transcribed region might be described as a gene, cistron, or operon.\" [GOC:txnOH]'],\n",
       "  'comment': ['Note that this term is meant to also capture non-specific binding to regulatory regions. Also, to minimize ambiguity in the use of the word \"promoter\" in GO, we have chosen the phrase \"transcription regulatory region\" to refer to all of the regulatory regions. Regulatory regions in the DNA which control initiation may include the \"core promoter\" where the basal transcription machinery binds, the \"core promoter proximal region\" where regulatory factors other than the basal machinery bind. There are also additional regulatory regions, in both the DNA and the RNA transcript, which regulate elongation or termination of transcription.'],\n",
       "  'synonym': ['\"bacterial-type RNA polymerase regulatory region DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase transcription regulatory region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"eubacterial-type RNA polymerase regulatory region DNA binding\" NARROW []',\n",
       "   '\"eubacterial-type RNA polymerase regulatory region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"regulatory region DNA binding\" BROAD []',\n",
       "   '\"transcription regulatory region DNA binding\" RELATED []',\n",
       "   '\"transcription regulatory region sequence-specific DNA binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001067 ! transcription regulatory region nucleic acid binding',\n",
       "   'GO:1990837 ! sequence-specific double-stranded DNA binding']},\n",
       " 'GO:0000977': {'name': ['RNA polymerase II transcription regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001012'],\n",
       "  'def': ['\"Binding to a specific sequence of DNA that is part of a regulatory region that controls the transcription of a gene or cistron by RNA polymerase II.\" [GOC:txnOH]'],\n",
       "  'comment': ['To minimize ambiguity in the use of the word \"promoter\" in GO, we have chosen the phrase \"transcription regulatory region\" to refer to all of the regulatory regions. Regulatory regions in the DNA which control initiation may include the \"core promoter\" where the basal transcription machinery binds, the \"core promoter proximal region\" where regulatory factors other than the basal machinery bind. There are also additional regulatory regions, in both the DNA and the RNA transcript, which regulate elongation or termination of transcription.'],\n",
       "  'synonym': ['\"RNA polymerase II regulatory region DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0000978': {'name': ['RNA polymerase II cis-regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000980'],\n",
       "  'def': ['\"Binding to a specific upstream regulatory DNA sequence (transcription factor recognition sequence or binding site) located in cis relative to the transcription start site (i.e., on the same strand of DNA) of a gene transcribed by RNA polymerase II.\" [GOC:txnOH-2018]'],\n",
       "  'comment': ['Note that the phrase \"upstream activating sequence\", or UAS is often used in S. cerevisiae literature to refer to cis-regulatory sequences. In bacteria such as E. coli, the phrase \"upstream activating sequence\", or UAS is usually a synonym for \"enhancer\".'],\n",
       "  'synonym': ['\"RNA polymerase II core promoter proximal region sequence-specific DNA binding\" RELATED []',\n",
       "   '\"RNA polymerase II distal enhancer sequence-specific DNA binding\" RELATED []',\n",
       "   '\"RNA polymerase II promoter proximal region sequence-specific DNA binding\" RELATED []',\n",
       "   '\"RNA polymerase II proximal promoter sequence-specific DNA binding\" RELATED []',\n",
       "   '\"RNA polymerase II upstream activating sequence (UAS) sequence-specific DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000977 ! RNA polymerase II transcription regulatory region sequence-specific DNA binding',\n",
       "   'GO:0000987 ! cis-regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0000979': {'name': ['RNA polymerase II core promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA sequence that is part of the core promoter of a RNA polymerase II-transcribed gene.\" [GOC:pg, GOC:txnOH, PMID:12381658]'],\n",
       "  'is_a': ['GO:0000977 ! RNA polymerase II transcription regulatory region sequence-specific DNA binding',\n",
       "   'GO:0001046 ! core promoter sequence-specific DNA binding']},\n",
       " 'GO:0000981': {'name': ['DNA-binding transcription factor activity, RNA polymerase II-specific'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000982',\n",
       "   'GO:0001133',\n",
       "   'GO:0001200',\n",
       "   'GO:0001201',\n",
       "   'GO:0001202',\n",
       "   'GO:0001203',\n",
       "   'GO:0003705'],\n",
       "  'def': ['\"A DNA-binding transcription factor activity that modulates the transcription of specific gene sets transcribed by RNA polymerase II.\" [GOC:txnOH-2018]'],\n",
       "  'comment': ['For usage guidance, see comment in GO:0003700 ; DNA-binding transcription factor activity.'],\n",
       "  'synonym': ['\"copper ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity\" NARROW []',\n",
       "   '\"metal ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity\" NARROW []',\n",
       "   '\"metal ion regulated sequence-specific DNA binding RNA polymerase II transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase II core promoter proximal region sequence-specific DNA binding transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase II distal enhancer sequence-specific DNA binding transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, copper ion regulated core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, copper ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, metal ion regulated core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, metal ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, metal ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, sequence-specific DNA binding\" EXACT []',\n",
       "   '\"RNA polymerase II transcription factor activity, sequence-specific transcription regulatory region DNA binding\" EXACT []',\n",
       "   '\"RNA polymerase II transcription factor activity, zinc ion regulated core promoter proximal region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcription factor activity, zinc ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"sequence-specific distal enhancer binding RNA polymerase II transcription factor activity\" NARROW []',\n",
       "   '\"sequence-specific DNA binding RNA polymerase II transcription factor activity\" EXACT []',\n",
       "   '\"sequence-specific transcription regulatory region DNA binding RNA polymerase II transcription factor recruiting transcription factor activity\" RELATED []',\n",
       "   '\"transcription factor\" BROAD [GOC:vw]',\n",
       "   '\"transcription factor activity, RNA polymerase II core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"transcription factor activity, RNA polymerase II distal enhancer sequence-specific binding\" NARROW []',\n",
       "   '\"transcription factor activity, RNA polymerase II proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"zinc ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity\" NARROW []'],\n",
       "  'is_a': ['GO:0003700 ! DNA-binding transcription factor activity']},\n",
       " 'GO:0000987': {'name': ['cis-regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000986',\n",
       "   'GO:0001150',\n",
       "   'GO:0001158',\n",
       "   'GO:0001159',\n",
       "   'GO:0035326'],\n",
       "  'def': ['\"Binding to a specific upstream regulatory DNA sequence (transcription factor recognition sequence or binding site) located in cis relative to the transcription start site (i.e., on the same strand of DNA) of a gene transcribed by some RNA polymerase. The proximal promoter is in cis with and relatively close to the core promoter.\" [GOC:txnOH-2018]'],\n",
       "  'synonym': ['\"bacterial-type cis-regulatory region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase core promoter proximal region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase enhancer sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase upstream activating sequence (UAS) sequence-specific DNA binding\" NARROW []',\n",
       "   '\"cis-regulatory region binding\" RELATED []',\n",
       "   '\"core promoter proximal region DNA binding\" NARROW []',\n",
       "   '\"core promoter proximal region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"enhancer binding\" RELATED []',\n",
       "   '\"enhancer sequence-specific DNA binding\" RELATED []',\n",
       "   '\"eubacterial-type RNA polymerase regulatory transcription factor sequence-specific DNA binding\" NARROW []',\n",
       "   '\"promoter proximal region sequence-specific DNA binding\" NARROW []',\n",
       "   '\"proximal promoter sequence-specific DNA binding\" NARROW []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0000988': {'name': ['obsolete transcription factor activity, protein binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a protein or protein complex, to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"protein binding transcription factor activity\" EXACT []',\n",
       "   '\"transcription factor activity\" BROAD []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0005515', 'GO:0140110']},\n",
       " 'GO:0000989': {'name': ['obsolete transcription factor activity, transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a specific transcription factor, which may be a single protein or a complex, in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"transcription factor binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0008134', 'GO:0140110']},\n",
       " 'GO:0000990': {'name': ['obsolete transcription factor activity, core RNA polymerase binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"core RNA polymerase binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0043175', 'GO:0140110']},\n",
       " 'GO:0000991': {'name': ['obsolete transcription factor activity, core RNA polymerase II binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase II (Pol II) complex, typically composed of twelve subunits, in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"core RNA polymerase II binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0000993', 'GO:0140110']},\n",
       " 'GO:0000992': {'name': ['RNA polymerase III cis-regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a specific upstream regulatory DNA sequence (transcription factor recognition sequence or binding site) located in cis relative to the transcription start site (i.e., on the same strand of DNA) of a gene transcribed by RNA polymerase III. The transcribed region might be contain a single gene or a cistron containing multiple genes.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'is_a': ['GO:0000987 ! cis-regulatory region sequence-specific DNA binding',\n",
       "   'GO:0001016 ! RNA polymerase III transcription regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0000993': {'name': ['RNA polymerase II complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase II core enzyme, a multisubunit eukaryotic nuclear RNA polymerase typically composed of twelve subunits.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"RNA polymerase II core binding\" EXACT []',\n",
       "   '\"RNAP II core binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001099 ! basal RNA polymerase II transcription machinery binding',\n",
       "   'GO:0043175 ! RNA polymerase core enzyme binding']},\n",
       " 'GO:0000994': {'name': ['RNA polymerase III core binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase III core enzyme, a multisubunit eukaryotic nuclear RNA polymerase typically composed of seventeen subunits.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0043175 ! RNA polymerase core enzyme binding']},\n",
       " 'GO:0000995': {'name': ['RNA polymerase III general transcription initiation factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001034'],\n",
       "  'def': ['\"A general transcription initiation factor activity that contributes to transcription start site selection and transcription initiation of genes transcribed by RNA polymerase III. Factors required for RNA polymerase III transcription initiation include TFIIIA, TFIIIB and TFIIIC. RNA polymerase III transcribes genes encoding short RNAs, including tRNAs, 5S rRNA, U6 snRNA, the short ncRNA component of RNases P, the mitochondrial RNA processing (MRP) RNA, the signal recognition particle SRP RNA, and in higher eukaryotes a number of micro and other small RNAs, though there is some variability across species as to whether a given small noncoding RNA is transcribed by RNA polymerase II or RNA polymerase III.\" [GOC:txnOH-2018, PMID:12381659, PMID:17977614, PMID:20413673, PMID:27068803, Wikipedia:RNA_polymerase_III]'],\n",
       "  'synonym': ['\"core RNA polymerase III binding transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase III general initiation factor activity\" EXACT []',\n",
       "   '\"RNA polymerase III transcription factor activity, sequence-specific DNA binding\" NARROW []',\n",
       "   '\"sequence-specific DNA binding RNA polymerase III transcription factor activity\" NARROW []',\n",
       "   '\"transcription factor activity, core RNA polymerase III binding\" NARROW []'],\n",
       "  'is_a': ['GO:0140223 ! general transcription initiation factor activity']},\n",
       " 'GO:0001000': {'name': ['bacterial-type RNA polymerase core enzyme binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a bacterial-type RNA polymerase core enzyme, typically consisting of two alpha, one beta, one beta prime, and one omega subunit.\" [GOC:txnOH]'],\n",
       "  'comment': ['Should omega be included here?'],\n",
       "  'synonym': ['\"eubacterial-type RNA polymerase core enzyme binding\" EXACT []'],\n",
       "  'is_a': ['GO:0043175 ! RNA polymerase core enzyme binding']},\n",
       " 'GO:0001001': {'name': ['mitochondrial single-subunit type RNA polymerase binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a single subunit mitochondrial RNA polymerase enzyme, which is composed of a single catalytic subunit similar to the RNA polymerase enzymes from phages T3, T7, and SP6.\" [GOC:txnOH, PMID:20701995, PMID:2088182]'],\n",
       "  'is_a': ['GO:0001050 ! single-subunit type RNA polymerase binding']},\n",
       " 'GO:0001002': {'name': ['RNA polymerase III type 1 promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001030'],\n",
       "  'def': ['\"Binding to a sequence of DNA that is a part of a type 1 promoter that controls transcription by RNA polymerase III. Type 1 promoters are found in 5S rRNA genes, downstream of the transcription start site within the sequence of the mature RNA, and require TFIIIA for recognition.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'synonym': ['\"RNA polymerase III type 1 promoter DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000992 ! RNA polymerase III cis-regulatory region sequence-specific DNA binding',\n",
       "   'GO:0080084 ! 5S rDNA binding']},\n",
       " 'GO:0001003': {'name': ['RNA polymerase III type 2 promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001031'],\n",
       "  'def': ['\"Binding to a sequence of DNA that is a part of a type 2 promoter that controls transcription by RNA polymerase III. Type 2 promoters consist of an A box and a B box downstream of the transcription start site within the sequence within the sequence of the mature RNA. Type 2 promoters are found in many tRNA genes as well as in other small RNAs.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'synonym': ['\"RNA polymerase III type 2 promoter DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000992 ! RNA polymerase III cis-regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0001004': {'name': ['obsolete RNA polymerase III transcription regulator recruiting activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001005',\n",
       "   'GO:0001008',\n",
       "   'GO:0001033',\n",
       "   'GO:0001038',\n",
       "   'GO:0001157'],\n",
       "  'def': ['\"OBSOLETE. Initiating the assembly of the RNA polymerase III pre-initiation complex by binding to a control sequence in the intragenic region. This allows to recruit TFIIIB to the DNA at a site centered approximately 26 base pairs upstream of the start site of transcription. For tRNA genes, TFIIIC first associates with DNA, and then recruits TFIIIB. For 5S rRNA genes, TFIIIA binds to DNA first, followed by TFIIIC, which then recruits TFIIIB.\" [GOC:txnOH-2018, PMID:12381659, PMID:23063749, PMID:27911719, Wikipedia:RNA_polymerase_III]'],\n",
       "  'comment': [\"The reason for obsoletion is that this function is redundant with other terms, most often 'transcription co-regulator activity; GO:0003712'.\"],\n",
       "  'synonym': ['\"RNA polymerase III assembly factor activity\" RELATED []',\n",
       "   '\"RNA polymerase III assembly factor activity, TFIIIB recruiting\" RELATED []',\n",
       "   '\"RNA polymerase III hybrid type promoter TFIIIB recruiting transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase III promoter sequence-specific DNA binding TFIIIB recruiting transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase III transcription factor recruiting activity\" EXACT []',\n",
       "   '\"RNA polymerase III type 1 promoter sequence-specific DNA binding TFIIIB recruiting transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase III type 2 promoter sequence-specific DNA binding TFIIIB recruiting transcription factor activity\" NARROW []',\n",
       "   '\"RNA polymerase III type 3 promoter TFIIIB recruiting transcription factor activity\" NARROW []',\n",
       "   '\"SNAPc-type activity\" RELATED []',\n",
       "   '\"TFIIIA activity\" RELATED []',\n",
       "   '\"TFIIIC-type activity\" RELATED []',\n",
       "   '\"transcription factor activity, RNA polymerase III promoter sequence-specific binding, TFIIIB recruiting\" EXACT []',\n",
       "   '\"transcription factor activity, RNA polymerase III type 1 promoter sequence-specific binding, TFIIIB recruiting\" NARROW []',\n",
       "   '\"transcription factor activity, RNA polymerase III type 1 promoter TFIIIB\" RELATED []',\n",
       "   '\"transcription factor activity, RNA polymerase III type 2 promoter sequence-specific binding, TFIIIB recruiting\" NARROW []',\n",
       "   '\"transcription factor activity, RNA polymerase III type 3 promoter TFIIIB recruiting\" NARROW []',\n",
       "   '\"type 2 RNA polymerase III promoter recognition\" NARROW []',\n",
       "   '\"type 3 RNA polymerase III promoter recognition\" NARROW []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001006': {'name': ['RNA polymerase III type 3 promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001032'],\n",
       "  'def': ['\"Binding to a sequence of DNA that is a part of a type 3 promoter that controls transcription by RNA polymerase III (Pol III). A type 3 Pol III promoter is composed of elements upstream of the transcription start site, including a TATA box. The human U6 snRNA gene has a type 3 promoter. Type 3 Pol III promoters have not been observed in S. cerevisiae.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'synonym': ['\"RNA polymerase III type 3 promoter DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000992 ! RNA polymerase III cis-regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0001007': {'name': ['obsolete transcription factor activity, RNA polymerase III transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase III transcription factor, which may be a single protein or a complex, in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"RNA polymerase III transcription factor binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001010': {'name': ['RNA polymerase II sequence-specific DNA-binding transcription factor recruiting activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"The function of binding to a specific DNA sequence and recruiting another transcription factor to the DNA in order to modulate transcription. The recruited factor may bind DNA directly, or may be colocalized via protein-protein interactions.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"sequence-specific DNA binding transcription factor recruiting transcription factor activity\" EXACT []',\n",
       "   '\"transcription factor activity, sequence-specific DNA binding transcription factor recruiting\" EXACT []',\n",
       "   '\"transcription factor activity, sequence-specific DNA-binding transcription factor recruiting\" RELATED []'],\n",
       "  'is_a': ['GO:0030674 ! protein-macromolecule adaptor activity']},\n",
       " 'GO:0001011': {'name': ['obsolete transcription factor activity, sequence-specific DNA binding, RNA polymerase recruiting'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a specific DNA sequence and recruiting RNA polymerase to the DNA in order to form the preinitiation complex (PIC).\" [GOC:txnOH]'],\n",
       "  'comment': ['This term was obsoleted because it was not clearly defined and usage has been inconsistent.'],\n",
       "  'synonym': ['\"sequence-specific DNA binding RNA polymerase recruiting transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0140223']},\n",
       " 'GO:0001014': {'name': ['snoRNA transcription by RNA polymerase III'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The synthesis of small nucleolar RNA (snoRNA) from a DNA template by RNA polymerase III, originating at a type 2 RNA polymerase III promoter.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"snoRNA transcription from a type 2 RNA polymerase III promoter\" NARROW []'],\n",
       "  'is_a': ['GO:0006383 ! transcription by RNA polymerase III',\n",
       "   'GO:0009302 ! sno(s)RNA transcription']},\n",
       " 'GO:0001015': {'name': ['snoRNA transcription by RNA polymerase II'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The synthesis of small nucleolar RNA (snoRNA) from a DNA template by RNA polymerase II, originating at an RNA polymerase II promoter.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"snoRNA transcription from an RNA polymerase II promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0006366 ! transcription by RNA polymerase II',\n",
       "   'GO:0009302 ! sno(s)RNA transcription']},\n",
       " 'GO:0001016': {'name': ['RNA polymerase III transcription regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA region that controls the transcription of a gene by RNA polymerase III. Binding may occur as a sequence specific interaction or as an interaction observed only once a factor has been recruited to the DNA by other factors.\" [GOC:txnOH, GOC:vw, PMID:12381659]'],\n",
       "  'synonym': ['\"RNA polymerase III regulatory region DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001018': {'name': ['mitochondrial promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000997',\n",
       "   'GO:0001044',\n",
       "   'GO:0001045',\n",
       "   'GO:0070361',\n",
       "   'GO:0070362',\n",
       "   'GO:0070363',\n",
       "   'GO:0070364'],\n",
       "  'def': ['\"Binding to a DNA region that controls the transcription of the mitochondrial DNA.\" [GOC:txnOH, GOC:vw, PMID:20056105]'],\n",
       "  'synonym': ['\"HSP coding strand binding\" NARROW [PMID:9485316]',\n",
       "   '\"HSP non-coding strand binding\" NARROW [PMID:9485316]',\n",
       "   '\"HSPas binding\" NARROW [PMID:9485316]',\n",
       "   '\"HSPs binding\" NARROW [PMID:9485316]',\n",
       "   '\"LSP coding strand binding\" NARROW [PMID:9485316]',\n",
       "   '\"LSP non-coding strand binding\" NARROW [PMID:9485316]',\n",
       "   '\"LSPas binding\" NARROW [PMID:9485316]',\n",
       "   '\"LSPs binding\" NARROW [PMID:9485316]',\n",
       "   '\"mitochondrial heavy strand promoter anti-sense binding\" NARROW []',\n",
       "   '\"mitochondrial heavy strand promoter sense binding\" NARROW []',\n",
       "   '\"mitochondrial light strand promoter anti-sense binding\" NARROW []',\n",
       "   '\"mitochondrial light strand promoter sense binding\" NARROW []',\n",
       "   '\"mitochondrial proximal promoter sequence-specific DNA binding\" RELATED []',\n",
       "   '\"mitochondrial RNA polymerase core promoter proximal region sequence-specific DNA binding\" RELATED []',\n",
       "   '\"mitochondrial RNA polymerase core promoter sequence-specific DNA binding\" RELATED []',\n",
       "   '\"mitochondrial RNA polymerase regulatory region DNA binding\" RELATED []',\n",
       "   '\"mitochondrial RNA polymerase regulatory region sequence-specific DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001019': {'name': ['plastid promoter transcription regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a DNA region that controls transcription by a plastid RNA polymerase. Binding may occur as a sequence specific interaction or as an interaction observed only once a factor has been recruited to the DNA by other factors.\" [GOC:txnOH, GOC:vw]'],\n",
       "  'comment': ['plastid promoter regulatory region sequence-specific DNA binding'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001025': {'name': ['RNA polymerase III general transcription initiation factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase III transcription factor, a protein required to initiate or regulate transcription by RNA polymerase III.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"RNA polymerase III transcription factor binding\" BROAD []'],\n",
       "  'is_a': ['GO:0140296 ! general transcription initiation factor binding']},\n",
       " 'GO:0001026': {'name': ['obsolete TFIIIB-type transcription factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase III (Pol III) complex, typically composed of seventeen subunits, and with another protein, macromolecule, or complex, permitting those molecules to function in a coordinated way, Once recruited to an RNA polymerase III promoter by one or more other transcription factors, binds to DNA, recruits RNA polymerase III and facilitates the transition from the closed to the open complex.\" [GOC:txnOH]'],\n",
       "  'comment': ['This term was obsoleted because it represents a gene product.'],\n",
       "  'synonym': ['\"RNA polymerase III recruiting transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001027': {'name': ['obsolete RNA polymerase III type 1 promoter TFIIIB-type transcription factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase III (Pol III) complex, typically composed of seventeen subunits, and with another protein, macromolecule, or complex, permitting those molecules to function in a coordinated way. Once recruited to an RNA polymerase III type 1 promoter by TFIIIA-type and TFIIIC-type factors, binds to DNA, recruits RNA polymerase III and facilitates the transition from the closed to the open complex.\" [GOC:txnOH-2018, PMID:12381659]'],\n",
       "  'comment': ['The reason for obsoletion is that there is a single TFIIIB activity; this is not dependent on the promoter.'],\n",
       "  'synonym': ['\"RNA polymerase III type 1 promoter polymerase recruiting transcription factor activity\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001028': {'name': ['obsolete RNA polymerase III type 2 promoter TFIIIB-type transcription factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase III (Pol III) complex, typically composed of seventeen subunits, and with another protein, macromolecule, or complex, permitting those molecules to function in a coordinated way. Once recruited to an RNA polymerase III type 2 promoter by a TFIIIC-type factor, binds to DNA, recruits RNA polymerase III and facilitates the transition from the closed to the open complex.\" [GOC:txnOH-2018, PMID:12381659]'],\n",
       "  'comment': ['The reason for obsoletion is that there is a single TFIIIB activity; this is not dependent on the promoter.'],\n",
       "  'synonym': ['\"RNA polymerase III type 2 promoter polymerase recruiting transcription factor activity\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001029': {'name': ['obsolete RNA polymerase III type 3 promoter TFIIIB-type transcription factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase III (Pol III) complex, typically composed of seventeen subunits, and with another protein, macromolecule, or complex, permitting those molecules to function in a coordinated way. Once recruited to an RNA polymerase III type 3 promoter by SNAP-type and TFIIIC-type factors, binds to DNA, recruits RNA polymerase III and facilitates the transition from the closed to the open complex.\" [GOC:txnOH-2018, PMID:12381659]'],\n",
       "  'synonym': ['\"RNA polymerase III type 3 promoter polymerase recruiting transcription factor activity\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001039': {'name': ['RNA polymerase III hybrid type promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001037'],\n",
       "  'def': ['\"Binding to a sequence of DNA that is a part of a hybrid type promoter that controls transcription by RNA polymerase III (Pol III). A hybrid Pol III promoter contains both regulatory elements both upstream and downstream of the transcription initiation site. An example gene with such a promoter is the S. cerevisiae U6 gene.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'synonym': ['\"RNA polymerase III hybrid type promoter DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000992 ! RNA polymerase III cis-regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0001040': {'name': ['obsolete RNA polymerase III hybrid type promoter TFIIIB-type transcription factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase III (Pol III) complex, typically composed of seventeen subunits, and with another protein, macromolecule, or complex, permitting those molecules to function in a coordinated way. Once recruited to an RNA polymerase III hybrid type promoter, binds to DNA, recruits RNA polymerase III and facilitates the transition from the closed to the open complex.\" [GOC:txnOH-2018, PMID:12381659]'],\n",
       "  'comment': ['The reason for obsoletion is that there is a single TFIIIB activity; this is not dependent on the promoter.'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001042': {'name': ['RNA polymerase I core binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a RNA polymerase I core enzyme, a multisubunit eukaryotic nuclear RNA polymerase typically composed of seventeen subunits.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0043175 ! RNA polymerase core enzyme binding']},\n",
       " 'GO:0001046': {'name': ['core promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0000985', 'GO:0001047'],\n",
       "  'def': ['\"Binding to a sequence of DNA that is part of a core promoter region. The core promoter is composed of the transcription start site and binding sites for the RNA polymerase and the basal transcription machinery. The transcribed region might be described as a gene, cistron, or operon.\" [GOC:pg, GOC:txnOH]'],\n",
       "  'synonym': ['\"bacterial-type RNA polymerase core promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"core promoter binding\" RELATED []',\n",
       "   '\"eubacterial-type RNA polymerase core promoter sequence-specific DNA binding\" NARROW []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001048': {'name': ['RNA polymerase IV core binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to RNA polymerase IV core enzyme, a multisubunit eukaryotic nuclear RNA polymerase found in plants and involved in siRNA production.\" [GOC:txnOH, PMID:19110459]'],\n",
       "  'is_a': ['GO:0043175 ! RNA polymerase core enzyme binding']},\n",
       " 'GO:0001049': {'name': ['RNA polymerase V core binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to RNA polymerase V core enzyme, a multisubunit eukaryotic nuclear RNA polymerase found in plants and involved in production of noncoding transcripts at target loci for silencing.\" [GOC:txnOH, PMID:19110459]'],\n",
       "  'is_a': ['GO:0043175 ! RNA polymerase core enzyme binding']},\n",
       " 'GO:0001050': {'name': ['single-subunit type RNA polymerase binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a single subunit RNA polymerase enzyme, which is composed of a single catalytic subunit similar to the RNA polymerase enzymes from phages T3, T7, and SP6.\" [GOC:txnOH, PMID:20701995]'],\n",
       "  'synonym': ['\"SP6-type RNA polymerase binding\" EXACT []',\n",
       "   '\"T3-type RNA polymerase binding\" EXACT []',\n",
       "   '\"T3/T7 type RNA polymerase binding\" EXACT []',\n",
       "   '\"T7-type RNA polymerase binding\" EXACT []'],\n",
       "  'is_a': ['GO:0070063 ! RNA polymerase binding']},\n",
       " 'GO:0001051': {'name': ['plastid single-subunit type RNA polymerase binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a single subunit plastid RNA polymerase enzyme, which is composed of a single catalytic subunit similar to the RNA polymerase enzymes from phages T3, T7, and SP6.\" [GOC:txnOH, PMID:20701995]'],\n",
       "  'is_a': ['GO:0001050 ! single-subunit type RNA polymerase binding']},\n",
       " 'GO:0001052': {'name': ['plastid PEP RNA polymerase core enzyme binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a bacterial-type plastid PEP RNA polymerase core enzyme, typically consisting of two alpha, one beta, one beta prime, and one double prime subunit.\" [GOC:txnOH, PMID:20701995]'],\n",
       "  'is_a': ['GO:0001000 ! bacterial-type RNA polymerase core enzyme binding']},\n",
       " 'GO:0001054': {'name': ['RNA polymerase I activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase activity involved in transcription from RNA polymerase I promoter\" EXACT []'],\n",
       "  'is_a': [\"GO:0003899 ! DNA-directed 5'-3' RNA polymerase activity\"]},\n",
       " 'GO:0001055': {'name': ['RNA polymerase II activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase activity involved in transcription from RNA polymerase II promoter\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-6814549 \"Pre-snRNA transcript initiation, Integrator binding, LEC binding\"',\n",
       "   'Reactome:R-HSA-6814559 \"Pre-snRNA is elongated and capped with 7-methylguanosine\"'],\n",
       "  'is_a': [\"GO:0003899 ! DNA-directed 5'-3' RNA polymerase activity\"]},\n",
       " 'GO:0001056': {'name': ['RNA polymerase III activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase activity involved in transcription from RNA polymerase III promoter\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-1964482 \"RNA polymerase III transcribes microbial dsDNA to dsRNA\"'],\n",
       "  'is_a': [\"GO:0003899 ! DNA-directed 5'-3' RNA polymerase activity\"]},\n",
       " 'GO:0001057': {'name': ['RNA polymerase IV activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase activity involved in transcription from RNA polymerase IV promoter\" EXACT []'],\n",
       "  'is_a': [\"GO:0003899 ! DNA-directed 5'-3' RNA polymerase activity\"]},\n",
       " 'GO:0001058': {'name': ['RNA polymerase V activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"DNA-directed RNA polymerase activity involved in transcription from RNA polymerase V promoter\" EXACT []'],\n",
       "  'is_a': [\"GO:0003899 ! DNA-directed 5'-3' RNA polymerase activity\"]},\n",
       " 'GO:0001059': {'name': ['transcription by RNA polymerase IV'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The synthesis of RNA from a DNA template by RNA polymerase IV, originating at a Pol IV-specific promoter.\" [GOC:txnOH, PMID:19110459]'],\n",
       "  'synonym': ['\"transcription from RNA pol IV promoter\" EXACT []',\n",
       "   '\"transcription from RNA polymerase IV promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0006351 ! DNA-templated transcription']},\n",
       " 'GO:0001060': {'name': ['transcription by RNA polymerase V'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The synthesis of RNA from a DNA template by RNA polymerase V, originating at a Pol V-specific promoter.\" [GOC:txnOH, PMID:19110459]'],\n",
       "  'synonym': ['\"transcription from RNA pol V promoter\" EXACT []',\n",
       "   '\"transcription from RNA polymerase V promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0006351 ! DNA-templated transcription']},\n",
       " 'GO:0001061': {'name': ['obsolete bacterial-type RNA polymerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the reaction'],\n",
       "  'comment': ['This term was obsoleted because the activity is the same as its parent.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0003899']},\n",
       " 'GO:0001062': {'name': ['obsolete plastid PEP-A RNA polymerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the reaction'],\n",
       "  'comment': ['This term was obsoleted because the activity is the same as its parent.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0003899']},\n",
       " 'GO:0001063': {'name': ['obsolete plastid PEP-B RNA polymerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the reaction'],\n",
       "  'comment': ['This term was obsoleted because the activity is the same as its parent.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0003899']},\n",
       " 'GO:0001064': {'name': ['single subunit type RNA polymerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"T3/T7 type RNA polymerase activity\" EXACT []'],\n",
       "  'is_a': [\"GO:0003899 ! DNA-directed 5'-3' RNA polymerase activity\"]},\n",
       " 'GO:0001065': {'name': ['mitochondrial single subunit type RNA polymerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'is_a': ['GO:0001064 ! single subunit type RNA polymerase activity']},\n",
       " 'GO:0001066': {'name': ['plastid single subunit type RNA polymerase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'is_a': ['GO:0001064 ! single subunit type RNA polymerase activity']},\n",
       " 'GO:0001067': {'name': ['transcription regulatory region nucleic acid binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a nucleic acid region that regulates a nucleic acid-based process. Such processes include transcription, DNA replication, and DNA repair.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"regulatory region nucleic acid binding\" RELATED []'],\n",
       "  'is_a': ['GO:0003676 ! nucleic acid binding']},\n",
       " 'GO:0001068': {'name': ['transcription regulatory region RNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a RNA region within the transcript that regulates the transcription of a gene, cistron, or operon.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0001069 ! regulatory region RNA binding']},\n",
       " 'GO:0001069': {'name': ['regulatory region RNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a RNA region that regulates a nucleic acid-based process. Such processes include transcription, DNA replication, and DNA repair.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0001067 ! transcription regulatory region nucleic acid binding',\n",
       "   'GO:0003723 ! RNA binding']},\n",
       " 'GO:0001070': {'name': ['RNA-binding transcription regulator activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A transcription regulator activity that modulates the transcription of specific gene sets via selective and non-covalent binding to a specific RNA sequence. This function is known to occur in phages and viruses, for example the lambda N and the HIV tat proteins are necessary to allow RNA polymerase to read through terminator sequences.\" [GOC:txnOH-2018, PMID:1756726]'],\n",
       "  'synonym': ['\"RNA binding transcription factor activity\" RELATED []',\n",
       "   '\"RNA binding transcription regulator activity\" EXACT []'],\n",
       "  'is_a': ['GO:0140110 ! transcription regulator activity']},\n",
       " 'GO:0001072': {'name': ['transcription antitermination factor activity, RNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binds to RNA, typically within the nascent RNA transcript, to promote readthrough of a transcription termination site and thus extending the length of the RNA transcript produced. Examples of antitermination factors which bind the nascent RNA include the lambda N protein and the HIV-1 tat protein.\" [GOC:txnOH, PMID:8332211]'],\n",
       "  'synonym': ['\"RNA binding transcription antitermination factor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0140110 ! transcription regulator activity']},\n",
       " 'GO:0001073': {'name': ['transcription antitermination factor activity, DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binds to DNA, typically within region of the promoter and transcribed region, to promote readthrough of a transcription termination site and thus extending the length of the RNA transcript produced. Examples of antitermination factors which bind DNA include the lambda Q protein.\" [GOC:txnOH, PMID:8332211]'],\n",
       "  'synonym': ['\"DNA binding transcription antitermination factor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0140110 ! transcription regulator activity']},\n",
       " 'GO:0001074': {'name': ['obsolete transcription factor activity, RNA polymerase II proximal promoter sequence-specific DNA binding involved in preinitiation complex assembly'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a sequence of DNA that is in cis with and relatively close to a core promoter for RNA polymerase II (RNAP II) in order to promote assembly of the transcriptional preinitiation complex (PIC), the formation of which is a prerequisite for transcription from an RNA polymerase II promoter.\" [GOC:txnOH]'],\n",
       "  'comment': ['This term has been obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"RNA polymerase II core promoter proximal region sequence-specific DNA binding transcription factor activity involved in preinitiation complex assembly\" EXACT []',\n",
       "   '\"transcription factor activity, RNA polymerase II core promoter proximal region sequence-specific binding involved in preinitiation complex assembly\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001076': {'name': ['obsolete transcription factor activity, RNA polymerase II transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase II transcription factor, which may be a single protein or a complex, in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"RNA polymerase II transcription factor binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0061629', 'GO:0140110']},\n",
       " 'GO:0001079': {'name': ['nitrogen catabolite regulation of transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of one nitrogen source leads to the modulation of the frequency, rate, or extent of transcription, from an RNA polymerase II promoter, of specific genes involved in the metabolism of other nitrogen sources.\" [GOC:mah, GOC:txnOH, PMID:19104072]'],\n",
       "  'synonym': ['\"regulation of transcription from RNA polymerase II promoter by nitrogen catabolites\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0006357 ! regulation of transcription by RNA polymerase II',\n",
       "   'GO:0090293 ! nitrogen catabolite regulation of transcription']},\n",
       " 'GO:0001080': {'name': ['nitrogen catabolite activation of transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of one nitrogen source leads to an increase in the frequency, rate, or extent of transcription, from an RNA polymerase II promoter, of specific genes involved in the metabolism of other nitrogen sources.\" [GOC:mah, GOC:txnOH, PMID:19104072]'],\n",
       "  'synonym': ['\"positive regulation of transcription from RNA polymerase II promoter by nitrogen catabolites\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0001079 ! nitrogen catabolite regulation of transcription from RNA polymerase II promoter',\n",
       "   'GO:0090294 ! nitrogen catabolite activation of transcription']},\n",
       " 'GO:0001081': {'name': ['nitrogen catabolite repression of transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A transcription regulation process in which the presence of one nitrogen source leads to a decrease in the frequency, rate, or extent of transcription, from an RNA polymerase II promoter, of specific genes involved in the metabolism of other nitrogen sources.\" [GOC:mah, GOC:txnOH, PMID:19104072]'],\n",
       "  'synonym': ['\"negative regulation of transcription from RNA polymerase II promoter by nitrogen catabolites\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0001079 ! nitrogen catabolite regulation of transcription from RNA polymerase II promoter',\n",
       "   'GO:0090295 ! nitrogen catabolite repression of transcription']},\n",
       " 'GO:0001082': {'name': ['obsolete transcription factor activity, RNA polymerase I transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase I transcription factor, which may be a single protein or a complex, in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"RNA polymerase I transcription factor binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001083': {'name': ['obsolete transcription factor activity, RNA polymerase II basal transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor, which may be a single protein or a complex, in order to modulate transcription. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"RNA polymerase II basal transcription factor binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001084': {'name': ['obsolete transcription factor activity, TFIID-class binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor of the TFIID class in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"TFIID-class binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001094']},\n",
       " 'GO:0001086': {'name': ['obsolete transcription factor activity, TFIIA-class binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor of the TFIIA class in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"TFIIA-class binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001092']},\n",
       " 'GO:0001087': {'name': ['obsolete transcription factor activity, TFIIB-class binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor of the TFIIB class in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"TFIIB-class binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001093']},\n",
       " 'GO:0001088': {'name': ['obsolete transcription factor activity, TFIIE-class binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor of the TFIIE class in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"TFIIE-class binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001095']},\n",
       " 'GO:0001089': {'name': ['obsolete transcription factor activity, TFIIF-class transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor of the TFIIF class in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"TFIIF-class binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001096']},\n",
       " 'GO:0001090': {'name': ['obsolete transcription factor activity, TFIIH-class binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a basal RNA polymerase II transcription factor of the TFIIH class in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"TFIIH-class binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001091': {'name': ['RNA polymerase II general transcription initiation factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a basal RNA polymerase II transcription factor, any of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase II and defined as a basal or general transcription factor.\" [GOC:txnOH, PMID:16858867]'],\n",
       "  'synonym': ['\"RNA polymerase II basal transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001099 ! basal RNA polymerase II transcription machinery binding',\n",
       "   'GO:0140296 ! general transcription initiation factor binding']},\n",
       " 'GO:0001092': {'name': ['TFIIA-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase II transcription factor belonging to the TFIIA complex, one of the complexes involved in formation of the preinitiation complex (PIC) by RNA polymerase II and defined as a basal or general transcription factor.\" [GOC:krc, PMID:16858867]'],\n",
       "  'synonym': ['\"TFIIA-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001091 ! RNA polymerase II general transcription initiation factor binding',\n",
       "   'GO:0044877 ! protein-containing complex binding']},\n",
       " 'GO:0001093': {'name': ['TFIIB-class transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase II transcription factor of the TFIIB class, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase II.\" [GOC:krc, PMID:16858867]'],\n",
       "  'is_a': ['GO:0001091 ! RNA polymerase II general transcription initiation factor binding']},\n",
       " 'GO:0001094': {'name': ['TFIID-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase II transcription factor belonging to the TFIID complex, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase II.\" [GOC:krc, PMID:16858867]'],\n",
       "  'synonym': ['\"TFIID-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001091 ! RNA polymerase II general transcription initiation factor binding',\n",
       "   'GO:0044877 ! protein-containing complex binding']},\n",
       " 'GO:0001095': {'name': ['TFIIE-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase II transcription factor belonging to the TFIIE complex, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase II.\" [GOC:krc, PMID:16858867]'],\n",
       "  'synonym': ['\"TFIIE-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001091 ! RNA polymerase II general transcription initiation factor binding',\n",
       "   'GO:0044877 ! protein-containing complex binding']},\n",
       " 'GO:0001096': {'name': ['TFIIF-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase II transcription factor belonging to the TFIIF complex, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase II.\" [GOC:krc, PMID:16858867]'],\n",
       "  'synonym': ['\"TFIIF-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001091 ! RNA polymerase II general transcription initiation factor binding',\n",
       "   'GO:0044877 ! protein-containing complex binding']},\n",
       " 'GO:0001097': {'name': ['TFIIH-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase II transcription factor belonging to the TFIIH complex, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase II.\" [GOC:krc, PMID:16858867]'],\n",
       "  'synonym': ['\"TFIIH-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001091 ! RNA polymerase II general transcription initiation factor binding',\n",
       "   'GO:0044877 ! protein-containing complex binding']},\n",
       " 'GO:0001098': {'name': ['basal transcription machinery binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a component of the basal transcription machinery which is composed of the RNA polymerase core enzyme and the basal transcription factor(s), the minimal set of factors required for formation of the preinitiation complex (PIC) by the RNA polymerase.\" [GOC:txnOH]'],\n",
       "  'comment': ['Note that the definition of basal, or general, transcription factors has typically been done at a small number of well characterized activator-independent promoters. At an activator-dependent promoter, one or more additional factors are generally required in addition to the basal factors.'],\n",
       "  'is_a': ['GO:0005515 ! protein binding']},\n",
       " 'GO:0001099': {'name': ['basal RNA polymerase II transcription machinery binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a component of the basal transcription machinery for RNA polymerase II which is composed of the RNA polymerase II core enzyme, a multisubunit eukaryotic nuclear RNA polymerase typically composed of twelve subunits, and the basal RNA polymerase II transcription factors, the minimal set of factors required for formation of the preinitiation complex (PIC) by the RNA polymerase.\" [GOC:txnOH]'],\n",
       "  'comment': ['Note that the definition of basal, or general, transcription factors has typically been done at a small number of well characterized activator-independent promoters. At an activator-dependent promoter, one or more additional factors are generally required in addition to the basal factors.'],\n",
       "  'synonym': ['\"basal RNAP II transcription machinery binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001098 ! basal transcription machinery binding']},\n",
       " 'GO:0001100': {'name': ['negative regulation of exit from mitosis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the inhibition of progression from anaphase/telophase (high mitotic CDK activity) to G1 (low mitotic CDK activity).\" [GOC:rn]'],\n",
       "  'synonym': ['\"down regulation of exit from mitosis\" EXACT []',\n",
       "   '\"down-regulation of exit from mitosis\" EXACT []',\n",
       "   '\"downregulation of exit from mitosis\" EXACT []',\n",
       "   '\"inhibition of exit from mitosis\" NARROW []'],\n",
       "  'is_a': ['GO:0007096 ! regulation of exit from mitosis',\n",
       "   'GO:1901991 ! negative regulation of mitotic cell cycle phase transition'],\n",
       "  'relationship': ['negatively_regulates GO:0010458 ! exit from mitosis']},\n",
       " 'GO:0001101': {'name': ['response to acid chemical'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus by the chemical structure of the anion portion of a dissociated acid (rather than the acid acting as a proton donor). The acid chemical may be in gaseous, liquid or solid form.\" [GOC:go_curators, GOC:rn]'],\n",
       "  'comment': [\"This term should be used to describe a response to a specific acid as a chemical. E.g., if an organism were responding to glutamate, then the response would be glutamate-specific; the organism is actually responding to the chemical structure of the anion portion of the dissociated acid. Note that this term is in the subset of terms that should not be used for direct gene product annotation. Instead, select a child term or, if no appropriate child term exists, please request a new term. Direct annotations to this term may be amended during annotation QC. If annotating experiments where an acid is playing a role as a proton donor, please annotate to GO:0010447 'response to acidic pH' instead.\"],\n",
       "  'subset': ['gocheck_do_not_manually_annotate'],\n",
       "  'synonym': ['\"response to acid\" BROAD []',\n",
       "   '\"response to acid anion\" RELATED []',\n",
       "   '\"response to oxoanion\" RELATED []'],\n",
       "  'is_a': ['GO:0042221 ! response to chemical']},\n",
       " 'GO:0001108': {'name': ['bacterial-type RNA polymerase holo enzyme binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a component of the basal transcription machinery which is composed of a bacterial-type RNA polymerase core enzyme and a sigma factor, the minimal set of factors required for formation of the preinitiation complex (PIC) by a bacterial-type RNA polymerase.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"basal bacterial-type RNA polymerase transcription machinery binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001098 ! basal transcription machinery binding']},\n",
       " 'GO:0001109': {'name': ['promoter clearance during DNA-templated transcription'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001122'],\n",
       "  'def': ['\"Any process involved in the transition from the initiation to the elongation phases of transcription by a DNA-dependent RNA polymerase, generally including a conformational change from the initiation conformation to the elongation conformation. Promoter clearance often involves breaking contact with transcription factors involved only in the initiation phase and making contacts with elongation specific factors.\" [GOC:txnOH, PMID:15020047, PMID:18280161]'],\n",
       "  'synonym': ['\"promoter clearance during DNA-dependent transcription\" EXACT [GOC:txnOH]',\n",
       "   '\"promoter clearance from bacterial-type RNA polymerase promoter\" NARROW []',\n",
       "   '\"promoter escape\" BROAD []'],\n",
       "  'is_a': ['GO:0032774 ! RNA biosynthetic process'],\n",
       "  'relationship': ['part_of GO:0006351 ! DNA-templated transcription']},\n",
       " 'GO:0001110': {'name': ['RNA polymerase III promoter clearance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A process that mediates the transition from the initiation to the elongation phases of transcription by RNA polymerase III, generally including a conformational change from the initiation conformation to the elongation conformation. Promoter clearance often involves breaking contact with transcription factors involved only in the initiation phase.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"promoter clearance from RNA polymerase III promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0001109 ! promoter clearance during DNA-templated transcription'],\n",
       "  'relationship': ['part_of GO:0006383 ! transcription by RNA polymerase III']},\n",
       " 'GO:0001111': {'name': ['RNA polymerase II promoter clearance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A process that mediates the transition from the initiation to the elongation phases of transcription by RNA polymerase II, generally including a conformational change from the initiation conformation to the elongation conformation. Promoter clearance often involves breaking contact with transcription factors involved only in the initiation phase and making contacts with elongation specific factors.\" [GOC:txnOH, PMID:15020047, PMID:22982364, PMID:31628251]'],\n",
       "  'synonym': ['\"promoter clearance from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"promoter escape from RNA polymerase II promoter\" EXACT []',\n",
       "   '\"promoter-paused RNA polymerase II release\" EXACT []'],\n",
       "  'is_a': ['GO:0001109 ! promoter clearance during DNA-templated transcription'],\n",
       "  'relationship': ['part_of GO:0006366 ! transcription by RNA polymerase II']},\n",
       " 'GO:0001112': {'name': ['DNA-templated transcription open complex formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001127'],\n",
       "  'def': ['\"Any process involved in the melting of the DNA hybrid of the core promoter region within the transcriptional closed complex of an RNA polymerase preinitiation complex (PIC) to produce an open complex where the DNA duplex around the transcription initiation site is unwound to form the transcription bubble.\" [GOC:txnOH, PMID:15020047, PMID:18280161]'],\n",
       "  'synonym': ['\"DNA-dependent transcriptional open complex formation\" EXACT [GOC:txnOH]',\n",
       "   '\"DNA-templated transcriptional open complex formation\" EXACT []',\n",
       "   '\"promoter melting\" BROAD []',\n",
       "   '\"transcription open complex formation at bacterial-type RNA polymerase promoter\" NARROW []',\n",
       "   '\"transcriptional open complex formation at bacterial-type RNA polymerase promoter\" NARROW []'],\n",
       "  'is_a': ['GO:0001120 ! protein-DNA complex remodeling'],\n",
       "  'relationship': ['part_of GO:0006352 ! DNA-templated transcription initiation']},\n",
       " 'GO:0001113': {'name': ['transcription open complex formation at RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the melting of the DNA hybrid of the core promoter region within the transcriptional closed complex of an RNA polymerase II preinitiation complex (PIC) to produce an open complex where the DNA duplex around the transcription initiation site is unwound to form the transcription bubble.\" [GOC:txnOH, PMID:15020047, PMID:18280161]'],\n",
       "  'synonym': ['\"RNA polymerase II promoter melting\" EXACT []',\n",
       "   '\"transcriptional open complex formation at RNA polymerase II promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0001112 ! DNA-templated transcription open complex formation'],\n",
       "  'relationship': ['part_of GO:0006367 ! transcription initiation at RNA polymerase II promoter']},\n",
       " 'GO:0001114': {'name': ['protein-DNA-RNA complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A macromolecular complex containing protein, DNA, and RNA molecules.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex']},\n",
       " 'GO:0001115': {'name': ['protein-DNA-RNA complex organization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process in which macromolecules aggregate, disaggregate, or are modified, resulting in the formation, disassembly, or alteration of a protein-DNA-RNA complex.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"protein-DNA-RNA complex subunit organisation\" EXACT [GOC:mah]'],\n",
       "  'is_a': ['GO:0043933 ! protein-containing complex organization']},\n",
       " 'GO:0001116': {'name': ['protein-DNA-RNA complex assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of proteins, DNA, and RNA molecules to form a protein-DNA-RNA complex.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0001115 ! protein-DNA-RNA complex organization',\n",
       "   'GO:0065003 ! protein-containing complex assembly']},\n",
       " 'GO:0001117': {'name': ['protein-DNA-RNA complex disassembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The disaggregation of a protein-DNA-RNA complex into its constituent components.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0001115 ! protein-DNA-RNA complex organization',\n",
       "   'GO:0032984 ! protein-containing complex disassembly']},\n",
       " 'GO:0001118': {'name': ['transcription ternary complex disassembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The disaggregation of a transcription ternary complex, composed of RNA polymerase, template DNA, and an RNA transcript, into its constituent components.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"transcription protein-DNA-RNA complex disassembly\" EXACT []'],\n",
       "  'is_a': ['GO:0001117 ! protein-DNA-RNA complex disassembly'],\n",
       "  'relationship': ['part_of GO:0006353 ! DNA-templated transcription termination']},\n",
       " 'GO:0001119': {'name': ['protein-DNA-RNA complex remodeling'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The acquisition, loss, or modification of macromolecules within a protein-DNA-RNA complex, resulting in the alteration of an existing complex.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0001115 ! protein-DNA-RNA complex organization',\n",
       "   'GO:0034367 ! protein-containing complex remodeling']},\n",
       " 'GO:0001120': {'name': ['protein-DNA complex remodeling'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The acquisition, loss, or modification of macromolecules within a protein-DNA complex, resulting in the alteration of an existing complex.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0034367 ! protein-containing complex remodeling',\n",
       "   'GO:0071824 ! protein-DNA complex organization']},\n",
       " 'GO:0001128': {'name': ['obsolete RNA polymerase II transcription coactivator activity involved in preinitiation complex assembly'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase II (RNAP II) regulatory transcription factor and also with the RNAP II basal transcription machinery in order to increase the frequency, rate or stability of the aggregation, arrangement and bonding together of proteins on RNA polymerase II promoter DNA to form the transcriptional preinitiation complex (PIC). Cofactors generally do not bind DNA, but rather mediate protein-protein interactions between activating transcription factors and the basal RNAP II transcription machinery.\" [GOC:txnOH, PMID:16858867]'],\n",
       "  'comment': ['This term was obsoleted because it represented a mix of function and process and is not annotated consistently'],\n",
       "  'synonym': ['\"RNA polymerase II transcription coactivator activity involved in preinitiation complex assembly\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001129': {'name': ['obsolete RNA polymerase II transcription factor activity, TBP-class protein binding, involved in preinitiation complex assembly'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a member of the class of TATA-binding proteins (TBP), including any of the TBP-related factors (TRFs), to facilitate the aggregation, arrangement and bonding together of proteins on RNA polymerase II promoter DNA to form the transcriptional preinitiation complex (PIC), the formation of which is a prerequisite for transcription by RNA polymerase.\" [GOC:txnOH, PMID:16858867]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"TATA-binding protein binding RNA polymerase II transcription factor activity involved in preinitiation complex assembly\" NARROW []',\n",
       "   '\"TBP-class protein binding RNA polymerase II transcription factor activity involved in preinitiation complex assembly\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001132': {'name': ['obsolete RNA polymerase II transcription factor activity, TBP-class protein binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a member of the class of TATA-binding proteins (TBP), including any of the TBP-related factors (TRFs), in order to modulate transcription. The transcription factor may or may not also interact selectively with DNA as well.\" [GOC:txnOH, PMID:16858867]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"TATA-binding protein binding RNA polymerase II transcription factor activity\" NARROW []',\n",
       "   '\"TBP-class protein binding RNA polymerase II transcription factor activity\" EXACT []',\n",
       "   '\"TRF protein binding RNA polymerase II transcription factor activity\" NARROW []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001134': {'name': ['obsolete transcription regulator recruiting activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. The function of binding to an RNA polymerase (RNAP) transcription regulator and recruiting it to the general transcription machinery complex in order to modulate transcription initiation.\" [GOC:txnOH-2018, PMID:16858867]'],\n",
       "  'comment': [\"The reason for obsoletion is that this function is redundant with other terms, most often 'transcription co-regulator activity; GO:0003712'.\"],\n",
       "  'synonym': ['\"transcription factor activity, transcription factor recruiting\" RELATED []',\n",
       "   '\"transcription factor recruiting activity\" EXACT []',\n",
       "   '\"transcription factor recruiting transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0003712']},\n",
       " 'GO:0001135': {'name': ['obsolete RNA polymerase II transcription regulator recruiting activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001136', 'GO:0001137', 'GO:0001138'],\n",
       "  'def': ['\"OBSOLETE. The function of binding to an RNA polymerase II (RNAP II) transcription regulator and recruiting it to the general transcription machinery complex in order to modulate transcription initiation.\" [GOC:txnOH-2018, PMID:16858867]'],\n",
       "  'comment': [\"The reason for obsoletion is that this function is redundant with other terms, most often 'transcription co-regulator activity; GO:0003712'.\"],\n",
       "  'synonym': ['\"RNA polymerase II transcription factor recruiting activity\" EXACT []',\n",
       "   '\"RNA polymerase II transcription factor recruiting transcription factor activity\" EXACT []',\n",
       "   '\"TFIIE-class transcription factor recruiting transcription factor activity\" NARROW []',\n",
       "   '\"TFIIF-class transcription factor recruiting transcription factor activity\" NARROW []',\n",
       "   '\"TFIIH-class transcription factor recruiting transcription factor activity\" NARROW []',\n",
       "   '\"transcription factor activity, RNA polymerase II transcription factor recruiting\" RELATED []',\n",
       "   '\"transcription factor activity, TFIIE-class transcription factor recruiting\" NARROW []',\n",
       "   '\"transcription factor activity, TFIIF-class transcription factor recruiting\" NARROW []',\n",
       "   '\"transcription factor activity, TFIIH-class transcription factor recruiting\" NARROW []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0003712']},\n",
       " 'GO:0001139': {'name': ['RNA polymerase II complex recruiting activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase II (Pol II) complex, typically composed of twelve subunits, and with another protein, macromolecule, or complex, permitting those molecules to function in a coordinated way in order to facilitate the aggregation, arrangement and bonding together of proteins on an RNA polymerase II promoter DNA to form the transcriptional preinitiation complex (PIC), the formation of which is a prerequisite for transcription by RNA polymerase.\" [GOC:txnOH, PMID:16858867]'],\n",
       "  'synonym': ['\"core RNA polymerase II recruiting transcription factor activity\" EXACT []',\n",
       "   '\"transcription factor activity, core RNA polymerase II recruiting\" RELATED []'],\n",
       "  'is_a': ['GO:0000993 ! RNA polymerase II complex binding',\n",
       "   'GO:0030674 ! protein-macromolecule adaptor activity']},\n",
       " 'GO:0001147': {'name': ['transcription termination site sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001145', 'GO:0001146', 'GO:0001148', 'GO:0001160'],\n",
       "  'def': ['\"Binding to a sequence of DNA that promotes termination by RNA polymerase. The transcribed region might be described as a gene, cistron, or operon.\" [GOC:txnOH, PMID:18280161, PMID:18391175]'],\n",
       "  'comment': ['Transcription termination sites can be recognized by the RNA polymerase (RNAP) itself or by another protein which interacts with the RNAP to promote transcription termination. Note that not all genes have a DNA specific sequence that functions as a termination site; for most mRNAs transcribed by RNAP II termination is not mediated by a specific termination sequence, but is coupled to polyadenylation.'],\n",
       "  'synonym': ['\"bacterial-type RNA polymerase termination site sequence-specific DNA binding\" NARROW []',\n",
       "   '\"mitochondrial RNA polymerase termination site sequence-specific DNA binding\" NARROW []',\n",
       "   '\"mitochondrial RNA polymerase terminator site sequence-specific DNA binding transcription factor activity\" NARROW []',\n",
       "   '\"transcription factor activity, mitochondrial RNA polymerase terminator site sequence-specific binding\" NARROW []',\n",
       "   '\"transcription termination site DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001149': {'name': ['obsolete transcription factor activity, bacterial-type RNA polymerase termination site sequence-specific binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a sequence of DNA that is a termination site for bacterial-type RNA polymerase in order to promote transcription termination by bacterial-type RNA polymerase.\" [GOC:txnOH, PMID:18280161]'],\n",
       "  'synonym': ['\"bacterial-type RNA polymerase termination site sequence-specific DNA binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001152': {'name': ['obsolete transcription factor activity, RNA polymerase III type 1 promoter sequence-specific binding, TFIIIC recruiting'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. The function of binding to a specific DNA sequence motif in a type 1 RNA polymerase III (Pol III) promoter in order to recruit the transcription factor TFIIIC to the promoter.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'comment': ['This term was made obsolete in the 2018 Transcription branch revision. This represents the activity of TFIIIA only.'],\n",
       "  'synonym': ['\"RNA polymerase III type 1 promoter sequence-specific DNA binding TFIIIC recruiting transcription factor activity\" EXACT []',\n",
       "   '\"TFIIIA activity\" RELATED []',\n",
       "   '\"type 1 RNA polymerase III promoter recognition\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001153': {'name': ['obsolete transcription factor activity, RNA polymerase III transcription factor recruiting'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. The function of binding to an RNA polymerase III (RNAP III) transcription factor and recruiting it to the transcription machinery complex in order to modulate transcription by RNAP III.\" [GOC:txnOH]'],\n",
       "  'comment': ['This term was made obsolete in the 2018 Transcription branch revision. This term was redundant with other terms.'],\n",
       "  'synonym': ['\"RNA polymerase III transcription factor recruiting transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001154': {'name': ['TFIIIB-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase III transcription factor belonging to the TFIIB complex, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase III.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'synonym': ['\"TFIIIB-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001025 ! RNA polymerase III general transcription initiation factor binding']},\n",
       " 'GO:0001155': {'name': ['TFIIIA-class transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase III transcription factor of the TFIIIA class, one of the factors involved in formation of the preinitiation complex (PIC) at RNA polymerase III promoters.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'is_a': ['GO:0001025 ! RNA polymerase III general transcription initiation factor binding']},\n",
       " 'GO:0001156': {'name': ['TFIIIC-class transcription factor complex binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a general RNA polymerase III transcription factor belonging to the TFIIC complex, one of the factors involved in formation of the preinitiation complex (PIC) by RNA polymerase III.\" [GOC:txnOH, PMID:12381659]'],\n",
       "  'synonym': ['\"TFIIIC-class transcription factor binding\" EXACT []'],\n",
       "  'is_a': ['GO:0001025 ! RNA polymerase III general transcription initiation factor binding']},\n",
       " 'GO:0001161': {'name': ['intronic transcription regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0044213'],\n",
       "  'def': ['\"Binding to an intronic DNA sequence that regulates the transcription of the transcript it is contained within.\" [GOC:txnOH]'],\n",
       "  'comment': ['To minimize ambiguity in the use of the word \"promoter\" in GO, we have chosen the phrase \"transcription regulatory region\" to refer to all of the regulatory regions. Regulatory regions in the DNA which control initiation may include the \"core promoter\" where the basal transcription machinery binds, the \"core promoter proximal region\" where regulatory factors other than the basal machinery bind. There are also additional regulatory regions, in both the DNA and the RNA transcript, which regulate elongation or termination of transcription.'],\n",
       "  'synonym': ['\"intronic transcription regulatory region DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001162': {'name': ['RNA polymerase II intronic transcription regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase II intronic DNA sequence that regulates the transcription of the transcript it is contained within.\" [GOC:txnOH]'],\n",
       "  'comment': ['To minimize ambiguity in the use of the word \"promoter\" in GO, we have chosen the phrase \"transcription regulatory region\" to refer to all of the regulatory regions. Regulatory regions in the DNA which control initiation may include the \"core promoter\" where the basal transcription machinery binds, the \"core promoter proximal region\" where regulatory factors other than the basal machinery bind. There are also additional regulatory regions, in both the DNA and the RNA transcript, which regulate elongation or termination of transcription.'],\n",
       "  'is_a': ['GO:0000977 ! RNA polymerase II transcription regulatory region sequence-specific DNA binding',\n",
       "   'GO:0001161 ! intronic transcription regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0001163': {'name': ['RNA polymerase I transcription regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001013'],\n",
       "  'def': ['\"Binding to a specific sequence of DNA that is part of a regulatory region that controls the transcription of a gene or cistron by RNA polymerase I.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"RNA polymerase I regulatory region DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0000976 ! transcription cis-regulatory region binding']},\n",
       " 'GO:0001164': {'name': ['RNA polymerase I core promoter sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001187'],\n",
       "  'def': ['\"Binding to a regulatory region composed of the transcription start site and binding sites for transcription factors of the RNA polymerase I transcription machinery. This site is often referred to as the CORE element. In mammalian cells, the CORE element functions in conjunction with the Upstream Control Element (UCE), while in fungi, protozoa, and plants, the CORE element functions without a UCE.\" [GOC:txnOH, PMID:12865296, PMID:14969726, PMID:8057832]'],\n",
       "  'synonym': ['\"RNA polymerase I CORE element sequence-specific DNA binding\" EXACT []',\n",
       "   '\"RNA polymerase I CORE element sequence-specific DNA binding transcription factor recruiting transcription factor activity\" RELATED []',\n",
       "   '\"transcription factor activity, RNA polymerase I CORE element binding transcription factor recruiting\" RELATED []'],\n",
       "  'is_a': ['GO:0001046 ! core promoter sequence-specific DNA binding',\n",
       "   'GO:0001163 ! RNA polymerase I transcription regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0001165': {'name': ['RNA polymerase I cis-regulatory region sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001166'],\n",
       "  'def': ['\"Binding to a specific upstream regulatory DNA sequence (transcription factor recognition sequence or binding site) located in cis relative to the transcription start site (i.e., on the same strand of DNA) of a gene transcribed by RNA polymerase I. RNA polymerase I elements are referred to either enhancers or upstream control element (UCE, or alternately referred to as the upstream element).\" [GOC:txnOH, PMID:12865296, PMID:14969726, PMID:8057832]'],\n",
       "  'synonym': ['\"RNA polymerase I enhancer sequence-specific DNA binding\" RELATED []',\n",
       "   '\"RNA polymerase I upstream control element sequence-specific DNA binding\" RELATED []',\n",
       "   '\"RNA polymerase I upstream element sequence-specific DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0001163 ! RNA polymerase I transcription regulatory region sequence-specific DNA binding']},\n",
       " 'GO:0001167': {'name': ['obsolete RNA polymerase I transcription factor activity, sequence-specific DNA binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a specific DNA sequence in order to modulate transcription by RNA polymerase I. The transcription factor may or may not also interact selectively with a protein or macromolecular complex.\" [GOC:txnOH]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"sequence-specific DNA binding RNA polymerase I transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001168': {'name': ['obsolete transcription factor activity, RNA polymerase I upstream control element sequence-specific binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a upstream control element (UCE, or alternately referred to as the upstream element, UE), a sequence of DNA that is in cis with and relatively close to a core promoter for RNA polymerase I in order to modulate transcription by RNA polymerase I.\" [GOC:txnOH, PMID:12865296, PMID:14969726, PMID:8057832]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"RNA polymerase I upstream control element sequence-specific DNA binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001169': {'name': ['obsolete transcription factor activity, RNA polymerase I CORE element sequence-specific binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to CORE element, a regulatory region composed of the transcription start site and binding sites for transcription factors of the RNA polymerase I transcription machinery in order to modulate transcription by RNA polymerase I.\" [GOC:txnOH, PMID:12865296, PMID:14969726, PMID:8057832]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"RNA polymerase I CORE element sequence-specific DNA binding transcription factor activity\" EXACT []',\n",
       "   '\"RNA polymerase I core promoter sequence-specific DNA binding transcription factor\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001170': {'name': ['obsolete transcription factor activity, RNA polymerase I enhancer sequence-specific binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to a RNA polymerase I (Pol I) enhancer in order to modulate transcription by RNA polymerase I.\" [GOC:txnOH, PMID:12865296, PMID:14969726, PMID:8057832]'],\n",
       "  'comment': [\"This term was obsoleted because it represents two separate functions. The corresponding 'binding' term should be used in combination with a transcription regulator activity child.\"],\n",
       "  'synonym': ['\"RNA polymerase I enhancer sequence-specific DNA binding transcription factor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001171': {'name': ['reverse transcription'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A DNA synthesis process that uses RNA as the initial template for synthesis of DNA, but which also includes an RNase activity to remove the RNA strand of an RNA-DNA heteroduplex produced by the RNA-dependent synthesis step and use of the initial DNA strand as a template for DNA synthesis.\" [GOC:txnOH, PMID:20358252]'],\n",
       "  'xref': ['Wikipedia:Reverse_transcription'],\n",
       "  'is_a': ['GO:0006278 ! RNA-templated DNA biosynthetic process']},\n",
       " 'GO:0001172': {'name': ['RNA-templated transcription'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The synthesis of an RNA transcript from an RNA template.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"transcription, RNA-dependent\" EXACT [GOC:txnOH]',\n",
       "   '\"transcription, RNA-templated\" EXACT []'],\n",
       "  'is_a': ['GO:0032774 ! RNA biosynthetic process']},\n",
       " 'GO:0001173': {'name': ['DNA-templated transcriptional start site selection'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001176'],\n",
       "  'def': ['\"Any process involved in the selection of the specific location within the template strand of a DNA-dependent RNA polymerase promoter for hybridization of the cognate ribonucleotides and formation of first phosphodiester bond within the nascent transcript.\" [GOC:txnOH, PMID:16826228, PMID:18846104]'],\n",
       "  'synonym': ['\"DNA-dependent transcriptional start site selection\" EXACT [GOC:txnOH]',\n",
       "   '\"transcriptional start site selection at bacterial-type RNA polymerase promoter\" NARROW []'],\n",
       "  'is_a': ['GO:0032774 ! RNA biosynthetic process'],\n",
       "  'relationship': ['part_of GO:0006352 ! DNA-templated transcription initiation']},\n",
       " 'GO:0001174': {'name': ['transcriptional start site selection at RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the selection of the specific location within the template strand of an RNA polymerase II promoter for hybridization of the cognate ribonucleotides and formation of first phosphodiester bond within the nascent transcript.\" [GOC:txnOH, PMID:16826228, PMID:18846104]'],\n",
       "  'is_a': ['GO:0001173 ! DNA-templated transcriptional start site selection'],\n",
       "  'relationship': ['part_of GO:0006367 ! transcription initiation at RNA polymerase II promoter']},\n",
       " 'GO:0001175': {'name': ['transcriptional start site selection at RNA polymerase III promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process involved in the selection of the specific location within the template strand of an RNA polymerase III promoter for hybridization of the cognate ribonucleotides and formation of first phosphodiester bond within the nascent transcript.\" [GOC:txnOH]'],\n",
       "  'is_a': ['GO:0001173 ! DNA-templated transcriptional start site selection'],\n",
       "  'relationship': ['part_of GO:0006384 ! transcription initiation at RNA polymerase III promoter']},\n",
       " 'GO:0001177': {'name': ['regulation of transcription open complex formation at RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the rate, frequency or extent of a process involved the melting of the DNA hybrid of the core promoter region within the transcriptional closed complex of an RNA polymerase II preinitiation complex (PIC) to produce an open complex where the DNA duplex around the transcription initiation site is unwound to form the transcription bubble.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"regulation of transcriptional open complex formation at RNA polymerase II promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0051128 ! regulation of cellular component organization'],\n",
       "  'relationship': ['regulates GO:0001113 ! transcription open complex formation at RNA polymerase II promoter']},\n",
       " 'GO:0001178': {'name': ['regulation of transcriptional start site selection at RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the rate, frequency or extent of a process involved in the selection of the specific location within the template strand of an RNA polymerase II promoter for hybridization of the cognate ribonucleotides and formation of first phosphodiester bond within the nascent transcript.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"regulation of transcription start site selection at RNA polymerase II promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0060260 ! regulation of transcription initiation by RNA polymerase II'],\n",
       "  'relationship': ['regulates GO:0001174 ! transcriptional start site selection at RNA polymerase II promoter']},\n",
       " 'GO:0001179': {'name': ['RNA polymerase I general transcription initiation factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an RNA polymerase I transcription factor, a protein required to initiate or regulate transcription by RNA polymerase I.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"RNA polymerase I transcription factor binding\" BROAD []'],\n",
       "  'is_a': ['GO:0140296 ! general transcription initiation factor binding']},\n",
       " 'GO:0001181': {'name': ['RNA polymerase I general transcription initiation factor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A general transcription initiation factor activity that contributes to transcription start site selection and transcription initiation of genes transcribed by RNA polymerase I. Factors required for RNA polymerase I transcription initiation include upstream activation factor (UAF), core factor (CF), TATA binding protein (TBP) and RRN3. In all species characterized, RNA polymerase I transcribes a large polycistronic transcript that is processed into several mature rRNAs (3 or 4 depending on the species), including the large subunit rRNA (28S in humans), the small subunit rRNA (18S in humans), as well as one or two additional smaller rRNAs (the 5.8S rRNA in humans). In most species, this large rRNA transcript is the sole product of RNA polymerase I. However there are rare exceptions, such as Trypanosoma brucei, where RNA polymerase I also transcribes certain mRNAs.\" [GOC:txnOH-2018, PMID:11500378, PMID:17972917, PMID:25346433, PMID:28340337, PMID:28842442, PMID:31358304]'],\n",
       "  'synonym': ['\"core RNA polymerase I binding transcription factor activity\" BROAD []',\n",
       "   '\"general RNA polymerase I transcription factor activity\" EXACT []',\n",
       "   '\"RNA polymerase I transcription general initiation factor activity\" EXACT []',\n",
       "   '\"transcription factor activity, core RNA polymerase I binding\" BROAD []'],\n",
       "  'is_a': ['GO:0140223 ! general transcription initiation factor activity']},\n",
       " 'GO:0001182': {'name': ['RNA polymerase I promoter clearance'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001184'],\n",
       "  'def': ['\"A process that mediates the transition from the initiation to the elongation phases of transcription by RNA polymerase I, generally including a conformational change from the initiation conformation to the elongation conformation. Promoter clearance often involves breaking contact with transcription factors involved only in the initiation phase and making contacts with elongation specific factors.\" [GOC:txnOH, PMID:18559419]'],\n",
       "  'synonym': ['\"promoter clearance from RNA polymerase I promoter\" EXACT []',\n",
       "   '\"promoter clearance from RNA polymerase I promoter for nuclear large rRNA transcript\" RELATED []',\n",
       "   '\"RNA polymerase I promoter escape\" EXACT []'],\n",
       "  'is_a': ['GO:0001109 ! promoter clearance during DNA-templated transcription'],\n",
       "  'relationship': ['part_of GO:0006360 ! transcription by RNA polymerase I']},\n",
       " 'GO:0001186': {'name': ['obsolete RNA polymerase I transcription regulator recruiting activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. The function of binding to an RNA polymerase I (RNAP I) transcription regulator and recruiting it to the general transcription machinery complex in order to modulate transcription initiation.\" [GOC:txnOH-2018, PMID:12381659, PMID:14969726, PMID:8057832]'],\n",
       "  'comment': [\"The reason for obsoletion is that this function is redundant with other terms, most often 'transcription co-regulator activity; GO:0003712'.\"],\n",
       "  'synonym': ['\"RNA polymerase I transcription factor recruiting activity\" EXACT []',\n",
       "   '\"RNA polymerase I transcription factor recruiting transcription factor activity\" EXACT []',\n",
       "   '\"transcription factor activity, RNA polymerase I transcription factor recruiting\" RELATED []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0003712']},\n",
       " 'GO:0001188': {'name': ['RNA polymerase I preinitiation complex assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001189'],\n",
       "  'def': ['\"The aggregation, arrangement and bonding together of proteins on promoter DNA to form the transcriptional preinitiation complex (PIC), the formation of which is a prerequisite for transcription from an RNA polymerase I promoter.\" [GOC:txnOH, PMID:12381659, PMID:14969726, PMID:8057832]'],\n",
       "  'synonym': ['\"RNA polymerase I transcriptional preinitiation complex assembly\" EXACT []',\n",
       "   '\"RNA polymerase I transcriptional preinitiation complex assembly at the promoter for the nuclear large rRNA transcript\" NARROW []',\n",
       "   '\"RNA polymerase I transcriptional preinitiation complex assembly at the promoter for the nucleolar primary rRNA transcript\" NARROW []'],\n",
       "  'is_a': ['GO:0070897 ! transcription preinitiation complex assembly'],\n",
       "  'relationship': ['part_of GO:0006361 ! transcription initiation at RNA polymerase I promoter']},\n",
       " 'GO:0001190': {'name': ['obsolete transcriptional activator activity, RNA polymerase II transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase II transcription factor, which may be a single protein or a complex, in order to increase the frequency, rate or extent of transcription from an RNA polymerase II promoter. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"RNA polymerase II transcription factor binding transcription activator activity\" EXACT []',\n",
       "   '\"RNA polymerase II transcription factor binding transcription factor activity involved in positive regulation of transcription\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001228', 'GO:0061629']},\n",
       " 'GO:0001191': {'name': ['obsolete transcriptional repressor activity, RNA polymerase II transcription factor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Binding to an RNA polymerase II transcription factor, which may be a single protein or a complex, in order to stop, prevent, or reduce the frequency, rate or extent of transcription from an RNA polymerase II promoter. A protein binding transcription factor may or may not also interact with the template nucleic acid (either DNA or RNA) as well.\" [GOC:txnOH, PMID:9811836]'],\n",
       "  'comment': ['The reason for obsoletion is that this concept is partly covered by other concepts in the ontology and usage has been inconsistent.'],\n",
       "  'synonym': ['\"RNA polymerase II transcription factor binding transcription factor activity involved in negative regulation of transcription\" EXACT []',\n",
       "   '\"RNA polymerase II transcription factor binding transcription repressor activity\" EXACT []',\n",
       "   '\"RNA polymerase II transcription repressor activity\" BROAD []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001227', 'GO:0061629']},\n",
       " 'GO:0001192': {'name': ['maintenance of transcriptional fidelity during transcription elongation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001194'],\n",
       "  'def': ['\"Suppression of the occurrence of transcriptional errors, such as substitutions and/or insertions of nucleotides that do not correctly match the template base, during the process of transcription elongation on a DNA template.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"maintenance of transcriptional fidelity during DNA-dependent transcription elongation\" EXACT [GOC:txnOH]',\n",
       "   '\"maintenance of transcriptional fidelity during DNA-dependent transcription elongation from bacterial-type RNA polymerase promoter\" EXACT [GOC:txnOH]',\n",
       "   '\"maintenance of transcriptional fidelity during DNA-templated transcription elongation\" EXACT []',\n",
       "   '\"maintenance of transcriptional fidelity during DNA-templated transcription elongation from bacterial-type RNA polymerase promoter\" NARROW []'],\n",
       "  'is_a': ['GO:0032774 ! RNA biosynthetic process'],\n",
       "  'relationship': ['part_of GO:0006354 ! DNA-templated transcription elongation']},\n",
       " 'GO:0001193': {'name': ['maintenance of transcriptional fidelity during transcription elongation by RNA polymerase II'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Suppression of the occurrence of transcriptional errors, such as substitutions and/or insertions of nucleotides that do not correctly match the template base, during the process of transcription elongation from an RNA polymerase II promoter.\" [GOC:txnOH, PMID:14531857, PMID:16492753, PMID:17535246]'],\n",
       "  'synonym': ['\"maintenance of transcriptional fidelity during DNA-dependent transcription elongation from RNA polymerase II promoter\" EXACT [GOC:txnOH]',\n",
       "   '\"maintenance of transcriptional fidelity during DNA-templated transcription elongation from RNA polymerase II promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0001192 ! maintenance of transcriptional fidelity during transcription elongation'],\n",
       "  'relationship': ['part_of GO:0006368 ! transcription elongation by RNA polymerase II']},\n",
       " 'GO:0001195': {'name': ['maintenance of transcriptional fidelity during transcription elongation by RNA polymerase III'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Suppression of the occurrence of transcriptional errors, such as substitutions and/or insertions of nucleotides that do not correctly match the template base, during the process of transcription elongation from a RNA polymerase III promoter.\" [GOC:txnOH]'],\n",
       "  'synonym': ['\"maintenance of transcriptional fidelity during DNA-dependent transcription elongation from RNA polymerase III promoter\" EXACT [GOC:txnOH]',\n",
       "   '\"maintenance of transcriptional fidelity during DNA-templated transcription elongation from RNA polymerase III promoter\" EXACT []'],\n",
       "  'is_a': ['GO:0001192 ! maintenance of transcriptional fidelity during transcription elongation'],\n",
       "  'relationship': ['part_of GO:0006385 ! transcription elongation by RNA polymerase III']},\n",
       " 'GO:0001196': {'name': ['obsolete regulation of mating-type specific transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any mating-type specific process that modulates the frequency, rate or extent of transcription from an RNA polymerase II promoter.\" [GOC:txnOH]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001197': {'name': ['obsolete positive regulation of mating-type specific transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any mating-type specific process that activates or increases the rate of transcription from an RNA polymerase II promoter.\" [GOC:txnOH]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"activation of transcription from an RNA polymerase II promoter, mating-type specific\" NARROW []',\n",
       "   '\"stimulation of transcription from an RNA polymerase II promoter, mating-type specific\" NARROW []',\n",
       "   '\"up regulation of transcription from an RNA polymerase II promoter, mating-type specific\" EXACT []',\n",
       "   '\"up-regulation of transcription from an RNA polymerase II promoter, mating-type specific\" EXACT []',\n",
       "   '\"upregulation of transcription from an RNA polymerase II promoter, mating-type specific\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001198': {'name': ['obsolete negative regulation of mating-type specific transcription from RNA polymerase II promoter'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any mating-type specific process that stops, prevents or reduces the rate of transcription from an RNA polymerase II promoter.\" [GOC:txnOH]'],\n",
       "  'comment': ['This term was obsoleted because it represents a GO-CAM model.'],\n",
       "  'synonym': ['\"down regulation of RNA polymerase II transcription, mating-type specific\" EXACT []',\n",
       "   '\"down-regulation of RNA polymerase II transcription, mating-type specific\" EXACT []',\n",
       "   '\"downregulation of RNA polymerase II transcription, mating-type specific\" EXACT []',\n",
       "   '\"inhibition of RNA polymerase II transcription, mating-type specific\" NARROW []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001207': {'name': ['obsolete histone displacement'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The removal of histones, including histone dimers, from nucleosomes within chromatin.\" [GOC:krc, PMID:15525516, PMID:17496903, PMID:21807128]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0140713']},\n",
       " 'GO:0001208': {'name': ['obsolete histone H2A-H2B dimer displacement'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The removal of a H2A-H2B histone dimer from a nucleosome within chromatin.\" [GOC:krc, PMID:15525516, PMID:17496903, PMID:21807128]'],\n",
       "  'comment': ['This term was obsoleted because it represents a molecular function.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0140713']},\n",
       " 'GO:0001216': {'name': ['DNA-binding transcription activator activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001140', 'GO:0001215'],\n",
       "  'def': ['\"A DNA-binding transcription factor activity that activates or increases transcription of specific gene sets.\" [GOC:txnOH-2018]'],\n",
       "  'comment': ['For usage guidance, see comment in GO:0003700 ; DNA-binding transcription factor activity.'],\n",
       "  'synonym': ['\"bacterial-type RNA polymerase core promoter proximal region sequence-specific DNA binding transcription factor activity involved in positive regulation of transcription\" RELATED []',\n",
       "   '\"bacterial-type RNA polymerase transcriptional activator activity, metal ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase transcriptional activator activity, sequence-specific DNA binding\" NARROW []',\n",
       "   '\"metal ion regulated sequence-specific DNA binding bacterial-type RNA polymerase transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"sequence-specific DNA binding bacterial-type RNA polymerase transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"transcriptional activator activity, bacterial-type RNA polymerase core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"transcriptional activator activity, bacterial-type RNA polymerase proximal promoter sequence-specific DNA binding\" RELATED []'],\n",
       "  'is_a': ['GO:0003700 ! DNA-binding transcription factor activity']},\n",
       " 'GO:0001217': {'name': ['DNA-binding transcription repressor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001141', 'GO:0001218', 'GO:0001219', 'GO:0001220'],\n",
       "  'def': ['\"A DNA-binding transcription factor activity that represses or decreases the transcription of specific gene sets.\" [GOC:txnOH-2018]'],\n",
       "  'comment': ['For usage guidance, see comment in GO:0003700 ; DNA-binding transcription factor activity.'],\n",
       "  'synonym': ['\"bacterial-type DNA binding transcription repressor activity\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase core promoter proximal region sequence-specific DNA binding transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase transcriptional repressor activity, cadmium ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase transcriptional repressor activity, copper ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase transcriptional repressor activity, metal ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"bacterial-type RNA polymerase transcriptional repressor activity, sequence-specific DNA binding\" NARROW []',\n",
       "   '\"cadmium ion regulated sequence-specific DNA binding bacterial-type RNA polymerase transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"copper ion regulated sequence-specific DNA binding bacterial-type RNA polymerase transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"metal ion regulated sequence-specific DNA binding bacterial-type RNA polymerase transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"sequence-specific DNA binding bacterial-type RNA polymerase transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"transcriptional repressor activity, bacterial-type RNA polymerase core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"transcriptional repressor activity, bacterial-type RNA polymerase proximal promoter sequence-specific DNA binding\" NARROW []'],\n",
       "  'is_a': ['GO:0003700 ! DNA-binding transcription factor activity']},\n",
       " 'GO:0001221': {'name': ['transcription coregulator binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001224'],\n",
       "  'def': ['\"Binding to a transcription coregulator, a protein involved in regulation of transcription via protein-protein interactions with transcription factors and other transcription regulatory proteins. Cofactors do not bind DNA directly, but rather mediate protein-protein interactions between regulatory transcription factors and the basal transcription machinery.\" [GOC:krc]'],\n",
       "  'synonym': ['\"RNA polymerase II transcription cofactor binding\" RELATED []',\n",
       "   '\"RNA polymerase II transcription coregulator binding\" RELATED []',\n",
       "   '\"transcription cofactor binding\" RELATED []'],\n",
       "  'is_a': ['GO:0008134 ! transcription factor binding']},\n",
       " 'GO:0001222': {'name': ['transcription corepressor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001226'],\n",
       "  'def': ['\"Binding to a transcription corepressor, a protein involved in negative regulation of transcription via protein-protein interactions with transcription factors and other proteins that negatively regulate transcription. Transcription corepressors do not bind DNA directly, but rather mediate protein-protein interactions between repressing transcription factors and the basal transcription machinery.\" [GOC:krc]'],\n",
       "  'synonym': ['\"RNA polymerase II transcription corepressor binding\" RELATED []'],\n",
       "  'is_a': ['GO:0001221 ! transcription coregulator binding']},\n",
       " 'GO:0001223': {'name': ['transcription coactivator binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001225'],\n",
       "  'def': ['\"Binding to a transcription coactivator, a protein involved in positive regulation of transcription via protein-protein interactions with transcription factors and other proteins that positively regulate transcription. Transcription coactivators do not bind DNA directly, but rather mediate protein-protein interactions between activating transcription factors and the basal transcription machinery.\" [GOC:krc]'],\n",
       "  'synonym': ['\"RNA polymerase II transcription coactivator binding\" RELATED []'],\n",
       "  'is_a': ['GO:0001221 ! transcription coregulator binding']},\n",
       " 'GO:0001227': {'name': ['DNA-binding transcription repressor activity, RNA polymerase II-specific'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001078', 'GO:0001206', 'GO:0001210', 'GO:0001214'],\n",
       "  'def': ['\"A DNA-binding transcription factor activity that represses or decreases the transcription of specific gene sets transcribed by RNA polymerase II.\" [GOC:txnOH-2018]'],\n",
       "  'comment': ['For usage guidance, see comment in GO:0003700 ; DNA-binding transcription factor activity.'],\n",
       "  'synonym': ['\"distal enhancer DNA-binding transcription repressor activity, RNA polymerase II-specific\" NARROW []',\n",
       "   '\"metal ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"metal ion regulated sequence-specific DNA binding transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"proximal promoter DNA-binding transcription repressor activity, RNA polymerase II-specific\" NARROW []',\n",
       "   '\"RNA polymerase II core promoter proximal region sequence-specific DNA binding transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"RNA polymerase II distal enhancer sequence-specific DNA-binding transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"RNA polymerase II transcription regulatory region sequence-specific DNA binding transcription factor activity involved in negative regulation of transcription\" EXACT []',\n",
       "   '\"RNA polymerase II transcriptional repressor activity, metal ion regulated core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcriptional repressor activity, metal ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"sequence-specific distal enhancer binding RNA polymerase II transcription factor activity involved in negative regulation of transcription\" NARROW []',\n",
       "   '\"transcriptional repressor activity, metal ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"transcriptional repressor activity, RNA polymerase II distal enhancer sequence-specific binding\" NARROW []',\n",
       "   '\"transcriptional repressor activity, RNA polymerase II proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"transcriptional repressor activity, RNA polymerase II transcription regulatory region sequence-specific DNA binding\" EXACT []'],\n",
       "  'is_a': ['GO:0000981 ! DNA-binding transcription factor activity, RNA polymerase II-specific',\n",
       "   'GO:0001217 ! DNA-binding transcription repressor activity']},\n",
       " 'GO:0001228': {'name': ['DNA-binding transcription activator activity, RNA polymerase II-specific'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001077',\n",
       "   'GO:0001205',\n",
       "   'GO:0001209',\n",
       "   'GO:0001211',\n",
       "   'GO:0001212',\n",
       "   'GO:0001213'],\n",
       "  'def': ['\"A DNA-binding transcription factor activity that activates or increases transcription of specific gene sets transcribed by RNA polymerase II.\" [GOC:aruk, GOC:txnOH-2018, PMID:20737563, PMID:27145859]'],\n",
       "  'comment': ['For usage guidance, see comment in GO:0003700 ; DNA-binding transcription factor activity.'],\n",
       "  'synonym': ['\"copper ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"distal enhancer DNA-binding transcription activator activity, RNA polymerase II-specific\" NARROW []',\n",
       "   '\"metal ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"metal ion regulated sequence-specific DNA binding transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"proximal promoter DNA-binding transcription activator activity, RNA polymerase II-specific\" NARROW []',\n",
       "   '\"RNA polymerase II core promoter proximal region sequence-specific DNA binding transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"RNA polymerase II distal enhancer sequence-specific DNA-binding transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"RNA polymerase II transcription regulatory region sequence-specific DNA binding transcription factor activity involved in positive regulation of transcription\" EXACT []',\n",
       "   '\"RNA polymerase II transcriptional activator activity, copper ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcriptional activator activity, metal ion regulated core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcriptional activator activity, metal ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcriptional activator activity, zinc ion regulated core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"RNA polymerase II transcriptional activator activity, zinc ion regulated proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"sequence-specific distal enhancer binding RNA polymerase II transcription factor activity involved in positive regulation of transcription\" NARROW []',\n",
       "   '\"transcriptional activator activity, metal ion regulated sequence-specific DNA binding\" NARROW []',\n",
       "   '\"transcriptional activator activity, RNA polymerase II core promoter proximal region sequence-specific binding\" NARROW []',\n",
       "   '\"transcriptional activator activity, RNA polymerase II distal enhancer sequence-specific DNA binding\" NARROW []',\n",
       "   '\"transcriptional activator activity, RNA polymerase II proximal promoter sequence-specific DNA binding\" NARROW []',\n",
       "   '\"transcriptional activator activity, RNA polymerase II transcription regulatory region sequence-specific DNA binding\" EXACT []',\n",
       "   '\"zinc ion regulated core promoter proximal region sequence-specific DNA binding RNA polymerase II transcription factor activity involved in positive regulation of transcription\" NARROW []'],\n",
       "  'is_a': ['GO:0000981 ! DNA-binding transcription factor activity, RNA polymerase II-specific',\n",
       "   'GO:0001216 ! DNA-binding transcription activator activity']},\n",
       " 'GO:0001300': {'name': ['obsolete chronological cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The process associated with progression of the cell from its inception to the end of its lifespan that occurs when the cell is in a non-dividing, or quiescent, state.\" [GOC:jh, PMID:12044934]'],\n",
       "  'comment': ['This term was obsoleted because it represent an assay - how long the cell lives -, not a true biological process.'],\n",
       "  'synonym': ['\"chronological cell ageing\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044838', 'GO:0090398']},\n",
       " 'GO:0001301': {'name': ['obsolete progressive alteration of chromatin involved in cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any chromatin organization process that occurs during the lifespan of the cell that results in changes in chromatin structure. Such changes may lead to gene dysregulation and ultimately to a loss in cell homeostasis, bringing about an aging phenotype.\" [GOC:jh, GOC:vw, PMID:12044938]'],\n",
       "  'comment': ['This term was obsoleted because it represents a phenotype.'],\n",
       "  'synonym': ['\"age-dependent accumulation of genetic damage\" RELATED []',\n",
       "   '\"progressive alteration of chromatin during cell ageing\" EXACT []',\n",
       "   '\"progressive alteration of chromatin during cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001302': {'name': ['obsolete replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The process associated with progression of the cell from its inception to the end of its lifespan that occurs as the cell continues cycles of growth and division.\" [GOC:jh, PMID:12044934]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"replicative cell ageing\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0090399']},\n",
       " 'GO:0001303': {'name': ['obsolete nucleolar fragmentation involved in replicative aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A nucleolar fragmentation process that gives rise to multiple rounded structures and that occurs in conjunction with increasing age in dividing cells.\" [GOC:jh, PMID:9891807]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"nucleolar fragmentation during replicative ageing\" RELATED []',\n",
       "   '\"nucleolar fragmentation during replicative aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001304': {'name': ['obsolete progressive alteration of chromatin involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A process that results in changes in chromatin structure contributing to replicative cell aging.\" [GOC:dph, GOC:jh, GOC:tb]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"progressive alteration of chromatin during replicative cell ageing\" EXACT []',\n",
       "   '\"progressive alteration of chromatin during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001305': {'name': ['obsolete progressive alteration of chromatin involved in chronological cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A process that results in changes in chromatin structure contributing to chronological cell aging, occurring in non-dividing cells.\" [GOC:dph, GOC:jh, GOC:tb]'],\n",
       "  'comment': ['This term was obsoleted because it represent an assay - how long the cell lives -, not a true biological process.'],\n",
       "  'synonym': ['\"progressive alteration of chromatin during chronological cell ageing\" RELATED []',\n",
       "   '\"progressive alteration of chromatin during chronological cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044838', 'GO:0090398']},\n",
       " 'GO:0001306': {'name': ['obsolete age-dependent response to oxidative stress'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of oxidative stress, a state often resulting from exposure to high levels of reactive oxygen species, where the change varies according to the age of the cell or organism.\" [GOC:jh, PMID:12044938]'],\n",
       "  'comment': ['This term was obsoleted because it represents a phenotype.'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0034599']},\n",
       " 'GO:0001307': {'name': ['obsolete extrachromosomal circular DNA accumulation involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Increase in abundance of circular DNA molecules in dividing cells as they age. These molecules originate in the chromosome but are excised and circularized, often by intramolecular homologous recombination between direct tandem repeats, and then replicated independently of chromosomal replication.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"extrachromosomal circular DNA accumulation during replicative cell ageing\" RELATED []',\n",
       "   '\"extrachromosomal circular DNA accumulation during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001308': {'name': ['obsolete negative regulation of chromatin silencing involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The process, which occurs as a dividing cell ages, leading to expression of genes that are typically not expressed due to silencing by regulatory proteins.\" [GOC:jh, PMID:12044934]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"loss of chromatin silencing during replicative cell ageing\" RELATED []',\n",
       "   '\"loss of chromatin silencing involved in replicative cell aging\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001309': {'name': ['obsolete age-dependent telomere shortening'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Progressive reduction in length of the telomeres, the termini of eukaryotic chromosomes, that occurs as part of the cellular aging process.\" [GOC:jh, PMID:9891807]'],\n",
       "  'comment': ['This term was made obsolete because it represents a phenotype.'],\n",
       "  'synonym': ['\"age-dependent telomere shortening\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:1903824']},\n",
       " 'GO:0001310': {'name': ['obsolete extrachromosomal rDNA circle accumulation involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Increase in abundance of circular DNA molecules containing ribosomal DNA repeats in dividing cells as they age. These molecules originate in the chromosome but are excised and circularized, often by intramolecular homologous recombination between direct tandem repeats, and then replicated independently of chromosomal replication.\" [GOC:jh, PMID:12044934]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"extrachromosomal rDNA circle accumulation during replicative cell ageing\" RELATED []',\n",
       "   '\"extrachromosomal rDNA circle accumulation during replicative cell aging\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"extrachromosomal ribosomal DNA circle accumulation during replicative cell aging\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001311': {'name': ['obsolete formation of extrachromosomal circular rDNA by homologous recombination involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Excision from the chromosome and circularization of DNA molecules encoding ribosomal RNA in dividing cells as they age.\" [GOC:jh, PMID:12044934]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"assembly of extrachromosomal circular rDNA by homologous recombination involved in replicative cell aging\" RELATED []',\n",
       "   '\"formation of extrachromosomal circular rDNA by homologous recombination during replicative cell ageing\" RELATED []',\n",
       "   '\"formation of extrachromosomal circular rDNA by homologous recombination during replicative cell aging\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"formation of extrachromosomal circular ribosomal DNA by homologous recombination during replicative cell aging\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001312': {'name': ['obsolete replication of extrachromosomal rDNA circles involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Replication of rDNA following its excision from the chromosome of dividing cells as they age. Extrachromosomal rDNA forms a circle that contains at least one autonomously replicating sequence (ARS), which supports replication independent of chromosomal replication.\" [GOC:jh, PMID:12044934]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"replication of extrachromosomal rDNA circles during replicative cell ageing\" RELATED []',\n",
       "   '\"replication of extrachromosomal rDNA circles during replicative cell aging\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"replication of extrachromosomal ribosomal DNA circles during replicative cell aging\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001313': {'name': ['obsolete formation of extrachromosomal circular DNA involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Excision from the chromosome and circularization of a region of chromosomal DNA, generally, but not always, via homologous recombination between direct tandem repeats, in dividing cells as they age.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"assembly of extrachromosomal circular DNA involved in replicative cell aging\" EXACT []',\n",
       "   '\"formation of extrachromosomal circular DNA during replicative cell ageing\" RELATED []',\n",
       "   '\"formation of extrachromosomal circular DNA during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001314': {'name': ['obsolete replication of extrachromosomal circular DNA involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Replication of circular DNA following excision from the chromosome of dividing cells as they age; replication of extrachromosomal circular DNA generally occurs independently of chromosomal replication.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"replication of extrachromosomal circular DNA during replicative cell ageing\" EXACT []',\n",
       "   '\"replication of extrachromosomal circular DNA during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001315': {'name': ['obsolete age-dependent response to reactive oxygen species'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of reactive oxygen species, where the change varies according to the age of the cell or organism.\" [GOC:jh, PMID:12044938]'],\n",
       "  'comment': ['This term was obsoleted because it represents a phenotype.'],\n",
       "  'synonym': ['\"age-dependent response to active oxygen species\" EXACT []',\n",
       "   '\"age-dependent response to AOS\" EXACT []',\n",
       "   '\"age-dependent response to reactive oxidative species\" EXACT []',\n",
       "   '\"age-dependent response to reactive oxygen intermediate\" EXACT []',\n",
       "   '\"age-dependent response to ROI\" EXACT []',\n",
       "   '\"age-dependent response to ROS\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0034599']},\n",
       " 'GO:0001316': {'name': ['obsolete age-dependent response to reactive oxygen species involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) occurring during the process of replicative cell aging as a result of reactive oxygen species, where the change varies according to the age of the cell or organism.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"age-dependent response to active oxygen species during replicative cell aging\" RELATED []',\n",
       "   '\"age-dependent response to AOS during replicative cell aging\" EXACT []',\n",
       "   '\"age-dependent response to reactive oxidative species during replicative cell aging\" EXACT []',\n",
       "   '\"age-dependent response to reactive oxygen intermediate during replicative cell aging\" RELATED []',\n",
       "   '\"age-dependent response to reactive oxygen species during replicative cell ageing\" EXACT []',\n",
       "   '\"age-dependent response to reactive oxygen species during replicative cell aging\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"age-dependent response to ROI during replicative cell aging\" RELATED []',\n",
       "   '\"age-dependent response to ROS during replicative cell aging\" RELATED []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001317': {'name': ['obsolete accumulation of oxidatively modified proteins involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Accumulation of proteins that have undergone reactions with reactive oxygen species in aging dividing cells and exhibit modifications such as increased protein carbonyl content, oxidized methionine, protein hydrophobicity, and cross-linking.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"accumulation of oxidatively modified proteins during replicative cell ageing\" RELATED []',\n",
       "   '\"accumulation of oxidatively modified proteins during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001318': {'name': ['obsolete formation of oxidatively modified proteins involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Chemical reaction, between proteins and reactive oxygen species, that occurs in dividing cells as they age and leads to a variety of changes in the affected proteins, including increases in protein carbonyl content, oxidized methionine, protein hydrophobicity, and cross-linking.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"formation of oxidatively modified proteins during replicative cell ageing\" RELATED []',\n",
       "   '\"formation of oxidatively modified proteins during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001319': {'name': ['obsolete inheritance of oxidatively modified proteins involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A protein localization process in which progeny cells acquire, or are barred from acquiring, proteins that have been altered by reaction with reactive oxygen species in dividing aging cells.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"inheritance of oxidatively modified proteins during replicative cell ageing\" RELATED []',\n",
       "   '\"inheritance of oxidatively modified proteins during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001320': {'name': ['obsolete age-dependent response to reactive oxygen species involved in chronological cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) occurring in non-dividing cells as they age as a result of reactive oxygen species, where the change varies according to the age of the cell or organism.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represent an assay - how long the cell lives -, not a true biological process.'],\n",
       "  'synonym': ['\"age-dependent response to active oxygen species during chronological cell aging\" RELATED []',\n",
       "   '\"age-dependent response to AOS during chronological cell aging\" RELATED []',\n",
       "   '\"age-dependent response to reactive oxidative species during chronological cell aging\" RELATED []',\n",
       "   '\"age-dependent response to reactive oxygen intermediate during chronological cell aging\" RELATED []',\n",
       "   '\"age-dependent response to reactive oxygen species during chronological cell ageing\" RELATED []',\n",
       "   '\"age-dependent response to reactive oxygen species during chronological cell aging\" RELATED [GOC:dph, GOC:tb]',\n",
       "   '\"age-dependent response to ROI during chronological cell aging\" RELATED []',\n",
       "   '\"age-dependent response to ROS during chronological cell aging\" RELATED []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044838', 'GO:0090398']},\n",
       " 'GO:0001321': {'name': ['obsolete age-dependent general metabolic decline involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A process of general metabolic decline that arises in dividing cells as they age, and alters cellular metabolism to cause a decline in cell function.\" [GOC:jh, GOC:mah]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"age-dependent general metabolic decline during replicative cell ageing\" RELATED []',\n",
       "   '\"age-dependent general metabolic decline involved in replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001322': {'name': ['obsolete age-dependent response to oxidative stress involved in replicative cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) occurring in dividing cells as they age as a result of oxidative stress, where the change varies according to the age of the cell or organism.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represents an assay, not a true biological process.'],\n",
       "  'synonym': ['\"age-dependent response to oxidative stress during replicative cell ageing\" RELATED []',\n",
       "   '\"age-dependent response to oxidative stress during replicative cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001323': {'name': ['obsolete age-dependent general metabolic decline involved in chronological cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. A process of general metabolic decline that arises in non-dividing cells as they age, and alters cellular metabolism to cause a decline in cell function.\" [GOC:jh, GOC:mah]'],\n",
       "  'comment': ['This term was obsoleted because it represent an assay - how long the cell lives -, not a true biological process.'],\n",
       "  'synonym': ['\"age-dependent general metabolic decline during chronological cell ageing\" RELATED []',\n",
       "   '\"age-dependent general metabolic decline during chronological cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044838', 'GO:0090398']},\n",
       " 'GO:0001324': {'name': ['obsolete age-dependent response to oxidative stress involved in chronological cell aging'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) occurring in non-dividing cells as they age as a result of oxidative stress, where the change varies according to the age of the cell or organism.\" [GOC:jh]'],\n",
       "  'comment': ['This term was obsoleted because it represent an assay - how long the cell lives -, not a true biological process.'],\n",
       "  'synonym': ['\"age-dependent response to oxidative stress during chronological cell ageing\" RELATED []',\n",
       "   '\"age-dependent response to oxidative stress during chronological cell aging\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044838', 'GO:0090398']},\n",
       " 'GO:0001325': {'name': ['formation of extrachromosomal circular DNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Excision from the chromosome and circularization of a region of chromosomal DNA, generally, but not always, via homologous recombination between direct tandem repeats.\" [GOC:jh, PMID:12044938]'],\n",
       "  'synonym': ['\"assembly of extrachromosomal circular DNA\" EXACT []'],\n",
       "  'is_a': ['GO:0006259 ! DNA metabolic process',\n",
       "   'GO:0022607 ! cellular component assembly']},\n",
       " 'GO:0001326': {'name': ['replication of extrachromosomal circular DNA'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Replication of circular DNA following excision from the chromosome; replication of extrachromosomal circular DNA generally occurs independently of chromosomal replication.\" [GOC:jh, PMID:33867825]'],\n",
       "  'is_a': ['GO:0006261 ! DNA-templated DNA replication']},\n",
       " 'GO:0001400': {'name': ['mating projection base'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The region where the mating projection meets the bulk of the cell, in unicellular fungi exposed to mating pheromone.\" [GOC:mcc]'],\n",
       "  'synonym': ['\"base of shmoo tip\" NARROW []',\n",
       "   '\"conjugation tube base\" NARROW []'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005937 ! mating projection']},\n",
       " 'GO:0001401': {'name': ['SAM complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A large complex of the mitochondrial outer membrane that mediates sorting of some imported proteins to the outer membrane and their assembly in the membrane; functions after import of incoming proteins by the mitochondrial outer membrane translocase complex.\" [PMID:12891361]'],\n",
       "  'comment': [\"See also the cellular component term 'mitochondrial outer membrane translocase complex ; GO:0005742'.\"],\n",
       "  'synonym': ['\"mitochondrial sorting and assembly machinery complex\" EXACT []',\n",
       "   '\"TOB complex\" EXACT []'],\n",
       "  'is_a': ['GO:0005742 ! mitochondrial outer membrane translocase complex']},\n",
       " 'GO:0001402': {'name': ['signal transduction involved in filamentous growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Relaying of environmental signals promoting filamentous growth.\" [GOC:mcc, PMID:9728395]'],\n",
       "  'synonym': ['\"MAPKKK cascade (pseudohyphal growth)\" RELATED []',\n",
       "   '\"signal transduction during filamentous growth\" RELATED [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0007165 ! signal transduction'],\n",
       "  'relationship': ['part_of GO:0030447 ! filamentous growth']},\n",
       " 'GO:0001403': {'name': ['invasive growth in response to glucose limitation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A growth pattern exhibited by budding haploid cells under certain growth conditions, in which cells retain the typical axial budding pattern of haploids, but become elongated and fail to separate after division; during growth on a solid substrate, this results in penetration of cells into the agar medium. An example of this process is found in Saccharomyces cerevisiae.\" [GOC:mcc, PMID:9728395]'],\n",
       "  'comment': [\"Note that this term should not be used to describe the invasion of host tissues by pathogenic organisms, which is described by the biological process term 'entry into host ; GO:0044409', nor should it be used to describe growth of diseased cells of an organism into the surrounding normal tissue, which is outside of the scope of GO.\"],\n",
       "  'subset': ['goslim_yeast'],\n",
       "  'synonym': ['\"colony morphology\" RELATED []'],\n",
       "  'is_a': ['GO:0036267 ! invasive filamentous growth']},\n",
       " 'GO:0001404': {'name': ['obsolete invasive growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Growth of a pathogenic organism that results in penetration into cells or tissues of the host organism. This often (but not necessarily) includes a filamentous growth form, and also can include secretion of proteases and lipases to break down host tissue.\" [GOC:mcc, PMID:9728395]'],\n",
       "  'comment': ['This term was made obsolete because more appropriate terms were created.'],\n",
       "  'synonym': ['\"invasive growth\" EXACT []', '\"tissue invasion\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0044409']},\n",
       " 'GO:0001405': {'name': ['PAM complex, Tim23 associated import motor'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Protein complex located on the matrix side of the mitochondrial inner membrane and associated with the TIM23 mitochondrial import inner membrane translocase complex (GO:0005744); ATPase motor activity to drive import of proteins into the mitochondrial matrix.\" [GOC:mcc, GOC:vw, PMID:14517234, PMID:14638855]'],\n",
       "  'synonym': ['\"mitochondrial import motor\" EXACT []',\n",
       "   '\"PAM complex\" EXACT []',\n",
       "   '\"pre-sequence translocase-associated import motor\" EXACT []',\n",
       "   '\"presequence translocase-associated import motor\" EXACT []'],\n",
       "  'is_a': ['GO:0098800 ! inner mitochondrial membrane protein complex'],\n",
       "  'relationship': ['part_of GO:0005744 ! TIM23 mitochondrial import inner membrane translocase complex',\n",
       "   'part_of GO:0005759 ! mitochondrial matrix']},\n",
       " 'GO:0001406': {'name': ['glycerophosphodiester transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of glycerophosphodiesters from one side of a membrane to the other. Glycerophosphodiesters are small molecules composed of glycerol-3-phosphate and an alcohol, for example, glycerophosphoinositol.\" [GOC:mcc, PMID:12912892]'],\n",
       "  'is_a': ['GO:0008514 ! organic anion transmembrane transporter activity',\n",
       "   'GO:0015605 ! organophosphate ester transmembrane transporter activity']},\n",
       " 'GO:0001407': {'name': ['glycerophosphodiester transmembrane transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process in which a glycerophosphodiester is transported across a membrane. Glycerophosphodiesters are small molecules composed of glycerol-3-phosphate and an alcohol, for example, glycerophosphoinositol.\" [GOC:mcc, PMID:12912892]'],\n",
       "  'is_a': ['GO:0015711 ! organic anion transport',\n",
       "   'GO:0015748 ! organophosphate ester transport',\n",
       "   'GO:0055085 ! transmembrane transport']},\n",
       " 'GO:0001408': {'name': ['guanine nucleotide transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of guanine nucleotides, GTP, GDP, and/or GMP, into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore.\" [GOC:mcc]'],\n",
       "  'is_a': ['GO:0015865 ! purine nucleotide transport']},\n",
       " 'GO:0001409': {'name': ['guanine nucleotide transmembrane transporter activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Enables the transfer of guanine nucleotides (GMP, GDP, and GTP) from one side of a membrane to the other.\" [GOC:mcc]'],\n",
       "  'is_a': ['GO:0015216 ! purine nucleotide transmembrane transporter activity']},\n",
       " 'GO:0001410': {'name': ['chlamydospore formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0055027'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the chlamydospore over time, from its formation to the mature structure. A chlamydospores is a mitotic (asexual) one-celled spore, produced primarily for survival, not dispersal, originating endogenously and singly within part of a pre-existing cell and possessing an inner secondary and often thickened cell wall. An example of this is found in Candida albicans.\" [GOC:mcc, GOC:mtg_sensu, ISBN:085199377X, PMID:14663094]'],\n",
       "  'synonym': ['\"chlamydospore development\" EXACT []'],\n",
       "  'is_a': ['GO:0043936 ! asexual sporulation resulting in formation of a cellular spore',\n",
       "   'GO:0048468 ! cell development']},\n",
       " 'GO:0001411': {'name': ['hyphal tip'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"The end, or tip, of a fungal hypha, where polarized growth occurs during hyphal elongation.\" [GOC:mcc]'],\n",
       "  'subset': ['goslim_candida'],\n",
       "  'is_a': ['GO:0030427 ! site of polarized growth', 'GO:0051286 ! cell tip']},\n",
       " 'GO:0001501': {'name': ['skeletal system development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the skeleton over time, from its formation to the mature structure. The skeleton is the bony framework of the body in vertebrates (endoskeleton) or the hard outer envelope of insects (exoskeleton or dermoskeleton).\" [GOC:dph, GOC:jid, GOC:tb]'],\n",
       "  'synonym': ['\"skeletal development\" EXACT []'],\n",
       "  'is_a': ['GO:0048731 ! system development']},\n",
       " 'GO:0001502': {'name': ['cartilage condensation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The condensation of mesenchymal cells that have been committed to differentiate into chondrocytes.\" [ISBN:0878932437]'],\n",
       "  'is_a': ['GO:0098743 ! cell aggregation'],\n",
       "  'relationship': ['part_of GO:0048705 ! skeletal system morphogenesis',\n",
       "   'part_of GO:0051216 ! cartilage development']},\n",
       " 'GO:0001503': {'name': ['ossification'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of bone or of a bony substance, or the conversion of fibrous tissue or of cartilage into bone or a bony substance.\" [GOC:mtg_mpo, PMID:17572649]'],\n",
       "  'comment': [\"Note that this term does not have a 'developmental process' parent because ossification isn't necessarily developmental, can also occur as part of bone remodeling. Instead use 'ossification involved in bone maturation ; GO:0043931'.\"],\n",
       "  'synonym': ['\"bone biosynthesis\" EXACT []',\n",
       "   '\"bone formation\" EXACT []',\n",
       "   '\"osteogenesis\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Ossification'],\n",
       "  'is_a': ['GO:0032501 ! multicellular organismal process']},\n",
       " 'GO:0001504': {'name': ['neurotransmitter uptake'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of neurotransmitters into neurons or glial cells. This process leads to inactivation and recycling of neurotransmitters.\" [ISBN:0123668387]'],\n",
       "  'subset': ['goslim_synapse'],\n",
       "  'synonym': ['\"neurotransmitter import\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"neurotransmitter import into glial cell\" NARROW []',\n",
       "   '\"neurotransmitter import into neuron\" NARROW []',\n",
       "   '\"neurotransmitter recycling\" BROAD []'],\n",
       "  'is_a': ['GO:0001505 ! regulation of neurotransmitter levels',\n",
       "   'GO:0006836 ! neurotransmitter transport',\n",
       "   'GO:0098657 ! import into cell']},\n",
       " 'GO:0001505': {'name': ['regulation of neurotransmitter levels'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates levels of neurotransmitter.\" [GOC:jl]'],\n",
       "  'is_a': ['GO:0065008 ! regulation of biological quality']},\n",
       " 'GO:0001506': {'name': ['obsolete neurotransmitter biosynthetic process and storage'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The chemical reactions and pathways resulting in the formation of neurotransmitters and the storage of the synthesized molecules.\" [GOC:go_curators, ISBN:0123668387]'],\n",
       "  'comment': ['This term was made obsolete because it is an amalgamation of its two children.'],\n",
       "  'synonym': ['\"neurotransmitter anabolism and storage\" EXACT []',\n",
       "   '\"neurotransmitter biosynthetic process and storage\" EXACT []',\n",
       "   '\"neurotransmitter formation and storage\" EXACT []',\n",
       "   '\"neurotransmitter synthesis and storage\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0042136', 'GO:0042137']},\n",
       " 'GO:0001507': {'name': ['acetylcholine catabolic process in synaptic cleft'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of acetylcholine that occurs in the synaptic cleft during synaptic transmission.\" [GOC:ai]'],\n",
       "  'synonym': ['\"acetylcholine breakdown in synaptic cleft\" EXACT []',\n",
       "   '\"acetylcholine degradation in synaptic cleft\" EXACT []'],\n",
       "  'is_a': ['GO:0006581 ! acetylcholine catabolic process'],\n",
       "  'relationship': ['part_of GO:0007271 ! synaptic transmission, cholinergic']},\n",
       " 'GO:0001508': {'name': ['action potential'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A process in which membrane potential cycles through a depolarizing spike, triggered in response to depolarization above some threshold, followed by repolarization. This cycle is driven by the flow of ions through various voltage gated channels with different thresholds and ion specificities.\" [GOC:dph, GOC:go_curators, GOC:tb, ISBN:978-0-07-139011-8]'],\n",
       "  'comment': ['Action potentials typically propagate across excitable membranes. This class covers both action potentials that propagate and those that fail to do so.'],\n",
       "  'is_a': ['GO:0042391 ! regulation of membrane potential']},\n",
       " 'GO:0001509': {'name': ['obsolete legumain activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Catalysis of the hydrolysis of proteins and small-molecule substrates at Asn-Xaa bonds.\" [EC:3.4.22.34]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"asparaginyl endopeptidase activity\" RELATED [EC:3.4.22.34]',\n",
       "   '\"bean endopeptidase activity\" RELATED [EC:3.4.22.34]',\n",
       "   '\"citvac\" RELATED [EC:3.4.22.34]',\n",
       "   '\"hemoglobinase activity\" RELATED [EC:3.4.22.34]',\n",
       "   '\"legumain activity\" EXACT []',\n",
       "   '\"phaseolin activity\" RELATED [EC:3.4.22.34]',\n",
       "   '\"proteinase B\" RELATED [EC:3.4.22.34]',\n",
       "   '\"PRSC1 gene product (Homo sapiens)\" RELATED [EC:3.4.22.34]',\n",
       "   '\"vicilin peptidohydrolase activity\" RELATED [EC:3.4.22.34]'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0004197']},\n",
       " 'GO:0001510': {'name': ['RNA methylation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Posttranscriptional addition of a methyl group to either a nucleotide or 2\\'-O ribose in a polyribonucleotide. Usually uses S-adenosylmethionine as a cofactor.\" [GOC:hjd, PMID:21823225]'],\n",
       "  'is_a': ['GO:0009451 ! RNA modification',\n",
       "   'GO:0043414 ! macromolecule methylation']},\n",
       " 'GO:0001511': {'name': ['obsolete fibrillin'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Large glycoprotein that is a calcium binding component of connective tissue microfibrils containing 34 six-cysteine (EGF-like) repeats and five eight-cysteine (TGFbeta-1 binding protein-like) repeats. Defects associated with Marfan syndrome.\" [ISBN:0198506732]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"fibrillin\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0001527', 'GO:0005509']},\n",
       " 'GO:0001512': {'name': ['dihydronicotinamide riboside quinone reductase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'comment': ['Formerly EC:1.10.99.2.'],\n",
       "  'synonym': ['\"N-ribosyldihydronicotinamide dehydrogenase (quinone) activity\" RELATED [EC:1.10.5.1]',\n",
       "   '\"NQO2\" RELATED [EC:1.10.5.1]',\n",
       "   '\"NQO2 activity\" RELATED [EC:1.10.5.1]',\n",
       "   '\"NRH:quinone oxidoreductase 2 activity\" NARROW [EC:1.10.5.1]',\n",
       "   '\"QR2 activity\" RELATED [EC:1.10.5.1]',\n",
       "   '\"quinone reductase 2 activity\" RELATED [EC:1.10.5.1]',\n",
       "   '\"ribosyldihydronicotinamide dehydrogenase (quinone) activity\" EXACT []'],\n",
       "  'xref': ['EC:1.10.5.1',\n",
       "   'MetaCyc:1.10.99.2-RXN',\n",
       "   'Reactome:R-HSA-8936519 \"NQO2:FAD dimer reduces quinones to hydroquinones\"',\n",
       "   'RHEA:12364'],\n",
       "  'is_a': ['GO:0016679 ! oxidoreductase activity, acting on diphenols and related substances as donors']},\n",
       " 'GO:0001514': {'name': ['selenocysteine incorporation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The incorporation of selenocysteine into a peptide; uses a special tRNA that recognizes the UGA codon as selenocysteine, rather than as a termination codon. Selenocysteine is synthesized from serine before its incorporation; it is not a posttranslational modification of peptidyl-cysteine.\" [RESID:AA0022]'],\n",
       "  'xref': ['RESID:AA0022'],\n",
       "  'is_a': ['GO:0006451 ! translational readthrough']},\n",
       " 'GO:0001515': {'name': ['opioid peptide activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Naturally occurring peptide that is an opioid (any non-alkaloid having an opiate-like effect that can be reversed by naloxone or other recognized morphine antagonist). These include Leu- and Met-enkephalin, dynorphin and neoendorphin, alpha, beta, gamma and delta endorphins formed from beta-lipotropin, various pronase-resistant peptides such as beta casamorphin, and other peptides whose opiate-like action seems to be indirect.\" [ISBN:0198506732]'],\n",
       "  'subset': ['goslim_chembl'],\n",
       "  'is_a': ['GO:0048018 ! receptor ligand activity']},\n",
       " 'GO:0001516': {'name': ['prostaglandin biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of prostaglandins, any of a group of biologically active metabolites which contain a cyclopentane ring.\" [GOC:ai]'],\n",
       "  'synonym': ['\"prostaglandin anabolism\" EXACT []',\n",
       "   '\"prostaglandin biosynthesis\" EXACT []',\n",
       "   '\"prostaglandin formation\" EXACT []',\n",
       "   '\"prostaglandin synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0006693 ! prostaglandin metabolic process',\n",
       "   'GO:0046457 ! prostanoid biosynthetic process']},\n",
       " 'GO:0001517': {'name': ['N-acetylglucosamine 6-O-sulfotransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'subset': ['goslim_chembl'],\n",
       "  'synonym': ['\"N-acetylglucosamine 6-O-sulphotransferase activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-2046222 \"CHST2,5,6 transfer SO4(2-) to GlcNAc residues on keratan-PG to form KSPG\"',\n",
       "   'Reactome:R-HSA-3656269 \"Defective CHST6 does not transfer SO4(2-) to GlcNAc residues on keratan-PG\"',\n",
       "   'Reactome:R-HSA-6786012 \"CHST4 transfers SO4(2-) from PAPS to Core 2 mucins\"'],\n",
       "  'is_a': ['GO:0008146 ! sulfotransferase activity']},\n",
       " 'GO:0001518': {'name': ['voltage-gated sodium channel complex'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A sodium channel in a cell membrane whose opening is governed by the membrane potential.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"voltage gated sodium channel complex\" EXACT []',\n",
       "   '\"voltage-dependent sodium channel complex\" EXACT []',\n",
       "   '\"voltage-sensitive sodium channel complex\" EXACT []'],\n",
       "  'xref': ['NIF_Subcellular:sao785001660'],\n",
       "  'is_a': ['GO:0034706 ! sodium channel complex',\n",
       "   'GO:0098797 ! plasma membrane protein complex']},\n",
       " 'GO:0001519': {'name': ['peptide amidation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The posttranslational conversion of C-terminal glycine-extended peptides to C-terminal alpha-amidated peptides. Occurs to over half of all peptide hormones to give bioactive peptides. This is a two step process catalyzed by a peptidyl-glycine alpha-hydroxylating monooxygenase and a peptidyl-alpha-hydroxyglycine alpha-amidating lyase. In some organisms, this process is catalyzed by two separate enzymes, whereas in higher organisms, one polypeptide catalyzes both reactions.\" [PMID:11028916]'],\n",
       "  'is_a': ['GO:0031179 ! peptide modification']},\n",
       " 'GO:0001520': {'name': ['outer dense fiber'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A supramolecular fiber found in the flagella of mammalian sperm that surrounds the nine microtubule doublets. These dense fibers are stiff and noncontractile. In human, they consist of about 10 major and at least 15 minor proteins, where all major proteins are ODF1, ODF2 or ODF2-related proteins.\" [GOC:cilia, GOC:krc, ISBN:0824072820, PMID:10381817, PMID:21586547, PMID:25361759]'],\n",
       "  'synonym': ['\"outer dense fibre\" EXACT []'],\n",
       "  'is_a': ['GO:0099513 ! polymeric cytoskeletal fiber'],\n",
       "  'relationship': ['part_of GO:0036126 ! sperm flagellum']},\n",
       " 'GO:0001522': {'name': ['pseudouridine synthesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The intramolecular conversion of uridine to pseudouridine within an RNA molecule.\" [GOC:hjd, GOC:mah]'],\n",
       "  'synonym': ['\"pseudouridylation\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Pseudouridine', 'Wikipedia:Pseudouridylation'],\n",
       "  'is_a': ['GO:0009451 ! RNA modification']},\n",
       " 'GO:0001523': {'name': ['retinoid metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving retinoids, any member of a class of isoprenoids that contain or are derived from four prenyl groups linked head-to-tail. Retinoids include retinol and retinal and structurally similar natural derivatives or synthetic compounds, but need not have vitamin A activity.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"retinoid metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0016101 ! diterpenoid metabolic process']},\n",
       " 'GO:0001524': {'name': ['obsolete globin'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. The colorless and basic protein moiety of hemoglobin and myoglobins.\" [GOC:ai, ISBN:0198506732]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"globin\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005344']},\n",
       " 'GO:0001525': {'name': ['angiogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Blood vessel formation when new vessels emerge from the proliferation of pre-existing blood vessels.\" [ISBN:0878932453]'],\n",
       "  'synonym': ['\"blood vessel formation from pre-existing blood vessels\" EXACT systematic_synonym []'],\n",
       "  'xref': ['Wikipedia:Angiogenesis'],\n",
       "  'is_a': ['GO:0048646 ! anatomical structure formation involved in morphogenesis'],\n",
       "  'relationship': ['part_of GO:0048514 ! blood vessel morphogenesis']},\n",
       " 'GO:0001526': {'name': ['obsolete proteoglycan sulfate transfer'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Transfer of sulfate to a proteoglycan (a glycoprotein whose carbohydrate units are glycosaminoglycans) using 3\\'-phosphoadenyl sulfate.\" [GOC:hjd]'],\n",
       "  'comment': ['This term was made obsolete because it represents a group of molecular functions.'],\n",
       "  'synonym': ['\"proteoglycan sulfate transfer\" EXACT []',\n",
       "   '\"proteoglycan sulphate transfer\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0006029', 'GO:0006790', 'GO:0050698']},\n",
       " 'GO:0001527': {'name': ['microfibril'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Extracellular matrix components occurring independently or along with elastin. Thought to have force-bearing functions in tendon. In addition to fibrillins, microfibrils may contain other associated proteins.\" [PMID:27026396]'],\n",
       "  'synonym': ['\"extended fibrils\" EXACT []', '\"fibrillin\" RELATED []'],\n",
       "  'is_a': ['GO:0099512 ! supramolecular fiber'],\n",
       "  'relationship': ['part_of GO:0031012 ! extracellular matrix']},\n",
       " 'GO:0001528': {'name': ['obsolete elastin'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. A major structural protein of mammalian connective tissues; composed of one third glycine, and also rich in proline, alanine, and valine. Chains are cross-linked together via lysine residues.\" [ISBN:0198506732]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"elastin\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0030023']},\n",
       " 'GO:0001529': {'name': ['obsolete elastin'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product.'],\n",
       "  'synonym': ['\"elastin\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0030023']},\n",
       " 'GO:0001530': {'name': ['lipopolysaccharide binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a lipopolysaccharide.\" [PMID:11079463]'],\n",
       "  'synonym': ['\"endotoxin binding\" BROAD []', '\"LPS binding\" EXACT []'],\n",
       "  'is_a': ['GO:0008289 ! lipid binding',\n",
       "   'GO:0097367 ! carbohydrate derivative binding']},\n",
       " 'GO:0001531': {'name': ['interleukin-21 receptor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an interleukin-21 receptor.\" [GOC:ai]'],\n",
       "  'synonym': ['\"IL-21\" NARROW []',\n",
       "   '\"interleukin-21 receptor ligand\" NARROW []'],\n",
       "  'is_a': ['GO:0005126 ! cytokine receptor binding']},\n",
       " 'GO:0001532': {'name': ['interleukin-21 receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with interleukin-21 and transmitting the signal from one side of the membrane to the other to initiate a change in cell activity.\" [GOC:jl, GOC:signaling]'],\n",
       "  'synonym': ['\"IL-21 receptor activity\" EXACT [GOC:mah]',\n",
       "   '\"IL-21R\" EXACT []'],\n",
       "  'is_a': ['GO:0004896 ! cytokine receptor activity']},\n",
       " 'GO:0001533': {'name': ['cornified envelope'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A type of plasma membrane that has been modified through addition of distinct intracellular and extracellular components, including ceramide, found in cornifying epithelial cells (corneocytes).\" [GOC:add, PMID:11112355, PMID:11590230, PMID:15803139]'],\n",
       "  'is_a': ['GO:0005886 ! plasma membrane']},\n",
       " 'GO:0001534': {'name': ['radial spoke'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Protein complex that links the outer microtubule doublet of a 9+2 type ciliary or flagellar axoneme with the sheath that surrounds the central pair of microtubules. Composed of a stalk that attaches to each doublet microtubule and a globular structure (spoke head) that projects toward the central pair of microtubules.\" [GOC:krc, ISBN:0124325653, PMID:22118931, PMID:25694453, PMID:34871179, PMID:9450971]'],\n",
       "  'comment': ['Radial spokes are involved in some types of beating motions of the cilium. The radial spoke is usually a T-shaped structure comprised of a short base that attaches to the A-microtubule of an axonemal outer microtubule doublet (MTD) of a cilium, an elongated stalk, a neck complex, and an orthogonal head structure that extends perpendicularly towards the inner sheath and the central pair (CP) microtubules. Groups of radial spokes (RSs) repeat along the MTD with regular spacing. In most organisms, each group of radial spokes is comprised of a triplet of spokes'],\n",
       "  'subset': ['goslim_pir'],\n",
       "  'xref': ['Wikipedia:Radial_spoke'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0005930 ! axoneme']},\n",
       " 'GO:0001535': {'name': ['radial spoke head'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Protein complex forming portion of the radial spoke that is orthogonal to the elongated stalk and which projects towards the central pair of microtubules within the ciliary or flagellum axoneme.\" [GOC:cilia, GOC:hjd, GOC:krc, PMID:22754630, PMID:34871179]'],\n",
       "  'synonym': ['\"radial spokehead\" EXACT []'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0001534 ! radial spoke']},\n",
       " 'GO:0001536': {'name': ['radial spoke stalk'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"Protein complex forming the elongated portion of the radial spoke between the base which binds to the A-tubule of each microtubule outer doublet and the neck which connects to the spoke head within the ciliary or flagellum axoneme.\" [GOC:hjd, GOC:krc, PMID:22754630, PMID:34871179]'],\n",
       "  'is_a': ['GO:0032991 ! protein-containing complex'],\n",
       "  'relationship': ['part_of GO:0001534 ! radial spoke']},\n",
       " 'GO:0001537': {'name': ['N-acetylgalactosamine 4-O-sulfotransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"N-acetylgalactosamine 4-O-sulphotransferase activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-2022063 \"CHST14 transfers SO4(2-) to GalNAc in dermatan or DS\"',\n",
       "   'Reactome:R-HSA-3636919 \"Defective CHST14 does not transfer SO4(2-) to GalNAc in dermatan or DS\"',\n",
       "   'Reactome:R-HSA-6786034 \"CHST8 transfers SO4(2-) from PAPS to glyco-Lutropin\"'],\n",
       "  'is_a': ['GO:0008146 ! sulfotransferase activity']},\n",
       " 'GO:0001539': {'name': ['cilium or flagellum-dependent cell motility'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Cell motility due to movement of eukaryotic cilia or bacterial-type flagella or archaeal-type flagella.\" [GOC:cilia, GOC:hjd, GOC:krc]'],\n",
       "  'comment': [\"Note that we deem eukaryotic cilia and microtubule-based flagella to be equivalent, while the bacterial- and archaeal-type flagella have a different structure. The former are microtubule-based structures that lash back and forth and are present only in eukaryotes, while the latter achieve motility by rotation. Bacterial- and archaeal-type flagella are superficially similar but have a different molecular composition and fine structure. These three structures never co-exist in the same organism. Therefore, GO:0001539 'cilium or flagellum-dependent cell motility' is in the subset of terms that should not be used for direct gene product annotation. Instead, select a child term. Direct annotations to GO:0001539 'cilium or flagellum-dependent cell motility' may be amended during annotation QC.\"],\n",
       "  'subset': ['gocheck_do_not_manually_annotate'],\n",
       "  'synonym': ['\"ciliary or bacterial-type flagellar motility\" RELATED []',\n",
       "   '\"ciliary/flagellar motility\" EXACT []'],\n",
       "  'is_a': ['GO:0048870 ! cell motility']},\n",
       " 'GO:0001540': {'name': ['amyloid-beta binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to an amyloid-beta peptide/protein.\" [GOC:hjd]'],\n",
       "  'subset': ['goslim_chembl'],\n",
       "  'synonym': ['\"beta-amyloid binding\" EXACT []'],\n",
       "  'is_a': ['GO:0042277 ! peptide binding']},\n",
       " 'GO:0001541': {'name': ['ovarian follicle development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the ovarian follicle over time, from its formation to the mature structure.\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'synonym': ['\"follicular phase\" RELATED []'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction',\n",
       "   'GO:0048856 ! anatomical structure development'],\n",
       "  'relationship': ['part_of GO:0008585 ! female gonad development']},\n",
       " 'GO:0001542': {'name': ['ovulation from ovarian follicle'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process leading to the rupture of the follicle, releasing the centrally located oocyte into the oviduct. An example of this is found in Mus musculus.\" [GOC:mtg_sensu, https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process', 'GO:0030728 ! ovulation'],\n",
       "  'relationship': ['part_of GO:0008585 ! female gonad development']},\n",
       " 'GO:0001543': {'name': ['ovarian follicle rupture'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Disruption of theca cell layer releasing follicular fluid and/or the oocyte.\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process'],\n",
       "  'relationship': ['part_of GO:0001542 ! ovulation from ovarian follicle']},\n",
       " 'GO:0001544': {'name': ['initiation of primordial ovarian follicle growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Increase in size of primordial follicles including proliferation and shape changes of granulosa and/or theca cells until oocyte is surrounded by one layer of cuboidal shaped granulosa cells (primary follicle).\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process'],\n",
       "  'relationship': ['part_of GO:0001541 ! ovarian follicle development']},\n",
       " 'GO:0001545': {'name': ['primary ovarian follicle growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Increase in size of primary follicles including oocyte growth and granulosa and/or theca cell proliferation until more than one layer of granulosa cells is present (preantral follicle).\" [GOC:mtg_mpo, https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process',\n",
       "   'GO:0048589 ! developmental growth'],\n",
       "  'relationship': ['part_of GO:0001541 ! ovarian follicle development']},\n",
       " 'GO:0001546': {'name': ['preantral ovarian follicle growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Increase in size of follicles surrounded by two or more layers of granulosa cells up to the onset of antrum formation.\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process',\n",
       "   'GO:0048589 ! developmental growth'],\n",
       "  'relationship': ['part_of GO:0001541 ! ovarian follicle development',\n",
       "   'part_of GO:0048162 ! multi-layer follicle stage']},\n",
       " 'GO:0001547': {'name': ['antral ovarian follicle growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Increase in size of antral follicles due to cell proliferation and/or growth of the antral cavity.\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction',\n",
       "   'GO:0022602 ! ovulation cycle process',\n",
       "   'GO:0048589 ! developmental growth'],\n",
       "  'relationship': ['part_of GO:0001541 ! ovarian follicle development']},\n",
       " 'GO:0001548': {'name': ['follicular fluid formation in ovarian follicle antrum'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The menstrual cycle process that results in the formation of one central cavity separating the oocyte/cumulus complex from mural granulosa and theca cells during the various stages of oogenesis.\" [GOC:dph, GOC:tb, https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'synonym': ['\"ovarian follicle antrum/follicular fluid biosynthesis\" EXACT []',\n",
       "   '\"ovarian follicle antrum/follicular fluid formation\" EXACT []'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process'],\n",
       "  'relationship': ['part_of GO:0001547 ! antral ovarian follicle growth']},\n",
       " 'GO:0001549': {'name': ['cumulus cell differentiation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process in which a subpopulation of granulosa cells surrounding the oocyte acquires the specialized features of an ovarian cumulus cell.\" [PMID:30010832]'],\n",
       "  'synonym': ['\"ovarian cumulus cell differentiation\" EXACT []'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction',\n",
       "   'GO:0022412 ! cellular process involved in reproduction in multicellular organism',\n",
       "   'GO:0022602 ! ovulation cycle process',\n",
       "   'GO:0030855 ! epithelial cell differentiation'],\n",
       "  'relationship': ['part_of GO:0001547 ! antral ovarian follicle growth',\n",
       "   'part_of GO:0048165 ! fused antrum stage']},\n",
       " 'GO:0001550': {'name': ['ovarian cumulus expansion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Increase in size of the cumulus surrounding the oocyte including change in morphology due to proliferation and dispersion of cumulus cells.\" [PMID:30010832]'],\n",
       "  'synonym': ['\"ovarian cumulus growth\" RELATED []'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process',\n",
       "   'GO:0048589 ! developmental growth'],\n",
       "  'relationship': ['part_of GO:0001547 ! antral ovarian follicle growth',\n",
       "   'part_of GO:0048165 ! fused antrum stage']},\n",
       " 'GO:0001551': {'name': ['ovarian follicle endowment'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Association of oocytes with supporting epithelial granulosa cells to form primordial follicles.\" [PMID:30010832]'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process'],\n",
       "  'relationship': ['part_of GO:0001541 ! ovarian follicle development']},\n",
       " 'GO:0001552': {'name': ['ovarian follicle atresia'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A periodic process in which immature ovarian follicles degenerate and are subsequently re-absorbed.\" [GOC:mtg_apoptosis, PMID:18638134]'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction'],\n",
       "  'relationship': ['part_of GO:0008585 ! female gonad development']},\n",
       " 'GO:0001553': {'name': ['luteinization'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The set of processes resulting in differentiation of theca and granulosa cells into luteal cells and in the formation of a corpus luteum after ovulation.\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'synonym': ['\"luteal phase\" RELATED []'],\n",
       "  'xref': ['Wikipedia:Luteal_phase', 'Wikipedia:Luteinization'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process'],\n",
       "  'relationship': ['part_of GO:0008585 ! female gonad development']},\n",
       " 'GO:0001554': {'name': ['luteolysis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The lysis or structural demise of the corpus luteum. During normal luteolysis, two closely related events occur. First, there is loss of the capacity to synthesize and secrete progesterone (functional luteolysis) followed by loss of the cells that comprise the corpus luteum (structural luteolysis). Preventing luteolysis is crucial to maintain pregnancy.\" [PMID:10617764]'],\n",
       "  'xref': ['Wikipedia:Luteolysis'],\n",
       "  'is_a': ['GO:0022602 ! ovulation cycle process'],\n",
       "  'relationship': ['part_of GO:0008585 ! female gonad development']},\n",
       " 'GO:0001555': {'name': ['oocyte growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The developmental growth process in which an oocyte irreversibly increases in size over time by accretion and biosynthetic production of matter similar to that already present.\" [https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction',\n",
       "   'GO:0022412 ! cellular process involved in reproduction in multicellular organism',\n",
       "   'GO:0048588 ! developmental cell growth'],\n",
       "  'relationship': ['part_of GO:0048599 ! oocyte development']},\n",
       " 'GO:0001556': {'name': ['oocyte maturation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A developmental process, independent of morphogenetic (shape) change, that is required for an oocyte to attain its fully functional state. Oocyte maturation commences after reinitiation of meiosis commonly starting with germinal vesicle breakdown, and continues up to the second meiotic arrest prior to fertilization.\" [GOC:devbiol, https://www.ncbi.nlm.nih.gov/books/NBK279054/]'],\n",
       "  'xref': ['Wikipedia:Oocyte_maturation'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction',\n",
       "   'GO:0022412 ! cellular process involved in reproduction in multicellular organism',\n",
       "   'GO:0048469 ! cell maturation'],\n",
       "  'relationship': ['part_of GO:0048599 ! oocyte development']},\n",
       " 'GO:0001557': {'name': ['obsolete metabolic process resulting in cell growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. The chemical reactions that occur in living organisms that result in an increase in the mass (size) of a cell.\" [GOC:dph]'],\n",
       "  'comment': ['This term was made obsolete as part of the metabolism rearrangements, because it is redundant with other terms.'],\n",
       "  'synonym': ['\"metabolic process resulting in cell growth\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0008152', 'GO:0016049']},\n",
       " 'GO:0001558': {'name': ['regulation of cell growth'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that modulates the frequency, rate, extent or direction of cell growth.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0040008 ! regulation of growth',\n",
       "   'GO:0051128 ! regulation of cellular component organization'],\n",
       "  'relationship': ['regulates GO:0016049 ! cell growth']},\n",
       " 'GO:0001559': {'name': ['obsolete regulation of cell growth by detection of nuclear:cytoplasmic ratio'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. Any process in which the size of the nucleus with respect to the cytoplasm modulates the frequency, rate or extent of cell growth, the irreversible increase in size of a cell over time.\" [GOC:dph]'],\n",
       "  'comment': ['This term was obsoleted because there is no evidence that this process exists.'],\n",
       "  'synonym': ['\"detection of nuclear:cytoplasmic ratio to regulate cell growth\" EXACT []',\n",
       "   '\"interpretation of nuclear:cytoplasmic ratio to regulate cell growth\" EXACT []',\n",
       "   '\"regulation of cell growth by nuclear:cytoplasmic ratio\" EXACT []',\n",
       "   '\"regulation of cell growth by sensing of nuclear:cytoplasmic ratio\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001560': {'name': ['regulation of cell growth by extracellular stimulus'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process in which external signals modulate the frequency, rate or extent of cell growth, the irreversible increase in size of a cell over time.\" [GOC:dph]'],\n",
       "  'synonym': ['\"interpretation of external signals that regulate cell growth\" EXACT []',\n",
       "   '\"regulation of cell growth by detection of exogenous stimulus\" EXACT []',\n",
       "   '\"regulation of cell growth by sensing of exogenous stimulus\" EXACT []',\n",
       "   '\"regulation of growth by exogenous signal\" EXACT []',\n",
       "   '\"regulation of growth by exogenous stimuli\" EXACT []',\n",
       "   '\"regulation of growth by exogenous stimulus\" EXACT []',\n",
       "   '\"regulation of growth by external signal\" EXACT []',\n",
       "   '\"regulation of growth by external stimuli\" EXACT []',\n",
       "   '\"regulation of growth by external stimulus\" EXACT []'],\n",
       "  'is_a': ['GO:0001558 ! regulation of cell growth'],\n",
       "  'relationship': ['part_of GO:0031668 ! cellular response to extracellular stimulus']},\n",
       " 'GO:0001561': {'name': ['fatty acid alpha-oxidation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A metabolic pathway by which 3-methyl branched fatty acids are degraded. These compounds are not degraded by the normal peroxisomal beta-oxidation pathway, because the 3-methyl blocks the dehydrogenation of the hydroxyl group by hydroxyacyl-CoA dehydrogenase. The 3-methyl branched fatty acid is converted in several steps to pristenic acid, which can then feed into the beta-oxidative pathway.\" [PMID:10198260]'],\n",
       "  'xref': ['MetaCyc:PWY-2501'],\n",
       "  'is_a': ['GO:0009062 ! fatty acid catabolic process',\n",
       "   'GO:0019395 ! fatty acid oxidation']},\n",
       " 'GO:0001562': {'name': ['response to protozoan'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0042833'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus from a protozoan.\" [GOC:ai]'],\n",
       "  'synonym': ['\"resistance to pathogenic protozoa\" RELATED []',\n",
       "   '\"response to protozoa\" EXACT []',\n",
       "   '\"response to protozoon\" EXACT []'],\n",
       "  'is_a': ['GO:0051707 ! response to other organism']},\n",
       " 'GO:0001563': {'name': ['detection of protozoan'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The series of events in which a stimulus from a protozoan is received and converted into a molecular signal.\" [GOC:ai]'],\n",
       "  'synonym': ['\"detection of protozoa\" EXACT []',\n",
       "   '\"detection of protozoon\" EXACT []',\n",
       "   '\"perception of protozoa\" RELATED []'],\n",
       "  'is_a': ['GO:0001562 ! response to protozoan',\n",
       "   'GO:0098543 ! detection of other organism']},\n",
       " 'GO:0001564': {'name': ['obsolete resistance to pathogenic protozoa'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:go_curators]'],\n",
       "  'comment': [\"This term was made obsolete because 'resistance' implies a phenotype rather than a biological process.\"],\n",
       "  'synonym': ['\"resistance to pathogenic protozoa\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0001562']},\n",
       " 'GO:0001565': {'name': ['phorbol ester receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a phorbol ester and transmitting the signal to initiate a change in cell activity.\" [GOC:ai, GOC:signaling, PMID:10506570]'],\n",
       "  'is_a': ['GO:0038023 ! signaling receptor activity']},\n",
       " 'GO:0001566': {'name': ['non-kinase phorbol ester receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a phorbol ester and transmitting the signal by a mechanism independent of kinase activity.\" [PMID:10506570]'],\n",
       "  'is_a': ['GO:0001565 ! phorbol ester receptor activity']},\n",
       " 'GO:0001567': {'name': ['cholesterol 25-hydroxylase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"cholesterol 25-monooxygenase activity\" RELATED [EC:1.14.99.38]',\n",
       "   '\"cholesterol,hydrogen-donor:oxygen oxidoreductase (25-hydroxylating) activity\" RELATED [EC:1.14.99.38]'],\n",
       "  'xref': ['EC:1.14.99.38',\n",
       "   'KEGG_REACTION:R07218',\n",
       "   'MetaCyc:1.14.99.38-RXN',\n",
       "   'RHEA:21104'],\n",
       "  'is_a': ['GO:0008395 ! steroid hydroxylase activity',\n",
       "   'GO:0016705 ! oxidoreductase activity, acting on paired donors, with incorporation or reduction of molecular oxygen']},\n",
       " 'GO:0001568': {'name': ['blood vessel development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of a blood vessel over time, from its formation to the mature structure. The blood vessel is the vasculature carrying blood.\" [GOC:hjd, UBERON:0001981]'],\n",
       "  'is_a': ['GO:0048856 ! anatomical structure development'],\n",
       "  'relationship': ['part_of GO:0001944 ! vasculature development']},\n",
       " 'GO:0001569': {'name': ['branching involved in blood vessel morphogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process of coordinated growth and sprouting of blood vessels giving rise to the organized vascular system.\" [GOC:dph]'],\n",
       "  'synonym': ['\"patterning of blood vessels\" BROAD [GOC:dph]'],\n",
       "  'is_a': ['GO:0048754 ! branching morphogenesis of an epithelial tube'],\n",
       "  'relationship': ['part_of GO:0001525 ! angiogenesis']},\n",
       " 'GO:0001570': {'name': ['vasculogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The differentiation of endothelial cells from progenitor cells during blood vessel development, and the de novo formation of blood vessels and tubes.\" [PMID:8999798]'],\n",
       "  'synonym': ['\"vascular morphogenesis\" EXACT []'],\n",
       "  'xref': ['Wikipedia:Vasculogenesis'],\n",
       "  'is_a': ['GO:0030154 ! cell differentiation'],\n",
       "  'relationship': ['part_of GO:0048514 ! blood vessel morphogenesis']},\n",
       " 'GO:0001571': {'name': ['non-tyrosine kinase fibroblast growth factor receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with fibroblast growth factor (FGF) and transmitting the signal from one side of the membrane to the other by a mechanism independent of tyrosine kinase activity.\" [GOC:signaling, PMID:11418238]'],\n",
       "  'synonym': ['\"non-tyrosine kinase FGF receptor activity\" EXACT []',\n",
       "   '\"non-tyrosine kinase FGFR activity\" EXACT []'],\n",
       "  'is_a': ['GO:0004888 ! transmembrane signaling receptor activity']},\n",
       " 'GO:0001572': {'name': ['lactosylceramide biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of lactosylceramides, Gal-beta-(1->4)-Glc-beta(1->1\\') ceramides, any compound formed by the replacement of the glycosidic C1 hydroxyl group of lactose by a ceramide group. They are the precursors of both gangliosides and globosides.\" [ISBN:0198506732, ISBN:0471586501]'],\n",
       "  'synonym': ['\"lactosylceramide anabolism\" EXACT []',\n",
       "   '\"lactosylceramide biosynthesis\" EXACT []',\n",
       "   '\"lactosylceramide formation\" EXACT []',\n",
       "   '\"lactosylceramide synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0006688 ! glycosphingolipid biosynthetic process',\n",
       "   'GO:0046478 ! lactosylceramide metabolic process',\n",
       "   'GO:0046513 ! ceramide biosynthetic process']},\n",
       " 'GO:0001573': {'name': ['ganglioside metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving ceramide oligosaccharides carrying in addition to other sugar residues, one or more sialic acid residues.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"ganglioside metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006672 ! ceramide metabolic process',\n",
       "   'GO:0006687 ! glycosphingolipid metabolic process']},\n",
       " 'GO:0001574': {'name': ['ganglioside biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of ceramide oligosaccharides carrying in addition to other sugar residues, one or more sialic acid residues.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"ganglioside anabolism\" EXACT []',\n",
       "   '\"ganglioside biosynthesis\" EXACT []',\n",
       "   '\"ganglioside formation\" EXACT []',\n",
       "   '\"ganglioside synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0001573 ! ganglioside metabolic process',\n",
       "   'GO:0006688 ! glycosphingolipid biosynthetic process',\n",
       "   'GO:0046513 ! ceramide biosynthetic process']},\n",
       " 'GO:0001575': {'name': ['globoside metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving globosides, globotetraosylceramides, ceramides containing a core structure of GalNAc-beta-(1->3)-Gal-alpha-(1->4)-Glc(I). Globosides are the major neutral glycosphingolipid in normal kidneys and erythrocytes.\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"globoside metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006687 ! glycosphingolipid metabolic process']},\n",
       " 'GO:0001576': {'name': ['globoside biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of a ceramide with a core structure of GalNAc-beta-(1->3)-Gal-alpha-(1->4)-Glc(I).\" [ISBN:0198506732]'],\n",
       "  'synonym': ['\"globoside anabolism\" EXACT []',\n",
       "   '\"globoside biosynthesis\" EXACT []',\n",
       "   '\"globoside formation\" EXACT []',\n",
       "   '\"globoside synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0001575 ! globoside metabolic process',\n",
       "   'GO:0006688 ! glycosphingolipid biosynthetic process']},\n",
       " 'GO:0001577': {'name': ['obsolete galectin'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. A lectin that exhibits calcium independent binding of beta-galactoside sugars.\" [PMID:9786891]'],\n",
       "  'comment': ['This term was made obsolete because it refers to a class of gene products.'],\n",
       "  'synonym': ['\"galectin\" EXACT []', '\"S-type lectin\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0007157', 'GO:0016936']},\n",
       " 'GO:0001578': {'name': ['microtubule bundle formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A process that results in a parallel arrangement of microtubules.\" [GOC:dph]'],\n",
       "  'synonym': ['\"microtubule bundling\" EXACT []'],\n",
       "  'is_a': ['GO:0000226 ! microtubule cytoskeleton organization']},\n",
       " 'GO:0001579': {'name': ['medium-chain fatty acid transport'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The directed movement of a medium-chain fatty acid into, out of or within a cell, or between cells, by means of some agent such as a transporter or pore. A medium-chain fatty acid is a fatty acid with an aliphatic tail of 6 to 12 carbons.\" [GOC:ai]'],\n",
       "  'is_a': ['GO:0015908 ! fatty acid transport']},\n",
       " 'GO:0001580': {'name': ['detection of chemical stimulus involved in sensory perception of bitter taste'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The series of events required for a bitter taste stimulus to be received and converted to a molecular signal.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"bitter taste detection\" EXACT []',\n",
       "   '\"perception of bitter taste, detection of chemical stimulus\" EXACT []',\n",
       "   '\"perception of bitter taste, sensory transduction of chemical stimulus\" EXACT []',\n",
       "   '\"sensory detection of bitter taste\" EXACT []',\n",
       "   '\"sensory detection of chemical stimulus during perception of bitter taste\" EXACT []',\n",
       "   '\"sensory transduction of bitter taste\" EXACT []',\n",
       "   '\"sensory transduction of chemical stimulus during perception of bitter taste\" EXACT []'],\n",
       "  'is_a': ['GO:0050912 ! detection of chemical stimulus involved in sensory perception of taste'],\n",
       "  'relationship': ['part_of GO:0050913 ! sensory perception of bitter taste']},\n",
       " 'GO:0001581': {'name': ['detection of chemical stimulus involved in sensory perception of sour taste'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The series of events required for a sour taste stimulus to be received and converted to a molecular signal.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"perception of sour taste, detection of chemical stimulus\" EXACT []',\n",
       "   '\"perception of sour taste, sensory transduction of chemical stimulus\" EXACT []',\n",
       "   '\"sensory detection of chemical stimulus during perception of sour taste\" EXACT []',\n",
       "   '\"sensory detection of sour taste\" EXACT []',\n",
       "   '\"sensory transduction of chemical stimulus during perception of sour taste\" EXACT []',\n",
       "   '\"sensory transduction of sour taste\" EXACT []',\n",
       "   '\"sour taste detection\" EXACT []'],\n",
       "  'is_a': ['GO:0050912 ! detection of chemical stimulus involved in sensory perception of taste'],\n",
       "  'relationship': ['part_of GO:0050915 ! sensory perception of sour taste']},\n",
       " 'GO:0001582': {'name': ['detection of chemical stimulus involved in sensory perception of sweet taste'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The series of events required for a sweet taste stimulus to be received and converted to a molecular signal.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"perception of sweet taste, detection of chemical stimulus\" EXACT []',\n",
       "   '\"perception of sweet taste, sensory transduction of chemical stimulus\" EXACT []',\n",
       "   '\"sensory detection of chemical stimulus during perception of sweet taste\" EXACT []',\n",
       "   '\"sensory detection of sweet taste\" EXACT []',\n",
       "   '\"sensory transduction of chemical stimulus during perception of sweet taste\" EXACT []',\n",
       "   '\"sensory transduction of sweet taste\" EXACT []',\n",
       "   '\"sweet taste detection\" EXACT []'],\n",
       "  'is_a': ['GO:0050912 ! detection of chemical stimulus involved in sensory perception of taste'],\n",
       "  'relationship': ['part_of GO:0050916 ! sensory perception of sweet taste']},\n",
       " 'GO:0001583': {'name': ['detection of chemical stimulus involved in sensory perception of salty taste'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The series of events required for a salty taste stimulus to be received and converted to a molecular signal.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"perception of salty taste, detection of chemical stimulus\" EXACT []',\n",
       "   '\"perception of salty taste, sensory transduction of chemical stimulus\" EXACT []',\n",
       "   '\"salty taste detection\" EXACT []',\n",
       "   '\"sensory detection of chemical stimulus during perception of salty taste\" EXACT []',\n",
       "   '\"sensory detection of salty taste\" EXACT []',\n",
       "   '\"sensory transduction of chemical stimulus during perception of salty taste\" EXACT []',\n",
       "   '\"sensory transduction of salty taste\" EXACT []'],\n",
       "  'is_a': ['GO:0050912 ! detection of chemical stimulus involved in sensory perception of taste'],\n",
       "  'relationship': ['part_of GO:0050914 ! sensory perception of salty taste']},\n",
       " 'GO:0001584': {'name': ['obsolete rhodopsin-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001620'],\n",
       "  'def': ['\"OBSOLETE. A G protein-coupled receptor that is structurally/functionally related to the rhodopsin receptor.\" [GOC:dph, GOC:mah, GOC:tb, IUPHAR_GPCR:1505]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product and is named based on protein features.'],\n",
       "  'synonym': ['\"Class A G protein coupled receptor\" EXACT []',\n",
       "   '\"Class A G-protein coupled receptor\" EXACT []',\n",
       "   '\"Class A GPCR\" EXACT []',\n",
       "   '\"class A orphan receptor activity\" EXACT []',\n",
       "   '\"rhodopsin-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001586': {'name': ['Gi/o-coupled serotonin receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with serotonin and transmitting the signal across the membrane by activation of the Gi/o subunit of an associated cytoplasmic heterotrimeric G protein complex. The Gi/o subunit subsequently inhibits adenylate cyclase and results in a decrease in cyclic AMP (cAMP) levels.\" [GOC:mah, PMID:18571247]'],\n",
       "  'synonym': ['\"5-HT1 receptor activity\" EXACT [GOC:bf]',\n",
       "   '\"serotonin receptor activity, coupled via Gi/o\" EXACT [GOC:bf]'],\n",
       "  'is_a': ['GO:0004993 ! G protein-coupled serotonin receptor activity']},\n",
       " 'GO:0001587': {'name': ['Gq/11-coupled serotonin receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with serotonin and transmitting the signal across the membrane by activation of the Gq/11 subunit of an associated cytoplasmic heterotrimeric G protein complex. The Gq/11 subunit subsequently activates phospholipase C and results in an increase in inositol triphosphate (IP3) levels.\" [GOC:bf, GOC:mah, PMID:18571247, PMID:18703043]'],\n",
       "  'synonym': ['\"5-HT2 receptor activity\" BROAD [GOC:bf]',\n",
       "   '\"serotonin receptor activity, coupled via Gq/11\" EXACT [GOC:bf]'],\n",
       "  'is_a': ['GO:0004993 ! G protein-coupled serotonin receptor activity']},\n",
       " 'GO:0001588': {'name': ['dopamine neurotransmitter receptor activity, coupled via Gs'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001589', 'GO:0001590'],\n",
       "  'def': ['\"Combining with the neurotransmitter dopamine and activating adenylate cyclase via coupling to Gs to initiate a change in cell activity.\" [GOC:mah, ISBN:0953351033, IUPHAR_RECEPTOR:2252, IUPHAR_RECEPTOR:2260]'],\n",
       "  'synonym': ['\"dopamine D1 receptor activity\" NARROW [GOC:bf]',\n",
       "   '\"dopamine D5 receptor activity\" NARROW [GOC:bf]'],\n",
       "  'is_a': ['GO:0004952 ! dopamine neurotransmitter receptor activity']},\n",
       " 'GO:0001591': {'name': ['dopamine neurotransmitter receptor activity, coupled via Gi/Go'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001592', 'GO:0001593', 'GO:0001670'],\n",
       "  'def': ['\"Combining with the neurotransmitter dopamine and activating adenylate cyclase via coupling to Gi/Go to initiate a change in cell activity.\" [GOC:mah, ISBN:0953351033, IUPHAR_RECEPTOR:2254, IUPHAR_RECEPTOR:2256, IUPHAR_RECEPTOR:2258]'],\n",
       "  'synonym': ['\"dopamine D2 receptor activity\" NARROW [GOC:bf]',\n",
       "   '\"dopamine D3 receptor activity\" NARROW [GOC:bf]',\n",
       "   '\"dopamine D4 receptor activity\" NARROW [GOC:bf]'],\n",
       "  'is_a': ['GO:0004952 ! dopamine neurotransmitter receptor activity']},\n",
       " 'GO:0001594': {'name': ['trace-amine receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a trace amine to initiate a change in cell activity. Trace amines are biogenic amines that are synthesized from aromatic amino acids and are substrates for monoamine oxidase, and are therefore detectable only at trace levels in mammals.\" [GOC:mah, PMID:19325074]'],\n",
       "  'is_a': ['GO:0008227 ! G protein-coupled amine receptor activity']},\n",
       " 'GO:0001595': {'name': ['angiotensin receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with angiotensin to initiate a change in cell activity.\" [GOC:ai]'],\n",
       "  'is_a': ['GO:0008528 ! G protein-coupled peptide receptor activity']},\n",
       " 'GO:0001596': {'name': ['angiotensin type I receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"An angiotensin receptor activity that acts via Gq-mediated activation of phospholipase C followed by phosphoinositide hydrolysis and Ca2+ signaling, and may act via additional signaling mechanisms.\" [GOC:mah, PMID:10977869]'],\n",
       "  'synonym': ['\"PLC-activating angiotensin receptor activity\" EXACT [GOC:bf]'],\n",
       "  'is_a': ['GO:0001595 ! angiotensin receptor activity']},\n",
       " 'GO:0001597': {'name': ['obsolete apelin-like receptor'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because the function it represents does not exist.'],\n",
       "  'synonym': ['\"apelin-like receptor\" EXACT []',\n",
       "   '\"APJ-like receptor\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0031704']},\n",
       " 'GO:0001598': {'name': ['obsolete chemokine receptor-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"chemokine receptor-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001601': {'name': ['peptide YY receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with gut peptide YY to initiate a change in cell activity.\" [PMID:9315606]'],\n",
       "  'is_a': ['GO:0004983 ! neuropeptide Y receptor activity']},\n",
       " 'GO:0001602': {'name': ['pancreatic polypeptide receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with pancreatic polypeptide PP to initiate a change in cell activity.\" [PMID:9315606]'],\n",
       "  'is_a': ['GO:0004983 ! neuropeptide Y receptor activity']},\n",
       " 'GO:0001603': {'name': ['obsolete vasopressin-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:curators]'],\n",
       "  'synonym': ['\"vasopressin-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001604': {'name': ['urotensin II receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with urotensin II to initiate a change in cell activity.\" [GOC:mah, PMID:15102493]'],\n",
       "  'is_a': ['GO:0008528 ! G protein-coupled peptide receptor activity']},\n",
       " 'GO:0001605': {'name': ['adrenomedullin receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with adrenomedullin to initiate a change in cell activity.\" [GOC:ai]'],\n",
       "  'synonym': ['\"G10D receptor\" NARROW []'],\n",
       "  'is_a': ['GO:0097642 ! calcitonin family receptor activity']},\n",
       " 'GO:0001606': {'name': ['obsolete GPR37/endothelin B-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"GPR37/endothelin B-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001607': {'name': ['neuromedin U receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with neuromedin U to initiate a change in cell activity.\" [GOC:ai]'],\n",
       "  'synonym': ['\"NMUR activity\" EXACT []'],\n",
       "  'is_a': ['GO:0008188 ! neuropeptide receptor activity']},\n",
       " 'GO:0001608': {'name': ['G protein-coupled nucleotide receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a nucleotide and transmitting the signal across the membrane by activating an associated G-protein; promotes the exchange of GDP for GTP on the alpha subunit of a heterotrimeric G-protein complex.\" [GOC:bf, GOC:dph, IUPHAR_GPCR:1294]'],\n",
       "  'synonym': ['\"G protein coupled nucleotide receptor activity\" EXACT []',\n",
       "   '\"G-protein coupled nucleotide receptor activity\" EXACT []',\n",
       "   '\"nucleotide receptor activity, G protein coupled\" EXACT []',\n",
       "   '\"nucleotide receptor activity, G-protein coupled\" EXACT [GOC:bf]'],\n",
       "  'is_a': ['GO:0004930 ! G protein-coupled receptor activity']},\n",
       " 'GO:0001609': {'name': ['G protein-coupled adenosine receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001610',\n",
       "   'GO:0001611',\n",
       "   'GO:0001612',\n",
       "   'GO:0001613',\n",
       "   'GO:0008501'],\n",
       "  'def': ['\"Combining with adenosine and transmitting the signal across the membrane by activating an associated G-protein; promotes the exchange of GDP for GTP on the alpha subunit of a heterotrimeric G-protein complex.\" [GOC:bf, GOC:mah, PMID:9755289]'],\n",
       "  'synonym': ['\"A1 adenosine receptor activity, G protein coupled\" NARROW []',\n",
       "   '\"A1 adenosine receptor activity, G-protein coupled\" NARROW []',\n",
       "   '\"A2A adenosine receptor activity, G protein coupled\" NARROW []',\n",
       "   '\"A2A adenosine receptor activity, G-protein coupled\" NARROW []',\n",
       "   '\"A2B adenosine receptor activity, G protein coupled\" NARROW []',\n",
       "   '\"A2B adenosine receptor activity, G-protein coupled\" NARROW []',\n",
       "   '\"A3 adenosine receptor activity, G protein coupled\" NARROW []',\n",
       "   '\"A3 adenosine receptor activity, G-protein coupled\" NARROW []',\n",
       "   '\"adenosine nucleotide receptor\" BROAD []',\n",
       "   '\"adenosine receptor activity, G protein coupled\" EXACT []',\n",
       "   '\"adenosine receptor activity, G-protein coupled\" EXACT [GOC:bf]',\n",
       "   '\"G protein coupled A1 adenosine receptor activity\" NARROW []',\n",
       "   '\"G protein coupled A2A adenosine receptor activity\" NARROW []',\n",
       "   '\"G protein coupled A2B adenosine receptor activity\" NARROW []',\n",
       "   '\"G protein coupled A3 adenosine receptor activity\" NARROW []',\n",
       "   '\"G protein coupled adenosine receptor activity\" EXACT []',\n",
       "   '\"G-protein coupled adenosine receptor activity\" EXACT []',\n",
       "   '\"G-protein-coupled A1 adenosine receptor activity\" NARROW []',\n",
       "   '\"G-protein-coupled A2A adenosine receptor activity\" NARROW []',\n",
       "   '\"G-protein-coupled A2B adenosine receptor activity\" NARROW []',\n",
       "   '\"G-protein-coupled A3 adenosine receptor activity\" NARROW []',\n",
       "   '\"P1 receptor\" EXACT [PMID:9755289]'],\n",
       "  'is_a': ['GO:0004930 ! G protein-coupled receptor activity']},\n",
       " 'GO:0001614': {'name': ['purinergic nucleotide receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0035586'],\n",
       "  'def': ['\"Combining with a purine nucleotide and transmitting the signal from one side of the membrane to the other to initiate a change in cell activity.\" [GOC:mah, GOC:signaling]'],\n",
       "  'synonym': ['\"P2 receptor\" RELATED [PMID:9755289]',\n",
       "   '\"purinergic receptor activity\" RELATED []',\n",
       "   '\"purinoceptor\" EXACT [PMID:9755289]',\n",
       "   '\"purinoreceptor\" EXACT [PMID:9755289]'],\n",
       "  'is_a': ['GO:0016502 ! nucleotide receptor activity']},\n",
       " 'GO:0001615': {'name': ['obsolete thyrotropin releasing hormone and secretagogue-like receptors activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"thyrotropin releasing hormone and secretagogue-like receptors activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001616': {'name': ['growth hormone secretagogue receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with ghrelin to initiate a change in cell activity.\" [GOC:mah, PMID:17983853]'],\n",
       "  'synonym': ['\"ghrelin receptor activity\" EXACT [GOC:mah, PMID:17983853]'],\n",
       "  'is_a': ['GO:0004930 ! G protein-coupled receptor activity']},\n",
       " 'GO:0001617': {'name': ['obsolete growth hormone secretagogue-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"growth hormone secretagogue-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001618': {'name': ['virus receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a virus component and mediating entry of the virus into the cell.\" [GOC:bf, GOC:dph, PMID:7621403, UniProtKB-KW:KW-1183]'],\n",
       "  'subset': ['goslim_chembl', 'goslim_generic'],\n",
       "  'synonym': ['\"viral receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0140272 ! exogenous protein binding']},\n",
       " 'GO:0001619': {'name': ['obsolete lysosphingolipid and lysophosphatidic acid receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. Combining with lysosphingolipid or lysophosphatidic acid to initiate a change in cell activity.\" [GOC:dph]'],\n",
       "  'synonym': ['\"lysosphingolipid and lysophosphatidic acid receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'consider': ['GO:0038036', 'GO:0070915']},\n",
       " 'GO:0001621': {'name': ['G protein-coupled ADP receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0045032'],\n",
       "  'def': ['\"Combining with ADP and transmitting the signal across the membrane by activating an associated G-protein; promotes the exchange of GDP for GTP on the alpha subunit of a heterotrimeric G-protein complex.\" [GOC:mah, GOC:signaling, PMID:11196645]'],\n",
       "  'synonym': ['\"ADP receptor activity\" BROAD []',\n",
       "   '\"ADP-activated adenosine receptor activity\" RELATED []',\n",
       "   '\"ADP-activated nucleotide receptor activity\" BROAD []',\n",
       "   '\"K101 receptor\" NARROW []',\n",
       "   '\"platelet ADP receptor activity\" NARROW [GOC:mah]'],\n",
       "  'is_a': ['GO:0045028 ! G protein-coupled purinergic nucleotide receptor activity']},\n",
       " 'GO:0001626': {'name': ['nociceptin receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with the peptide nociceptin, and transmitting the signal across the membrane by activating an associated G-protein.\" [GOC:bf, GOC:mah, PMID:18670432]'],\n",
       "  'synonym': ['\"nociceptin/orphanin-FQ receptor activity\" EXACT [GOC:bf]',\n",
       "   '\"OFQ receptor activity\" EXACT [PR:000012940]',\n",
       "   '\"ORPH receptor\" EXACT []',\n",
       "   '\"orphanin-FQ receptor activity\" EXACT [GOC:bf]',\n",
       "   '\"X-opioid receptor activity\" RELATED [GOC:bf]'],\n",
       "  'is_a': ['GO:0004985 ! G protein-coupled opioid receptor activity',\n",
       "   'GO:0008528 ! G protein-coupled peptide receptor activity']},\n",
       " 'GO:0001627': {'name': ['obsolete leucine-rich G-protein receptor-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"leucine-rich G protein receptor-like receptor activity\" EXACT []',\n",
       "   '\"leucine-rich G-protein receptor-like receptor activity\" EXACT []',\n",
       "   '\"LGR-like receptor\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001628': {'name': ['obsolete gastropyloric receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because the gastropyloric receptor is a type of neuron.'],\n",
       "  'synonym': ['\"gastropyloric receptor activity\" EXACT []',\n",
       "   '\"GPR receptor\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001629': {'name': ['obsolete G-protein receptor 45-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"G protein receptor 45-like receptor activity\" EXACT []',\n",
       "   '\"G-protein receptor 45-like receptor activity\" EXACT []',\n",
       "   '\"GPR45-like receptor\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001630': {'name': ['obsolete GP40-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"GP40-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001631': {'name': ['cysteinyl leukotriene receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a cysteinyl leukotriene to initiate a change in cell activity. Cysteinyl leukotrienes are leukotrienes that contain a peptide group based on cysteine.\" [GOC:ai, ISBN:0198506732]'],\n",
       "  'synonym': ['\"CysLT receptor\" EXACT []'],\n",
       "  'is_a': ['GO:0004974 ! leukotriene receptor activity']},\n",
       " 'GO:0001632': {'name': ['leukotriene B4 receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with leukotriene B4, LTB4, to initiate a change in cell activity. Leukotriene B4 is also known as (6Z, 8E, 10E, 14Z)-(5S, 12R)-5,12-dihydroxyicosa-6,8,10,14-tetraen-1-oate.\" [GOC:ai, ISBN:0198506732]'],\n",
       "  'synonym': ['\"BLT receptor\" NARROW []'],\n",
       "  'is_a': ['GO:0004974 ! leukotriene receptor activity']},\n",
       " 'GO:0001633': {'name': ['obsolete secretin-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001638'],\n",
       "  'def': ['\"OBSOLETE. A G protein-coupled receptor that is structurally/functionally related to the secretin receptor.\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product and is named based on protein features.'],\n",
       "  'synonym': ['\"class B G protein coupled receptor\" EXACT []',\n",
       "   '\"class B G-protein coupled receptor\" EXACT []',\n",
       "   '\"class B GPCR\" EXACT []',\n",
       "   '\"class B orphan receptor activity\" EXACT []',\n",
       "   '\"secretin-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001634': {'name': ['pituitary adenylate cyclase-activating polypeptide receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0016522'],\n",
       "  'def': ['\"A G protein-coupled receptor that interacts with pituitary adenylate cyclase-activating polypeptide.\" [GOC:dph, GOC:tb]'],\n",
       "  'synonym': ['\"PACAP receptor\" EXACT []',\n",
       "   '\"pituitary adenylate cyclase activating polypeptide receptor\" EXACT []',\n",
       "   '\"pituitary adenylate cyclase activating protein receptor activity\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"pituitary adenylate cyclase-activating peptide receptor activity\" RELATED []',\n",
       "   '\"pituitary adenylyl cyclase activating protein receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0004930 ! G protein-coupled receptor activity']},\n",
       " 'GO:0001635': {'name': ['calcitonin gene-related peptide receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a calcitonin gene-related polypeptide (CGRP) to initiate a change in cell activity.\" [GOC:mah, PMID:12037140]'],\n",
       "  'synonym': ['\"calcitonin gene-related polypeptide receptor activity\" EXACT []',\n",
       "   '\"CGRP receptor\" EXACT []'],\n",
       "  'xref': ['Wikipedia:CALCRL'],\n",
       "  'is_a': ['GO:0097642 ! calcitonin family receptor activity']},\n",
       " 'GO:0001636': {'name': ['obsolete corticotrophin-releasing factor gastric inhibitory peptide-like receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:mah]'],\n",
       "  'comment': ['This term was made obsolete because it represents a gene product, was not defined, and is named based on protein features.'],\n",
       "  'synonym': ['\"corticotrophin-releasing factor gastric inhibitory peptide-like receptor activity\" EXACT []'],\n",
       "  'is_obsolete': ['true']},\n",
       " 'GO:0001637': {'name': ['G protein-coupled chemoattractant receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with a chemoattractant and transmitting the signal across the membrane by activating an associated G-protein; promotes the exchange of GDP for GTP on the alpha subunit of a heterotrimeric G-protein complex.\" [GOC:bf, GOC:mah]'],\n",
       "  'synonym': ['\"G protein chemoattractant receptor activity\" EXACT []',\n",
       "   '\"G-protein chemoattractant receptor activity\" EXACT [GOC:bf]',\n",
       "   '\"G-protein coupled chemoattractant receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0004930 ! G protein-coupled receptor activity']},\n",
       " 'GO:0001639': {'name': ['PLC activating G protein-coupled glutamate receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A G protein-coupled receptor that binds glutamate and is linked to the inositol 1,4,5-trisphosphate/calcium signaling system.\" [PMID:9016303]'],\n",
       "  'synonym': ['\"Group I metabotropic glutamate receptor\" RELATED []',\n",
       "   '\"group I metabotropic glutamate receptor activity\" RELATED []',\n",
       "   '\"phospholipase C activating metabotropic glutamate receptor activity\" EXACT []',\n",
       "   '\"PLC activating G-protein coupled glutamate receptor activity\" EXACT []',\n",
       "   '\"PLC activating metabotropic glutamate receptor activity\" EXACT [GOC:bf]'],\n",
       "  'is_a': ['GO:0098988 ! G protein-coupled glutamate receptor activity']},\n",
       " 'GO:0001640': {'name': ['adenylate cyclase inhibiting G protein-coupled glutamate receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with glutamate and transmitting the signal across the membrane by activating the alpha-subunit of an associated heterotrimeric G-protein complex to inhibit downstream adenylate cyclase activity.\" [GOC:bf, GOC:dph]'],\n",
       "  'synonym': ['\"adenylate cyclase inhibiting metabotropic glutamate receptor activity\" RELATED [GOC:bf]',\n",
       "   '\"adenylyl cyclase inhibiting metabotropic glutamate receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0098988 ! G protein-coupled glutamate receptor activity']},\n",
       " 'GO:0001641': {'name': ['group II metabotropic glutamate receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A G protein-coupled receptor that is activated by trans-1-aminocyclopentane-1,3-dicarboxylic acid (t-ACPD) and inhibits adenylate cyclase activity.\" [GOC:dph]'],\n",
       "  'is_a': ['GO:0001640 ! adenylate cyclase inhibiting G protein-coupled glutamate receptor activity']},\n",
       " 'GO:0001642': {'name': ['group III metabotropic glutamate receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A G protein-coupled receptor that is activated by L-AP-4 and inhibits adenylate cyclase activity.\" [PMID:9016303]'],\n",
       "  'is_a': ['GO:0001640 ! adenylate cyclase inhibiting G protein-coupled glutamate receptor activity']},\n",
       " 'GO:0001646': {'name': ['cAMP receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'alt_id': ['GO:0001644'],\n",
       "  'def': ['\"Combining with cAMP (cyclic AMP, adenosine 3\\',5\\'-cyclophosphate) and transmitting the signal from one side of the membrane to the other to initiate a change in cell activity.\" [GOC:pg]'],\n",
       "  'synonym': ['\"3\\',5\\' cAMP receptor activity\" EXACT []',\n",
       "   '\"3\\',5\\'-cAMP receptor activity\" EXACT []',\n",
       "   '\"adenosine 3\\',5\\'-cyclophosphate receptor activity\" EXACT []',\n",
       "   '\"class E G protein coupled receptor\" BROAD []',\n",
       "   '\"class E G-protein coupled receptor\" BROAD []',\n",
       "   '\"class E GPCR\" BROAD []',\n",
       "   '\"cyclic AMP receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0045028 ! G protein-coupled purinergic nucleotide receptor activity']},\n",
       " 'GO:0001647': {'name': ['G protein-coupled cytokinin receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with cytokinin and transmitting the signal across the membrane by activating an associated G-protein; promotes the exchange of GDP for GTP on the alpha subunit of a heterotrimeric G-protein complex.\" [GOC:bf, GOC:dph]'],\n",
       "  'synonym': ['\"cytokinin receptor activity, G-protein coupled\" EXACT [GOC:bf]',\n",
       "   '\"G protein coupled cytokinin receptor activity\" EXACT []',\n",
       "   '\"G-protein coupled cytokinin receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0004930 ! G protein-coupled receptor activity',\n",
       "   'GO:0009884 ! cytokinin receptor activity']},\n",
       " 'GO:0001648': {'name': ['proteinase-activated receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"A G protein-coupled receptor activity that is activated by cleavage by a serine protease, exposing a tethered ligand corresponding to the new N-terminus, which binds to the receptor and activates it.\" [GOC:mah, PMID:11356985, PMID:33742547]'],\n",
       "  'synonym': ['\"protease-activated receptor activity\" EXACT []',\n",
       "   '\"proteinase activated receptor activity\" EXACT []'],\n",
       "  'is_a': ['GO:0008528 ! G protein-coupled peptide receptor activity']},\n",
       " 'GO:0001649': {'name': ['osteoblast differentiation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whereby a relatively unspecialized cell acquires the specialized features of an osteoblast, a mesodermal or neural crest cell that gives rise to bone.\" [CL:0000062, GO_REF:0000034, GOC:jid]'],\n",
       "  'synonym': ['\"osteoblast cell differentiation\" EXACT []'],\n",
       "  'is_a': ['GO:0030154 ! cell differentiation'],\n",
       "  'relationship': ['part_of GO:0001503 ! ossification']},\n",
       " 'GO:0001650': {'name': ['fibrillar center'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A structure found most metazoan nucleoli, but not usually found in lower eukaryotes; surrounded by the dense fibrillar component; the zone of transcription from multiple copies of the pre-rRNA genes is in the border region between these two structures.\" [PMID:10754561]'],\n",
       "  'synonym': ['\"fibrillar centre\" EXACT []'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005730 ! nucleolus']},\n",
       " 'GO:0001651': {'name': ['dense fibrillar component'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A structure found in the nucleolus, which contains newly synthesized preribosomal RNA (pre-rRNA) and a collection of proteins.\" [PMID:10754561]'],\n",
       "  'synonym': ['\"pars fibrosa\" BROAD [NIF_Subcellular:sao1841764412]'],\n",
       "  'xref': ['NIF_Subcellular:sao1841764412'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005730 ! nucleolus']},\n",
       " 'GO:0001652': {'name': ['granular component'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A structure found in the nucleolus, which contains nearly completed preribosomal particles destined for the cytoplasm.\" [PMID:10754561]'],\n",
       "  'synonym': ['\"pars granulosa\" BROAD [NIF_Subcellular:sao1217793903]'],\n",
       "  'xref': ['NIF_Subcellular:sao1217793903'],\n",
       "  'is_a': ['GO:0110165 ! cellular anatomical entity'],\n",
       "  'relationship': ['part_of GO:0005730 ! nucleolus']},\n",
       " 'GO:0001653': {'name': ['peptide receptor activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Combining with an extracellular or intracellular peptide to initiate a change in cell activity.\" [GOC:jl]'],\n",
       "  'synonym': ['\"endogenous peptide receptor activity\" NARROW []',\n",
       "   '\"exogenous peptide receptor activity\" NARROW []'],\n",
       "  'is_a': ['GO:0038023 ! signaling receptor activity']},\n",
       " 'GO:0001654': {'name': ['eye development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0042460'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the eye over time, from its formation to the mature structure. The eye is the organ of sight.\" [GOC:jid, GOC:jl]'],\n",
       "  'xref': ['Wikipedia:Eye_development'],\n",
       "  'is_a': ['GO:0007423 ! sensory organ development'],\n",
       "  'relationship': ['part_of GO:0150063 ! visual system development']},\n",
       " 'GO:0001655': {'name': ['urogenital system development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the urogenital system over time, from its formation to the mature structure.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0048731 ! system development']},\n",
       " 'GO:0001656': {'name': ['metanephros development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the metanephros over time, from its formation to the mature structure. In mammals, the metanephros is the excretory organ of the fetus, which develops into the mature kidney and is formed from the rear portion of the nephrogenic cord. The metanephros is an endocrine and metabolic organ that filters the blood and excretes the end products of body metabolism in the form of urine.\" [GOC:bf, ISBN:0192800752]'],\n",
       "  'is_a': ['GO:0001822 ! kidney development']},\n",
       " 'GO:0001657': {'name': ['ureteric bud development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the ureteric bud over time, from its formation to the mature structure.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0072164 ! mesonephric tubule development']},\n",
       " 'GO:0001658': {'name': ['branching involved in ureteric bud morphogenesis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process in which the branching structure of the ureteric bud is generated and organized. The ureteric bud is an epithelial tube that grows out from the metanephric duct. The bud elongates and branches to give rise to the ureter and kidney collecting tubules.\" [GOC:dph, PMID:16916378]'],\n",
       "  'synonym': ['\"ureteric bud branching\" EXACT [GOC:dph]'],\n",
       "  'is_a': ['GO:0048754 ! branching morphogenesis of an epithelial tube'],\n",
       "  'relationship': ['part_of GO:0060675 ! ureteric bud morphogenesis']},\n",
       " 'GO:0001659': {'name': ['temperature homeostasis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A homeostatic process in which an organism modulates its internal body temperature.\" [GOC:jl]'],\n",
       "  'synonym': ['\"thermoregulation\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'xref': ['Wikipedia:Thermoregulation'],\n",
       "  'is_a': ['GO:0048871 ! multicellular organismal-level homeostasis']},\n",
       " 'GO:0001660': {'name': ['fever generation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The heat generation process that results in a rise in body temperature above the normal, often as a response to infection.\" [GOC:dph, GOC:jl]'],\n",
       "  'synonym': ['\"pyrexia\" BROAD []'],\n",
       "  'xref': ['Wikipedia:Fever'],\n",
       "  'is_a': ['GO:0006953 ! acute-phase response',\n",
       "   'GO:0031649 ! heat generation']},\n",
       " 'GO:0001661': {'name': ['conditioned taste aversion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A conditioned aversion to a specific chemical compound as a result of that compound being coupled with a noxious stimulus.\" [GOC:dph, PMID:9920659]'],\n",
       "  'xref': ['Wikipedia:Conditioned_taste_aversion', 'Wikipedia:Taste_aversion'],\n",
       "  'is_a': ['GO:0007631 ! feeding behavior',\n",
       "   'GO:0008306 ! associative learning']},\n",
       " 'GO:0001662': {'name': ['behavioral fear response'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"An acute behavioral change resulting from a perceived external threat.\" [GOC:dph, PMID:9920659]'],\n",
       "  'synonym': ['\"behavioural fear response\" EXACT []'],\n",
       "  'is_a': ['GO:0002209 ! behavioral defense response',\n",
       "   'GO:0042596 ! fear response']},\n",
       " 'GO:0001664': {'name': ['G protein-coupled receptor binding'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binding to a G protein-coupled receptor.\" [GOC:ceb, GOC:dph]'],\n",
       "  'synonym': ['\"G protein coupled receptor binding\" EXACT []',\n",
       "   '\"G protein coupled receptor ligand\" NARROW []',\n",
       "   '\"G-protein coupled receptor binding\" EXACT [GOC:bf]',\n",
       "   '\"G-protein-coupled receptor ligand\" NARROW []'],\n",
       "  'xref': ['Reactome:R-HSA-500717 \"Activation of GRIK3 homomer\"'],\n",
       "  'is_a': ['GO:0005102 ! signaling receptor binding']},\n",
       " 'GO:0001665': {'name': ['alpha-N-acetylgalactosaminide alpha-2,6-sialyltransferase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"GalNAc alpha-2,6-sialyltransferase I activity\" RELATED [EC:2.4.3.3]'],\n",
       "  'xref': ['EC:2.4.3.3',\n",
       "   'MetaCyc:2.4.99.3-RXN',\n",
       "   'Reactome:R-HSA-4084980 \"ST6GALNAC1-6 transfer Neu5Ac to terminal GalNAc (alpha-2,6 link)\"',\n",
       "   'Reactome:R-HSA-9603991 \"ST6GALNAC6 transfers Neu5Ac to Type 1 MSGG to form Type 1 DSGG\"',\n",
       "   'RHEA:11136'],\n",
       "  'is_a': ['GO:0008373 ! sialyltransferase activity']},\n",
       " 'GO:0001666': {'name': ['response to hypoxia'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Any process that results in a change in state or activity of a cell or an organism (in terms of movement, secretion, enzyme production, gene expression, etc.) as a result of a stimulus indicating lowered oxygen tension. Hypoxia, defined as a decline in O2 levels below normoxic levels of 20.8 - 20.95%, results in metabolic adaptation at both the cellular and organismal level.\" [GOC:hjd]'],\n",
       "  'comment': [\"Note that this term should not be confused with 'response to anoxia ; GO:0034059'. Note that in laboratory studies, hypoxia is typically studied at O2 concentrations ranging from 0.1 - 5%.\"],\n",
       "  'synonym': ['\"response to hypoxic stress\" EXACT []',\n",
       "   '\"response to intermittent hypoxia\" NARROW []',\n",
       "   '\"response to lowered oxygen tension\" EXACT []',\n",
       "   '\"response to sustained hypoxia\" NARROW []'],\n",
       "  'is_a': ['GO:0006950 ! response to stress',\n",
       "   'GO:0036293 ! response to decreased oxygen levels']},\n",
       " 'GO:0001667': {'name': ['ameboidal-type cell migration'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Cell migration that is accomplished by extension and retraction of a pseudopodium.\" [GOC:dph]'],\n",
       "  'comment': ['Note that this term refers to a mode of migration rather than to any particular cell type.'],\n",
       "  'synonym': ['\"ameboid cell migration\" EXACT []',\n",
       "   '\"amoeboid cell migration\" EXACT []',\n",
       "   '\"amoeboidal cell migration\" EXACT []'],\n",
       "  'is_a': ['GO:0016477 ! cell migration']},\n",
       " 'GO:0001669': {'name': ['acrosomal vesicle'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"A structure in the head of a spermatozoon that contains acid hydrolases, and is concerned with the breakdown of the outer membrane of the ovum during fertilization. It lies just beneath the plasma membrane and is derived from the lysosome.\" [ISBN:0124325653, ISBN:0198506732]'],\n",
       "  'synonym': ['\"acrosomal granule\" EXACT [GOC:sart]',\n",
       "   '\"acrosome\" EXACT [GOC:dph]'],\n",
       "  'xref': ['Wikipedia:Acrosome'],\n",
       "  'is_a': ['GO:0030141 ! secretory granule']},\n",
       " 'GO:0001671': {'name': ['ATPase activator activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Binds to and increases the activity of an ATP hydrolysis activity.\" [GOC:ajp]'],\n",
       "  'synonym': ['\"ATPase stimulator activity\" EXACT []'],\n",
       "  'xref': ['Reactome:R-HSA-5251955 \"HSP40s activate intrinsic ATPase activity of HSP70s in the nucleoplasm\"',\n",
       "   'Reactome:R-HSA-5251959 \"HSP40s activate intrinsic ATPase activity of HSP70s in the cytosol\"'],\n",
       "  'is_a': ['GO:0060590 ! ATPase regulator activity',\n",
       "   'GO:0140677 ! molecular function activator activity']},\n",
       " 'GO:0001673': {'name': ['male germ cell nucleus'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0043081'],\n",
       "  'def': ['\"The nucleus of a male germ cell, a reproductive cell in males.\" [CL:0000015, GOC:hjd, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"male germ-cell nucleus\" EXACT []'],\n",
       "  'is_a': ['GO:0043073 ! germ cell nucleus']},\n",
       " 'GO:0001674': {'name': ['female germ cell nucleus'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'alt_id': ['GO:0043080'],\n",
       "  'def': ['\"The nucleus of the female germ cell, a reproductive cell in females.\" [CL:0000021, GOC:hjd]'],\n",
       "  'synonym': ['\"female germ-cell nucleus\" EXACT []'],\n",
       "  'is_a': ['GO:0043073 ! germ cell nucleus']},\n",
       " 'GO:0001675': {'name': ['acrosome assembly'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of the acrosome from the spermatid Golgi.\" [GOC:dph, GOC:hjd, GOC:tb]'],\n",
       "  'synonym': ['\"acrosome formation\" EXACT [GOC:dph, GOC:tb]'],\n",
       "  'is_a': ['GO:0003006 ! developmental process involved in reproduction',\n",
       "   'GO:0010927 ! cellular component assembly involved in morphogenesis',\n",
       "   'GO:0022412 ! cellular process involved in reproduction in multicellular organism',\n",
       "   'GO:0033363 ! secretory granule organization',\n",
       "   'GO:0070925 ! organelle assembly'],\n",
       "  'relationship': ['part_of GO:0007286 ! spermatid development']},\n",
       " 'GO:0001676': {'name': ['long-chain fatty acid metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving a long-chain fatty acid, a fatty acid with an aliphatic tail of 13 to 21 carbons.\" [GOC:ajp]'],\n",
       "  'synonym': ['\"long-chain fatty acid metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006631 ! fatty acid metabolic process']},\n",
       " 'GO:0001677': {'name': ['formation of translation initiation ternary complex'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Formation of a complex between aminoacylated initiator methionine tRNA, GTP, and initiation factor 2 (either eIF2 in eukaryotes, or IF2 in prokaryotes). In prokaryotes, fMet-tRNA (initiator) is used rather than Met-tRNA (initiator).\" [GOC:hjd]'],\n",
       "  'synonym': ['\"translation initiation ternary complex assembly\" EXACT []'],\n",
       "  'is_a': ['GO:0022618 ! protein-RNA complex assembly'],\n",
       "  'relationship': ['part_of GO:0006413 ! translational initiation']},\n",
       " 'GO:0001678': {'name': ['intracellular glucose homeostasis'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A homeostatic process involved in the maintenance of a steady state level of glucose within a cell.\" [GOC:dph, GOC:go_curators, GOC:tb]'],\n",
       "  'synonym': ['\"cell glucose homeostasis\" EXACT [GOC:dph, GOC:tb]',\n",
       "   '\"cellular glucose homeostasis\" EXACT []'],\n",
       "  'is_a': ['GO:0042593 ! glucose homeostasis',\n",
       "   'GO:0055082 ! intracellular chemical homeostasis']},\n",
       " 'GO:0001680': {'name': [\"tRNA 3'-terminal CCA addition\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Post-transcriptional addition of the terminal 3\\' CCA sequence to a tRNA which does not encode this sequence within the primary transcript. CCA addition proceeds by the sequential addition of CTP, CTP, and then ATP to the 3\\' end of the tRNA, yielding a diphosphate with each nucleotide addition.\" [GOC:go_curators, PMID:2247609]'],\n",
       "  'is_a': [\"GO:0042780 ! tRNA 3'-end processing\"]},\n",
       " 'GO:0001681': {'name': ['sialate O-acetylesterase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Catalysis of the reaction'],\n",
       "  'synonym': ['\"N-acetylneuraminate acetyltransferase activity\" RELATED [EC:3.1.1.53]',\n",
       "   '\"N-acyl-O-acetylneuraminate O-acetylhydrolase activity\" RELATED [EC:3.1.1.53]',\n",
       "   '\"sialate 9(4)-O-acetylesterase activity\" EXACT []'],\n",
       "  'xref': ['EC:3.1.1.53', 'MetaCyc:SIALATE-O-ACETYLESTERASE-RXN'],\n",
       "  'is_a': ['GO:0008126 ! acetylesterase activity']},\n",
       " 'GO:0001682': {'name': [\"tRNA 5'-leader removal\"],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"Generation of the mature 5\\'-end of the tRNA, usually via an endonucleolytic cleavage by RNase P.\" [PMID:11592395]'],\n",
       "  'synonym': ['\"tRNA 5\\' leader removal\" EXACT []'],\n",
       "  'is_a': [\"GO:0099116 ! tRNA 5'-end processing\"]},\n",
       " 'GO:0001683': {'name': ['obsolete axonemal dynein heavy chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"axonemal dynein heavy chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005858']},\n",
       " 'GO:0001684': {'name': ['obsolete axonemal dynein intermediate chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"axonemal dynein intermediate chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005858']},\n",
       " 'GO:0001685': {'name': ['obsolete axonemal dynein intermediate light chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"axonemal dynein intermediate light chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005858']},\n",
       " 'GO:0001686': {'name': ['obsolete axonemal dynein light chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"axonemal dynein light chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005858']},\n",
       " 'GO:0001687': {'name': ['obsolete cytoplasmic dynein heavy chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"cytoplasmic dynein heavy chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005868']},\n",
       " 'GO:0001688': {'name': ['obsolete cytoplasmic dynein intermediate chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"cytoplasmic dynein intermediate chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005868']},\n",
       " 'GO:0001689': {'name': ['obsolete cytoplasmic dynein intermediate light chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"cytoplasmic dynein intermediate light chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005868']},\n",
       " 'GO:0001690': {'name': ['obsolete cytoplasmic dynein light chain'],\n",
       "  'namespace': ['cellular_component'],\n",
       "  'def': ['\"OBSOLETE. (Was not defined before being made obsolete).\" [GOC:jl]'],\n",
       "  'comment': ['This term was made obsolete because it represents a single gene product and not a complex.'],\n",
       "  'synonym': ['\"cytoplasmic dynein light chain\" EXACT []'],\n",
       "  'is_obsolete': ['true'],\n",
       "  'replaced_by': ['GO:0005868']},\n",
       " 'GO:0001691': {'name': ['pseudophosphatase activity'],\n",
       "  'namespace': ['molecular_function'],\n",
       "  'def': ['\"Maintains the phosphorylation state of certain molecules by associating with them and preventing them from associating with active phosphatases, and thus inhibiting the enzyme activity without interacting with the enzyme. Often pertains to proteins belonging to dual-specificity phosphatase family but lacking critical active site residues.\" [GOC:ajp]'],\n",
       "  'is_a': ['GO:0019212 ! phosphatase inhibitor activity']},\n",
       " 'GO:0001692': {'name': ['histamine metabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways involving histamine, a physiologically active amine, found in plant and animal tissue and released from mast cells as part of an allergic reaction in humans.\" [GOC:jl, ISBN:0395825172]'],\n",
       "  'synonym': ['\"histamine metabolism\" EXACT []'],\n",
       "  'is_a': ['GO:0006576 ! biogenic amine metabolic process',\n",
       "   'GO:0006796 ! phosphate-containing compound metabolic process',\n",
       "   'GO:0052803 ! imidazole-containing compound metabolic process',\n",
       "   'GO:0097164 ! ammonium ion metabolic process']},\n",
       " 'GO:0001694': {'name': ['histamine biosynthetic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0001693'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the formation of histamine, a physiologically active amine, found in plant and animal tissue and released from mast cells as part of an allergic reaction in humans.\" [GOC:jl, ISBN:0395825172]'],\n",
       "  'synonym': ['\"histamine anabolism\" EXACT []',\n",
       "   '\"histamine biosynthesis\" EXACT []',\n",
       "   '\"histamine formation\" EXACT []',\n",
       "   '\"histamine synthesis\" EXACT []'],\n",
       "  'is_a': ['GO:0001692 ! histamine metabolic process',\n",
       "   'GO:0018130 ! heterocycle biosynthetic process',\n",
       "   'GO:0019438 ! aromatic compound biosynthetic process',\n",
       "   'GO:0042401 ! biogenic amine biosynthetic process',\n",
       "   'GO:1901362 ! organic cyclic compound biosynthetic process']},\n",
       " 'GO:0001695': {'name': ['histamine catabolic process'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The chemical reactions and pathways resulting in the breakdown of histamine, a physiologically active amine, found in plant and animal tissue and released from mast cells as part of an allergic reaction in humans.\" [GOC:jl, ISBN:0395825172]'],\n",
       "  'synonym': ['\"histamine breakdown\" EXACT []',\n",
       "   '\"histamine catabolism\" EXACT []',\n",
       "   '\"histamine degradation\" EXACT []'],\n",
       "  'is_a': ['GO:0001692 ! histamine metabolic process',\n",
       "   'GO:0042135 ! neurotransmitter catabolic process',\n",
       "   'GO:0042402 ! cellular biogenic amine catabolic process',\n",
       "   'GO:0052805 ! imidazole-containing compound catabolic process']},\n",
       " 'GO:0001696': {'name': ['gastric acid secretion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The regulated release of gastric acid (hydrochloric acid) by parietal or oxyntic cells during digestion.\" [GOC:hjd]'],\n",
       "  'synonym': ['\"hydrochloric acid secretion\" NARROW []'],\n",
       "  'is_a': ['GO:0022600 ! digestive system process',\n",
       "   'GO:0046717 ! acid secretion']},\n",
       " 'GO:0001697': {'name': ['histamine-induced gastric acid secretion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The regulated release of gastric acid induced by the interaction of histamine with H2 type receptor receptors with subsequent activation of adenylate cyclase and elevation of intracellular cyclic AMP.\" [GOC:hjd]'],\n",
       "  'is_a': ['GO:0001696 ! gastric acid secretion']},\n",
       " 'GO:0001698': {'name': ['gastrin-induced gastric acid secretion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The regulated release of gastric acid induced by the interaction of gastrin with its receptor.\" [GOC:hjd]'],\n",
       "  'is_a': ['GO:0001696 ! gastric acid secretion']},\n",
       " 'GO:0001699': {'name': ['acetylcholine-induced gastric acid secretion'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The regulated release of gastric acid by parietal cells in response to acetylcholine.\" [GOC:hjd]'],\n",
       "  'is_a': ['GO:0001696 ! gastric acid secretion'],\n",
       "  'relationship': ['part_of GO:1905145 ! cellular response to acetylcholine']},\n",
       " 'GO:0001700': {'name': ['embryonic development via the syncytial blastoderm'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the embryo over time, from zygote formation through syncytial blastoderm to the hatching of the first instar larva. An example of this process is found in Drosophila melanogaster.\" [GOC:go_curators, GOC:mtg_sensu]'],\n",
       "  'is_a': ['GO:0009792 ! embryo development ending in birth or egg hatching']},\n",
       " 'GO:0001701': {'name': ['in utero embryonic development'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process whose specific outcome is the progression of the embryo in the uterus over time, from formation of the zygote in the oviduct, to birth. An example of this process is found in Mus musculus.\" [GOC:go_curators, GOC:mtg_sensu]'],\n",
       "  'is_a': ['GO:0043009 ! chordate embryonic development']},\n",
       " 'GO:0001702': {'name': ['gastrulation with mouth forming second'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'alt_id': ['GO:0010003', 'GO:0048276'],\n",
       "  'def': ['\"A gastrulation process in which the initial invagination becomes the anus and the mouth forms second.\" [GOC:go_curators, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"deuterostomic gastrulation\" EXACT [GOC:dph]'],\n",
       "  'is_a': ['GO:0007369 ! gastrulation']},\n",
       " 'GO:0001703': {'name': ['gastrulation with mouth forming first'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"A gastrulation process in which the initial invagination becomes the mouth and the anus forms second.\" [GOC:go_curators, GOC:mtg_sensu]'],\n",
       "  'synonym': ['\"protostomic gastrulation\" NARROW [GOC:dph]'],\n",
       "  'is_a': ['GO:0007369 ! gastrulation']},\n",
       " 'GO:0001704': {'name': ['formation of primary germ layer'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of the ectoderm, mesoderm and endoderm during gastrulation.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0048646 ! anatomical structure formation involved in morphogenesis'],\n",
       "  'relationship': ['part_of GO:0007369 ! gastrulation']},\n",
       " 'GO:0001705': {'name': ['ectoderm formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of ectoderm during gastrulation.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0001704 ! formation of primary germ layer'],\n",
       "  'relationship': ['part_of GO:0007398 ! ectoderm development']},\n",
       " 'GO:0001706': {'name': ['endoderm formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The formation of the endoderm during gastrulation.\" [GOC:go_curators]'],\n",
       "  'synonym': ['\"endoblast formation\" NARROW [GOC:dph, GOC:sdb_2009, GOC:tb]'],\n",
       "  'is_a': ['GO:0001704 ! formation of primary germ layer'],\n",
       "  'relationship': ['part_of GO:0007492 ! endoderm development']},\n",
       " 'GO:0001707': {'name': ['mesoderm formation'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The process that gives rise to the mesoderm. This process pertains to the initial formation of the structure from unspecified parts.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0001704 ! formation of primary germ layer'],\n",
       "  'relationship': ['part_of GO:0048332 ! mesoderm morphogenesis']},\n",
       " 'GO:0001708': {'name': ['cell fate specification'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cellular developmental process involved in cell fate commitment in which the cell is designated to follow a developmental path, unless they receive extrinsic cues that direct an alternative fate.\" [GOC:go_curators]'],\n",
       "  'is_a': ['GO:0048869 ! cellular developmental process'],\n",
       "  'relationship': ['part_of GO:0045165 ! cell fate commitment']},\n",
       " 'GO:0001709': {'name': ['cell fate determination'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cellular developmental process involved in cell fate commitment that occurs after cell fate specification,  in which a cell is irreversibly committed to a cellular developmental fate which is heritable on cell division.\" [ISBN:0878932437]'],\n",
       "  'xref': ['Wikipedia:Cell_fate_determination'],\n",
       "  'is_a': ['GO:0048869 ! cellular developmental process'],\n",
       "  'relationship': ['part_of GO:0045165 ! cell fate commitment']},\n",
       " 'GO:0001710': {'name': ['mesodermal cell fate commitment'],\n",
       "  'namespace': ['biological_process'],\n",
       "  'def': ['\"The cell differentiation process that results in commitment of a cell to become part of the mesoderm.\" [GOC:go_curators, ISBN:0878932437]'],\n",
       "  'synonym': ['\"mesoderm cell fate commitment\" EXACT []'],\n",
       "  'is_a': ['GO:0060795 ! cell fate commitment involved in formation of primary germ layer'],\n",
       "  'relationship': ['part_of GO:0048333 ! mesodermal cell differentiation']},\n",
       " ...}"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Convert GO obo file to dict\n",
    "ID = ''\n",
    "go_dict = dict()\n",
    "with open('input/go-basic.obo') as fin:\n",
    "    for line in fin:\n",
    "        if line.startswith('id: '):\n",
    "            ID = line.split('id: ')[1].strip('\\n')\n",
    "            continue\n",
    "        if ': ' in line and ID != '':\n",
    "            k = line.split(': ')[0]\n",
    "            v = line.split(': ')[1].strip('\\n')\n",
    "            go_dict.setdefault(ID,dict()).setdefault(k,[]).append(v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "go_id_to_go_ont = dict()\n",
    "for go_id, values in go_dict.items():\n",
    "    go_ont_name = values['namespace'][0]\n",
    "    go_id_to_go_ont[go_id] = go_ont_name\n",
    "    \n",
    "go_ontology_names = set(go_id_to_go_ont.values())\n",
    "go_ontology_names.remove('external')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'biological_process', 'cellular_component', 'molecular_function'}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_name = 'CC_MF_BP_(GO)_2_CC_MF_BP_(GO).csv'\n",
    "with open(f'output/go2go/{file_name}', 'w') as fout:  \n",
    "    writer = csv.writer(fout)\n",
    "    writer.writerow(['CC_MF_BP_(GO)', 'CC_MF_BP_(GO)', 'Relationship'])\n",
    "    \n",
    "    for go_id, values in go_dict.items():\n",
    "        if not go_id.startswith('GO:'):\n",
    "            continue\n",
    "        go_id = go_id_to_go_ont[go_id]+':'+go_id.split('GO:')[1]\n",
    "            \n",
    "        # 'is_a' relationships\n",
    "        try:\n",
    "            is_a_gos = [go.split(' !')[0] for go in values['is_a']]\n",
    "            for other_go in is_a_gos:\n",
    "                ont_type = go_id_to_go_ont[other_go]\n",
    "                other_go = ont_type+':'+other_go.split('GO:')[1]\n",
    "            writer.writerow([go_id, other_go, '-is_a-'])\n",
    "        except:\n",
    "            pass\n",
    "\n",
    "        # 'part_of', 'regulates', etc. relationships\n",
    "        try:\n",
    "            rel_gos = [r_g.split(' !')[0].split(' ') for r_g in values['relationship']]\n",
    "            for rel, other_go in rel_gos:\n",
    "                other_go = ont_type+':'+other_go.split('GO:')[1]\n",
    "                writer.writerow([go_id, other_go, f'-{rel}->'])\n",
    "        except:\n",
    "            pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for first, second in zip(df['CC_MF_BP_(GO)'], df['CC_MF_BP_(GO).1']):\n",
    "    if first.split(':')[0] != second.split(':')[0]:\n",
    "        print(first, second)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Relationship\n",
       "-is_a-                     43005\n",
       "-part_of->                  6802\n",
       "-regulates->                3127\n",
       "-negatively_regulates->     2708\n",
       "-positively_regulates->     2695\n",
       "dtype: int64"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(f'output/go2go/{file_name}')\n",
    "df.value_counts('Relationship')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "compound_to_compound_alignment.ipynb:    \"db2keggdrug, db2keggcompound, db2oldttd, db2uniprot, db2pharmgkb, db2wikipedia = dict(), dict(), dict(), dict(), dict(), dict()\\n\",\r\n",
      "compound_to_compound_alignment.ipynb:    \"keggdrug2db, keggcompound2db, oldttd2db, uniprot2db, pharmgkb2db, wikipedia2db = dict(), dict(), dict(), dict(), dict(), dict()\\n\",\r\n",
      "compound_to_compound_alignment.ipynb:    \"        db2pharmgkb[db] = onts['PharmGKB']  \\n\",\r\n",
      "compound_to_compound_alignment.ipynb:    \"            pharmgkb2db.setdefault(pharm,set()).add(db)\\n\",\r\n",
      "compound_to_compound_alignment.ipynb:    \"pharmgkb2db = switch_dictset_to_dictlist(pharmgkb2db)\\n\",\r\n",
      "compound_to_compound_alignment.ipynb:    \"         db2keggcompound, db2oldttd, db2uniprot, db2pharmgkb,\\n\",\r\n",
      "compound_to_compound_alignment.ipynb:    \"         uniprot2db, pharmgkb2db, wikipedia2db]\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"with open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv','w') as fout:\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"! cp 'output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv' 'output/edges/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"df1 = pd.read_csv(open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'))\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"with open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv','w') as fout:\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"#! cp 'output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH-disease_pharmgkb.csv' 'output/edges/edges_gene-NOT-ASSOCIATED_WITH->disease_pharmgkb.csv'\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"df1 = pd.read_csv(open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'))\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"with open('output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH-disease_pharmgkb.csv','w') as fout:\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"! cp 'output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH-disease_pharmgkb.csv' 'output/edges/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"df1 = pd.read_csv(open('output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH->disease_pharmgkb.csv'))\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"pharmgkb_df = pd.read_csv('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv').append(pd.read_csv('output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH->disease_pharmgkb.csv')).drop_duplicates()\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"#display(pharmgkb_df.head())\\n\",\r\n",
      "gene_to_disease-old_5-24-23.ipynb:    \"df = pd.concat([disgenet_df, clingen_df, clinvar_df, pharmgkb_df, kegg_df]).drop_duplicates()\\n\",\r\n",
      "gene_to_disease.ipynb:    \"with open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv','w') as fout:\\n\",\r\n",
      "gene_to_disease.ipynb:    \"! cp 'output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv' 'output/edges/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'\\n\",\r\n",
      "gene_to_disease.ipynb:    \"df1 = pd.read_csv(open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'))\\n\",\r\n",
      "gene_to_disease.ipynb:    \"with open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv','w') as fout:\\n\",\r\n",
      "gene_to_disease.ipynb:    \"#! cp 'output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH-disease_pharmgkb.csv' 'output/edges/edges_gene-NOT-ASSOCIATED_WITH->disease_pharmgkb.csv'\\n\",\r\n",
      "gene_to_disease.ipynb:    \"df1 = pd.read_csv(open('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'))\\n\",\r\n",
      "gene_to_disease.ipynb:    \"with open('output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH-disease_pharmgkb.csv','w') as fout:\\n\",\r\n",
      "gene_to_disease.ipynb:    \"! cp 'output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH-disease_pharmgkb.csv' 'output/edges/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv'\\n\",\r\n",
      "gene_to_disease.ipynb:    \"df1 = pd.read_csv(open('output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH->disease_pharmgkb.csv'))\\n\",\r\n",
      "gene_to_disease.ipynb:    \"pharmgkb_df = pd.read_csv('output/gene2disease/edges_gene-ASSOCIATED_WITH->disease_pharmgkb.csv').append(pd.read_csv('output/gene2disease/edges_gene-NOT-ASSOCIATED_WITH->disease_pharmgkb.csv')).drop_duplicates()\\n\",\r\n",
      "gene_to_disease.ipynb:    \"#display(pharmgkb_df.head())\\n\",\r\n",
      "gene_to_disease.ipynb:    \"df = pd.concat([disgenet_df, clingen_df, clinvar_df, pharmgkb_df, kegg_df]).drop_duplicates()\\n\",\r\n"
     ]
    }
   ],
   "source": [
    "!grep -r \"pharmgkb\" *.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(f'output/go2go/{file_name}')\n",
    "df.to_csv(f'output/edges/{file_name}', index=False)\n",
    "df.to_csv(f'output/edges to use/{file_name}', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "for first, second in zip(df['CC_MF_BP_(GO)'], df['CC_MF_BP_(GO).1']):\n",
    "    if first.split(':')[0] != second.split(':')[0]:\n",
    "        print(first, second)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "molecular_function\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'-is_a->': 13736, '-part_of->': 11}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Molecular_Function (GO)</th>\n",
       "      <th>Molecular_Function (GO).1</th>\n",
       "      <th>Relationship</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>molecular_function:0000006</td>\n",
       "      <td>molecular_function:0005385</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>molecular_function:0000007</td>\n",
       "      <td>molecular_function:0005385</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>molecular_function:0000009</td>\n",
       "      <td>molecular_function:0000030</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Molecular_Function (GO)   Molecular_Function (GO).1 Relationship\n",
       "0  molecular_function:0000006  molecular_function:0005385      -is_a->\n",
       "1  molecular_function:0000007  molecular_function:0005385      -is_a->\n",
       "2  molecular_function:0000009  molecular_function:0000030      -is_a->"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "biological_process\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'-is_a->': 50938,\n",
       " '-regulates->': 3157,\n",
       " '-part_of->': 5018,\n",
       " '-negatively_regulates->': 2729,\n",
       " '-positively_regulates->': 2718}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biological_Process (GO)</th>\n",
       "      <th>Biological_Process (GO).1</th>\n",
       "      <th>Relationship</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>biological_process:0000001</td>\n",
       "      <td>biological_process:0048308</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>biological_process:0000001</td>\n",
       "      <td>biological_process:0048311</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>biological_process:0000002</td>\n",
       "      <td>biological_process:0007005</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Biological_Process (GO)   Biological_Process (GO).1 Relationship\n",
       "0  biological_process:0000001  biological_process:0048308      -is_a->\n",
       "1  biological_process:0000001  biological_process:0048311      -is_a->\n",
       "2  biological_process:0000002  biological_process:0007005      -is_a->"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cellular_component\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'-is_a->': 4676, '-part_of->': 1822}"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Cellular_Component (GO)</th>\n",
       "      <th>Cellular_Component (GO).1</th>\n",
       "      <th>Relationship</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cellular_component:0000015</td>\n",
       "      <td>cellular_component:1902494</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>cellular_component:0000015</td>\n",
       "      <td>cellular_component:0005829</td>\n",
       "      <td>-part_of-&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cellular_component:0000109</td>\n",
       "      <td>cellular_component:0140513</td>\n",
       "      <td>-is_a-&gt;</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Cellular_Component (GO)   Cellular_Component (GO).1 Relationship\n",
       "0  cellular_component:0000015  cellular_component:1902494      -is_a->\n",
       "1  cellular_component:0000015  cellular_component:0005829   -part_of->\n",
       "2  cellular_component:0000109  cellular_component:0140513      -is_a->"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "for go2go_type in go_ontology_names:\n",
    "    cap_go2go_type = '_'.join([w.capitalize() for w in go2go_type.split('_')])\n",
    "    \n",
    "    # Output GO-[rel]->GO\n",
    "    with open(f'output/go2go/{cap_go2go_type}_(GO)_2_{cap_go2go_type}_(GO).csv','w') as fout:\n",
    "        writer = csv.writer(fout)\n",
    "        writer.writerow([f'{cap_go2go_type} (GO)', f'{cap_go2go_type} (GO)','Relationship'])\n",
    "        go2go = dict()\n",
    "        rel_counts = dict()\n",
    "\n",
    "        for go, values in go_dict.items():\n",
    "            if values['namespace'][0] == go2go_type:\n",
    "                \n",
    "                for value in values:\n",
    "                    ### Obsolete? ###\n",
    "                    if value == 'is_obsolete' and go_dict[go][value] == ['true']:\n",
    "                        continue\n",
    "                    if value == 'replaced_by':\n",
    "                        continue\n",
    "                    if 'GO' not in go:\n",
    "                        continue\n",
    "\n",
    "                        \n",
    "                    ### Relationships ###\n",
    "                    if value == 'is_a':\n",
    "                        rel_type = '-'+value+'->'\n",
    "                        start_node = go.split('GO:')[1]\n",
    "                        start_node = go2go_type+':'+start_node\n",
    "                        end_nodes = go_dict[go][value]\n",
    "                        for end_node in end_nodes:\n",
    "                            end_node = end_node.split('GO:')[1]\n",
    "                            end_node = go2go_type+':'+end_node\n",
    "                            go2go.setdefault(start_node, dict()).setdefault(rel_type,[]).append(end_node.split(' !')[0])\n",
    "                            writer.writerow([start_node, end_node.split(' !')[0], rel_type])\n",
    "                            rel_counts[rel_type] = rel_counts.setdefault(rel_type,0) + 1\n",
    "\n",
    "                    elif value == 'relationship':\n",
    "                        rels = go_dict[go][value]\n",
    "                        for rel in rels:\n",
    "                            rel_type = '-'+rel.split(' ')[0]+'->'\n",
    "                            start_node = go.split('GO:')[1]\n",
    "                            start_node = go2go_type+':'+start_node\n",
    "                            end_node = rel.split(' ')[1]\n",
    "                            end_node = end_node.split('GO:')[1]\n",
    "                            end_node = go2go_type+':'+end_node\n",
    "                            go2go.setdefault(start_node, dict()).setdefault(rel_type,[]).append(end_node)\n",
    "                            writer.writerow([start_node, end_node, rel_type])\n",
    "                            rel_counts[rel_type] = rel_counts.setdefault(rel_type,0) + 1    \n",
    "\n",
    "    df = pd.read_csv(f'output/go2go/{cap_go2go_type}_(GO)_2_{cap_go2go_type}_(GO).csv')\n",
    "    df.to_csv(f'output/go2go/{cap_go2go_type}_(GO)_2_{cap_go2go_type}_(GO).csv', index=False)\n",
    "    \n",
    "    df.to_csv(f'output/go2go/{cap_go2go_type}_(GO)_2_{cap_go2go_type}_(GO).csv', index=False)\n",
    "    df.to_csv(f'output/edges/{cap_go2go_type}_(GO)_2_{cap_go2go_type}_(GO).csv', index=False)\n",
    "    df.to_csv(f'output/edges to use/{cap_go2go_type}_(GO)_2_{cap_go2go_type}_(GO).csv', index=False)\n",
    "    \n",
    "    \n",
    "    print(go2go_type)\n",
    "    display(rel_counts) # Relationship counts\n",
    "    display(df.head(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GO:0018130',\n",
       " 'GO:0034309',\n",
       " 'GO:0042181',\n",
       " 'GO:0120255',\n",
       " 'GO:1901362',\n",
       " 'GO:2001316']"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "is_a_gos"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Node to Text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "go2text = dict()\n",
    "\n",
    "for the_go_term in go_dict:\n",
    "    try:\n",
    "        namespace = go_dict[the_go_term]['namespace'][0]\n",
    "        if namespace in go_ontology_names:\n",
    "            go_term = namespace+':'+the_go_term.split('GO:')[1]\n",
    "            text_description = go_dict[the_go_term]['def'][0].split('\\\"')[1]\n",
    "            go2text[go_term] = text_description\n",
    "    except:\n",
    "        continue\n",
    "        \n",
    "json.dump(go2text, open('output/nodes/node features/node2text/go2text_description.json','w'))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
